{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3da6228f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import logging\n",
    "import os\n",
    "import re\n",
    "from pathlib import Path\n",
    "\n",
    "import anndata\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import scanpy as sc\n",
    "from sklearn.metrics import classification_report, confusion_matrix\n",
    "from sklearn.calibration import CalibratedClassifierCV\n",
    "from sklearn.svm import SVC, LinearSVC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3c8e732c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pollock.utils as utils\n",
    "from pollock.dataloaders import normalize, get_train_dataloaders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fd8e4a5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import torch"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5639ace6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'snRNAseq': {'ccrcc': {'train': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/ccrcc_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/ccrcc_val.h5ad'},\n",
       "  'brca': {'train': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/brca_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/brca_val.h5ad'},\n",
       "  'gbm': {'val': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/gbm_val.h5ad',\n",
       "   'train': '/data/pollock/benchmarking/pollock_datasets/snRNAseq/gbm_train.h5ad'}},\n",
       " 'snATACseq': {'brca': {'train': '/data/pollock/benchmarking/pollock_datasets/snATACseq/brca_gene_activity_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/snATACseq/brca_gene_activity_val.h5ad'},\n",
       "  'ccrcc': {'train': '/data/pollock/benchmarking/pollock_datasets/snATACseq/ccrcc_gene_activity_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/snATACseq/ccrcc_gene_activity_val.h5ad'},\n",
       "  'gbm': {'val': '/data/pollock/benchmarking/pollock_datasets/snATACseq/gbm_gene_activity_val.h5ad',\n",
       "   'train': '/data/pollock/benchmarking/pollock_datasets/snATACseq/gbm_gene_activity_train.h5ad'}},\n",
       " 'scRNAseq': {'cesc': {'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/cesc_val.h5ad',\n",
       "   'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/cesc_train.h5ad'},\n",
       "  'myeloma': {'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/myeloma_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/myeloma_val.h5ad'},\n",
       "  'brca': {'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/brca_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/brca_val.h5ad'},\n",
       "  'hnscc': {'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/hnscc_train.h5ad',\n",
       "   'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/hnscc_val.h5ad'},\n",
       "  'pdac': {'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/pdac_val.h5ad',\n",
       "   'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/pdac_train.h5ad'},\n",
       "  'melanoma': {'val': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/melanoma_val.h5ad',\n",
       "   'train': '/data/pollock/benchmarking/pollock_datasets/scRNAseq/melanoma_train.h5ad'}}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_dir = '/data/pollock/benchmarking/pollock_datasets/'\n",
    "fps = utils.listfiles(data_dir, regex=r'.h5ad')\n",
    "fps = [fp for fp in fps if '_train' in fp or '_val' in fp]\n",
    "\n",
    "fmap = {}\n",
    "for fp in fps:\n",
    "    dtype = fp.split('/')[-2]\n",
    "    disease = fp.split('/')[-1].split('_')[0]\n",
    "    partition = 'train' if '_train' in fp else 'val'\n",
    "    \n",
    "    if dtype not in fmap:\n",
    "        fmap[dtype] = {}\n",
    "    if disease not in fmap[dtype]:\n",
    "        fmap[dtype][disease] = {}\n",
    "    \n",
    "    if not (dtype == 'snATACseq' and ('peaks' in fp or 'motif' in fp)):\n",
    "        fmap[dtype][disease][partition] = fp\n",
    "fmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "71977e7b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'snRNAseq': {'ccrcc': {'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold3_val.h5ad'},\n",
       "   'fold1': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold1_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold1_train.h5ad'},\n",
       "   'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold2_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold0_train.h5ad'},\n",
       "   'fold4': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold4_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold4_train.h5ad'}},\n",
       "  'brca': {'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold2_val.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold1_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold3_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold0_train.h5ad'},\n",
       "   'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold4_val.h5ad'}},\n",
       "  'gbm': {'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold4_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold3_val.h5ad'},\n",
       "   'fold0': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold0_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold0_val.h5ad'},\n",
       "   'fold2': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold2_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold2_train.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold1_val.h5ad'}}},\n",
       " 'snATACseq': {'ccrcc': {'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold3_val.h5ad'},\n",
       "   'fold1': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold1_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold1_train.h5ad'},\n",
       "   'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold2_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold0_train.h5ad'},\n",
       "   'fold4': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold4_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold4_train.h5ad'}},\n",
       "  'brca': {'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold2_val.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold1_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold3_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold0_train.h5ad'},\n",
       "   'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold4_val.h5ad'}},\n",
       "  'gbm': {'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold4_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold3_val.h5ad'},\n",
       "   'fold0': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold0_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold0_val.h5ad'},\n",
       "   'fold2': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold2_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold2_train.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold1_val.h5ad'}}},\n",
       " 'scRNAseq': {'hnscc': {'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold0_train.h5ad'},\n",
       "   'fold4': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold4_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold4_train.h5ad'},\n",
       "   'fold1': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold1_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold1_train.h5ad'},\n",
       "   'fold3': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold3_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold3_train.h5ad'},\n",
       "   'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold2_val.h5ad'}},\n",
       "  'cesc': {'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold4_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold3_val.h5ad'},\n",
       "   'fold2': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold2_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold2_train.h5ad'},\n",
       "   'fold0': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold0_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold0_val.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold1_val.h5ad'}},\n",
       "  'brca': {'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_val.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold1_val.h5ad'},\n",
       "   'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold3_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold0_train.h5ad'},\n",
       "   'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold4_val.h5ad'}},\n",
       "  'myeloma': {'fold0': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold0_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold0_val.h5ad'},\n",
       "   'fold1': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold1_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold1_train.h5ad'},\n",
       "   'fold3': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold3_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold3_train.h5ad'},\n",
       "   'fold4': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold4_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold4_train.h5ad'},\n",
       "   'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold2_val.h5ad'}},\n",
       "  'melanoma': {'fold3': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold3_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold3_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold0_train.h5ad'},\n",
       "   'fold2': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold2_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold2_train.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold1_val.h5ad'},\n",
       "   'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold4_val.h5ad'}},\n",
       "  'pdac': {'fold2': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold2_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold2_val.h5ad'},\n",
       "   'fold3': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold3_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold3_train.h5ad'},\n",
       "   'fold1': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold1_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold1_val.h5ad'},\n",
       "   'fold4': {'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold4_train.h5ad',\n",
       "    'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold4_val.h5ad'},\n",
       "   'fold0': {'val': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold0_val.h5ad',\n",
       "    'train': '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold0_train.h5ad'}}}}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_dir = '/data/pollock/benchmarking/pollock_datasets_with_folds/'\n",
    "fps = utils.listfiles(data_dir, regex=r'.h5ad')\n",
    "fps = [fp for fp in fps if '_train' in fp or '_val' in fp]\n",
    "\n",
    "folds_fmap = {}\n",
    "for fp in fps:\n",
    "    dtype = fp.split('/')[-2]\n",
    "    disease = fp.split('/')[-1].split('_')[0]\n",
    "    partition = 'train' if '_train' in fp else 'val'\n",
    "    fold = fp.split('/')[-1].split('_')[-2]\n",
    "    \n",
    "    if dtype not in folds_fmap:\n",
    "        folds_fmap[dtype] = {}\n",
    "    if disease not in folds_fmap[dtype]:\n",
    "        folds_fmap[dtype][disease] = {}\n",
    "    if fold not in folds_fmap[dtype][disease]:\n",
    "        folds_fmap[dtype][disease][fold] = {}\n",
    "    \n",
    "    if not (dtype == 'snATACseq' and ('peaks' in fp or 'motif' in fp)):\n",
    "        folds_fmap[dtype][disease][fold][partition] = fp\n",
    "folds_fmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "923b22da",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_metrics(y_true_labels, y_pred_labels, overlapping_only=False):\n",
    "    classes = sorted(set(y_true_labels))\n",
    "    \n",
    "    if overlapping_only:\n",
    "        classes = [c for c in classes if c in y_pred_labels]\n",
    "        y_pred_labels, y_true_labels = zip(*[(p, t) for p, t in zip(y_pred_labels, y_true_labels)\n",
    "                              if t in classes])\n",
    "        classes = sorted(set(y_true_labels).union(set(y_pred_labels)))\n",
    "\n",
    "    y_pred = [classes.index(c) for c in y_pred_labels]\n",
    "    y_true = [classes.index(c) for c in y_true_labels]\n",
    "\n",
    "    d = classification_report(y_true, y_pred, labels=list(range(len(classes))), target_names=classes,\n",
    "                             output_dict=True)\n",
    "    d = pd.DataFrame.from_dict(d)\n",
    "    cm = confusion_matrix(y_true, y_pred, labels=list(range(len(classes))))\n",
    "    cm = pd.DataFrame(data=cm, index=classes, columns=classes)\n",
    "    \n",
    "    d.index.name = 'metric'\n",
    "    cm.index.name = 'cell_type'\n",
    "    \n",
    "    return d, cm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a825061",
   "metadata": {},
   "source": [
    "#### predict cell types"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64cb6223",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### scanpy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "849cec71",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/scanpy/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8b6f2e8b",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def normalize_for_scanpy(a, highly_variable=True):\n",
    "    sc.pp.normalize_total(a, target_sum=1e4)\n",
    "    sc.pp.log1p(a)\n",
    "    \n",
    "    if highly_variable:\n",
    "        sc.pp.highly_variable_genes(a)\n",
    "        a.raw = a\n",
    "        a = a[:, a.var.highly_variable]\n",
    "    \n",
    "    sc.pp.scale(a)\n",
    "    \n",
    "    return a\n",
    "\n",
    "\n",
    "\n",
    "def predict_scanpy(train, val):\n",
    "    val.obs['orig_cell_type'] = val.obs['cell_type'].to_list()\n",
    "    \n",
    "    train, val = normalize_for_scanpy(train), normalize_for_scanpy(val, highly_variable=False)\n",
    "    \n",
    "    var_names = train.var_names.intersection(val.var_names)\n",
    "    train = train[:, var_names]\n",
    "    val = val[:, var_names]\n",
    "    \n",
    "    sc.pp.pca(train)\n",
    "    sc.pp.neighbors(train)\n",
    "    sc.tl.umap(train)\n",
    "    \n",
    "    sc.tl.ingest(val, train, obs='cell_type')\n",
    "    \n",
    "    val.obs['predicted_cell_type'] = val.obs['cell_type'].to_list()\n",
    "    val.obs['cell_type'] = val.obs['orig_cell_type'].to_list()\n",
    "    \n",
    "    return val\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "28267c6a",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        train = sc.read_h5ad(d2['train'])\n",
    "        val = sc.read_h5ad(d2['val'])\n",
    "        \n",
    "        to_prediction[dtype][disease] = predict_scanpy(train, val)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2e7f8731",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, a in d1.items():\n",
    "        a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f2a04e57",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n",
      "snRNAseq brca\n",
      "snRNAseq gbm\n",
      "snATACseq brca\n",
      "snATACseq ccrcc\n",
      "snATACseq gbm\n",
      "scRNAseq cesc\n",
      "scRNAseq myeloma\n",
      "scRNAseq brca\n",
      "scRNAseq hnscc\n",
      "scRNAseq pdac\n",
      "scRNAseq melanoma\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in fmap.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))\n",
    "        y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "        y_true = adata.obs['cell_type'].to_list()\n",
    "        d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "        d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "        cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c378ffd6",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# with folds\n",
    "result_dir = '/data/pollock/benchmarking/results/scanpy/interdataset_with_folds'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0ab5b3f1",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/scanpy/preprocessing/_simple.py:843: UserWarning: Revieved a view of an AnnData. Making a copy.\n",
      "  view_to_actual(adata)\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in folds_fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        to_prediction[dtype][disease] = {}\n",
    "        for fold, d3 in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            train = sc.read_h5ad(d3['train'])\n",
    "            val = sc.read_h5ad(d3['val'])\n",
    "\n",
    "            to_prediction[dtype][disease][fold] = predict_scanpy(train, val)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "206af6b4",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "99fd1246",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold4\n",
      "snRNAseq gbm fold4\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold1\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold1\n",
      "snATACseq brca fold3\n",
      "snATACseq brca fold0\n",
      "snATACseq brca fold4\n",
      "snATACseq gbm fold4\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold1\n",
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold2\n",
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq brca fold2\n",
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold4\n",
      "scRNAseq myeloma fold0\n",
      "scRNAseq myeloma fold1\n",
      "scRNAseq myeloma fold3\n",
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n",
      "scRNAseq melanoma fold3\n",
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold1\n",
      "scRNAseq melanoma fold4\n",
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold1\n",
      "scRNAseq pdac fold4\n",
      "scRNAseq pdac fold0\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))\n",
    "            y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "            y_true = adata.obs['cell_type'].to_list()\n",
    "            d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "            d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "            cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a3561fd",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "21331e20",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### svm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "9037be6c",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def predict_svm(train, val, method='linear'):    \n",
    "    train, val = normalize(train), normalize(val)\n",
    "    \n",
    "    var_names = train.var_names.intersection(val.var_names)\n",
    "    train = train[:, var_names]\n",
    "    val = val[:, var_names]\n",
    "    \n",
    "    X = train.X.toarray() if 'sparse' in str(type(train.X)).lower() else train.X\n",
    "    labels = train.obs['cell_type'].to_list()\n",
    "    to_label = sorted(set(labels))\n",
    "    y = [to_label.index(l) for l in labels]\n",
    "\n",
    "    clf = LinearSVC() if method == 'linear' else SVC()\n",
    "    clf = CalibratedClassifierCV(clf)\n",
    "    \n",
    "    clf.fit(X, y)\n",
    "    \n",
    "    X = val.X.toarray() if 'sparse' in str(type(val.X)).lower() else val.X\n",
    "    preds = clf.predict(X)\n",
    "    probs = clf.predict_proba(X)\n",
    "\n",
    "    val.obs['predicted_cell_type'] = [to_label[c] for c in preds]\n",
    "    \n",
    "    prob_df = pd.DataFrame(data=probs, index=val.obs.index.to_list(),\n",
    "                          columns=[f'probability {ct}' for ct in to_label])\n",
    "    \n",
    "    val.obs = pd.concat((val.obs, prob_df), axis=1)\n",
    "    \n",
    "    return val\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "40c74d3f",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/svm_linear/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3d0ba5b7",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        train = sc.read_h5ad(d2['train'])\n",
    "        val = sc.read_h5ad(d2['val'])\n",
    "        \n",
    "        to_prediction[dtype][disease] = predict_svm(train, val, method='linear')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "496447de",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, a in d1.items():\n",
    "        a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dba4ae89",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n",
      "snRNAseq brca\n",
      "snRNAseq gbm\n",
      "snATACseq brca\n",
      "snATACseq ccrcc\n",
      "snATACseq gbm\n",
      "scRNAseq cesc\n",
      "scRNAseq myeloma\n",
      "scRNAseq brca\n",
      "scRNAseq hnscc\n",
      "scRNAseq pdac\n",
      "scRNAseq melanoma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in fmap.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))\n",
    "        y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "        y_true = adata.obs['cell_type'].to_list()\n",
    "        d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "        d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "        cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "27ca4c92",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# with folds\n",
    "result_dir = '/data/pollock/benchmarking/results/svm_linear/interdataset_with_folds'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "c6eba31a",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in folds_fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        to_prediction[dtype][disease] = {}\n",
    "        for fold, d3 in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            train = sc.read_h5ad(d3['train'])\n",
    "            val = sc.read_h5ad(d3['val'])\n",
    "\n",
    "            to_prediction[dtype][disease][fold] = predict_svm(train, val, method='linear')\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "7f025569",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4378cfe9",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold4\n",
      "snRNAseq gbm fold4\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold1\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold1\n",
      "snATACseq brca fold3\n",
      "snATACseq brca fold0\n",
      "snATACseq brca fold4\n",
      "snATACseq gbm fold4\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold1\n",
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold2\n",
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq brca fold2\n",
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold4\n",
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold3\n",
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold1\n",
      "scRNAseq melanoma fold4\n",
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold1\n",
      "scRNAseq pdac fold4\n",
      "scRNAseq pdac fold0\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))\n",
    "            y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "            y_true = adata.obs['cell_type'].to_list()\n",
    "            d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "            d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "            cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fbf7fac0",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### MLP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "eedf6a89",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "class MLP(torch.nn.Module):\n",
    "    def __init__(self, genes, classes, method='small'):\n",
    "        \"\"\"\n",
    "        MLP\n",
    "        \"\"\"\n",
    "        super(MLP, self).__init__()\n",
    "        self.genes = genes\n",
    "        self.n_genes = len(genes)\n",
    "        self.classes = classes\n",
    "        self.n_classes = len(classes)\n",
    "\n",
    "        if method == 'small':\n",
    "            self.layers = torch.nn.Sequential(\n",
    "                torch.nn.Linear(self.n_genes, 128),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(128, 32),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(32, 16),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(16, 8),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(8, self.n_classes),\n",
    "                torch.nn.Softmax(dim=1)\n",
    "            )\n",
    "        else:\n",
    "            self.layers = torch.nn.Sequential(\n",
    "                torch.nn.Linear(self.n_genes, 64),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(64, 16),\n",
    "                torch.nn.ReLU(),\n",
    "                torch.nn.Dropout(.1),\n",
    "                torch.nn.Linear(16, self.n_classes),\n",
    "                torch.nn.Softmax(dim=1)\n",
    "            )\n",
    "            \n",
    "    def forward(self, x):\n",
    "        x = self.layers(x)\n",
    "        return x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "01f662b8",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "def train_mlp(train_dl, val_dl, lr=1e-4, epochs=25):\n",
    "    model = MLP(train_dl.dataset.adata.var.index.to_list(), train_dl.dataset.cell_types,\n",
    "                method='small' if train_dl.dataset.adata.shape[0] < 5000 else 'large')\n",
    "    model = model.cuda()\n",
    "\n",
    "    criteria = torch.nn.CrossEntropyLoss()\n",
    "\n",
    "    opt = torch.optim.Adam(model.parameters(), lr=lr)\n",
    "    \n",
    "    \n",
    "    for epoch in range(epochs):\n",
    "        train_loss, val_loss = 0., 0.\n",
    "        start = time.time()\n",
    "        model.train()\n",
    "        for i, b in enumerate(train_dl):\n",
    "            x, y = b['x'], b['y']\n",
    "            x, y = x.cuda(), y.cuda()\n",
    "            opt.zero_grad()\n",
    "            out = model(x)\n",
    "            loss = criteria(out, y)\n",
    "            loss.backward()\n",
    "            opt.step()\n",
    "\n",
    "            train_loss += float(loss.detach().cpu())\n",
    "        train_loss = train_loss / len(train_dl)\n",
    "\n",
    "        time_delta = time.time() - start\n",
    "        model.eval()\n",
    "        with torch.no_grad():\n",
    "            for i, b in enumerate(val_dl):\n",
    "                x, y = b['x'], b['y']\n",
    "                x, y = x.cuda(), y.cuda()\n",
    "\n",
    "                out = model(x)\n",
    "                loss = criteria(out, y)\n",
    "                val_loss += float(loss.detach().cpu())\n",
    "\n",
    "\n",
    "        val_loss = val_loss / len(val_dl)\n",
    "\n",
    "        print(f'epoch: {epoch}, train loss: {train_loss:.3f}, val loss: {val_loss:.3f}, time: {time_delta:.2f}')\n",
    "    \n",
    "    return model\n",
    "\n",
    "def mlp_predict_dl(dl, model):\n",
    "    y_prob = None\n",
    "    model.eval()\n",
    "    with torch.no_grad():\n",
    "        for i, b in enumerate(dl):\n",
    "            x, y = b['x'], b['y']\n",
    "            x, y = x.cuda(), y.cuda()\n",
    "\n",
    "            out = model(x).detach().cpu().numpy()\n",
    "\n",
    "            if y_prob is None:\n",
    "                y_prob = out\n",
    "            else:\n",
    "                y_prob = np.concatenate((y_prob, out), axis=0)\n",
    "                \n",
    "    labels = [model.classes[i] for i in np.argmax(y_prob, axis=1)]\n",
    "    \n",
    "    return labels, y_prob\n",
    "\n",
    "def mlp_predict(train, val, epochs=25):\n",
    "    train_dl, val_dl = get_train_dataloaders(train, val)\n",
    "    model = train_mlp(train_dl, val_dl, epochs=epochs)\n",
    "    labels, probs = mlp_predict_dl(val_dl, model)\n",
    "    val.obs['predicted_cell_type'] = labels\n",
    "\n",
    "    prob_df = pd.DataFrame(data=probs, index=val.obs.index.to_list(),\n",
    "                          columns=[f'probability {ct}' for ct in model.classes])\n",
    "\n",
    "    val.obs = pd.concat((val.obs, prob_df), axis=1)\n",
    "    \n",
    "    return val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "4e1e9bc8",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/mlp/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ea1bf78c",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:28:55,625 24684 genes overlap with model after filtering\n",
      "2022-01-19 07:28:55,627 1318 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.384, val loss: 2.351, time: 1.74\n",
      "epoch: 1, train loss: 2.283, val loss: 2.203, time: 1.36\n",
      "epoch: 2, train loss: 2.190, val loss: 2.136, time: 1.33\n",
      "epoch: 3, train loss: 2.135, val loss: 2.078, time: 1.41\n",
      "epoch: 4, train loss: 2.082, val loss: 2.014, time: 1.31\n",
      "epoch: 5, train loss: 2.032, val loss: 1.959, time: 1.31\n",
      "epoch: 6, train loss: 1.985, val loss: 1.910, time: 1.31\n",
      "epoch: 7, train loss: 1.948, val loss: 1.879, time: 1.31\n",
      "epoch: 8, train loss: 1.918, val loss: 1.850, time: 1.31\n",
      "epoch: 9, train loss: 1.886, val loss: 1.825, time: 1.29\n",
      "epoch: 10, train loss: 1.865, val loss: 1.803, time: 1.37\n",
      "epoch: 11, train loss: 1.835, val loss: 1.785, time: 1.34\n",
      "epoch: 12, train loss: 1.815, val loss: 1.770, time: 1.31\n",
      "epoch: 13, train loss: 1.802, val loss: 1.757, time: 1.30\n",
      "epoch: 14, train loss: 1.789, val loss: 1.750, time: 1.30\n",
      "epoch: 15, train loss: 1.775, val loss: 1.741, time: 1.30\n",
      "epoch: 16, train loss: 1.760, val loss: 1.731, time: 1.30\n",
      "epoch: 17, train loss: 1.758, val loss: 1.727, time: 1.30\n",
      "epoch: 18, train loss: 1.747, val loss: 1.723, time: 1.29\n",
      "epoch: 19, train loss: 1.742, val loss: 1.719, time: 1.30\n",
      "epoch: 20, train loss: 1.733, val loss: 1.714, time: 1.38\n",
      "epoch: 21, train loss: 1.734, val loss: 1.709, time: 1.33\n",
      "epoch: 22, train loss: 1.732, val loss: 1.706, time: 1.31\n",
      "epoch: 23, train loss: 1.718, val loss: 1.699, time: 1.31\n",
      "epoch: 24, train loss: 1.711, val loss: 1.687, time: 1.30\n",
      "snRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:30:00,411 25360 genes overlap with model after filtering\n",
      "2022-01-19 07:30:00,412 1183 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.475, val loss: 2.330, time: 1.30\n",
      "epoch: 1, train loss: 2.233, val loss: 2.145, time: 1.28\n",
      "epoch: 2, train loss: 2.098, val loss: 2.045, time: 1.26\n",
      "epoch: 3, train loss: 1.993, val loss: 1.985, time: 1.28\n",
      "epoch: 4, train loss: 1.928, val loss: 1.945, time: 1.27\n",
      "epoch: 5, train loss: 1.871, val loss: 1.894, time: 1.32\n",
      "epoch: 6, train loss: 1.831, val loss: 1.873, time: 1.36\n",
      "epoch: 7, train loss: 1.812, val loss: 1.861, time: 1.25\n",
      "epoch: 8, train loss: 1.793, val loss: 1.853, time: 1.25\n",
      "epoch: 9, train loss: 1.783, val loss: 1.842, time: 1.26\n",
      "epoch: 10, train loss: 1.764, val loss: 1.834, time: 1.24\n",
      "epoch: 11, train loss: 1.749, val loss: 1.826, time: 1.24\n",
      "epoch: 12, train loss: 1.743, val loss: 1.823, time: 1.24\n",
      "epoch: 13, train loss: 1.738, val loss: 1.819, time: 1.26\n",
      "epoch: 14, train loss: 1.733, val loss: 1.818, time: 1.23\n",
      "epoch: 15, train loss: 1.728, val loss: 1.815, time: 1.29\n",
      "epoch: 16, train loss: 1.727, val loss: 1.814, time: 1.31\n",
      "epoch: 17, train loss: 1.722, val loss: 1.813, time: 1.44\n",
      "epoch: 18, train loss: 1.720, val loss: 1.813, time: 1.27\n",
      "epoch: 19, train loss: 1.716, val loss: 1.810, time: 1.27\n",
      "epoch: 20, train loss: 1.717, val loss: 1.810, time: 1.27\n",
      "epoch: 21, train loss: 1.715, val loss: 1.809, time: 1.24\n",
      "epoch: 22, train loss: 1.711, val loss: 1.811, time: 1.26\n",
      "epoch: 23, train loss: 1.709, val loss: 1.806, time: 1.27\n",
      "epoch: 24, train loss: 1.710, val loss: 1.805, time: 1.24\n",
      "snRNAseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:31:01,326 25009 genes overlap with model after filtering\n",
      "2022-01-19 07:31:01,327 1142 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.189, val loss: 2.159, time: 1.01\n",
      "epoch: 1, train loss: 2.090, val loss: 1.989, time: 1.02\n",
      "epoch: 2, train loss: 1.978, val loss: 1.900, time: 1.08\n",
      "epoch: 3, train loss: 1.909, val loss: 1.840, time: 1.08\n",
      "epoch: 4, train loss: 1.848, val loss: 1.769, time: 1.09\n",
      "epoch: 5, train loss: 1.799, val loss: 1.716, time: 1.09\n",
      "epoch: 6, train loss: 1.745, val loss: 1.649, time: 1.09\n",
      "epoch: 7, train loss: 1.701, val loss: 1.617, time: 1.09\n",
      "epoch: 8, train loss: 1.653, val loss: 1.559, time: 1.10\n",
      "epoch: 9, train loss: 1.616, val loss: 1.535, time: 1.10\n",
      "epoch: 10, train loss: 1.595, val loss: 1.527, time: 1.10\n",
      "epoch: 11, train loss: 1.586, val loss: 1.523, time: 1.11\n",
      "epoch: 12, train loss: 1.584, val loss: 1.519, time: 1.09\n",
      "epoch: 13, train loss: 1.573, val loss: 1.513, time: 1.03\n",
      "epoch: 14, train loss: 1.564, val loss: 1.507, time: 1.09\n",
      "epoch: 15, train loss: 1.564, val loss: 1.499, time: 1.09\n",
      "epoch: 16, train loss: 1.546, val loss: 1.485, time: 1.10\n",
      "epoch: 17, train loss: 1.527, val loss: 1.458, time: 1.09\n",
      "epoch: 18, train loss: 1.499, val loss: 1.426, time: 1.09\n",
      "epoch: 19, train loss: 1.488, val loss: 1.412, time: 1.10\n",
      "epoch: 20, train loss: 1.480, val loss: 1.406, time: 1.01\n",
      "epoch: 21, train loss: 1.477, val loss: 1.403, time: 1.09\n",
      "epoch: 22, train loss: 1.468, val loss: 1.402, time: 1.09\n",
      "epoch: 23, train loss: 1.470, val loss: 1.401, time: 1.03\n",
      "epoch: 24, train loss: 1.468, val loss: 1.400, time: 1.08\n",
      "snATACseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:31:52,531 19699 genes overlap with model after filtering\n",
      "2022-01-19 07:31:52,532 17 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.059, val loss: 2.021, time: 0.94\n",
      "epoch: 1, train loss: 1.979, val loss: 1.928, time: 1.02\n",
      "epoch: 2, train loss: 1.904, val loss: 1.857, time: 1.01\n",
      "epoch: 3, train loss: 1.838, val loss: 1.767, time: 1.02\n",
      "epoch: 4, train loss: 1.772, val loss: 1.735, time: 1.02\n",
      "epoch: 5, train loss: 1.747, val loss: 1.720, time: 1.02\n",
      "epoch: 6, train loss: 1.726, val loss: 1.702, time: 1.02\n",
      "epoch: 7, train loss: 1.702, val loss: 1.676, time: 1.03\n",
      "epoch: 8, train loss: 1.670, val loss: 1.648, time: 1.00\n",
      "epoch: 9, train loss: 1.644, val loss: 1.627, time: 1.03\n",
      "epoch: 10, train loss: 1.621, val loss: 1.605, time: 1.02\n",
      "epoch: 11, train loss: 1.582, val loss: 1.569, time: 1.02\n",
      "epoch: 12, train loss: 1.553, val loss: 1.539, time: 1.03\n",
      "epoch: 13, train loss: 1.527, val loss: 1.519, time: 1.03\n",
      "epoch: 14, train loss: 1.510, val loss: 1.507, time: 1.03\n",
      "epoch: 15, train loss: 1.495, val loss: 1.493, time: 1.03\n",
      "epoch: 16, train loss: 1.482, val loss: 1.478, time: 1.03\n",
      "epoch: 17, train loss: 1.466, val loss: 1.458, time: 1.02\n",
      "epoch: 18, train loss: 1.442, val loss: 1.435, time: 1.02\n",
      "epoch: 19, train loss: 1.411, val loss: 1.411, time: 1.02\n",
      "epoch: 20, train loss: 1.395, val loss: 1.395, time: 1.02\n",
      "epoch: 21, train loss: 1.381, val loss: 1.383, time: 1.03\n",
      "epoch: 22, train loss: 1.367, val loss: 1.375, time: 1.05\n",
      "epoch: 23, train loss: 1.362, val loss: 1.371, time: 0.99\n",
      "epoch: 24, train loss: 1.360, val loss: 1.368, time: 1.01\n",
      "snATACseq ccrcc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:32:42,152 19677 genes overlap with model after filtering\n",
      "2022-01-19 07:32:42,153 40 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.785, val loss: 1.776, time: 0.78\n",
      "epoch: 1, train loss: 1.755, val loss: 1.722, time: 0.86\n",
      "epoch: 2, train loss: 1.648, val loss: 1.566, time: 0.85\n",
      "epoch: 3, train loss: 1.513, val loss: 1.460, time: 0.85\n",
      "epoch: 4, train loss: 1.439, val loss: 1.403, time: 0.79\n",
      "epoch: 5, train loss: 1.396, val loss: 1.378, time: 0.86\n",
      "epoch: 6, train loss: 1.376, val loss: 1.366, time: 0.85\n",
      "epoch: 7, train loss: 1.363, val loss: 1.359, time: 0.85\n",
      "epoch: 8, train loss: 1.355, val loss: 1.353, time: 0.85\n",
      "epoch: 9, train loss: 1.345, val loss: 1.349, time: 0.91\n",
      "epoch: 10, train loss: 1.336, val loss: 1.345, time: 0.86\n",
      "epoch: 11, train loss: 1.334, val loss: 1.341, time: 0.86\n",
      "epoch: 12, train loss: 1.321, val loss: 1.334, time: 0.85\n",
      "epoch: 13, train loss: 1.311, val loss: 1.319, time: 0.85\n",
      "epoch: 14, train loss: 1.291, val loss: 1.284, time: 0.85\n",
      "epoch: 15, train loss: 1.263, val loss: 1.251, time: 0.85\n",
      "epoch: 16, train loss: 1.236, val loss: 1.240, time: 0.85\n",
      "epoch: 17, train loss: 1.226, val loss: 1.235, time: 0.86\n",
      "epoch: 18, train loss: 1.226, val loss: 1.232, time: 0.85\n",
      "epoch: 19, train loss: 1.215, val loss: 1.229, time: 0.86\n",
      "epoch: 20, train loss: 1.217, val loss: 1.228, time: 0.85\n",
      "epoch: 21, train loss: 1.220, val loss: 1.227, time: 0.86\n",
      "epoch: 22, train loss: 1.213, val loss: 1.225, time: 0.85\n",
      "epoch: 23, train loss: 1.210, val loss: 1.222, time: 0.89\n",
      "epoch: 24, train loss: 1.211, val loss: 1.220, time: 0.87\n",
      "snATACseq gbm\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:33:23,713 19698 genes overlap with model after filtering\n",
      "2022-01-19 07:33:23,714 30 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.065, val loss: 2.048, time: 0.88\n",
      "epoch: 1, train loss: 2.019, val loss: 1.963, time: 0.96\n",
      "epoch: 2, train loss: 1.928, val loss: 1.852, time: 0.96\n",
      "epoch: 3, train loss: 1.858, val loss: 1.794, time: 0.96\n",
      "epoch: 4, train loss: 1.777, val loss: 1.723, time: 0.97\n",
      "epoch: 5, train loss: 1.701, val loss: 1.663, time: 0.97\n",
      "epoch: 6, train loss: 1.660, val loss: 1.635, time: 0.97\n",
      "epoch: 7, train loss: 1.635, val loss: 1.621, time: 0.97\n",
      "epoch: 8, train loss: 1.619, val loss: 1.612, time: 0.97\n",
      "epoch: 9, train loss: 1.612, val loss: 1.604, time: 0.90\n",
      "epoch: 10, train loss: 1.602, val loss: 1.595, time: 0.95\n",
      "epoch: 11, train loss: 1.593, val loss: 1.583, time: 1.00\n",
      "epoch: 12, train loss: 1.583, val loss: 1.565, time: 0.99\n",
      "epoch: 13, train loss: 1.568, val loss: 1.537, time: 0.96\n",
      "epoch: 14, train loss: 1.549, val loss: 1.506, time: 0.96\n",
      "epoch: 15, train loss: 1.531, val loss: 1.493, time: 0.95\n",
      "epoch: 16, train loss: 1.514, val loss: 1.482, time: 0.96\n",
      "epoch: 17, train loss: 1.501, val loss: 1.472, time: 0.96\n",
      "epoch: 18, train loss: 1.482, val loss: 1.460, time: 0.96\n",
      "epoch: 19, train loss: 1.471, val loss: 1.439, time: 0.96\n",
      "epoch: 20, train loss: 1.449, val loss: 1.402, time: 0.96\n",
      "epoch: 21, train loss: 1.427, val loss: 1.372, time: 0.96\n",
      "epoch: 22, train loss: 1.414, val loss: 1.359, time: 0.97\n",
      "epoch: 23, train loss: 1.400, val loss: 1.354, time: 0.96\n",
      "epoch: 24, train loss: 1.398, val loss: 1.349, time: 0.96\n",
      "scRNAseq cesc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:34:07,252 21146 genes overlap with model after filtering\n",
      "2022-01-19 07:34:07,253 703 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.385, val loss: 2.350, time: 1.23\n",
      "epoch: 1, train loss: 2.282, val loss: 2.158, time: 1.24\n",
      "epoch: 2, train loss: 2.134, val loss: 1.992, time: 1.24\n",
      "epoch: 3, train loss: 2.035, val loss: 1.900, time: 1.24\n",
      "epoch: 4, train loss: 1.989, val loss: 1.863, time: 1.22\n",
      "epoch: 5, train loss: 1.953, val loss: 1.841, time: 1.23\n",
      "epoch: 6, train loss: 1.924, val loss: 1.820, time: 1.22\n",
      "epoch: 7, train loss: 1.898, val loss: 1.801, time: 1.24\n",
      "epoch: 8, train loss: 1.873, val loss: 1.780, time: 1.24\n",
      "epoch: 9, train loss: 1.853, val loss: 1.767, time: 1.26\n",
      "scRNAseq myeloma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:34:31,628 18733 genes overlap with model after filtering\n",
      "2022-01-19 07:34:31,629 1290 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.388, val loss: 2.370, time: 0.96\n",
      "epoch: 1, train loss: 2.339, val loss: 2.286, time: 1.00\n",
      "epoch: 2, train loss: 2.266, val loss: 2.226, time: 1.01\n",
      "epoch: 3, train loss: 2.205, val loss: 2.184, time: 1.00\n",
      "epoch: 4, train loss: 2.158, val loss: 2.121, time: 1.00\n",
      "epoch: 5, train loss: 2.072, val loss: 1.999, time: 1.00\n",
      "epoch: 6, train loss: 1.988, val loss: 1.911, time: 1.01\n",
      "epoch: 7, train loss: 1.903, val loss: 1.793, time: 1.00\n",
      "epoch: 8, train loss: 1.824, val loss: 1.723, time: 1.00\n",
      "epoch: 9, train loss: 1.780, val loss: 1.685, time: 1.00\n",
      "scRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:34:51,449 22285 genes overlap with model after filtering\n",
      "2022-01-19 07:34:51,450 1268 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.441, val loss: 2.269, time: 1.56\n",
      "epoch: 1, train loss: 2.163, val loss: 2.093, time: 1.52\n",
      "epoch: 2, train loss: 2.016, val loss: 1.980, time: 1.51\n",
      "epoch: 3, train loss: 1.920, val loss: 1.927, time: 1.54\n",
      "epoch: 4, train loss: 1.861, val loss: 1.897, time: 1.56\n",
      "epoch: 5, train loss: 1.813, val loss: 1.880, time: 1.54\n",
      "epoch: 6, train loss: 1.796, val loss: 1.869, time: 1.56\n",
      "epoch: 7, train loss: 1.775, val loss: 1.862, time: 1.48\n",
      "epoch: 8, train loss: 1.767, val loss: 1.858, time: 1.65\n",
      "epoch: 9, train loss: 1.755, val loss: 1.852, time: 1.54\n",
      "scRNAseq hnscc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:35:22,191 21915 genes overlap with model after filtering\n",
      "2022-01-19 07:35:22,193 1008 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.281, val loss: 2.093, time: 1.32\n",
      "epoch: 1, train loss: 1.952, val loss: 1.837, time: 1.29\n",
      "epoch: 2, train loss: 1.777, val loss: 1.758, time: 1.32\n",
      "epoch: 3, train loss: 1.704, val loss: 1.727, time: 1.34\n",
      "epoch: 4, train loss: 1.667, val loss: 1.711, time: 1.33\n",
      "epoch: 5, train loss: 1.642, val loss: 1.702, time: 1.37\n",
      "epoch: 6, train loss: 1.626, val loss: 1.696, time: 1.42\n",
      "epoch: 7, train loss: 1.613, val loss: 1.693, time: 1.43\n",
      "epoch: 8, train loss: 1.605, val loss: 1.689, time: 1.37\n",
      "epoch: 9, train loss: 1.594, val loss: 1.687, time: 1.33\n",
      "scRNAseq pdac\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:35:51,104 23831 genes overlap with model after filtering\n",
      "2022-01-19 07:35:51,105 1188 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.647, val loss: 2.451, time: 1.98\n",
      "epoch: 1, train loss: 2.368, val loss: 2.269, time: 2.15\n",
      "epoch: 2, train loss: 2.237, val loss: 2.200, time: 1.97\n",
      "epoch: 3, train loss: 2.179, val loss: 2.171, time: 2.03\n",
      "epoch: 4, train loss: 2.135, val loss: 2.154, time: 2.10\n",
      "epoch: 5, train loss: 2.114, val loss: 2.145, time: 1.98\n",
      "epoch: 6, train loss: 2.101, val loss: 2.141, time: 1.97\n",
      "epoch: 7, train loss: 2.090, val loss: 2.137, time: 1.98\n",
      "epoch: 8, train loss: 2.083, val loss: 2.135, time: 1.98\n",
      "epoch: 9, train loss: 2.081, val loss: 2.133, time: 1.97\n",
      "scRNAseq melanoma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-01-19 07:36:30,740 19219 genes overlap with model after filtering\n",
      "2022-01-19 07:36:30,741 1385 genes missing from dataset after filtering\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/torch/nn/modules/container.py:141: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.\n",
      "  input = module(input)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.293, val loss: 2.272, time: 1.07\n",
      "epoch: 1, train loss: 2.227, val loss: 2.173, time: 1.16\n",
      "epoch: 2, train loss: 2.134, val loss: 2.083, time: 1.16\n",
      "epoch: 3, train loss: 2.069, val loss: 2.034, time: 1.01\n",
      "epoch: 4, train loss: 2.027, val loss: 1.996, time: 1.03\n",
      "epoch: 5, train loss: 1.988, val loss: 1.938, time: 1.03\n",
      "epoch: 6, train loss: 1.918, val loss: 1.806, time: 1.03\n",
      "epoch: 7, train loss: 1.824, val loss: 1.711, time: 1.03\n",
      "epoch: 8, train loss: 1.786, val loss: 1.681, time: 1.02\n",
      "epoch: 9, train loss: 1.748, val loss: 1.658, time: 1.02\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        train = sc.read_h5ad(d2['train'])\n",
    "        val = sc.read_h5ad(d2['val'])\n",
    "        \n",
    "        to_prediction[dtype][disease] = mlp_predict(train, val,\n",
    "            epochs=10 if dtype=='scRNAseq' else 25)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a2f319fa",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, a in d1.items():\n",
    "        a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "508922e9",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc\n",
      "snRNAseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm\n",
      "snATACseq brca\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc\n",
      "snATACseq gbm\n",
      "scRNAseq cesc\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma\n",
      "scRNAseq brca\n",
      "scRNAseq hnscc\n",
      "scRNAseq pdac\n",
      "scRNAseq melanoma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in fmap.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        print(dtype, disease)\n",
    "        adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}.h5ad'))\n",
    "        y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "        y_true = adata.obs['cell_type'].to_list()\n",
    "        d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "        d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "        cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "d6f30658",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# with folds\n",
    "result_dir = '/data/pollock/benchmarking/results/mlp/interdataset_with_folds'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "15645f63",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:10:45,824 24602 genes overlap with model after filtering\n",
      "2022-03-25 10:10:45,825 1513 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.269, val loss: 2.086, time: 1.71\n",
      "epoch: 1, train loss: 1.974, val loss: 1.915, time: 1.22\n",
      "epoch: 2, train loss: 1.838, val loss: 1.814, time: 1.31\n",
      "epoch: 3, train loss: 1.741, val loss: 1.763, time: 1.25\n",
      "epoch: 4, train loss: 1.689, val loss: 1.721, time: 1.35\n",
      "epoch: 5, train loss: 1.642, val loss: 1.678, time: 1.24\n",
      "epoch: 6, train loss: 1.609, val loss: 1.656, time: 1.21\n",
      "epoch: 7, train loss: 1.594, val loss: 1.649, time: 1.24\n",
      "epoch: 8, train loss: 1.580, val loss: 1.640, time: 1.27\n",
      "epoch: 9, train loss: 1.572, val loss: 1.636, time: 1.23\n",
      "epoch: 10, train loss: 1.568, val loss: 1.633, time: 1.23\n",
      "epoch: 11, train loss: 1.564, val loss: 1.632, time: 1.23\n",
      "epoch: 12, train loss: 1.561, val loss: 1.629, time: 1.25\n",
      "epoch: 13, train loss: 1.560, val loss: 1.629, time: 1.34\n",
      "epoch: 14, train loss: 1.556, val loss: 1.629, time: 1.21\n",
      "epoch: 15, train loss: 1.555, val loss: 1.627, time: 1.21\n",
      "epoch: 16, train loss: 1.555, val loss: 1.626, time: 1.34\n",
      "epoch: 17, train loss: 1.553, val loss: 1.624, time: 1.29\n",
      "epoch: 18, train loss: 1.553, val loss: 1.624, time: 1.21\n",
      "epoch: 19, train loss: 1.553, val loss: 1.623, time: 1.26\n",
      "epoch: 20, train loss: 1.553, val loss: 1.624, time: 1.24\n",
      "epoch: 21, train loss: 1.552, val loss: 1.623, time: 1.33\n",
      "epoch: 22, train loss: 1.550, val loss: 1.624, time: 1.34\n",
      "epoch: 23, train loss: 1.551, val loss: 1.623, time: 1.23\n",
      "epoch: 24, train loss: 1.550, val loss: 1.621, time: 1.22\n",
      "snRNAseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:11:49,239 24613 genes overlap with model after filtering\n",
      "2022-03-25 10:11:49,241 1496 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.268, val loss: 2.080, time: 1.20\n",
      "epoch: 1, train loss: 1.936, val loss: 1.875, time: 1.24\n",
      "epoch: 2, train loss: 1.787, val loss: 1.768, time: 1.25\n",
      "epoch: 3, train loss: 1.697, val loss: 1.703, time: 1.21\n",
      "epoch: 4, train loss: 1.657, val loss: 1.679, time: 1.30\n",
      "epoch: 5, train loss: 1.628, val loss: 1.662, time: 1.20\n",
      "epoch: 6, train loss: 1.602, val loss: 1.651, time: 1.23\n",
      "epoch: 7, train loss: 1.591, val loss: 1.645, time: 1.24\n",
      "epoch: 8, train loss: 1.579, val loss: 1.640, time: 1.25\n",
      "epoch: 9, train loss: 1.575, val loss: 1.636, time: 1.35\n",
      "epoch: 10, train loss: 1.572, val loss: 1.633, time: 1.22\n",
      "epoch: 11, train loss: 1.565, val loss: 1.630, time: 1.24\n",
      "epoch: 12, train loss: 1.564, val loss: 1.628, time: 1.25\n",
      "epoch: 13, train loss: 1.562, val loss: 1.626, time: 1.22\n",
      "epoch: 14, train loss: 1.561, val loss: 1.625, time: 1.24\n",
      "epoch: 15, train loss: 1.558, val loss: 1.625, time: 1.27\n",
      "epoch: 16, train loss: 1.558, val loss: 1.624, time: 1.21\n",
      "epoch: 17, train loss: 1.558, val loss: 1.624, time: 1.22\n",
      "epoch: 18, train loss: 1.557, val loss: 1.624, time: 1.29\n",
      "epoch: 19, train loss: 1.555, val loss: 1.622, time: 1.30\n",
      "epoch: 20, train loss: 1.555, val loss: 1.623, time: 1.21\n",
      "epoch: 21, train loss: 1.555, val loss: 1.624, time: 1.21\n",
      "epoch: 22, train loss: 1.555, val loss: 1.621, time: 1.24\n",
      "epoch: 23, train loss: 1.553, val loss: 1.621, time: 1.22\n",
      "epoch: 24, train loss: 1.554, val loss: 1.621, time: 1.24\n",
      "snRNAseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:12:47,672 24633 genes overlap with model after filtering\n",
      "2022-03-25 10:12:47,673 1406 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.252, val loss: 2.010, time: 1.19\n",
      "epoch: 1, train loss: 1.944, val loss: 1.843, time: 1.29\n",
      "epoch: 2, train loss: 1.802, val loss: 1.749, time: 1.31\n",
      "epoch: 3, train loss: 1.694, val loss: 1.689, time: 1.25\n",
      "epoch: 4, train loss: 1.634, val loss: 1.664, time: 1.22\n",
      "epoch: 5, train loss: 1.612, val loss: 1.653, time: 1.23\n",
      "epoch: 6, train loss: 1.594, val loss: 1.644, time: 1.22\n",
      "epoch: 7, train loss: 1.585, val loss: 1.638, time: 1.30\n",
      "epoch: 8, train loss: 1.575, val loss: 1.633, time: 1.28\n",
      "epoch: 9, train loss: 1.571, val loss: 1.632, time: 1.26\n",
      "epoch: 10, train loss: 1.566, val loss: 1.630, time: 1.27\n",
      "epoch: 11, train loss: 1.565, val loss: 1.627, time: 1.27\n",
      "epoch: 12, train loss: 1.560, val loss: 1.626, time: 1.39\n",
      "epoch: 13, train loss: 1.559, val loss: 1.623, time: 1.28\n",
      "epoch: 14, train loss: 1.560, val loss: 1.623, time: 1.30\n",
      "epoch: 15, train loss: 1.556, val loss: 1.623, time: 1.28\n",
      "epoch: 16, train loss: 1.556, val loss: 1.622, time: 1.30\n",
      "epoch: 17, train loss: 1.554, val loss: 1.623, time: 1.26\n",
      "epoch: 18, train loss: 1.555, val loss: 1.623, time: 1.28\n",
      "epoch: 19, train loss: 1.552, val loss: 1.622, time: 1.29\n",
      "epoch: 20, train loss: 1.554, val loss: 1.623, time: 1.21\n",
      "epoch: 21, train loss: 1.553, val loss: 1.621, time: 1.35\n",
      "epoch: 22, train loss: 1.551, val loss: 1.621, time: 1.30\n",
      "epoch: 23, train loss: 1.551, val loss: 1.621, time: 1.24\n",
      "epoch: 24, train loss: 1.552, val loss: 1.620, time: 1.22\n",
      "snRNAseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:13:46,971 24598 genes overlap with model after filtering\n",
      "2022-03-25 10:13:46,972 1535 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.278, val loss: 2.076, time: 1.19\n",
      "epoch: 1, train loss: 1.928, val loss: 1.856, time: 1.28\n",
      "epoch: 2, train loss: 1.757, val loss: 1.741, time: 1.20\n",
      "epoch: 3, train loss: 1.676, val loss: 1.696, time: 1.35\n",
      "epoch: 4, train loss: 1.636, val loss: 1.672, time: 1.32\n",
      "epoch: 5, train loss: 1.612, val loss: 1.659, time: 1.35\n",
      "epoch: 6, train loss: 1.595, val loss: 1.651, time: 1.28\n",
      "epoch: 7, train loss: 1.586, val loss: 1.645, time: 1.28\n",
      "epoch: 8, train loss: 1.578, val loss: 1.639, time: 1.29\n",
      "epoch: 9, train loss: 1.571, val loss: 1.635, time: 1.27\n",
      "epoch: 10, train loss: 1.565, val loss: 1.633, time: 1.30\n",
      "epoch: 11, train loss: 1.566, val loss: 1.631, time: 1.26\n",
      "epoch: 12, train loss: 1.563, val loss: 1.628, time: 1.26\n",
      "epoch: 13, train loss: 1.560, val loss: 1.627, time: 1.27\n",
      "epoch: 14, train loss: 1.560, val loss: 1.627, time: 1.32\n",
      "epoch: 15, train loss: 1.557, val loss: 1.624, time: 1.35\n",
      "epoch: 16, train loss: 1.556, val loss: 1.624, time: 1.27\n",
      "epoch: 17, train loss: 1.555, val loss: 1.621, time: 1.27\n",
      "epoch: 18, train loss: 1.555, val loss: 1.622, time: 1.31\n",
      "epoch: 19, train loss: 1.553, val loss: 1.622, time: 1.29\n",
      "epoch: 20, train loss: 1.553, val loss: 1.621, time: 1.29\n",
      "epoch: 21, train loss: 1.552, val loss: 1.620, time: 1.28\n",
      "epoch: 22, train loss: 1.552, val loss: 1.619, time: 1.28\n",
      "epoch: 23, train loss: 1.553, val loss: 1.617, time: 1.30\n",
      "epoch: 24, train loss: 1.552, val loss: 1.620, time: 1.25\n",
      "snRNAseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:14:46,805 24626 genes overlap with model after filtering\n",
      "2022-03-25 10:14:46,806 1508 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.281, val loss: 2.078, time: 1.22\n",
      "epoch: 1, train loss: 1.947, val loss: 1.834, time: 1.23\n",
      "epoch: 2, train loss: 1.790, val loss: 1.763, time: 1.25\n",
      "epoch: 3, train loss: 1.724, val loss: 1.734, time: 1.26\n",
      "epoch: 4, train loss: 1.695, val loss: 1.723, time: 1.21\n",
      "epoch: 5, train loss: 1.680, val loss: 1.712, time: 1.31\n",
      "epoch: 6, train loss: 1.651, val loss: 1.674, time: 1.22\n",
      "epoch: 7, train loss: 1.619, val loss: 1.649, time: 1.24\n",
      "epoch: 8, train loss: 1.598, val loss: 1.638, time: 1.26\n",
      "epoch: 9, train loss: 1.584, val loss: 1.635, time: 1.38\n",
      "epoch: 10, train loss: 1.577, val loss: 1.634, time: 1.25\n",
      "epoch: 11, train loss: 1.570, val loss: 1.628, time: 1.25\n",
      "epoch: 12, train loss: 1.565, val loss: 1.630, time: 1.26\n",
      "epoch: 13, train loss: 1.563, val loss: 1.628, time: 1.25\n",
      "epoch: 14, train loss: 1.560, val loss: 1.628, time: 1.27\n",
      "epoch: 15, train loss: 1.557, val loss: 1.623, time: 1.26\n",
      "epoch: 16, train loss: 1.555, val loss: 1.623, time: 1.26\n",
      "epoch: 17, train loss: 1.554, val loss: 1.624, time: 1.22\n",
      "epoch: 18, train loss: 1.552, val loss: 1.621, time: 1.28\n",
      "epoch: 19, train loss: 1.554, val loss: 1.622, time: 1.38\n",
      "epoch: 20, train loss: 1.553, val loss: 1.621, time: 1.28\n",
      "epoch: 21, train loss: 1.551, val loss: 1.619, time: 1.27\n",
      "epoch: 22, train loss: 1.549, val loss: 1.620, time: 1.29\n",
      "epoch: 23, train loss: 1.549, val loss: 1.618, time: 1.24\n",
      "epoch: 24, train loss: 1.549, val loss: 1.617, time: 1.28\n",
      "snRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:15:46,180 25314 genes overlap with model after filtering\n",
      "2022-03-25 10:15:46,181 1192 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.458, val loss: 2.251, time: 1.28\n",
      "epoch: 1, train loss: 2.144, val loss: 2.057, time: 1.31\n",
      "epoch: 2, train loss: 1.982, val loss: 1.966, time: 1.47\n",
      "epoch: 3, train loss: 1.894, val loss: 1.908, time: 1.32\n",
      "epoch: 4, train loss: 1.838, val loss: 1.874, time: 1.32\n",
      "epoch: 5, train loss: 1.804, val loss: 1.857, time: 1.32\n",
      "epoch: 6, train loss: 1.781, val loss: 1.847, time: 1.31\n",
      "epoch: 7, train loss: 1.763, val loss: 1.838, time: 1.30\n",
      "epoch: 8, train loss: 1.754, val loss: 1.830, time: 1.31\n",
      "epoch: 9, train loss: 1.740, val loss: 1.827, time: 1.37\n",
      "epoch: 10, train loss: 1.734, val loss: 1.822, time: 1.29\n",
      "epoch: 11, train loss: 1.725, val loss: 1.820, time: 1.35\n",
      "epoch: 12, train loss: 1.722, val loss: 1.819, time: 1.37\n",
      "epoch: 13, train loss: 1.720, val loss: 1.817, time: 1.46\n",
      "epoch: 14, train loss: 1.716, val loss: 1.815, time: 1.32\n",
      "epoch: 15, train loss: 1.715, val loss: 1.816, time: 1.33\n",
      "epoch: 16, train loss: 1.713, val loss: 1.814, time: 1.34\n",
      "epoch: 17, train loss: 1.712, val loss: 1.811, time: 1.30\n",
      "epoch: 18, train loss: 1.710, val loss: 1.809, time: 1.34\n",
      "epoch: 19, train loss: 1.709, val loss: 1.810, time: 1.29\n",
      "epoch: 20, train loss: 1.707, val loss: 1.808, time: 1.31\n",
      "epoch: 21, train loss: 1.706, val loss: 1.808, time: 1.30\n",
      "epoch: 22, train loss: 1.706, val loss: 1.808, time: 1.52\n",
      "epoch: 23, train loss: 1.706, val loss: 1.809, time: 1.29\n",
      "epoch: 24, train loss: 1.704, val loss: 1.807, time: 1.32\n",
      "snRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:16:49,087 25304 genes overlap with model after filtering\n",
      "2022-03-25 10:16:49,088 1237 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.463, val loss: 2.308, time: 1.29\n",
      "epoch: 1, train loss: 2.205, val loss: 2.104, time: 1.29\n",
      "epoch: 2, train loss: 2.025, val loss: 2.002, time: 1.35\n",
      "epoch: 3, train loss: 1.928, val loss: 1.945, time: 1.30\n",
      "epoch: 4, train loss: 1.864, val loss: 1.891, time: 1.35\n",
      "epoch: 5, train loss: 1.805, val loss: 1.855, time: 1.50\n",
      "epoch: 6, train loss: 1.775, val loss: 1.838, time: 1.33\n",
      "epoch: 7, train loss: 1.757, val loss: 1.830, time: 1.32\n",
      "epoch: 8, train loss: 1.747, val loss: 1.826, time: 1.33\n",
      "epoch: 9, train loss: 1.738, val loss: 1.821, time: 1.32\n",
      "epoch: 10, train loss: 1.731, val loss: 1.816, time: 1.32\n",
      "epoch: 11, train loss: 1.728, val loss: 1.815, time: 1.31\n",
      "epoch: 12, train loss: 1.723, val loss: 1.812, time: 1.33\n",
      "epoch: 13, train loss: 1.719, val loss: 1.811, time: 1.32\n",
      "epoch: 14, train loss: 1.718, val loss: 1.811, time: 1.31\n",
      "epoch: 15, train loss: 1.716, val loss: 1.809, time: 1.51\n",
      "epoch: 16, train loss: 1.712, val loss: 1.806, time: 1.31\n",
      "epoch: 17, train loss: 1.712, val loss: 1.806, time: 1.32\n",
      "epoch: 18, train loss: 1.710, val loss: 1.805, time: 1.32\n",
      "epoch: 19, train loss: 1.710, val loss: 1.804, time: 1.33\n",
      "epoch: 20, train loss: 1.708, val loss: 1.805, time: 1.31\n",
      "epoch: 21, train loss: 1.707, val loss: 1.803, time: 1.31\n",
      "epoch: 22, train loss: 1.705, val loss: 1.802, time: 1.29\n",
      "epoch: 23, train loss: 1.704, val loss: 1.802, time: 1.32\n",
      "epoch: 24, train loss: 1.704, val loss: 1.801, time: 1.30\n",
      "snRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:17:51,692 25340 genes overlap with model after filtering\n",
      "2022-03-25 10:17:51,693 1209 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.466, val loss: 2.321, time: 1.29\n",
      "epoch: 1, train loss: 2.239, val loss: 2.114, time: 1.31\n",
      "epoch: 2, train loss: 2.031, val loss: 1.970, time: 1.30\n",
      "epoch: 3, train loss: 1.927, val loss: 1.923, time: 1.28\n",
      "epoch: 4, train loss: 1.882, val loss: 1.903, time: 1.30\n",
      "epoch: 5, train loss: 1.857, val loss: 1.894, time: 1.29\n",
      "epoch: 6, train loss: 1.822, val loss: 1.857, time: 1.31\n",
      "epoch: 7, train loss: 1.783, val loss: 1.842, time: 1.31\n",
      "epoch: 8, train loss: 1.759, val loss: 1.831, time: 1.46\n",
      "epoch: 9, train loss: 1.748, val loss: 1.825, time: 1.31\n",
      "epoch: 10, train loss: 1.744, val loss: 1.822, time: 1.33\n",
      "epoch: 11, train loss: 1.735, val loss: 1.819, time: 1.32\n",
      "epoch: 12, train loss: 1.732, val loss: 1.816, time: 1.34\n",
      "epoch: 13, train loss: 1.725, val loss: 1.814, time: 1.33\n",
      "epoch: 14, train loss: 1.721, val loss: 1.812, time: 1.31\n",
      "epoch: 15, train loss: 1.720, val loss: 1.809, time: 1.30\n",
      "epoch: 16, train loss: 1.715, val loss: 1.807, time: 1.32\n",
      "epoch: 17, train loss: 1.715, val loss: 1.806, time: 1.30\n",
      "epoch: 18, train loss: 1.710, val loss: 1.805, time: 1.45\n",
      "epoch: 19, train loss: 1.710, val loss: 1.804, time: 1.33\n",
      "epoch: 20, train loss: 1.710, val loss: 1.804, time: 1.32\n",
      "epoch: 21, train loss: 1.708, val loss: 1.803, time: 1.30\n",
      "epoch: 22, train loss: 1.707, val loss: 1.802, time: 1.33\n",
      "epoch: 23, train loss: 1.706, val loss: 1.801, time: 1.29\n",
      "epoch: 24, train loss: 1.705, val loss: 1.801, time: 1.32\n",
      "snRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:18:53,888 25343 genes overlap with model after filtering\n",
      "2022-03-25 10:18:53,890 1195 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.462, val loss: 2.314, time: 1.29\n",
      "epoch: 1, train loss: 2.233, val loss: 2.149, time: 1.46\n",
      "epoch: 2, train loss: 2.081, val loss: 2.020, time: 1.35\n",
      "epoch: 3, train loss: 1.970, val loss: 1.926, time: 1.38\n",
      "epoch: 4, train loss: 1.885, val loss: 1.874, time: 1.37\n",
      "epoch: 5, train loss: 1.837, val loss: 1.852, time: 1.37\n",
      "epoch: 6, train loss: 1.796, val loss: 1.835, time: 1.39\n",
      "epoch: 7, train loss: 1.770, val loss: 1.827, time: 1.38\n",
      "epoch: 8, train loss: 1.757, val loss: 1.819, time: 1.35\n",
      "epoch: 9, train loss: 1.742, val loss: 1.818, time: 1.42\n",
      "epoch: 10, train loss: 1.736, val loss: 1.814, time: 1.36\n",
      "epoch: 11, train loss: 1.731, val loss: 1.812, time: 1.50\n",
      "epoch: 12, train loss: 1.725, val loss: 1.812, time: 1.39\n",
      "epoch: 13, train loss: 1.723, val loss: 1.810, time: 1.38\n",
      "epoch: 14, train loss: 1.721, val loss: 1.808, time: 1.41\n",
      "epoch: 15, train loss: 1.718, val loss: 1.805, time: 1.40\n",
      "epoch: 16, train loss: 1.715, val loss: 1.805, time: 1.39\n",
      "epoch: 17, train loss: 1.713, val loss: 1.803, time: 1.38\n",
      "epoch: 18, train loss: 1.715, val loss: 1.802, time: 1.38\n",
      "epoch: 19, train loss: 1.711, val loss: 1.807, time: 1.38\n",
      "epoch: 20, train loss: 1.709, val loss: 1.803, time: 1.41\n",
      "epoch: 21, train loss: 1.708, val loss: 1.802, time: 1.47\n",
      "epoch: 22, train loss: 1.707, val loss: 1.801, time: 1.40\n",
      "epoch: 23, train loss: 1.707, val loss: 1.801, time: 1.39\n",
      "epoch: 24, train loss: 1.706, val loss: 1.800, time: 1.41\n",
      "snRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:19:58,323 25346 genes overlap with model after filtering\n",
      "2022-03-25 10:19:58,325 1217 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.470, val loss: 2.293, time: 1.31\n",
      "epoch: 1, train loss: 2.206, val loss: 2.091, time: 1.35\n",
      "epoch: 2, train loss: 2.034, val loss: 1.987, time: 1.37\n",
      "epoch: 3, train loss: 1.936, val loss: 1.938, time: 1.38\n",
      "epoch: 4, train loss: 1.855, val loss: 1.874, time: 1.50\n",
      "epoch: 5, train loss: 1.801, val loss: 1.851, time: 1.42\n",
      "epoch: 6, train loss: 1.778, val loss: 1.837, time: 1.39\n",
      "epoch: 7, train loss: 1.762, val loss: 1.833, time: 1.39\n",
      "epoch: 8, train loss: 1.754, val loss: 1.827, time: 1.38\n",
      "epoch: 9, train loss: 1.742, val loss: 1.824, time: 1.37\n",
      "epoch: 10, train loss: 1.739, val loss: 1.820, time: 1.38\n",
      "epoch: 11, train loss: 1.733, val loss: 1.817, time: 1.36\n",
      "epoch: 12, train loss: 1.730, val loss: 1.816, time: 1.38\n",
      "epoch: 13, train loss: 1.728, val loss: 1.814, time: 1.39\n",
      "epoch: 14, train loss: 1.722, val loss: 1.812, time: 1.50\n",
      "epoch: 15, train loss: 1.720, val loss: 1.814, time: 1.36\n",
      "epoch: 16, train loss: 1.720, val loss: 1.810, time: 1.29\n",
      "epoch: 17, train loss: 1.717, val loss: 1.807, time: 1.29\n",
      "epoch: 18, train loss: 1.713, val loss: 1.806, time: 1.29\n",
      "epoch: 19, train loss: 1.710, val loss: 1.806, time: 1.28\n",
      "epoch: 20, train loss: 1.709, val loss: 1.806, time: 1.15\n",
      "epoch: 21, train loss: 1.707, val loss: 1.804, time: 1.21\n",
      "epoch: 22, train loss: 1.706, val loss: 1.802, time: 1.31\n",
      "epoch: 23, train loss: 1.707, val loss: 1.801, time: 1.28\n",
      "epoch: 24, train loss: 1.706, val loss: 1.802, time: 1.39\n",
      "snRNAseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:21:00,417 24927 genes overlap with model after filtering\n",
      "2022-03-25 10:21:00,418 1326 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.170, val loss: 2.092, time: 0.99\n",
      "epoch: 1, train loss: 2.033, val loss: 1.863, time: 0.99\n",
      "epoch: 2, train loss: 1.873, val loss: 1.697, time: 1.09\n",
      "epoch: 3, train loss: 1.760, val loss: 1.611, time: 1.07\n",
      "epoch: 4, train loss: 1.711, val loss: 1.577, time: 1.06\n",
      "epoch: 5, train loss: 1.681, val loss: 1.555, time: 1.06\n",
      "epoch: 6, train loss: 1.651, val loss: 1.523, time: 1.06\n",
      "epoch: 7, train loss: 1.610, val loss: 1.469, time: 1.05\n",
      "epoch: 8, train loss: 1.559, val loss: 1.433, time: 1.06\n",
      "epoch: 9, train loss: 1.524, val loss: 1.420, time: 1.06\n",
      "epoch: 10, train loss: 1.524, val loss: 1.415, time: 1.07\n",
      "epoch: 11, train loss: 1.495, val loss: 1.411, time: 1.17\n",
      "epoch: 12, train loss: 1.499, val loss: 1.409, time: 1.08\n",
      "epoch: 13, train loss: 1.493, val loss: 1.407, time: 1.08\n",
      "epoch: 14, train loss: 1.480, val loss: 1.406, time: 1.08\n",
      "epoch: 15, train loss: 1.476, val loss: 1.406, time: 1.07\n",
      "epoch: 16, train loss: 1.469, val loss: 1.405, time: 1.09\n",
      "epoch: 17, train loss: 1.473, val loss: 1.405, time: 1.06\n",
      "epoch: 18, train loss: 1.470, val loss: 1.403, time: 1.06\n",
      "epoch: 19, train loss: 1.462, val loss: 1.402, time: 1.05\n",
      "epoch: 20, train loss: 1.463, val loss: 1.401, time: 1.06\n",
      "epoch: 21, train loss: 1.457, val loss: 1.401, time: 1.06\n",
      "epoch: 22, train loss: 1.455, val loss: 1.401, time: 1.07\n",
      "epoch: 23, train loss: 1.451, val loss: 1.400, time: 1.06\n",
      "epoch: 24, train loss: 1.458, val loss: 1.400, time: 1.16\n",
      "snRNAseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:21:46,881 24983 genes overlap with model after filtering\n",
      "2022-03-25 10:21:46,882 1404 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.180, val loss: 2.134, time: 1.03\n",
      "epoch: 1, train loss: 2.077, val loss: 1.959, time: 1.08\n",
      "epoch: 2, train loss: 1.975, val loss: 1.873, time: 1.08\n",
      "epoch: 3, train loss: 1.921, val loss: 1.816, time: 1.07\n",
      "epoch: 4, train loss: 1.881, val loss: 1.760, time: 1.05\n",
      "epoch: 5, train loss: 1.831, val loss: 1.675, time: 1.06\n",
      "epoch: 6, train loss: 1.758, val loss: 1.562, time: 1.05\n",
      "epoch: 7, train loss: 1.686, val loss: 1.490, time: 1.05\n",
      "epoch: 8, train loss: 1.645, val loss: 1.464, time: 1.06\n",
      "epoch: 9, train loss: 1.604, val loss: 1.454, time: 1.05\n",
      "epoch: 10, train loss: 1.593, val loss: 1.449, time: 1.07\n",
      "epoch: 11, train loss: 1.586, val loss: 1.446, time: 1.17\n",
      "epoch: 12, train loss: 1.576, val loss: 1.441, time: 1.09\n",
      "epoch: 13, train loss: 1.579, val loss: 1.440, time: 1.07\n",
      "epoch: 14, train loss: 1.572, val loss: 1.438, time: 1.07\n",
      "epoch: 15, train loss: 1.575, val loss: 1.436, time: 1.07\n",
      "epoch: 16, train loss: 1.581, val loss: 1.435, time: 1.08\n",
      "epoch: 17, train loss: 1.559, val loss: 1.435, time: 0.98\n",
      "epoch: 18, train loss: 1.561, val loss: 1.434, time: 0.95\n",
      "epoch: 19, train loss: 1.557, val loss: 1.433, time: 0.95\n",
      "epoch: 20, train loss: 1.553, val loss: 1.433, time: 0.95\n",
      "epoch: 21, train loss: 1.548, val loss: 1.431, time: 1.04\n",
      "epoch: 22, train loss: 1.547, val loss: 1.431, time: 1.03\n",
      "epoch: 23, train loss: 1.532, val loss: 1.429, time: 1.06\n",
      "epoch: 24, train loss: 1.544, val loss: 1.428, time: 0.95\n",
      "snRNAseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:22:32,140 24962 genes overlap with model after filtering\n",
      "2022-03-25 10:22:32,141 1361 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.162, val loss: 2.074, time: 0.98\n",
      "epoch: 1, train loss: 2.039, val loss: 1.893, time: 0.99\n",
      "epoch: 2, train loss: 1.944, val loss: 1.833, time: 1.07\n",
      "epoch: 3, train loss: 1.895, val loss: 1.805, time: 1.07\n",
      "epoch: 4, train loss: 1.849, val loss: 1.787, time: 1.07\n",
      "epoch: 5, train loss: 1.825, val loss: 1.753, time: 1.07\n",
      "epoch: 6, train loss: 1.773, val loss: 1.675, time: 1.07\n",
      "epoch: 7, train loss: 1.712, val loss: 1.615, time: 1.06\n",
      "epoch: 8, train loss: 1.674, val loss: 1.579, time: 1.05\n",
      "epoch: 9, train loss: 1.654, val loss: 1.560, time: 0.97\n",
      "epoch: 10, train loss: 1.631, val loss: 1.547, time: 1.04\n",
      "epoch: 11, train loss: 1.603, val loss: 1.540, time: 1.05\n",
      "epoch: 12, train loss: 1.590, val loss: 1.535, time: 1.11\n",
      "epoch: 13, train loss: 1.587, val loss: 1.531, time: 1.13\n",
      "epoch: 14, train loss: 1.575, val loss: 1.528, time: 1.08\n",
      "epoch: 15, train loss: 1.567, val loss: 1.525, time: 1.07\n",
      "epoch: 16, train loss: 1.563, val loss: 1.523, time: 1.07\n",
      "epoch: 17, train loss: 1.558, val loss: 1.521, time: 1.07\n",
      "epoch: 18, train loss: 1.560, val loss: 1.521, time: 1.06\n",
      "epoch: 19, train loss: 1.549, val loss: 1.520, time: 1.06\n",
      "epoch: 20, train loss: 1.544, val loss: 1.520, time: 1.05\n",
      "epoch: 21, train loss: 1.540, val loss: 1.519, time: 1.05\n",
      "epoch: 22, train loss: 1.546, val loss: 1.517, time: 1.05\n",
      "epoch: 23, train loss: 1.546, val loss: 1.516, time: 1.06\n",
      "epoch: 24, train loss: 1.542, val loss: 1.516, time: 1.06\n",
      "snRNAseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:23:17,859 24988 genes overlap with model after filtering\n",
      "2022-03-25 10:23:17,861 1392 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.156, val loss: 2.046, time: 1.03\n",
      "epoch: 1, train loss: 2.024, val loss: 1.912, time: 1.08\n",
      "epoch: 2, train loss: 1.947, val loss: 1.813, time: 1.06\n",
      "epoch: 3, train loss: 1.882, val loss: 1.747, time: 1.07\n",
      "epoch: 4, train loss: 1.848, val loss: 1.723, time: 1.06\n",
      "epoch: 5, train loss: 1.831, val loss: 1.705, time: 1.06\n",
      "epoch: 6, train loss: 1.811, val loss: 1.680, time: 1.05\n",
      "epoch: 7, train loss: 1.782, val loss: 1.639, time: 1.05\n",
      "epoch: 8, train loss: 1.747, val loss: 1.602, time: 1.05\n",
      "epoch: 9, train loss: 1.729, val loss: 1.585, time: 1.05\n",
      "epoch: 10, train loss: 1.707, val loss: 1.575, time: 1.06\n",
      "epoch: 11, train loss: 1.693, val loss: 1.566, time: 1.05\n",
      "epoch: 12, train loss: 1.690, val loss: 1.557, time: 1.07\n",
      "epoch: 13, train loss: 1.655, val loss: 1.547, time: 1.09\n",
      "epoch: 14, train loss: 1.634, val loss: 1.540, time: 1.08\n",
      "epoch: 15, train loss: 1.618, val loss: 1.533, time: 1.07\n",
      "epoch: 16, train loss: 1.609, val loss: 1.528, time: 1.07\n",
      "epoch: 17, train loss: 1.598, val loss: 1.523, time: 1.09\n",
      "epoch: 18, train loss: 1.589, val loss: 1.515, time: 1.07\n",
      "epoch: 19, train loss: 1.576, val loss: 1.503, time: 1.20\n",
      "epoch: 20, train loss: 1.562, val loss: 1.487, time: 0.98\n",
      "epoch: 21, train loss: 1.538, val loss: 1.466, time: 0.97\n",
      "epoch: 22, train loss: 1.524, val loss: 1.447, time: 1.05\n",
      "epoch: 23, train loss: 1.507, val loss: 1.432, time: 1.06\n",
      "epoch: 24, train loss: 1.488, val loss: 1.422, time: 1.06\n",
      "snRNAseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:24:04,147 24951 genes overlap with model after filtering\n",
      "2022-03-25 10:24:04,147 1407 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.179, val loss: 2.124, time: 0.97\n",
      "epoch: 1, train loss: 2.064, val loss: 1.901, time: 1.07\n",
      "epoch: 2, train loss: 1.885, val loss: 1.685, time: 1.07\n",
      "epoch: 3, train loss: 1.774, val loss: 1.574, time: 1.06\n",
      "epoch: 4, train loss: 1.711, val loss: 1.513, time: 1.07\n",
      "epoch: 5, train loss: 1.663, val loss: 1.480, time: 1.06\n",
      "epoch: 6, train loss: 1.634, val loss: 1.459, time: 1.07\n",
      "epoch: 7, train loss: 1.599, val loss: 1.447, time: 1.06\n",
      "epoch: 8, train loss: 1.587, val loss: 1.440, time: 1.06\n",
      "epoch: 9, train loss: 1.571, val loss: 1.435, time: 1.05\n",
      "epoch: 10, train loss: 1.562, val loss: 1.431, time: 1.06\n",
      "epoch: 11, train loss: 1.549, val loss: 1.426, time: 1.06\n",
      "epoch: 12, train loss: 1.533, val loss: 1.418, time: 1.08\n",
      "epoch: 13, train loss: 1.513, val loss: 1.408, time: 1.17\n",
      "epoch: 14, train loss: 1.486, val loss: 1.406, time: 1.08\n",
      "epoch: 15, train loss: 1.486, val loss: 1.403, time: 1.10\n",
      "epoch: 16, train loss: 1.479, val loss: 1.403, time: 1.07\n",
      "epoch: 17, train loss: 1.474, val loss: 1.402, time: 1.08\n",
      "epoch: 18, train loss: 1.467, val loss: 1.402, time: 1.07\n",
      "epoch: 19, train loss: 1.459, val loss: 1.401, time: 1.06\n",
      "epoch: 20, train loss: 1.457, val loss: 1.401, time: 1.06\n",
      "epoch: 21, train loss: 1.453, val loss: 1.401, time: 1.06\n",
      "epoch: 22, train loss: 1.462, val loss: 1.400, time: 1.06\n",
      "epoch: 23, train loss: 1.452, val loss: 1.400, time: 1.06\n",
      "epoch: 24, train loss: 1.449, val loss: 1.399, time: 1.06\n",
      "snATACseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:24:50,846 19692 genes overlap with model after filtering\n",
      "2022-03-25 10:24:50,848 35 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.778, val loss: 1.759, time: 0.78\n",
      "epoch: 1, train loss: 1.717, val loss: 1.672, time: 0.78\n",
      "epoch: 2, train loss: 1.612, val loss: 1.557, time: 0.78\n",
      "epoch: 3, train loss: 1.516, val loss: 1.480, time: 0.78\n",
      "epoch: 4, train loss: 1.451, val loss: 1.416, time: 0.78\n",
      "epoch: 5, train loss: 1.400, val loss: 1.369, time: 0.79\n",
      "epoch: 6, train loss: 1.359, val loss: 1.339, time: 0.77\n",
      "epoch: 7, train loss: 1.329, val loss: 1.309, time: 0.76\n",
      "epoch: 8, train loss: 1.295, val loss: 1.280, time: 0.76\n",
      "epoch: 9, train loss: 1.272, val loss: 1.255, time: 0.77\n",
      "epoch: 10, train loss: 1.240, val loss: 1.228, time: 0.76\n",
      "epoch: 11, train loss: 1.202, val loss: 1.194, time: 0.76\n",
      "epoch: 12, train loss: 1.177, val loss: 1.166, time: 0.77\n",
      "epoch: 13, train loss: 1.150, val loss: 1.152, time: 0.77\n",
      "epoch: 14, train loss: 1.133, val loss: 1.144, time: 0.77\n",
      "epoch: 15, train loss: 1.132, val loss: 1.137, time: 0.76\n",
      "epoch: 16, train loss: 1.122, val loss: 1.133, time: 0.78\n",
      "epoch: 17, train loss: 1.112, val loss: 1.130, time: 0.80\n",
      "epoch: 18, train loss: 1.116, val loss: 1.127, time: 0.81\n",
      "epoch: 19, train loss: 1.104, val loss: 1.126, time: 0.78\n",
      "epoch: 20, train loss: 1.111, val loss: 1.125, time: 0.78\n",
      "epoch: 21, train loss: 1.100, val loss: 1.123, time: 0.77\n",
      "epoch: 22, train loss: 1.101, val loss: 1.122, time: 0.79\n",
      "epoch: 23, train loss: 1.104, val loss: 1.121, time: 0.77\n",
      "epoch: 24, train loss: 1.094, val loss: 1.119, time: 0.79\n",
      "snATACseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:25:28,808 19680 genes overlap with model after filtering\n",
      "2022-03-25 10:25:28,809 38 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.787, val loss: 1.776, time: 0.72\n",
      "epoch: 1, train loss: 1.738, val loss: 1.697, time: 0.76\n",
      "epoch: 2, train loss: 1.663, val loss: 1.645, time: 0.76\n",
      "epoch: 3, train loss: 1.607, val loss: 1.584, time: 0.76\n",
      "epoch: 4, train loss: 1.554, val loss: 1.539, time: 0.78\n",
      "epoch: 5, train loss: 1.510, val loss: 1.489, time: 0.72\n",
      "epoch: 6, train loss: 1.446, val loss: 1.412, time: 0.67\n",
      "epoch: 7, train loss: 1.377, val loss: 1.337, time: 0.69\n",
      "epoch: 8, train loss: 1.310, val loss: 1.275, time: 0.76\n",
      "epoch: 9, train loss: 1.257, val loss: 1.223, time: 0.78\n",
      "epoch: 10, train loss: 1.202, val loss: 1.186, time: 0.76\n",
      "epoch: 11, train loss: 1.179, val loss: 1.165, time: 0.77\n",
      "epoch: 12, train loss: 1.164, val loss: 1.152, time: 0.76\n",
      "epoch: 13, train loss: 1.137, val loss: 1.144, time: 0.77\n",
      "epoch: 14, train loss: 1.136, val loss: 1.139, time: 0.76\n",
      "epoch: 15, train loss: 1.129, val loss: 1.135, time: 0.76\n",
      "epoch: 16, train loss: 1.119, val loss: 1.133, time: 0.76\n",
      "epoch: 17, train loss: 1.111, val loss: 1.131, time: 0.76\n",
      "epoch: 18, train loss: 1.113, val loss: 1.128, time: 0.76\n",
      "epoch: 19, train loss: 1.108, val loss: 1.126, time: 0.76\n",
      "epoch: 20, train loss: 1.107, val loss: 1.125, time: 0.76\n",
      "epoch: 21, train loss: 1.101, val loss: 1.122, time: 0.76\n",
      "epoch: 22, train loss: 1.100, val loss: 1.121, time: 0.76\n",
      "epoch: 23, train loss: 1.096, val loss: 1.119, time: 0.84\n",
      "epoch: 24, train loss: 1.093, val loss: 1.119, time: 0.78\n",
      "snATACseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:26:05,778 19683 genes overlap with model after filtering\n",
      "2022-03-25 10:26:05,779 38 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.788, val loss: 1.774, time: 0.74\n",
      "epoch: 1, train loss: 1.740, val loss: 1.711, time: 0.78\n",
      "epoch: 2, train loss: 1.695, val loss: 1.691, time: 0.77\n",
      "epoch: 3, train loss: 1.680, val loss: 1.681, time: 0.77\n",
      "epoch: 4, train loss: 1.658, val loss: 1.638, time: 0.77\n",
      "epoch: 5, train loss: 1.604, val loss: 1.575, time: 0.77\n",
      "epoch: 6, train loss: 1.561, val loss: 1.528, time: 0.77\n",
      "epoch: 7, train loss: 1.503, val loss: 1.462, time: 0.76\n",
      "epoch: 8, train loss: 1.444, val loss: 1.404, time: 0.77\n",
      "epoch: 9, train loss: 1.398, val loss: 1.370, time: 0.77\n",
      "epoch: 10, train loss: 1.381, val loss: 1.356, time: 0.77\n",
      "epoch: 11, train loss: 1.361, val loss: 1.347, time: 0.77\n",
      "epoch: 12, train loss: 1.349, val loss: 1.338, time: 0.84\n",
      "epoch: 13, train loss: 1.339, val loss: 1.329, time: 0.77\n",
      "epoch: 14, train loss: 1.328, val loss: 1.318, time: 0.78\n",
      "epoch: 15, train loss: 1.303, val loss: 1.289, time: 0.79\n",
      "epoch: 16, train loss: 1.266, val loss: 1.235, time: 0.78\n",
      "epoch: 17, train loss: 1.228, val loss: 1.212, time: 0.77\n",
      "epoch: 18, train loss: 1.211, val loss: 1.202, time: 0.77\n",
      "epoch: 19, train loss: 1.201, val loss: 1.194, time: 0.77\n",
      "epoch: 20, train loss: 1.197, val loss: 1.185, time: 0.77\n",
      "epoch: 21, train loss: 1.181, val loss: 1.173, time: 0.76\n",
      "epoch: 22, train loss: 1.169, val loss: 1.159, time: 0.76\n",
      "epoch: 23, train loss: 1.155, val loss: 1.143, time: 0.77\n",
      "epoch: 24, train loss: 1.145, val loss: 1.130, time: 0.79\n",
      "snATACseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:26:44,042 19682 genes overlap with model after filtering\n",
      "2022-03-25 10:26:44,043 37 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.788, val loss: 1.781, time: 0.74\n",
      "epoch: 1, train loss: 1.770, val loss: 1.757, time: 0.76\n",
      "epoch: 2, train loss: 1.726, val loss: 1.688, time: 0.79\n",
      "epoch: 3, train loss: 1.639, val loss: 1.595, time: 0.79\n",
      "epoch: 4, train loss: 1.544, val loss: 1.491, time: 0.79\n",
      "epoch: 5, train loss: 1.435, val loss: 1.367, time: 0.77\n",
      "epoch: 6, train loss: 1.322, val loss: 1.259, time: 0.78\n",
      "epoch: 7, train loss: 1.256, val loss: 1.196, time: 0.69\n",
      "epoch: 8, train loss: 1.202, val loss: 1.165, time: 0.71\n",
      "epoch: 9, train loss: 1.175, val loss: 1.149, time: 0.70\n",
      "epoch: 10, train loss: 1.153, val loss: 1.140, time: 0.72\n",
      "epoch: 11, train loss: 1.142, val loss: 1.133, time: 0.77\n",
      "epoch: 12, train loss: 1.129, val loss: 1.130, time: 0.76\n",
      "epoch: 13, train loss: 1.123, val loss: 1.126, time: 0.77\n",
      "epoch: 14, train loss: 1.115, val loss: 1.125, time: 0.77\n",
      "epoch: 15, train loss: 1.112, val loss: 1.122, time: 0.77\n",
      "epoch: 16, train loss: 1.105, val loss: 1.121, time: 0.75\n",
      "epoch: 17, train loss: 1.105, val loss: 1.120, time: 0.77\n",
      "epoch: 18, train loss: 1.108, val loss: 1.120, time: 0.79\n",
      "epoch: 19, train loss: 1.100, val loss: 1.119, time: 0.78\n",
      "epoch: 20, train loss: 1.097, val loss: 1.118, time: 0.78\n",
      "epoch: 21, train loss: 1.093, val loss: 1.117, time: 0.78\n",
      "epoch: 22, train loss: 1.093, val loss: 1.117, time: 0.79\n",
      "epoch: 23, train loss: 1.094, val loss: 1.116, time: 0.79\n",
      "epoch: 24, train loss: 1.095, val loss: 1.116, time: 0.78\n",
      "snATACseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:27:21,681 19685 genes overlap with model after filtering\n",
      "2022-03-25 10:27:21,683 47 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 1.787, val loss: 1.779, time: 0.73\n",
      "epoch: 1, train loss: 1.757, val loss: 1.729, time: 0.79\n",
      "epoch: 2, train loss: 1.658, val loss: 1.596, time: 0.76\n",
      "epoch: 3, train loss: 1.538, val loss: 1.489, time: 0.77\n",
      "epoch: 4, train loss: 1.462, val loss: 1.430, time: 0.77\n",
      "epoch: 5, train loss: 1.418, val loss: 1.395, time: 0.76\n",
      "epoch: 6, train loss: 1.393, val loss: 1.378, time: 0.77\n",
      "epoch: 7, train loss: 1.383, val loss: 1.369, time: 0.83\n",
      "epoch: 8, train loss: 1.370, val loss: 1.362, time: 0.78\n",
      "epoch: 9, train loss: 1.355, val loss: 1.357, time: 0.78\n",
      "epoch: 10, train loss: 1.356, val loss: 1.353, time: 0.77\n",
      "epoch: 11, train loss: 1.347, val loss: 1.349, time: 0.78\n",
      "epoch: 12, train loss: 1.346, val loss: 1.345, time: 0.78\n",
      "epoch: 13, train loss: 1.347, val loss: 1.342, time: 0.80\n",
      "epoch: 14, train loss: 1.339, val loss: 1.339, time: 0.78\n",
      "epoch: 15, train loss: 1.331, val loss: 1.336, time: 0.77\n",
      "epoch: 16, train loss: 1.335, val loss: 1.331, time: 0.79\n",
      "epoch: 17, train loss: 1.329, val loss: 1.325, time: 0.72\n",
      "epoch: 18, train loss: 1.310, val loss: 1.313, time: 0.76\n",
      "epoch: 19, train loss: 1.301, val loss: 1.288, time: 0.76\n",
      "epoch: 20, train loss: 1.272, val loss: 1.244, time: 0.77\n",
      "epoch: 21, train loss: 1.232, val loss: 1.209, time: 0.77\n",
      "epoch: 22, train loss: 1.210, val loss: 1.181, time: 0.77\n",
      "epoch: 23, train loss: 1.175, val loss: 1.159, time: 0.81\n",
      "epoch: 24, train loss: 1.167, val loss: 1.149, time: 0.73\n",
      "snATACseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:28:00,427 19694 genes overlap with model after filtering\n",
      "2022-03-25 10:28:00,428 21 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.061, val loss: 2.025, time: 0.90\n",
      "epoch: 1, train loss: 1.987, val loss: 1.933, time: 0.95\n",
      "epoch: 2, train loss: 1.893, val loss: 1.822, time: 0.96\n",
      "epoch: 3, train loss: 1.800, val loss: 1.732, time: 0.95\n",
      "epoch: 4, train loss: 1.735, val loss: 1.683, time: 0.95\n",
      "epoch: 5, train loss: 1.697, val loss: 1.640, time: 0.95\n",
      "epoch: 6, train loss: 1.651, val loss: 1.601, time: 0.95\n",
      "epoch: 7, train loss: 1.621, val loss: 1.583, time: 0.90\n",
      "epoch: 8, train loss: 1.613, val loss: 1.574, time: 0.94\n",
      "epoch: 9, train loss: 1.600, val loss: 1.568, time: 0.93\n",
      "epoch: 10, train loss: 1.592, val loss: 1.564, time: 1.02\n",
      "epoch: 11, train loss: 1.585, val loss: 1.562, time: 0.94\n",
      "epoch: 12, train loss: 1.589, val loss: 1.561, time: 0.96\n",
      "epoch: 13, train loss: 1.578, val loss: 1.558, time: 1.07\n",
      "epoch: 14, train loss: 1.567, val loss: 1.556, time: 0.87\n",
      "epoch: 15, train loss: 1.564, val loss: 1.555, time: 0.96\n",
      "epoch: 16, train loss: 1.562, val loss: 1.553, time: 0.91\n",
      "epoch: 17, train loss: 1.561, val loss: 1.552, time: 0.94\n",
      "epoch: 18, train loss: 1.556, val loss: 1.550, time: 0.95\n",
      "epoch: 19, train loss: 1.548, val loss: 1.549, time: 0.80\n",
      "epoch: 20, train loss: 1.547, val loss: 1.546, time: 0.88\n",
      "epoch: 21, train loss: 1.541, val loss: 1.541, time: 0.89\n",
      "epoch: 22, train loss: 1.539, val loss: 1.529, time: 0.97\n",
      "epoch: 23, train loss: 1.517, val loss: 1.491, time: 0.97\n",
      "epoch: 24, train loss: 1.486, val loss: 1.460, time: 0.93\n",
      "snATACseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:28:47,555 19694 genes overlap with model after filtering\n",
      "2022-03-25 10:28:47,555 24 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.081, val loss: 2.074, time: 0.87\n",
      "epoch: 1, train loss: 2.035, val loss: 1.979, time: 0.96\n",
      "epoch: 2, train loss: 1.932, val loss: 1.872, time: 0.95\n",
      "epoch: 3, train loss: 1.797, val loss: 1.683, time: 0.94\n",
      "epoch: 4, train loss: 1.663, val loss: 1.563, time: 0.93\n",
      "epoch: 5, train loss: 1.579, val loss: 1.502, time: 0.93\n",
      "epoch: 6, train loss: 1.537, val loss: 1.474, time: 0.93\n",
      "epoch: 7, train loss: 1.504, val loss: 1.455, time: 0.93\n",
      "epoch: 8, train loss: 1.492, val loss: 1.439, time: 0.94\n",
      "epoch: 9, train loss: 1.467, val loss: 1.420, time: 1.00\n",
      "epoch: 10, train loss: 1.442, val loss: 1.398, time: 0.88\n",
      "epoch: 11, train loss: 1.418, val loss: 1.381, time: 0.99\n",
      "epoch: 12, train loss: 1.408, val loss: 1.372, time: 0.94\n",
      "epoch: 13, train loss: 1.400, val loss: 1.368, time: 0.94\n",
      "epoch: 14, train loss: 1.382, val loss: 1.364, time: 0.94\n",
      "epoch: 15, train loss: 1.383, val loss: 1.364, time: 0.95\n",
      "epoch: 16, train loss: 1.377, val loss: 1.362, time: 0.94\n",
      "epoch: 17, train loss: 1.374, val loss: 1.362, time: 0.94\n",
      "epoch: 18, train loss: 1.367, val loss: 1.360, time: 0.93\n",
      "epoch: 19, train loss: 1.368, val loss: 1.361, time: 0.97\n",
      "epoch: 20, train loss: 1.362, val loss: 1.363, time: 0.93\n",
      "epoch: 21, train loss: 1.360, val loss: 1.362, time: 0.92\n",
      "epoch: 22, train loss: 1.357, val loss: 1.358, time: 0.93\n",
      "epoch: 23, train loss: 1.352, val loss: 1.358, time: 0.93\n",
      "epoch: 24, train loss: 1.360, val loss: 1.359, time: 0.96\n",
      "snATACseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:29:35,784 19698 genes overlap with model after filtering\n",
      "2022-03-25 10:29:35,785 25 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.073, val loss: 2.062, time: 0.91\n",
      "epoch: 1, train loss: 2.034, val loss: 1.982, time: 0.94\n",
      "epoch: 2, train loss: 1.927, val loss: 1.845, time: 0.96\n",
      "epoch: 3, train loss: 1.831, val loss: 1.770, time: 0.96\n",
      "epoch: 4, train loss: 1.774, val loss: 1.708, time: 0.94\n",
      "epoch: 5, train loss: 1.717, val loss: 1.634, time: 0.99\n",
      "epoch: 6, train loss: 1.658, val loss: 1.591, time: 0.94\n",
      "epoch: 7, train loss: 1.621, val loss: 1.569, time: 0.92\n",
      "epoch: 8, train loss: 1.605, val loss: 1.554, time: 0.95\n",
      "epoch: 9, train loss: 1.596, val loss: 1.541, time: 0.96\n",
      "epoch: 10, train loss: 1.572, val loss: 1.525, time: 1.05\n",
      "epoch: 11, train loss: 1.557, val loss: 1.504, time: 0.98\n",
      "epoch: 12, train loss: 1.527, val loss: 1.483, time: 1.10\n",
      "epoch: 13, train loss: 1.520, val loss: 1.468, time: 0.89\n",
      "epoch: 14, train loss: 1.495, val loss: 1.462, time: 0.96\n",
      "epoch: 15, train loss: 1.488, val loss: 1.456, time: 0.95\n",
      "epoch: 16, train loss: 1.478, val loss: 1.452, time: 0.97\n",
      "epoch: 17, train loss: 1.470, val loss: 1.447, time: 0.97\n",
      "epoch: 18, train loss: 1.464, val loss: 1.444, time: 1.00\n",
      "epoch: 19, train loss: 1.459, val loss: 1.439, time: 1.02\n",
      "epoch: 20, train loss: 1.454, val loss: 1.433, time: 0.88\n",
      "epoch: 21, train loss: 1.445, val loss: 1.425, time: 0.92\n",
      "epoch: 22, train loss: 1.435, val loss: 1.416, time: 0.95\n",
      "epoch: 23, train loss: 1.418, val loss: 1.403, time: 0.97\n",
      "epoch: 24, train loss: 1.408, val loss: 1.388, time: 1.04\n",
      "snATACseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:30:24,317 19696 genes overlap with model after filtering\n",
      "2022-03-25 10:30:24,318 22 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.077, val loss: 2.063, time: 0.99\n",
      "epoch: 1, train loss: 2.022, val loss: 1.949, time: 1.03\n",
      "epoch: 2, train loss: 1.893, val loss: 1.808, time: 1.02\n",
      "epoch: 3, train loss: 1.790, val loss: 1.714, time: 1.02\n",
      "epoch: 4, train loss: 1.712, val loss: 1.641, time: 1.02\n",
      "epoch: 5, train loss: 1.666, val loss: 1.598, time: 1.02\n",
      "epoch: 6, train loss: 1.629, val loss: 1.576, time: 1.03\n",
      "epoch: 7, train loss: 1.608, val loss: 1.565, time: 1.03\n",
      "epoch: 8, train loss: 1.594, val loss: 1.555, time: 1.03\n",
      "epoch: 9, train loss: 1.581, val loss: 1.547, time: 1.05\n",
      "epoch: 10, train loss: 1.571, val loss: 1.541, time: 1.04\n",
      "epoch: 11, train loss: 1.558, val loss: 1.532, time: 1.03\n",
      "epoch: 12, train loss: 1.539, val loss: 1.515, time: 1.05\n",
      "epoch: 13, train loss: 1.525, val loss: 1.483, time: 1.05\n",
      "epoch: 14, train loss: 1.488, val loss: 1.441, time: 1.00\n",
      "epoch: 15, train loss: 1.471, val loss: 1.428, time: 1.02\n",
      "epoch: 16, train loss: 1.445, val loss: 1.418, time: 1.02\n",
      "epoch: 17, train loss: 1.434, val loss: 1.405, time: 1.02\n",
      "epoch: 18, train loss: 1.416, val loss: 1.393, time: 1.03\n",
      "epoch: 19, train loss: 1.401, val loss: 1.380, time: 1.04\n",
      "epoch: 20, train loss: 1.392, val loss: 1.374, time: 1.03\n",
      "epoch: 21, train loss: 1.377, val loss: 1.365, time: 1.01\n",
      "epoch: 22, train loss: 1.370, val loss: 1.362, time: 1.04\n",
      "epoch: 23, train loss: 1.370, val loss: 1.359, time: 1.00\n",
      "epoch: 24, train loss: 1.364, val loss: 1.356, time: 1.05\n",
      "snATACseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:31:15,139 19689 genes overlap with model after filtering\n",
      "2022-03-25 10:31:15,141 24 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.078, val loss: 2.075, time: 0.98\n",
      "epoch: 1, train loss: 2.066, val loss: 2.047, time: 1.04\n",
      "epoch: 2, train loss: 2.005, val loss: 1.948, time: 0.95\n",
      "epoch: 3, train loss: 1.916, val loss: 1.841, time: 0.95\n",
      "epoch: 4, train loss: 1.819, val loss: 1.746, time: 1.04\n",
      "epoch: 5, train loss: 1.752, val loss: 1.690, time: 0.95\n",
      "epoch: 6, train loss: 1.697, val loss: 1.645, time: 0.95\n",
      "epoch: 7, train loss: 1.639, val loss: 1.569, time: 0.88\n",
      "epoch: 8, train loss: 1.570, val loss: 1.513, time: 0.99\n",
      "epoch: 9, train loss: 1.531, val loss: 1.490, time: 1.01\n",
      "epoch: 10, train loss: 1.512, val loss: 1.478, time: 1.01\n",
      "epoch: 11, train loss: 1.503, val loss: 1.469, time: 0.94\n",
      "epoch: 12, train loss: 1.490, val loss: 1.464, time: 1.24\n",
      "epoch: 13, train loss: 1.478, val loss: 1.459, time: 0.86\n",
      "epoch: 14, train loss: 1.478, val loss: 1.456, time: 0.87\n",
      "epoch: 15, train loss: 1.465, val loss: 1.453, time: 0.89\n",
      "epoch: 16, train loss: 1.465, val loss: 1.451, time: 0.94\n",
      "epoch: 17, train loss: 1.459, val loss: 1.447, time: 0.93\n",
      "epoch: 18, train loss: 1.452, val loss: 1.443, time: 0.99\n",
      "epoch: 19, train loss: 1.454, val loss: 1.440, time: 0.98\n",
      "epoch: 20, train loss: 1.448, val loss: 1.435, time: 0.86\n",
      "epoch: 21, train loss: 1.441, val loss: 1.425, time: 0.87\n",
      "epoch: 22, train loss: 1.433, val loss: 1.404, time: 0.98\n",
      "epoch: 23, train loss: 1.394, val loss: 1.371, time: 0.93\n",
      "epoch: 24, train loss: 1.374, val loss: 1.357, time: 0.95\n",
      "snATACseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:32:01,965 19680 genes overlap with model after filtering\n",
      "2022-03-25 10:32:01,967 47 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.078, val loss: 2.074, time: 0.91\n",
      "epoch: 1, train loss: 2.049, val loss: 2.028, time: 0.96\n",
      "epoch: 2, train loss: 1.978, val loss: 1.938, time: 0.97\n",
      "epoch: 3, train loss: 1.889, val loss: 1.841, time: 0.99\n",
      "epoch: 4, train loss: 1.812, val loss: 1.783, time: 0.98\n",
      "epoch: 5, train loss: 1.731, val loss: 1.654, time: 1.02\n",
      "epoch: 6, train loss: 1.657, val loss: 1.621, time: 0.98\n",
      "epoch: 7, train loss: 1.617, val loss: 1.588, time: 0.97\n",
      "epoch: 8, train loss: 1.579, val loss: 1.539, time: 0.97\n",
      "epoch: 9, train loss: 1.523, val loss: 1.463, time: 0.97\n",
      "epoch: 10, train loss: 1.469, val loss: 1.410, time: 0.99\n",
      "epoch: 11, train loss: 1.444, val loss: 1.381, time: 0.96\n",
      "epoch: 12, train loss: 1.423, val loss: 1.366, time: 0.97\n",
      "epoch: 13, train loss: 1.410, val loss: 1.357, time: 0.97\n",
      "epoch: 14, train loss: 1.394, val loss: 1.352, time: 0.97\n",
      "epoch: 15, train loss: 1.386, val loss: 1.348, time: 0.96\n",
      "epoch: 16, train loss: 1.379, val loss: 1.346, time: 0.96\n",
      "epoch: 17, train loss: 1.383, val loss: 1.343, time: 0.97\n",
      "epoch: 18, train loss: 1.377, val loss: 1.341, time: 0.97\n",
      "epoch: 19, train loss: 1.372, val loss: 1.340, time: 0.97\n",
      "epoch: 20, train loss: 1.368, val loss: 1.336, time: 0.97\n",
      "epoch: 21, train loss: 1.361, val loss: 1.336, time: 1.07\n",
      "epoch: 22, train loss: 1.362, val loss: 1.335, time: 0.98\n",
      "epoch: 23, train loss: 1.363, val loss: 1.336, time: 1.00\n",
      "epoch: 24, train loss: 1.358, val loss: 1.334, time: 0.99\n",
      "snATACseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:32:47,566 19686 genes overlap with model after filtering\n",
      "2022-03-25 10:32:47,567 33 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.073, val loss: 2.059, time: 0.92\n",
      "epoch: 1, train loss: 2.022, val loss: 1.951, time: 1.01\n",
      "epoch: 2, train loss: 1.926, val loss: 1.845, time: 0.98\n",
      "epoch: 3, train loss: 1.828, val loss: 1.734, time: 0.98\n",
      "epoch: 4, train loss: 1.734, val loss: 1.643, time: 0.99\n",
      "epoch: 5, train loss: 1.670, val loss: 1.589, time: 0.99\n",
      "epoch: 6, train loss: 1.628, val loss: 1.559, time: 1.00\n",
      "epoch: 7, train loss: 1.612, val loss: 1.539, time: 1.19\n",
      "epoch: 8, train loss: 1.593, val loss: 1.524, time: 0.97\n",
      "epoch: 9, train loss: 1.575, val loss: 1.502, time: 0.99\n",
      "epoch: 10, train loss: 1.552, val loss: 1.467, time: 0.97\n",
      "epoch: 11, train loss: 1.518, val loss: 1.425, time: 0.98\n",
      "epoch: 12, train loss: 1.501, val loss: 1.401, time: 0.98\n",
      "epoch: 13, train loss: 1.473, val loss: 1.388, time: 0.97\n",
      "epoch: 14, train loss: 1.459, val loss: 1.382, time: 0.97\n",
      "epoch: 15, train loss: 1.461, val loss: 1.375, time: 0.97\n",
      "epoch: 16, train loss: 1.444, val loss: 1.373, time: 0.97\n",
      "epoch: 17, train loss: 1.433, val loss: 1.371, time: 0.96\n",
      "epoch: 18, train loss: 1.423, val loss: 1.366, time: 0.96\n",
      "epoch: 19, train loss: 1.423, val loss: 1.361, time: 0.96\n",
      "epoch: 20, train loss: 1.412, val loss: 1.357, time: 0.90\n",
      "epoch: 21, train loss: 1.402, val loss: 1.354, time: 0.91\n",
      "epoch: 22, train loss: 1.409, val loss: 1.354, time: 0.99\n",
      "epoch: 23, train loss: 1.404, val loss: 1.351, time: 0.93\n",
      "epoch: 24, train loss: 1.392, val loss: 1.351, time: 0.98\n",
      "snATACseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:33:33,068 19692 genes overlap with model after filtering\n",
      "2022-03-25 10:33:33,069 35 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.071, val loss: 2.065, time: 0.89\n",
      "epoch: 1, train loss: 2.055, val loss: 2.033, time: 0.96\n",
      "epoch: 2, train loss: 1.991, val loss: 1.903, time: 0.79\n",
      "epoch: 3, train loss: 1.892, val loss: 1.785, time: 0.81\n",
      "epoch: 4, train loss: 1.799, val loss: 1.672, time: 0.90\n",
      "epoch: 5, train loss: 1.697, val loss: 1.566, time: 0.98\n",
      "epoch: 6, train loss: 1.622, val loss: 1.510, time: 1.20\n",
      "epoch: 7, train loss: 1.573, val loss: 1.470, time: 0.96\n",
      "epoch: 8, train loss: 1.542, val loss: 1.434, time: 0.96\n",
      "epoch: 9, train loss: 1.512, val loss: 1.408, time: 0.97\n",
      "epoch: 10, train loss: 1.485, val loss: 1.387, time: 0.86\n",
      "epoch: 11, train loss: 1.463, val loss: 1.376, time: 0.92\n",
      "epoch: 12, train loss: 1.449, val loss: 1.365, time: 0.92\n",
      "epoch: 13, train loss: 1.444, val loss: 1.356, time: 0.91\n",
      "epoch: 14, train loss: 1.426, val loss: 1.352, time: 0.92\n",
      "epoch: 15, train loss: 1.420, val loss: 1.347, time: 0.92\n",
      "epoch: 16, train loss: 1.411, val loss: 1.342, time: 0.91\n",
      "epoch: 17, train loss: 1.391, val loss: 1.337, time: 1.00\n",
      "epoch: 18, train loss: 1.399, val loss: 1.334, time: 1.04\n",
      "epoch: 19, train loss: 1.395, val loss: 1.331, time: 1.11\n",
      "epoch: 20, train loss: 1.379, val loss: 1.329, time: 0.95\n",
      "epoch: 21, train loss: 1.381, val loss: 1.329, time: 1.00\n",
      "epoch: 22, train loss: 1.373, val loss: 1.327, time: 1.00\n",
      "epoch: 23, train loss: 1.373, val loss: 1.326, time: 1.00\n",
      "epoch: 24, train loss: 1.363, val loss: 1.325, time: 0.99\n",
      "snATACseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:34:16,533 19692 genes overlap with model after filtering\n",
      "2022-03-25 10:34:16,534 32 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.051, val loss: 1.977, time: 0.94\n",
      "epoch: 1, train loss: 1.939, val loss: 1.843, time: 0.95\n",
      "epoch: 2, train loss: 1.861, val loss: 1.767, time: 0.99\n",
      "epoch: 3, train loss: 1.793, val loss: 1.702, time: 0.98\n",
      "epoch: 4, train loss: 1.758, val loss: 1.668, time: 1.03\n",
      "epoch: 5, train loss: 1.719, val loss: 1.620, time: 1.02\n",
      "epoch: 6, train loss: 1.683, val loss: 1.580, time: 0.99\n",
      "epoch: 7, train loss: 1.646, val loss: 1.554, time: 1.00\n",
      "epoch: 8, train loss: 1.618, val loss: 1.530, time: 0.95\n",
      "epoch: 9, train loss: 1.574, val loss: 1.509, time: 0.98\n",
      "epoch: 10, train loss: 1.553, val loss: 1.495, time: 0.99\n",
      "epoch: 11, train loss: 1.530, val loss: 1.486, time: 0.99\n",
      "epoch: 12, train loss: 1.517, val loss: 1.481, time: 0.98\n",
      "epoch: 13, train loss: 1.506, val loss: 1.474, time: 0.98\n",
      "epoch: 14, train loss: 1.508, val loss: 1.466, time: 0.99\n",
      "epoch: 15, train loss: 1.491, val loss: 1.452, time: 0.96\n",
      "epoch: 16, train loss: 1.470, val loss: 1.414, time: 0.94\n",
      "epoch: 17, train loss: 1.440, val loss: 1.371, time: 0.99\n",
      "epoch: 18, train loss: 1.424, val loss: 1.352, time: 0.99\n",
      "epoch: 19, train loss: 1.416, val loss: 1.346, time: 1.08\n",
      "epoch: 20, train loss: 1.407, val loss: 1.345, time: 0.97\n",
      "epoch: 21, train loss: 1.400, val loss: 1.341, time: 1.00\n",
      "epoch: 22, train loss: 1.389, val loss: 1.340, time: 1.00\n",
      "epoch: 23, train loss: 1.382, val loss: 1.340, time: 0.90\n",
      "epoch: 24, train loss: 1.386, val loss: 1.340, time: 0.93\n",
      "snATACseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:35:00,552 19694 genes overlap with model after filtering\n",
      "2022-03-25 10:35:00,553 31 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.069, val loss: 2.055, time: 0.86\n",
      "epoch: 1, train loss: 2.011, val loss: 1.960, time: 0.90\n",
      "epoch: 2, train loss: 1.898, val loss: 1.853, time: 0.93\n",
      "epoch: 3, train loss: 1.815, val loss: 1.791, time: 0.92\n",
      "epoch: 4, train loss: 1.751, val loss: 1.751, time: 0.97\n",
      "epoch: 5, train loss: 1.704, val loss: 1.722, time: 0.99\n",
      "epoch: 6, train loss: 1.680, val loss: 1.699, time: 0.87\n",
      "epoch: 7, train loss: 1.649, val loss: 1.659, time: 0.86\n",
      "epoch: 8, train loss: 1.617, val loss: 1.610, time: 0.85\n",
      "epoch: 9, train loss: 1.590, val loss: 1.561, time: 0.93\n",
      "epoch: 10, train loss: 1.548, val loss: 1.523, time: 0.91\n",
      "epoch: 11, train loss: 1.526, val loss: 1.501, time: 0.99\n",
      "epoch: 12, train loss: 1.513, val loss: 1.492, time: 0.92\n",
      "epoch: 13, train loss: 1.498, val loss: 1.483, time: 0.94\n",
      "epoch: 14, train loss: 1.493, val loss: 1.479, time: 0.97\n",
      "epoch: 15, train loss: 1.484, val loss: 1.477, time: 0.84\n",
      "epoch: 16, train loss: 1.473, val loss: 1.472, time: 0.88\n",
      "epoch: 17, train loss: 1.479, val loss: 1.469, time: 0.86\n",
      "epoch: 18, train loss: 1.470, val loss: 1.467, time: 0.94\n",
      "epoch: 19, train loss: 1.470, val loss: 1.464, time: 0.94\n",
      "epoch: 20, train loss: 1.461, val loss: 1.461, time: 0.94\n",
      "epoch: 21, train loss: 1.457, val loss: 1.459, time: 0.94\n",
      "epoch: 22, train loss: 1.452, val loss: 1.455, time: 0.97\n",
      "epoch: 23, train loss: 1.447, val loss: 1.450, time: 0.96\n",
      "epoch: 24, train loss: 1.443, val loss: 1.442, time: 0.98\n",
      "scRNAseq hnscc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:35:42,994 21815 genes overlap with model after filtering\n",
      "2022-03-25 10:35:42,994 1132 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.249, val loss: 2.082, time: 1.33\n",
      "epoch: 1, train loss: 1.979, val loss: 1.945, time: 1.36\n",
      "epoch: 2, train loss: 1.862, val loss: 1.853, time: 1.54\n",
      "epoch: 3, train loss: 1.774, val loss: 1.786, time: 1.35\n",
      "epoch: 4, train loss: 1.709, val loss: 1.740, time: 1.35\n",
      "epoch: 5, train loss: 1.667, val loss: 1.719, time: 1.38\n",
      "epoch: 6, train loss: 1.641, val loss: 1.708, time: 1.33\n",
      "epoch: 7, train loss: 1.623, val loss: 1.700, time: 1.33\n",
      "epoch: 8, train loss: 1.611, val loss: 1.696, time: 1.32\n",
      "epoch: 9, train loss: 1.601, val loss: 1.691, time: 1.33\n",
      "scRNAseq hnscc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:36:11,770 21953 genes overlap with model after filtering\n",
      "2022-03-25 10:36:11,771 1017 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.244, val loss: 2.104, time: 1.55\n",
      "epoch: 1, train loss: 1.978, val loss: 1.903, time: 1.35\n",
      "epoch: 2, train loss: 1.830, val loss: 1.809, time: 1.37\n",
      "epoch: 3, train loss: 1.754, val loss: 1.774, time: 1.36\n",
      "epoch: 4, train loss: 1.721, val loss: 1.761, time: 1.30\n",
      "epoch: 5, train loss: 1.699, val loss: 1.753, time: 1.37\n",
      "epoch: 6, train loss: 1.687, val loss: 1.748, time: 1.33\n",
      "epoch: 7, train loss: 1.677, val loss: 1.743, time: 1.31\n",
      "epoch: 8, train loss: 1.647, val loss: 1.715, time: 1.38\n",
      "epoch: 9, train loss: 1.615, val loss: 1.700, time: 1.50\n",
      "scRNAseq hnscc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:36:41,570 21876 genes overlap with model after filtering\n",
      "2022-03-25 10:36:41,572 1029 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.258, val loss: 2.055, time: 1.34\n",
      "epoch: 1, train loss: 1.960, val loss: 1.888, time: 1.33\n",
      "epoch: 2, train loss: 1.816, val loss: 1.797, time: 1.32\n",
      "epoch: 3, train loss: 1.745, val loss: 1.762, time: 1.32\n",
      "epoch: 4, train loss: 1.710, val loss: 1.743, time: 1.32\n",
      "epoch: 5, train loss: 1.664, val loss: 1.716, time: 1.29\n",
      "epoch: 6, train loss: 1.634, val loss: 1.706, time: 1.39\n",
      "epoch: 7, train loss: 1.615, val loss: 1.701, time: 1.28\n",
      "epoch: 8, train loss: 1.604, val loss: 1.697, time: 1.27\n",
      "epoch: 9, train loss: 1.597, val loss: 1.694, time: 1.26\n",
      "scRNAseq hnscc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:37:09,575 21868 genes overlap with model after filtering\n",
      "2022-03-25 10:37:09,579 1038 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.288, val loss: 2.126, time: 1.05\n",
      "epoch: 1, train loss: 2.021, val loss: 1.950, time: 1.09\n",
      "epoch: 2, train loss: 1.903, val loss: 1.892, time: 1.14\n",
      "epoch: 3, train loss: 1.837, val loss: 1.848, time: 1.19\n",
      "epoch: 4, train loss: 1.776, val loss: 1.783, time: 1.15\n",
      "epoch: 5, train loss: 1.700, val loss: 1.732, time: 1.17\n",
      "epoch: 6, train loss: 1.658, val loss: 1.713, time: 1.29\n",
      "epoch: 7, train loss: 1.634, val loss: 1.703, time: 1.16\n",
      "epoch: 8, train loss: 1.617, val loss: 1.697, time: 1.18\n",
      "epoch: 9, train loss: 1.609, val loss: 1.692, time: 1.20\n",
      "scRNAseq hnscc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:37:34,301 21879 genes overlap with model after filtering\n",
      "2022-03-25 10:37:34,302 944 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.234, val loss: 2.101, time: 1.12\n",
      "epoch: 1, train loss: 2.019, val loss: 1.986, time: 1.21\n",
      "epoch: 2, train loss: 1.911, val loss: 1.913, time: 1.15\n",
      "epoch: 3, train loss: 1.825, val loss: 1.836, time: 1.13\n",
      "epoch: 4, train loss: 1.756, val loss: 1.783, time: 1.14\n",
      "epoch: 5, train loss: 1.715, val loss: 1.760, time: 1.15\n",
      "epoch: 6, train loss: 1.695, val loss: 1.747, time: 1.27\n",
      "epoch: 7, train loss: 1.679, val loss: 1.736, time: 1.24\n",
      "epoch: 8, train loss: 1.652, val loss: 1.717, time: 1.17\n",
      "epoch: 9, train loss: 1.623, val loss: 1.704, time: 1.18\n",
      "scRNAseq cesc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:37:59,297 21175 genes overlap with model after filtering\n",
      "2022-03-25 10:37:59,298 652 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.369, val loss: 2.273, time: 1.12\n",
      "epoch: 1, train loss: 2.237, val loss: 2.179, time: 1.13\n",
      "epoch: 2, train loss: 2.191, val loss: 2.139, time: 1.13\n",
      "epoch: 3, train loss: 2.135, val loss: 2.092, time: 1.13\n",
      "epoch: 4, train loss: 2.091, val loss: 2.056, time: 1.14\n",
      "epoch: 5, train loss: 2.059, val loss: 2.023, time: 1.15\n",
      "epoch: 6, train loss: 2.031, val loss: 1.994, time: 1.15\n",
      "epoch: 7, train loss: 2.018, val loss: 1.980, time: 1.16\n",
      "epoch: 8, train loss: 2.004, val loss: 1.972, time: 1.15\n",
      "epoch: 9, train loss: 1.992, val loss: 1.959, time: 1.22\n",
      "scRNAseq cesc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:38:22,607 21072 genes overlap with model after filtering\n",
      "2022-03-25 10:38:22,608 706 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.372, val loss: 2.305, time: 1.12\n",
      "epoch: 1, train loss: 2.264, val loss: 2.188, time: 1.16\n",
      "epoch: 2, train loss: 2.169, val loss: 2.074, time: 1.15\n",
      "epoch: 3, train loss: 2.075, val loss: 1.974, time: 1.15\n",
      "epoch: 4, train loss: 1.998, val loss: 1.905, time: 1.29\n",
      "epoch: 5, train loss: 1.937, val loss: 1.856, time: 1.09\n",
      "epoch: 6, train loss: 1.901, val loss: 1.824, time: 1.13\n",
      "epoch: 7, train loss: 1.866, val loss: 1.801, time: 1.15\n",
      "epoch: 8, train loss: 1.848, val loss: 1.784, time: 1.14\n",
      "epoch: 9, train loss: 1.834, val loss: 1.774, time: 1.13\n",
      "scRNAseq cesc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:38:45,960 21174 genes overlap with model after filtering\n",
      "2022-03-25 10:38:45,961 664 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.376, val loss: 2.304, time: 1.12\n",
      "epoch: 1, train loss: 2.264, val loss: 2.176, time: 1.17\n",
      "epoch: 2, train loss: 2.194, val loss: 2.111, time: 1.17\n",
      "epoch: 3, train loss: 2.142, val loss: 2.050, time: 1.17\n",
      "epoch: 4, train loss: 2.075, val loss: 1.945, time: 1.19\n",
      "epoch: 5, train loss: 1.985, val loss: 1.842, time: 1.17\n",
      "epoch: 6, train loss: 1.913, val loss: 1.776, time: 1.18\n",
      "epoch: 7, train loss: 1.860, val loss: 1.715, time: 1.17\n",
      "epoch: 8, train loss: 1.801, val loss: 1.658, time: 1.15\n",
      "epoch: 9, train loss: 1.760, val loss: 1.629, time: 1.16\n",
      "scRNAseq cesc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:39:09,339 21151 genes overlap with model after filtering\n",
      "2022-03-25 10:39:09,340 645 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.367, val loss: 2.293, time: 1.25\n",
      "epoch: 1, train loss: 2.266, val loss: 2.205, time: 1.18\n",
      "epoch: 2, train loss: 2.179, val loss: 2.107, time: 0.93\n",
      "epoch: 3, train loss: 2.091, val loss: 2.010, time: 0.83\n",
      "epoch: 4, train loss: 2.020, val loss: 1.931, time: 0.83\n",
      "epoch: 5, train loss: 1.974, val loss: 1.890, time: 1.03\n",
      "epoch: 6, train loss: 1.944, val loss: 1.855, time: 1.16\n",
      "epoch: 7, train loss: 1.916, val loss: 1.819, time: 1.15\n",
      "epoch: 8, train loss: 1.887, val loss: 1.790, time: 1.17\n",
      "epoch: 9, train loss: 1.857, val loss: 1.758, time: 1.17\n",
      "scRNAseq cesc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:39:31,189 21184 genes overlap with model after filtering\n",
      "2022-03-25 10:39:31,190 667 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.383, val loss: 2.325, time: 1.08\n",
      "epoch: 1, train loss: 2.241, val loss: 2.134, time: 1.08\n",
      "epoch: 2, train loss: 2.096, val loss: 1.999, time: 1.16\n",
      "epoch: 3, train loss: 2.003, val loss: 1.900, time: 1.12\n",
      "epoch: 4, train loss: 1.943, val loss: 1.854, time: 1.13\n",
      "epoch: 5, train loss: 1.910, val loss: 1.827, time: 1.14\n",
      "epoch: 6, train loss: 1.881, val loss: 1.811, time: 1.16\n",
      "epoch: 7, train loss: 1.863, val loss: 1.799, time: 1.17\n",
      "epoch: 8, train loss: 1.854, val loss: 1.786, time: 1.17\n",
      "epoch: 9, train loss: 1.822, val loss: 1.728, time: 1.15\n",
      "scRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:39:54,378 22276 genes overlap with model after filtering\n",
      "2022-03-25 10:39:54,379 1435 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.443, val loss: 2.288, time: 1.38\n",
      "epoch: 1, train loss: 2.189, val loss: 2.119, time: 1.38\n",
      "epoch: 2, train loss: 2.052, val loss: 2.042, time: 1.40\n",
      "epoch: 3, train loss: 1.967, val loss: 1.993, time: 1.17\n",
      "epoch: 4, train loss: 1.912, val loss: 1.953, time: 1.25\n",
      "epoch: 5, train loss: 1.844, val loss: 1.898, time: 1.42\n",
      "epoch: 6, train loss: 1.797, val loss: 1.871, time: 1.47\n",
      "epoch: 7, train loss: 1.770, val loss: 1.858, time: 1.45\n",
      "epoch: 8, train loss: 1.753, val loss: 1.851, time: 1.48\n",
      "epoch: 9, train loss: 1.742, val loss: 1.844, time: 1.44\n",
      "scRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:40:22,694 22307 genes overlap with model after filtering\n",
      "2022-03-25 10:40:22,695 1398 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.441, val loss: 2.252, time: 1.44\n",
      "epoch: 1, train loss: 2.178, val loss: 2.101, time: 1.41\n",
      "epoch: 2, train loss: 2.066, val loss: 2.050, time: 1.41\n",
      "epoch: 3, train loss: 1.997, val loss: 1.991, time: 1.59\n",
      "epoch: 4, train loss: 1.920, val loss: 1.931, time: 1.55\n",
      "epoch: 5, train loss: 1.848, val loss: 1.888, time: 1.42\n",
      "epoch: 6, train loss: 1.801, val loss: 1.866, time: 1.43\n",
      "epoch: 7, train loss: 1.774, val loss: 1.854, time: 1.49\n",
      "epoch: 8, train loss: 1.756, val loss: 1.847, time: 1.48\n",
      "epoch: 9, train loss: 1.745, val loss: 1.841, time: 1.31\n",
      "scRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:40:51,548 22275 genes overlap with model after filtering\n",
      "2022-03-25 10:40:51,549 1441 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.436, val loss: 2.243, time: 1.45\n",
      "epoch: 1, train loss: 2.164, val loss: 2.099, time: 1.42\n",
      "epoch: 2, train loss: 2.036, val loss: 2.036, time: 1.59\n",
      "epoch: 3, train loss: 1.948, val loss: 1.962, time: 1.48\n",
      "epoch: 4, train loss: 1.867, val loss: 1.911, time: 1.43\n",
      "epoch: 5, train loss: 1.815, val loss: 1.883, time: 1.43\n",
      "epoch: 6, train loss: 1.783, val loss: 1.867, time: 1.44\n",
      "epoch: 7, train loss: 1.767, val loss: 1.857, time: 1.43\n",
      "epoch: 8, train loss: 1.750, val loss: 1.851, time: 1.43\n",
      "epoch: 9, train loss: 1.744, val loss: 1.846, time: 1.45\n",
      "scRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:41:20,896 22254 genes overlap with model after filtering\n",
      "2022-03-25 10:41:20,897 1370 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.434, val loss: 2.270, time: 1.37\n",
      "epoch: 1, train loss: 2.187, val loss: 2.110, time: 1.33\n",
      "epoch: 2, train loss: 2.020, val loss: 1.996, time: 1.55\n",
      "epoch: 3, train loss: 1.915, val loss: 1.938, time: 1.40\n",
      "epoch: 4, train loss: 1.867, val loss: 1.917, time: 1.49\n",
      "epoch: 5, train loss: 1.840, val loss: 1.907, time: 1.41\n",
      "epoch: 6, train loss: 1.825, val loss: 1.899, time: 1.52\n",
      "epoch: 7, train loss: 1.805, val loss: 1.878, time: 1.47\n",
      "epoch: 8, train loss: 1.777, val loss: 1.855, time: 1.50\n",
      "epoch: 9, train loss: 1.754, val loss: 1.846, time: 1.48\n",
      "scRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:41:49,758 22221 genes overlap with model after filtering\n",
      "2022-03-25 10:41:49,759 1367 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.437, val loss: 2.265, time: 1.49\n",
      "epoch: 1, train loss: 2.119, val loss: 2.048, time: 1.64\n",
      "epoch: 2, train loss: 1.956, val loss: 1.947, time: 1.50\n",
      "epoch: 3, train loss: 1.870, val loss: 1.904, time: 1.48\n",
      "epoch: 4, train loss: 1.821, val loss: 1.881, time: 1.45\n",
      "epoch: 5, train loss: 1.790, val loss: 1.867, time: 1.52\n",
      "epoch: 6, train loss: 1.771, val loss: 1.857, time: 1.49\n",
      "epoch: 7, train loss: 1.757, val loss: 1.852, time: 1.48\n",
      "epoch: 8, train loss: 1.748, val loss: 1.846, time: 1.49\n",
      "epoch: 9, train loss: 1.739, val loss: 1.842, time: 1.49\n",
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:42:19,017 18620 genes overlap with model after filtering\n",
      "2022-03-25 10:42:19,018 1218 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.395, val loss: 2.389, time: 0.87\n",
      "epoch: 1, train loss: 2.357, val loss: 2.320, time: 0.90\n",
      "epoch: 2, train loss: 2.239, val loss: 2.159, time: 0.92\n",
      "epoch: 3, train loss: 2.111, val loss: 2.062, time: 1.02\n",
      "epoch: 4, train loss: 2.042, val loss: 2.020, time: 0.84\n",
      "epoch: 5, train loss: 2.019, val loss: 2.005, time: 0.86\n",
      "epoch: 6, train loss: 1.996, val loss: 1.993, time: 0.85\n",
      "epoch: 7, train loss: 1.987, val loss: 1.980, time: 0.88\n",
      "epoch: 8, train loss: 1.965, val loss: 1.970, time: 0.89\n",
      "epoch: 9, train loss: 1.958, val loss: 1.956, time: 0.94\n",
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:42:36,227 18583 genes overlap with model after filtering\n",
      "2022-03-25 10:42:36,229 1374 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.394, val loss: 2.385, time: 0.87\n",
      "epoch: 1, train loss: 2.347, val loss: 2.323, time: 0.88\n",
      "epoch: 2, train loss: 2.266, val loss: 2.246, time: 0.88\n",
      "epoch: 3, train loss: 2.192, val loss: 2.163, time: 0.88\n",
      "epoch: 4, train loss: 2.127, val loss: 2.111, time: 0.90\n",
      "epoch: 5, train loss: 2.096, val loss: 2.066, time: 0.89\n",
      "epoch: 6, train loss: 2.039, val loss: 2.012, time: 0.87\n",
      "epoch: 7, train loss: 2.008, val loss: 1.990, time: 0.92\n",
      "epoch: 8, train loss: 1.984, val loss: 1.970, time: 0.97\n",
      "epoch: 9, train loss: 1.958, val loss: 1.940, time: 0.92\n",
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:42:53,976 18628 genes overlap with model after filtering\n",
      "2022-03-25 10:42:53,977 1465 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.384, val loss: 2.370, time: 0.86\n",
      "epoch: 1, train loss: 2.318, val loss: 2.294, time: 0.88\n",
      "epoch: 2, train loss: 2.221, val loss: 2.190, time: 0.90\n",
      "epoch: 3, train loss: 2.102, val loss: 2.059, time: 0.90\n",
      "epoch: 4, train loss: 2.014, val loss: 1.973, time: 0.88\n",
      "epoch: 5, train loss: 1.960, val loss: 1.923, time: 0.88\n",
      "epoch: 6, train loss: 1.924, val loss: 1.891, time: 0.90\n",
      "epoch: 7, train loss: 1.893, val loss: 1.868, time: 0.90\n",
      "epoch: 8, train loss: 1.872, val loss: 1.853, time: 0.92\n",
      "epoch: 9, train loss: 1.850, val loss: 1.840, time: 0.92\n",
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:43:11,483 18710 genes overlap with model after filtering\n",
      "2022-03-25 10:43:11,484 1292 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.383, val loss: 2.324, time: 0.86\n",
      "epoch: 1, train loss: 2.278, val loss: 2.212, time: 0.93\n",
      "epoch: 2, train loss: 2.202, val loss: 2.136, time: 0.99\n",
      "epoch: 3, train loss: 2.139, val loss: 2.074, time: 0.91\n",
      "epoch: 4, train loss: 2.097, val loss: 2.037, time: 0.90\n",
      "epoch: 5, train loss: 2.056, val loss: 1.993, time: 0.91\n",
      "epoch: 6, train loss: 1.999, val loss: 1.905, time: 0.93\n",
      "epoch: 7, train loss: 1.903, val loss: 1.828, time: 0.92\n",
      "epoch: 8, train loss: 1.844, val loss: 1.770, time: 0.91\n",
      "epoch: 9, train loss: 1.785, val loss: 1.729, time: 0.98\n",
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:43:29,327 18683 genes overlap with model after filtering\n",
      "2022-03-25 10:43:29,328 1360 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.398, val loss: 2.389, time: 0.83\n",
      "epoch: 1, train loss: 2.363, val loss: 2.314, time: 0.82\n",
      "epoch: 2, train loss: 2.288, val loss: 2.245, time: 0.90\n",
      "epoch: 3, train loss: 2.237, val loss: 2.179, time: 0.89\n",
      "epoch: 4, train loss: 2.129, val loss: 2.027, time: 0.90\n",
      "epoch: 5, train loss: 2.002, val loss: 1.910, time: 0.87\n",
      "epoch: 6, train loss: 1.915, val loss: 1.843, time: 0.86\n",
      "epoch: 7, train loss: 1.859, val loss: 1.790, time: 0.90\n",
      "epoch: 8, train loss: 1.820, val loss: 1.741, time: 0.97\n",
      "epoch: 9, train loss: 1.775, val loss: 1.699, time: 0.91\n",
      "scRNAseq melanoma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:43:46,790 19162 genes overlap with model after filtering\n",
      "2022-03-25 10:43:46,791 1543 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.289, val loss: 2.252, time: 1.10\n",
      "epoch: 1, train loss: 2.213, val loss: 2.143, time: 1.14\n",
      "epoch: 2, train loss: 2.121, val loss: 2.031, time: 1.09\n",
      "epoch: 3, train loss: 2.042, val loss: 1.939, time: 1.07\n",
      "epoch: 4, train loss: 1.966, val loss: 1.875, time: 1.07\n",
      "epoch: 5, train loss: 1.915, val loss: 1.825, time: 1.03\n",
      "epoch: 6, train loss: 1.869, val loss: 1.781, time: 1.04\n",
      "epoch: 7, train loss: 1.847, val loss: 1.756, time: 1.05\n",
      "epoch: 8, train loss: 1.813, val loss: 1.724, time: 1.07\n",
      "epoch: 9, train loss: 1.793, val loss: 1.697, time: 1.08\n",
      "scRNAseq melanoma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:44:06,467 19131 genes overlap with model after filtering\n",
      "2022-03-25 10:44:06,468 1544 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.296, val loss: 2.284, time: 1.16\n",
      "epoch: 1, train loss: 2.226, val loss: 2.168, time: 1.10\n",
      "epoch: 2, train loss: 2.107, val loss: 2.076, time: 1.13\n",
      "epoch: 3, train loss: 2.023, val loss: 1.970, time: 1.09\n",
      "epoch: 4, train loss: 1.946, val loss: 1.883, time: 1.09\n",
      "epoch: 5, train loss: 1.893, val loss: 1.812, time: 1.11\n",
      "epoch: 6, train loss: 1.834, val loss: 1.727, time: 1.11\n",
      "epoch: 7, train loss: 1.778, val loss: 1.673, time: 1.09\n",
      "epoch: 8, train loss: 1.750, val loss: 1.640, time: 1.08\n",
      "epoch: 9, train loss: 1.716, val loss: 1.621, time: 1.09\n",
      "scRNAseq melanoma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:44:26,546 19149 genes overlap with model after filtering\n",
      "2022-03-25 10:44:26,547 1523 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.295, val loss: 2.281, time: 1.07\n",
      "epoch: 1, train loss: 2.225, val loss: 2.202, time: 1.09\n",
      "epoch: 2, train loss: 2.137, val loss: 2.117, time: 1.10\n",
      "epoch: 3, train loss: 2.053, val loss: 2.058, time: 1.23\n",
      "epoch: 4, train loss: 1.975, val loss: 1.992, time: 1.03\n",
      "epoch: 5, train loss: 1.924, val loss: 1.931, time: 1.15\n",
      "epoch: 6, train loss: 1.875, val loss: 1.877, time: 1.06\n",
      "epoch: 7, train loss: 1.825, val loss: 1.820, time: 1.07\n",
      "epoch: 8, train loss: 1.775, val loss: 1.751, time: 1.20\n",
      "epoch: 9, train loss: 1.724, val loss: 1.706, time: 1.06\n",
      "scRNAseq melanoma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:44:45,623 19164 genes overlap with model after filtering\n",
      "2022-03-25 10:44:45,624 1529 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.285, val loss: 2.262, time: 1.03\n",
      "epoch: 1, train loss: 2.209, val loss: 2.207, time: 1.09\n",
      "epoch: 2, train loss: 2.138, val loss: 2.119, time: 1.10\n",
      "epoch: 3, train loss: 2.038, val loss: 1.985, time: 1.09\n",
      "epoch: 4, train loss: 1.934, val loss: 1.899, time: 1.07\n",
      "epoch: 5, train loss: 1.859, val loss: 1.834, time: 1.09\n",
      "epoch: 6, train loss: 1.800, val loss: 1.792, time: 1.08\n",
      "epoch: 7, train loss: 1.765, val loss: 1.748, time: 1.12\n",
      "epoch: 8, train loss: 1.729, val loss: 1.715, time: 1.12\n",
      "epoch: 9, train loss: 1.699, val loss: 1.695, time: 1.22\n",
      "scRNAseq melanoma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:45:05,306 19103 genes overlap with model after filtering\n",
      "2022-03-25 10:45:05,307 1510 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.298, val loss: 2.288, time: 1.09\n",
      "epoch: 1, train loss: 2.266, val loss: 2.219, time: 1.11\n",
      "epoch: 2, train loss: 2.207, val loss: 2.160, time: 1.11\n",
      "epoch: 3, train loss: 2.139, val loss: 2.099, time: 1.13\n",
      "epoch: 4, train loss: 2.051, val loss: 1.967, time: 1.14\n",
      "epoch: 5, train loss: 1.967, val loss: 1.891, time: 1.12\n",
      "epoch: 6, train loss: 1.922, val loss: 1.843, time: 1.11\n",
      "epoch: 7, train loss: 1.864, val loss: 1.788, time: 1.11\n",
      "epoch: 8, train loss: 1.820, val loss: 1.727, time: 1.11\n",
      "epoch: 9, train loss: 1.772, val loss: 1.685, time: 1.11\n",
      "scRNAseq pdac fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:45:26,837 23822 genes overlap with model after filtering\n",
      "2022-03-25 10:45:26,838 1199 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.682, val loss: 2.494, time: 2.24\n",
      "epoch: 1, train loss: 2.456, val loss: 2.366, time: 2.14\n",
      "epoch: 2, train loss: 2.357, val loss: 2.281, time: 1.86\n",
      "epoch: 3, train loss: 2.267, val loss: 2.224, time: 1.81\n",
      "epoch: 4, train loss: 2.201, val loss: 2.182, time: 1.92\n",
      "epoch: 5, train loss: 2.149, val loss: 2.150, time: 1.84\n",
      "epoch: 6, train loss: 2.123, val loss: 2.138, time: 1.92\n",
      "epoch: 7, train loss: 2.103, val loss: 2.130, time: 1.93\n",
      "epoch: 8, train loss: 2.091, val loss: 2.127, time: 2.02\n",
      "epoch: 9, train loss: 2.083, val loss: 2.125, time: 2.08\n",
      "scRNAseq pdac fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:46:07,017 23837 genes overlap with model after filtering\n",
      "2022-03-25 10:46:07,018 1215 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.678, val loss: 2.453, time: 1.88\n",
      "epoch: 1, train loss: 2.388, val loss: 2.251, time: 1.87\n",
      "epoch: 2, train loss: 2.247, val loss: 2.177, time: 1.92\n",
      "epoch: 3, train loss: 2.177, val loss: 2.136, time: 1.90\n",
      "epoch: 4, train loss: 2.128, val loss: 2.118, time: 2.07\n",
      "epoch: 5, train loss: 2.102, val loss: 2.107, time: 1.99\n",
      "epoch: 6, train loss: 2.084, val loss: 2.100, time: 1.95\n",
      "epoch: 7, train loss: 2.073, val loss: 2.094, time: 1.95\n",
      "epoch: 8, train loss: 2.065, val loss: 2.091, time: 1.93\n",
      "epoch: 9, train loss: 2.059, val loss: 2.089, time: 1.68\n",
      "scRNAseq pdac fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:46:46,488 23823 genes overlap with model after filtering\n",
      "2022-03-25 10:46:46,489 1164 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.675, val loss: 2.444, time: 1.89\n",
      "epoch: 1, train loss: 2.395, val loss: 2.300, time: 2.21\n",
      "epoch: 2, train loss: 2.288, val loss: 2.242, time: 2.01\n",
      "epoch: 3, train loss: 2.216, val loss: 2.184, time: 2.02\n",
      "epoch: 4, train loss: 2.150, val loss: 2.150, time: 2.04\n",
      "epoch: 5, train loss: 2.118, val loss: 2.136, time: 2.01\n",
      "epoch: 6, train loss: 2.097, val loss: 2.128, time: 2.00\n",
      "epoch: 7, train loss: 2.087, val loss: 2.122, time: 2.01\n",
      "epoch: 8, train loss: 2.075, val loss: 2.118, time: 1.95\n",
      "epoch: 9, train loss: 2.062, val loss: 2.098, time: 2.11\n",
      "scRNAseq pdac fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:47:27,544 23781 genes overlap with model after filtering\n",
      "2022-03-25 10:47:27,545 1197 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.695, val loss: 2.545, time: 1.91\n",
      "epoch: 1, train loss: 2.468, val loss: 2.320, time: 1.91\n",
      "epoch: 2, train loss: 2.282, val loss: 2.201, time: 1.89\n",
      "epoch: 3, train loss: 2.184, val loss: 2.153, time: 2.04\n",
      "epoch: 4, train loss: 2.143, val loss: 2.140, time: 1.82\n",
      "epoch: 5, train loss: 2.112, val loss: 2.132, time: 2.11\n",
      "epoch: 6, train loss: 2.097, val loss: 2.127, time: 1.89\n",
      "epoch: 7, train loss: 2.082, val loss: 2.124, time: 1.94\n",
      "epoch: 8, train loss: 2.072, val loss: 2.119, time: 1.97\n",
      "epoch: 9, train loss: 2.065, val loss: 2.117, time: 1.95\n",
      "scRNAseq pdac fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2022-03-25 10:48:07,313 23764 genes overlap with model after filtering\n",
      "2022-03-25 10:48:07,315 1177 genes missing from dataset after filtering\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "epoch: 0, train loss: 2.667, val loss: 2.459, time: 1.93\n",
      "epoch: 1, train loss: 2.432, val loss: 2.339, time: 2.14\n",
      "epoch: 2, train loss: 2.322, val loss: 2.247, time: 1.95\n",
      "epoch: 3, train loss: 2.244, val loss: 2.189, time: 1.97\n",
      "epoch: 4, train loss: 2.205, val loss: 2.174, time: 1.98\n",
      "epoch: 5, train loss: 2.181, val loss: 2.155, time: 1.97\n",
      "epoch: 6, train loss: 2.135, val loss: 2.127, time: 1.92\n",
      "epoch: 7, train loss: 2.093, val loss: 2.102, time: 1.95\n",
      "epoch: 8, train loss: 2.055, val loss: 2.092, time: 1.94\n",
      "epoch: 9, train loss: 2.036, val loss: 2.086, time: 2.15\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for dtype, d1 in folds_fmap.items():\n",
    "    to_prediction[dtype] = {}\n",
    "    for disease, d2 in d1.items():\n",
    "        to_prediction[dtype][disease] = {}\n",
    "        for fold, d3 in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            train = sc.read_h5ad(d3['train'])\n",
    "            val = sc.read_h5ad(d3['val'])\n",
    "\n",
    "            to_prediction[dtype][disease][fold] = mlp_predict(\n",
    "                train, val, epochs=10 if dtype=='scRNAseq' else 25)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "8ef3a1f0",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            a.write_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "92b9bf4a",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold4\n",
      "snRNAseq gbm fold4\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold1\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold3\n",
      "snATACseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold4\n",
      "snATACseq gbm fold4\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold4\n",
      "scRNAseq myeloma fold0\n",
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n",
      "scRNAseq myeloma fold4\n",
      "scRNAseq myeloma fold2\n",
      "scRNAseq melanoma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold1\n",
      "scRNAseq melanoma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold4\n",
      "scRNAseq pdac fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    }
   ],
   "source": [
    "for dtype, d1 in to_prediction.items():\n",
    "    for disease, d2 in d1.items():\n",
    "        for fold, a in d2.items():\n",
    "            print(dtype, disease, fold)\n",
    "            adata = sc.read_h5ad(os.path.join(result_dir, f'{dtype}_{disease}_{fold}.h5ad'))\n",
    "            y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "            y_true = adata.obs['cell_type'].to_list()\n",
    "            d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "            d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "            cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4fd50399",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### actinn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fa0d47ae",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/actinn/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "0e5cc26e",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/brca/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/cesc/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/hnscc/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/pdac/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/brca_gene_activity/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/ccrcc_gene_activity/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/gbm_gene_activity/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/brca/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/ccrcc/actinn.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/gbm/actinn.tsv']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles('/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/', regex=r'actinn.tsv'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "6cd5ccae",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "for fp in fps:\n",
    "    disease = fp.split('/')[-2]\n",
    "    dtype = fp.split('/')[-3]\n",
    "    df = pd.read_csv(fp, sep='\\t')\n",
    "    y_pred = df['predicted'].to_list()\n",
    "    y_true = df['groundtruth'].to_list()\n",
    "    \n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e793d7eb",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "###### SingleCellNet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7206862c",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/singlecellnet'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c0627605",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/brca/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/cesc/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/hnscc/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/melanoma/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/myeloma/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/pdac/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/brca_gene_activity/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/ccrcc_gene_activity/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/gbm_gene_activity/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/brca/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/ccrcc/SingleCellNet.tsv',\n",
       " '/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/gbm/SingleCellNet.tsv']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles('/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/', regex=r'SingleCellNet.tsv'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6e93f8b6",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/brca/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/cesc/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/hnscc/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/melanoma/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/myeloma/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/scRNAseq/pdac/SingleCellNet.tsv\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/brca_gene_activity/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/ccrcc_gene_activity/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snATACseq/gbm_gene_activity/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/brca/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/ccrcc/SingleCellNet.tsv\n",
      "/home/estorrs/pollock/benchmarking/results/01272021_harmonized_v2/snRNAseq/gbm/SingleCellNet.tsv\n"
     ]
    }
   ],
   "source": [
    "for fp in fps:\n",
    "    print(fp)\n",
    "    disease = fp.split('/')[-2]\n",
    "    dtype = fp.split('/')[-3]\n",
    "    df = pd.read_csv(fp, sep='\\t')\n",
    "    df = df[df['predicted']!='rand']\n",
    "    y_pred = df['predicted'].to_list()\n",
    "    y_true = df['groundtruth'].to_list()\n",
    "    \n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "18ae5792",
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "551100e9",
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_brca_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_brca_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_brca_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_brca_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_brca_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_cesc_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_cesc_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_cesc_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_cesc_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_hnscc_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_hnscc_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_hnscc_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_hnscc_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_hnscc_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_melanoma_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_melanoma_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_melanoma_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_melanoma_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_melanoma_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_myeloma_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_myeloma_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_myeloma_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_myeloma_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_myeloma_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_pdac_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_pdac_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_pdac_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_pdac_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/scRNAseq_pdac_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_brca_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_brca_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_brca_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_brca_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_brca_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_ccrcc_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_ccrcc_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_ccrcc_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_ccrcc_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_ccrcc_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_gbm_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_gbm_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_gbm_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_gbm_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snATACseq_gbm_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_brca_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_brca_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_brca_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_brca_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_brca_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_ccrcc_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_ccrcc_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_ccrcc_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_ccrcc_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_ccrcc_fold4.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_gbm_fold0.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_gbm_fold1.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_gbm_fold2.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_gbm_fold3.txt',\n",
       " '/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/snRNAseq_gbm_fold4.txt']"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles(result_dir, regex=r'[0-9].txt'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "eada1175",
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq cesc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq hnscc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq melanoma fold4\n",
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rand cell count: 1, excluding from pred df\n",
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n",
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n",
      "scRNAseq pdac fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snATACseq gbm fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq brca fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq ccrcc fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'predicted_cell_type' as categorical\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "snRNAseq gbm fold4\n"
     ]
    }
   ],
   "source": [
    "for fp in fps:\n",
    "    dtype, disease, fold = fp.replace('.txt', '').split('/')[-1].split('_')\n",
    "    print(dtype, disease, fold)\n",
    "    \n",
    "    adata = sc.read_h5ad(folds_fmap[dtype][disease][fold]['val'])\n",
    "\n",
    "    df = pd.read_csv(fp, sep='\\t')\n",
    "    df = df[[c for c in df.columns if 'rand_' not in c]]\n",
    "    df = df.transpose()\n",
    "    df['predicted_cell_type'] = [df.columns[i] for i in np.argmax(df.values, axis=1)]\n",
    "    assert df.shape[0] == adata.obs.shape[0]\n",
    "    counts = Counter(df['predicted_cell_type'])\n",
    "    rand_count = counts.get('rand', 0)\n",
    "    rand_idxs = None\n",
    "    if rand_count:\n",
    "        print(f'rand cell count: {rand_count}, excluding from pred df')\n",
    "        rand_idxs = df[df['predicted_cell_type']=='rand'].index.to_list()\n",
    "        \n",
    "    \n",
    "    adata.obs = pd.merge(adata.obs, df, how='left', left_index=True, right_index=True)\n",
    "#     print(set(adata.obs['predicted_cell_type']))\n",
    "    adata.write_h5ad(fp.replace('.txt', '.h5ad'))\n",
    "    \n",
    "    if rand_idxs is not None:\n",
    "        adata = adata[[x for x in adata.obs.index if x not in rand_idxs]]\n",
    "    \n",
    "    y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "    y_true = adata.obs['cell_type'].to_list()\n",
    "    \n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "        \n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ec3f7b9",
   "metadata": {},
   "source": [
    "#### seurat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6500e99d",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/seurat/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "b46fdc9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/pollock/benchmarking/results/seurat/scRNAseq_brca.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/scRNAseq_cesc.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/scRNAseq_hnscc.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/scRNAseq_melanoma.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/scRNAseq_myeloma.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/scRNAseq_pdac.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snATACseq_brca.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snATACseq_ccrcc.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snATACseq_gbm.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snRNAseq_brca.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snRNAseq_ccrcc.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/snRNAseq_gbm.h5ad']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles(result_dir, regex=r'.h5ad$'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "7ae8d769",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca\n",
      "scRNAseq cesc\n",
      "scRNAseq hnscc\n",
      "scRNAseq melanoma\n",
      "scRNAseq myeloma\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac\n",
      "snATACseq brca\n",
      "snATACseq ccrcc\n",
      "snATACseq gbm\n",
      "snRNAseq brca\n",
      "snRNAseq ccrcc\n",
      "snRNAseq gbm\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for fp in fps:\n",
    "    root = fp.split('/')[-1].replace('.h5ad', '')\n",
    "    dtype, disease = root.split('_')\n",
    "    print(dtype, disease)\n",
    "    \n",
    "    adata = sc.read_h5ad(fp)\n",
    "\n",
    "    y_pred = adata.obs['predicted.id'].to_list()\n",
    "    y_true = adata.obs['cell_type'].to_list()\n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_confusion_matrix.txt'), sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "0818c2a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "1558a076",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_cesc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_cesc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_cesc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_cesc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_hnscc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_hnscc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_hnscc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_hnscc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_hnscc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_melanoma_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_melanoma_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_melanoma_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_melanoma_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_melanoma_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_myeloma_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_myeloma_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_myeloma_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_myeloma_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_myeloma_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_pdac_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_pdac_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_pdac_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_pdac_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/scRNAseq_pdac_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snATACseq_gbm_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/seurat/interdataset_with_folds/snRNAseq_gbm_fold4.h5ad']"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles(result_dir, regex=r'.h5ad$'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "aeaf4211",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold2\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold4\n",
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold1\n",
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold3\n",
      "scRNAseq melanoma fold4\n",
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold0\n",
      "scRNAseq pdac fold1\n",
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold4\n",
      "snATACseq brca fold0\n",
      "snATACseq brca fold1\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold3\n",
      "snATACseq brca fold4\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold1\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold4\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold4\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold1\n",
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold4\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for fp in fps:\n",
    "    root = fp.split('/')[-1].replace('.h5ad', '')\n",
    "    dtype, disease, fold = root.split('_')\n",
    "    print(dtype, disease, fold)\n",
    "    \n",
    "    adata = sc.read_h5ad(fp)\n",
    "\n",
    "    y_pred = adata.obs['predicted.id'].to_list()\n",
    "    y_true = adata.obs['cell_type'].to_list()\n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c318294e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "47fdbf3e",
   "metadata": {},
   "source": [
    "###### actinn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98257291",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "84b3c305",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "86ccf430",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "f970e151",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_cesc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_cesc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_cesc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_cesc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_cesc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_hnscc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_hnscc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_hnscc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_hnscc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_hnscc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_myeloma_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_myeloma_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_myeloma_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_myeloma_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_myeloma_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_pdac_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_pdac_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_pdac_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_pdac_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/scRNAseq_pdac_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snATACseq_gbm_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/actinn/interdataset_with_folds/snRNAseq_gbm_fold4.h5ad']"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles(result_dir, regex=r'.h5ad$'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "d95e22b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold2\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold4\n",
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq cesc fold2\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq myeloma fold0\n",
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n",
      "scRNAseq myeloma fold3\n",
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold0\n",
      "scRNAseq pdac fold1\n",
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold4\n",
      "snATACseq brca fold0\n",
      "snATACseq brca fold1\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold3\n",
      "snATACseq brca fold4\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold1\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold4\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold4\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold1\n",
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold4\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for fp in fps:\n",
    "    root = fp.split('/')[-1].replace('.h5ad', '')\n",
    "    dtype, disease, fold = root.split('_')\n",
    "    print(dtype, disease, fold)\n",
    "    \n",
    "    adata = sc.read_h5ad(fp)\n",
    "\n",
    "    y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "    y_true = adata.obs['cell_type'].to_list()\n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5678fb66",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be895541",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "c9f6bbb4",
   "metadata": {},
   "source": [
    "###### pollock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "de570c3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "result_dir = '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "cf27ca17",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_cesc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_cesc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_cesc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_cesc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_cesc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_hnscc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_hnscc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_hnscc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_hnscc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_hnscc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_melanoma_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_melanoma_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_melanoma_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_melanoma_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_melanoma_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_myeloma_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_myeloma_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_myeloma_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_myeloma_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_myeloma_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_pdac_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_pdac_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_pdac_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_pdac_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/scRNAseq_pdac_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snATACseq_gbm_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_brca_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_brca_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_brca_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_brca_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_brca_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_ccrcc_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_ccrcc_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_ccrcc_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_ccrcc_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_ccrcc_fold4.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_gbm_fold0.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_gbm_fold1.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_gbm_fold2.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_gbm_fold3.h5ad',\n",
       " '/data/pollock/benchmarking/results/pollock/interdataset_with_folds/snRNAseq_gbm_fold4.h5ad']"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fps = sorted(utils.listfiles(result_dir, regex=r'.h5ad$'))\n",
    "fps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "bf0e58cc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq brca fold0\n",
      "scRNAseq brca fold1\n",
      "scRNAseq brca fold2\n",
      "scRNAseq brca fold3\n",
      "scRNAseq brca fold4\n",
      "scRNAseq cesc fold0\n",
      "scRNAseq cesc fold1\n",
      "scRNAseq cesc fold2\n",
      "scRNAseq cesc fold3\n",
      "scRNAseq cesc fold4\n",
      "scRNAseq hnscc fold0\n",
      "scRNAseq hnscc fold1\n",
      "scRNAseq hnscc fold2\n",
      "scRNAseq hnscc fold3\n",
      "scRNAseq hnscc fold4\n",
      "scRNAseq melanoma fold0\n",
      "scRNAseq melanoma fold1\n",
      "scRNAseq melanoma fold2\n",
      "scRNAseq melanoma fold3\n",
      "scRNAseq melanoma fold4\n",
      "scRNAseq myeloma fold0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold1\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq myeloma fold4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n",
      "/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.\n",
      "  _warn_prf(average, modifier, msg_start, len(result))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "scRNAseq pdac fold0\n",
      "scRNAseq pdac fold1\n",
      "scRNAseq pdac fold2\n",
      "scRNAseq pdac fold3\n",
      "scRNAseq pdac fold4\n",
      "snATACseq brca fold0\n",
      "snATACseq brca fold1\n",
      "snATACseq brca fold2\n",
      "snATACseq brca fold3\n",
      "snATACseq brca fold4\n",
      "snATACseq ccrcc fold0\n",
      "snATACseq ccrcc fold1\n",
      "snATACseq ccrcc fold2\n",
      "snATACseq ccrcc fold3\n",
      "snATACseq ccrcc fold4\n",
      "snATACseq gbm fold0\n",
      "snATACseq gbm fold1\n",
      "snATACseq gbm fold2\n",
      "snATACseq gbm fold3\n",
      "snATACseq gbm fold4\n",
      "snRNAseq brca fold0\n",
      "snRNAseq brca fold1\n",
      "snRNAseq brca fold2\n",
      "snRNAseq brca fold3\n",
      "snRNAseq brca fold4\n",
      "snRNAseq ccrcc fold0\n",
      "snRNAseq ccrcc fold1\n",
      "snRNAseq ccrcc fold2\n",
      "snRNAseq ccrcc fold3\n",
      "snRNAseq ccrcc fold4\n",
      "snRNAseq gbm fold0\n",
      "snRNAseq gbm fold1\n",
      "snRNAseq gbm fold2\n",
      "snRNAseq gbm fold3\n",
      "snRNAseq gbm fold4\n"
     ]
    }
   ],
   "source": [
    "to_prediction = {}\n",
    "for fp in fps:\n",
    "    root = fp.split('/')[-1].replace('.h5ad', '')\n",
    "    dtype, disease, fold = root.split('_')\n",
    "    print(dtype, disease, fold)\n",
    "    \n",
    "    adata = sc.read_h5ad(fp)\n",
    "\n",
    "    y_pred = adata.obs['predicted_cell_type'].to_list()\n",
    "    y_true = adata.obs['cell_type'].to_list()\n",
    "    d, cm = get_metrics(y_true, y_pred)\n",
    "\n",
    "    d.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_clf_report.txt'), sep='\\t')\n",
    "    cm.to_csv(os.path.join(result_dir, f'{dtype}_{disease}_{fold}_confusion_matrix.txt'), sep='\\t')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14f1015c",
   "metadata": {},
   "source": [
    "#### multiclass roc for linear svm and pollock"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "e2fad757",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.metrics import roc_curve, auc, roc_auc_score\n",
    "from sklearn.preprocessing import label_binarize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "de5354f9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_type</th>\n",
       "      <th>barcode</th>\n",
       "      <th>sample</th>\n",
       "      <th>n_counts</th>\n",
       "      <th>size_factors</th>\n",
       "      <th>y_pred</th>\n",
       "      <th>predicted_cell_type_probability</th>\n",
       "      <th>predicted_cell_type</th>\n",
       "      <th>probability B cell</th>\n",
       "      <th>probability CD4 T cell</th>\n",
       "      <th>...</th>\n",
       "      <th>probability Dendritic</th>\n",
       "      <th>probability Endothelial</th>\n",
       "      <th>probability Erythrocyte</th>\n",
       "      <th>probability Fibroblast</th>\n",
       "      <th>probability Malignant</th>\n",
       "      <th>probability Mast</th>\n",
       "      <th>probability Monocyte</th>\n",
       "      <th>probability NK</th>\n",
       "      <th>probability Plasma</th>\n",
       "      <th>probability Treg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_AACCAACTCCACTAGA-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACCAACTCCACTAGA-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4712.0</td>\n",
       "      <td>1.213182</td>\n",
       "      <td>4</td>\n",
       "      <td>0.993027</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>0.000614</td>\n",
       "      <td>0.000489</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000855</td>\n",
       "      <td>0.993027</td>\n",
       "      <td>0.002185</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000936</td>\n",
       "      <td>0.000664</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>0.000299</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>0.000094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACCCAAAGACGAGCT-1</th>\n",
       "      <td>NK</td>\n",
       "      <td>AACCCAAAGACGAGCT-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4732.0</td>\n",
       "      <td>1.218332</td>\n",
       "      <td>10</td>\n",
       "      <td>0.995098</td>\n",
       "      <td>NK</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.000235</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001162</td>\n",
       "      <td>0.000163</td>\n",
       "      <td>0.000836</td>\n",
       "      <td>0.000269</td>\n",
       "      <td>0.000528</td>\n",
       "      <td>0.000188</td>\n",
       "      <td>0.000253</td>\n",
       "      <td>0.995098</td>\n",
       "      <td>0.000084</td>\n",
       "      <td>0.000575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACGTCAGTTTACACG-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACGTCAGTTTACACG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3294.0</td>\n",
       "      <td>0.848095</td>\n",
       "      <td>4</td>\n",
       "      <td>0.964709</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>0.004673</td>\n",
       "      <td>0.003166</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002698</td>\n",
       "      <td>0.964709</td>\n",
       "      <td>0.004423</td>\n",
       "      <td>0.003396</td>\n",
       "      <td>0.002712</td>\n",
       "      <td>0.004379</td>\n",
       "      <td>0.000911</td>\n",
       "      <td>0.001713</td>\n",
       "      <td>0.002370</td>\n",
       "      <td>0.001697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACTTCTGTCCAGAAG-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>AACTTCTGTCCAGAAG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3202.0</td>\n",
       "      <td>0.824408</td>\n",
       "      <td>2</td>\n",
       "      <td>0.939156</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>0.000410</td>\n",
       "      <td>0.018011</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002787</td>\n",
       "      <td>0.002750</td>\n",
       "      <td>0.009137</td>\n",
       "      <td>0.001392</td>\n",
       "      <td>0.002610</td>\n",
       "      <td>0.005852</td>\n",
       "      <td>0.002945</td>\n",
       "      <td>0.014044</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AAGCCATAGTGATTCC-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AAGCCATAGTGATTCC-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4088.0</td>\n",
       "      <td>1.052523</td>\n",
       "      <td>4</td>\n",
       "      <td>0.991898</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>0.001071</td>\n",
       "      <td>0.000682</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000795</td>\n",
       "      <td>0.991898</td>\n",
       "      <td>0.001467</td>\n",
       "      <td>0.000629</td>\n",
       "      <td>0.000686</td>\n",
       "      <td>0.001158</td>\n",
       "      <td>0.000116</td>\n",
       "      <td>0.000332</td>\n",
       "      <td>0.000332</td>\n",
       "      <td>0.000197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTATGCCAGCCCAGT-1</th>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>TTTATGCCAGCCCAGT-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4480.0</td>\n",
       "      <td>1.153450</td>\n",
       "      <td>6</td>\n",
       "      <td>0.946407</td>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>0.002872</td>\n",
       "      <td>0.004750</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003805</td>\n",
       "      <td>0.006681</td>\n",
       "      <td>0.005148</td>\n",
       "      <td>0.946407</td>\n",
       "      <td>0.003304</td>\n",
       "      <td>0.006503</td>\n",
       "      <td>0.000866</td>\n",
       "      <td>0.003830</td>\n",
       "      <td>0.006494</td>\n",
       "      <td>0.005201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCATGGTCGACTTA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCATGGTCGACTTA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4580.0</td>\n",
       "      <td>1.179197</td>\n",
       "      <td>12</td>\n",
       "      <td>0.995185</td>\n",
       "      <td>Treg</td>\n",
       "      <td>0.000739</td>\n",
       "      <td>0.001071</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000017</td>\n",
       "      <td>0.000058</td>\n",
       "      <td>0.000428</td>\n",
       "      <td>0.001124</td>\n",
       "      <td>0.000208</td>\n",
       "      <td>0.000238</td>\n",
       "      <td>0.000206</td>\n",
       "      <td>0.000412</td>\n",
       "      <td>0.995185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCCTCGTTAGCGGA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCCTCGTTAGCGGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4525.0</td>\n",
       "      <td>1.165036</td>\n",
       "      <td>12</td>\n",
       "      <td>0.980331</td>\n",
       "      <td>Treg</td>\n",
       "      <td>0.000945</td>\n",
       "      <td>0.001490</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.000246</td>\n",
       "      <td>0.001260</td>\n",
       "      <td>0.005305</td>\n",
       "      <td>0.000826</td>\n",
       "      <td>0.001646</td>\n",
       "      <td>0.000715</td>\n",
       "      <td>0.000591</td>\n",
       "      <td>0.980331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGACTCATGGGATG-1</th>\n",
       "      <td>Plasma</td>\n",
       "      <td>TTTGACTCATGGGATG-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4452.0</td>\n",
       "      <td>1.146241</td>\n",
       "      <td>11</td>\n",
       "      <td>0.998757</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0.000162</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.000225</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.998757</td>\n",
       "      <td>0.000162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGGTTCAGTTTCGA-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>TTTGGTTCAGTTTCGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>3414.0</td>\n",
       "      <td>0.878991</td>\n",
       "      <td>2</td>\n",
       "      <td>0.983060</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>0.000265</td>\n",
       "      <td>0.004784</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>0.000981</td>\n",
       "      <td>0.001974</td>\n",
       "      <td>0.000877</td>\n",
       "      <td>0.001331</td>\n",
       "      <td>0.001449</td>\n",
       "      <td>0.001938</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>0.001317</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5748 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         cell_type             barcode  sample  n_counts  \\\n",
       "0_AACCAACTCCACTAGA-1   Endothelial  AACCAACTCCACTAGA-1       0    4712.0   \n",
       "0_AACCCAAAGACGAGCT-1            NK  AACCCAAAGACGAGCT-1       0    4732.0   \n",
       "0_AACGTCAGTTTACACG-1   Endothelial  AACGTCAGTTTACACG-1       0    3294.0   \n",
       "0_AACTTCTGTCCAGAAG-1    CD8 T cell  AACTTCTGTCCAGAAG-1       0    3202.0   \n",
       "0_AAGCCATAGTGATTCC-1   Endothelial  AAGCCATAGTGATTCC-1       0    4088.0   \n",
       "...                            ...                 ...     ...       ...   \n",
       "29_TTTATGCCAGCCCAGT-1   Fibroblast  TTTATGCCAGCCCAGT-1      29    4480.0   \n",
       "29_TTTCATGGTCGACTTA-1         Treg  TTTCATGGTCGACTTA-1      29    4580.0   \n",
       "29_TTTCCTCGTTAGCGGA-1         Treg  TTTCCTCGTTAGCGGA-1      29    4525.0   \n",
       "29_TTTGACTCATGGGATG-1       Plasma  TTTGACTCATGGGATG-1      29    4452.0   \n",
       "29_TTTGGTTCAGTTTCGA-1   CD8 T cell  TTTGGTTCAGTTTCGA-1      29    3414.0   \n",
       "\n",
       "                       size_factors  y_pred  predicted_cell_type_probability  \\\n",
       "0_AACCAACTCCACTAGA-1       1.213182       4                         0.993027   \n",
       "0_AACCCAAAGACGAGCT-1       1.218332      10                         0.995098   \n",
       "0_AACGTCAGTTTACACG-1       0.848095       4                         0.964709   \n",
       "0_AACTTCTGTCCAGAAG-1       0.824408       2                         0.939156   \n",
       "0_AAGCCATAGTGATTCC-1       1.052523       4                         0.991898   \n",
       "...                             ...     ...                              ...   \n",
       "29_TTTATGCCAGCCCAGT-1      1.153450       6                         0.946407   \n",
       "29_TTTCATGGTCGACTTA-1      1.179197      12                         0.995185   \n",
       "29_TTTCCTCGTTAGCGGA-1      1.165036      12                         0.980331   \n",
       "29_TTTGACTCATGGGATG-1      1.146241      11                         0.998757   \n",
       "29_TTTGGTTCAGTTTCGA-1      0.878991       2                         0.983060   \n",
       "\n",
       "                      predicted_cell_type  probability B cell  \\\n",
       "0_AACCAACTCCACTAGA-1          Endothelial            0.000614   \n",
       "0_AACCCAAAGACGAGCT-1                   NK            0.000034   \n",
       "0_AACGTCAGTTTACACG-1          Endothelial            0.004673   \n",
       "0_AACTTCTGTCCAGAAG-1           CD8 T cell            0.000410   \n",
       "0_AAGCCATAGTGATTCC-1          Endothelial            0.001071   \n",
       "...                                   ...                 ...   \n",
       "29_TTTATGCCAGCCCAGT-1          Fibroblast            0.002872   \n",
       "29_TTTCATGGTCGACTTA-1                Treg            0.000739   \n",
       "29_TTTCCTCGTTAGCGGA-1                Treg            0.000945   \n",
       "29_TTTGACTCATGGGATG-1              Plasma            0.000250   \n",
       "29_TTTGGTTCAGTTTCGA-1          CD8 T cell            0.000265   \n",
       "\n",
       "                       probability CD4 T cell  ...  probability Dendritic  \\\n",
       "0_AACCAACTCCACTAGA-1                 0.000489  ...               0.000855   \n",
       "0_AACCCAAAGACGAGCT-1                 0.000235  ...               0.001162   \n",
       "0_AACGTCAGTTTACACG-1                 0.003166  ...               0.002698   \n",
       "0_AACTTCTGTCCAGAAG-1                 0.018011  ...               0.002787   \n",
       "0_AAGCCATAGTGATTCC-1                 0.000682  ...               0.000795   \n",
       "...                                       ...  ...                    ...   \n",
       "29_TTTATGCCAGCCCAGT-1                0.004750  ...               0.003805   \n",
       "29_TTTCATGGTCGACTTA-1                0.001071  ...               0.000006   \n",
       "29_TTTCCTCGTTAGCGGA-1                0.001490  ...               0.000018   \n",
       "29_TTTGACTCATGGGATG-1                0.000133  ...               0.000187   \n",
       "29_TTTGGTTCAGTTTCGA-1                0.004784  ...               0.000643   \n",
       "\n",
       "                       probability Endothelial  probability Erythrocyte  \\\n",
       "0_AACCAACTCCACTAGA-1                  0.993027                 0.002185   \n",
       "0_AACCCAAAGACGAGCT-1                  0.000163                 0.000836   \n",
       "0_AACGTCAGTTTACACG-1                  0.964709                 0.004423   \n",
       "0_AACTTCTGTCCAGAAG-1                  0.002750                 0.009137   \n",
       "0_AAGCCATAGTGATTCC-1                  0.991898                 0.001467   \n",
       "...                                        ...                      ...   \n",
       "29_TTTATGCCAGCCCAGT-1                 0.006681                 0.005148   \n",
       "29_TTTCATGGTCGACTTA-1                 0.000017                 0.000058   \n",
       "29_TTTCCTCGTTAGCGGA-1                 0.000062                 0.000246   \n",
       "29_TTTGACTCATGGGATG-1                 0.000019                 0.000010   \n",
       "29_TTTGGTTCAGTTTCGA-1                 0.000981                 0.001974   \n",
       "\n",
       "                       probability Fibroblast  probability Malignant  \\\n",
       "0_AACCAACTCCACTAGA-1                 0.000268               0.000936   \n",
       "0_AACCCAAAGACGAGCT-1                 0.000269               0.000528   \n",
       "0_AACGTCAGTTTACACG-1                 0.003396               0.002712   \n",
       "0_AACTTCTGTCCAGAAG-1                 0.001392               0.002610   \n",
       "0_AAGCCATAGTGATTCC-1                 0.000629               0.000686   \n",
       "...                                       ...                    ...   \n",
       "29_TTTATGCCAGCCCAGT-1                0.946407               0.003304   \n",
       "29_TTTCATGGTCGACTTA-1                0.000428               0.001124   \n",
       "29_TTTCCTCGTTAGCGGA-1                0.001260               0.005305   \n",
       "29_TTTGACTCATGGGATG-1                0.000162               0.000037   \n",
       "29_TTTGGTTCAGTTTCGA-1                0.000877               0.001331   \n",
       "\n",
       "                       probability Mast  probability Monocyte  probability NK  \\\n",
       "0_AACCAACTCCACTAGA-1           0.000664              0.000077        0.000299   \n",
       "0_AACCCAAAGACGAGCT-1           0.000188              0.000253        0.995098   \n",
       "0_AACGTCAGTTTACACG-1           0.004379              0.000911        0.001713   \n",
       "0_AACTTCTGTCCAGAAG-1           0.005852              0.002945        0.014044   \n",
       "0_AAGCCATAGTGATTCC-1           0.001158              0.000116        0.000332   \n",
       "...                                 ...                   ...             ...   \n",
       "29_TTTATGCCAGCCCAGT-1          0.006503              0.000866        0.003830   \n",
       "29_TTTCATGGTCGACTTA-1          0.000208              0.000238        0.000206   \n",
       "29_TTTCCTCGTTAGCGGA-1          0.000826              0.001646        0.000715   \n",
       "29_TTTGACTCATGGGATG-1          0.000042              0.000225        0.000015   \n",
       "29_TTTGGTTCAGTTTCGA-1          0.001449              0.001938        0.001314   \n",
       "\n",
       "                       probability Plasma  probability Treg  \n",
       "0_AACCAACTCCACTAGA-1             0.000135          0.000094  \n",
       "0_AACCCAAAGACGAGCT-1             0.000084          0.000575  \n",
       "0_AACGTCAGTTTACACG-1             0.002370          0.001697  \n",
       "0_AACTTCTGTCCAGAAG-1             0.000093          0.000811  \n",
       "0_AAGCCATAGTGATTCC-1             0.000332          0.000197  \n",
       "...                                   ...               ...  \n",
       "29_TTTATGCCAGCCCAGT-1            0.006494          0.005201  \n",
       "29_TTTCATGGTCGACTTA-1            0.000412          0.995185  \n",
       "29_TTTCCTCGTTAGCGGA-1            0.000591          0.980331  \n",
       "29_TTTGACTCATGGGATG-1            0.998757          0.000162  \n",
       "29_TTTGGTTCAGTTTCGA-1            0.000065          0.001317  \n",
       "\n",
       "[5748 rows x 21 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = sc.read_h5ad('/data/pollock/benchmarking/results/pollock/scRNAseq_brca.h5ad')\n",
    "a.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "ca8128c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_type</th>\n",
       "      <th>barcode</th>\n",
       "      <th>sample</th>\n",
       "      <th>n_counts</th>\n",
       "      <th>size_factors</th>\n",
       "      <th>predicted_cell_type</th>\n",
       "      <th>probability B cell</th>\n",
       "      <th>probability CD4 T cell</th>\n",
       "      <th>probability CD8 T cell</th>\n",
       "      <th>probability Dendritic</th>\n",
       "      <th>probability Endothelial</th>\n",
       "      <th>probability Erythrocyte</th>\n",
       "      <th>probability Fibroblast</th>\n",
       "      <th>probability Malignant</th>\n",
       "      <th>probability Mast</th>\n",
       "      <th>probability Monocyte</th>\n",
       "      <th>probability NK</th>\n",
       "      <th>probability Plasma</th>\n",
       "      <th>probability Treg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_AACCAACTCCACTAGA-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACCAACTCCACTAGA-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4712.0</td>\n",
       "      <td>1.213182</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-1.645374</td>\n",
       "      <td>-0.798354</td>\n",
       "      <td>-1.320047</td>\n",
       "      <td>-0.613273</td>\n",
       "      <td>9.651504</td>\n",
       "      <td>0.046297</td>\n",
       "      <td>-3.343092</td>\n",
       "      <td>-0.727528</td>\n",
       "      <td>-1.168746</td>\n",
       "      <td>-0.569176</td>\n",
       "      <td>-1.320954</td>\n",
       "      <td>-0.929309</td>\n",
       "      <td>-1.167010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACCCAAAGACGAGCT-1</th>\n",
       "      <td>NK</td>\n",
       "      <td>AACCCAAAGACGAGCT-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4732.0</td>\n",
       "      <td>1.218332</td>\n",
       "      <td>NK</td>\n",
       "      <td>-0.844792</td>\n",
       "      <td>-1.667611</td>\n",
       "      <td>-1.252446</td>\n",
       "      <td>-0.681986</td>\n",
       "      <td>-0.991126</td>\n",
       "      <td>-1.047975</td>\n",
       "      <td>-1.393238</td>\n",
       "      <td>0.279072</td>\n",
       "      <td>-0.840426</td>\n",
       "      <td>-0.061285</td>\n",
       "      <td>10.557428</td>\n",
       "      <td>-1.192751</td>\n",
       "      <td>-1.279777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACGTCAGTTTACACG-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACGTCAGTTTACACG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3296.0</td>\n",
       "      <td>0.848610</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-0.874611</td>\n",
       "      <td>-1.563723</td>\n",
       "      <td>-0.647155</td>\n",
       "      <td>-1.483668</td>\n",
       "      <td>7.808266</td>\n",
       "      <td>-0.778172</td>\n",
       "      <td>-0.275812</td>\n",
       "      <td>-0.957157</td>\n",
       "      <td>-0.471111</td>\n",
       "      <td>-0.729114</td>\n",
       "      <td>-1.426803</td>\n",
       "      <td>-0.044268</td>\n",
       "      <td>-0.892382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACTTCTGTCCAGAAG-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>AACTTCTGTCCAGAAG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3202.0</td>\n",
       "      <td>0.824408</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>-1.695771</td>\n",
       "      <td>-0.758672</td>\n",
       "      <td>1.230153</td>\n",
       "      <td>-1.391558</td>\n",
       "      <td>-1.419928</td>\n",
       "      <td>-0.788930</td>\n",
       "      <td>-0.927254</td>\n",
       "      <td>-0.453907</td>\n",
       "      <td>-0.026667</td>\n",
       "      <td>-0.981914</td>\n",
       "      <td>-0.534299</td>\n",
       "      <td>-1.087195</td>\n",
       "      <td>-2.054266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AAGCCATAGTGATTCC-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AAGCCATAGTGATTCC-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4088.0</td>\n",
       "      <td>1.052523</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-0.334221</td>\n",
       "      <td>-1.272656</td>\n",
       "      <td>-0.656085</td>\n",
       "      <td>-1.552842</td>\n",
       "      <td>14.972706</td>\n",
       "      <td>-0.857209</td>\n",
       "      <td>-1.206331</td>\n",
       "      <td>-0.683234</td>\n",
       "      <td>-0.559324</td>\n",
       "      <td>-1.068178</td>\n",
       "      <td>-0.820980</td>\n",
       "      <td>-1.125220</td>\n",
       "      <td>-1.229909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTATGCCAGCCCAGT-1</th>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>TTTATGCCAGCCCAGT-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4480.0</td>\n",
       "      <td>1.153450</td>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>-1.617480</td>\n",
       "      <td>-0.865152</td>\n",
       "      <td>-1.466225</td>\n",
       "      <td>-0.761909</td>\n",
       "      <td>-1.233621</td>\n",
       "      <td>-0.567406</td>\n",
       "      <td>4.051761</td>\n",
       "      <td>0.728402</td>\n",
       "      <td>-0.353512</td>\n",
       "      <td>-1.202108</td>\n",
       "      <td>-0.631813</td>\n",
       "      <td>-1.264545</td>\n",
       "      <td>-0.332282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCATGGTCGACTTA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCATGGTCGACTTA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4580.0</td>\n",
       "      <td>1.179197</td>\n",
       "      <td>Treg</td>\n",
       "      <td>-1.489965</td>\n",
       "      <td>-0.297774</td>\n",
       "      <td>-0.895083</td>\n",
       "      <td>-1.135407</td>\n",
       "      <td>-0.943314</td>\n",
       "      <td>-0.690525</td>\n",
       "      <td>-1.695587</td>\n",
       "      <td>-2.054170</td>\n",
       "      <td>-1.055325</td>\n",
       "      <td>-0.272783</td>\n",
       "      <td>-1.668621</td>\n",
       "      <td>-0.500289</td>\n",
       "      <td>6.091204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCCTCGTTAGCGGA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCCTCGTTAGCGGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4526.0</td>\n",
       "      <td>1.165293</td>\n",
       "      <td>Treg</td>\n",
       "      <td>-1.554138</td>\n",
       "      <td>-1.126189</td>\n",
       "      <td>0.067056</td>\n",
       "      <td>-1.275277</td>\n",
       "      <td>-1.274302</td>\n",
       "      <td>-1.661355</td>\n",
       "      <td>-1.579366</td>\n",
       "      <td>-0.442887</td>\n",
       "      <td>0.256902</td>\n",
       "      <td>-0.620646</td>\n",
       "      <td>-1.139950</td>\n",
       "      <td>-1.634987</td>\n",
       "      <td>5.346128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGACTCATGGGATG-1</th>\n",
       "      <td>Plasma</td>\n",
       "      <td>TTTGACTCATGGGATG-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4452.0</td>\n",
       "      <td>1.146241</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>-1.184166</td>\n",
       "      <td>-1.148166</td>\n",
       "      <td>-1.161676</td>\n",
       "      <td>-0.978821</td>\n",
       "      <td>-1.056848</td>\n",
       "      <td>-1.118930</td>\n",
       "      <td>-0.771188</td>\n",
       "      <td>-0.980366</td>\n",
       "      <td>-0.481189</td>\n",
       "      <td>-1.247963</td>\n",
       "      <td>-0.962314</td>\n",
       "      <td>9.834721</td>\n",
       "      <td>-1.319493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGGTTCAGTTTCGA-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>TTTGGTTCAGTTTCGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>3414.0</td>\n",
       "      <td>0.878991</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>-0.817018</td>\n",
       "      <td>-1.149525</td>\n",
       "      <td>1.255636</td>\n",
       "      <td>-1.025783</td>\n",
       "      <td>-0.896883</td>\n",
       "      <td>-0.954583</td>\n",
       "      <td>-1.481391</td>\n",
       "      <td>-0.913094</td>\n",
       "      <td>-0.755771</td>\n",
       "      <td>-0.679273</td>\n",
       "      <td>-1.546723</td>\n",
       "      <td>-0.758668</td>\n",
       "      <td>-1.101037</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5748 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         cell_type             barcode  sample  n_counts  \\\n",
       "0_AACCAACTCCACTAGA-1   Endothelial  AACCAACTCCACTAGA-1       0    4712.0   \n",
       "0_AACCCAAAGACGAGCT-1            NK  AACCCAAAGACGAGCT-1       0    4732.0   \n",
       "0_AACGTCAGTTTACACG-1   Endothelial  AACGTCAGTTTACACG-1       0    3296.0   \n",
       "0_AACTTCTGTCCAGAAG-1    CD8 T cell  AACTTCTGTCCAGAAG-1       0    3202.0   \n",
       "0_AAGCCATAGTGATTCC-1   Endothelial  AAGCCATAGTGATTCC-1       0    4088.0   \n",
       "...                            ...                 ...     ...       ...   \n",
       "29_TTTATGCCAGCCCAGT-1   Fibroblast  TTTATGCCAGCCCAGT-1      29    4480.0   \n",
       "29_TTTCATGGTCGACTTA-1         Treg  TTTCATGGTCGACTTA-1      29    4580.0   \n",
       "29_TTTCCTCGTTAGCGGA-1         Treg  TTTCCTCGTTAGCGGA-1      29    4526.0   \n",
       "29_TTTGACTCATGGGATG-1       Plasma  TTTGACTCATGGGATG-1      29    4452.0   \n",
       "29_TTTGGTTCAGTTTCGA-1   CD8 T cell  TTTGGTTCAGTTTCGA-1      29    3414.0   \n",
       "\n",
       "                       size_factors predicted_cell_type  probability B cell  \\\n",
       "0_AACCAACTCCACTAGA-1       1.213182         Endothelial           -1.645374   \n",
       "0_AACCCAAAGACGAGCT-1       1.218332                  NK           -0.844792   \n",
       "0_AACGTCAGTTTACACG-1       0.848610         Endothelial           -0.874611   \n",
       "0_AACTTCTGTCCAGAAG-1       0.824408          CD8 T cell           -1.695771   \n",
       "0_AAGCCATAGTGATTCC-1       1.052523         Endothelial           -0.334221   \n",
       "...                             ...                 ...                 ...   \n",
       "29_TTTATGCCAGCCCAGT-1      1.153450          Fibroblast           -1.617480   \n",
       "29_TTTCATGGTCGACTTA-1      1.179197                Treg           -1.489965   \n",
       "29_TTTCCTCGTTAGCGGA-1      1.165293                Treg           -1.554138   \n",
       "29_TTTGACTCATGGGATG-1      1.146241              Plasma           -1.184166   \n",
       "29_TTTGGTTCAGTTTCGA-1      0.878991          CD8 T cell           -0.817018   \n",
       "\n",
       "                       probability CD4 T cell  probability CD8 T cell  \\\n",
       "0_AACCAACTCCACTAGA-1                -0.798354               -1.320047   \n",
       "0_AACCCAAAGACGAGCT-1                -1.667611               -1.252446   \n",
       "0_AACGTCAGTTTACACG-1                -1.563723               -0.647155   \n",
       "0_AACTTCTGTCCAGAAG-1                -0.758672                1.230153   \n",
       "0_AAGCCATAGTGATTCC-1                -1.272656               -0.656085   \n",
       "...                                       ...                     ...   \n",
       "29_TTTATGCCAGCCCAGT-1               -0.865152               -1.466225   \n",
       "29_TTTCATGGTCGACTTA-1               -0.297774               -0.895083   \n",
       "29_TTTCCTCGTTAGCGGA-1               -1.126189                0.067056   \n",
       "29_TTTGACTCATGGGATG-1               -1.148166               -1.161676   \n",
       "29_TTTGGTTCAGTTTCGA-1               -1.149525                1.255636   \n",
       "\n",
       "                       probability Dendritic  probability Endothelial  \\\n",
       "0_AACCAACTCCACTAGA-1               -0.613273                 9.651504   \n",
       "0_AACCCAAAGACGAGCT-1               -0.681986                -0.991126   \n",
       "0_AACGTCAGTTTACACG-1               -1.483668                 7.808266   \n",
       "0_AACTTCTGTCCAGAAG-1               -1.391558                -1.419928   \n",
       "0_AAGCCATAGTGATTCC-1               -1.552842                14.972706   \n",
       "...                                      ...                      ...   \n",
       "29_TTTATGCCAGCCCAGT-1              -0.761909                -1.233621   \n",
       "29_TTTCATGGTCGACTTA-1              -1.135407                -0.943314   \n",
       "29_TTTCCTCGTTAGCGGA-1              -1.275277                -1.274302   \n",
       "29_TTTGACTCATGGGATG-1              -0.978821                -1.056848   \n",
       "29_TTTGGTTCAGTTTCGA-1              -1.025783                -0.896883   \n",
       "\n",
       "                       probability Erythrocyte  probability Fibroblast  \\\n",
       "0_AACCAACTCCACTAGA-1                  0.046297               -3.343092   \n",
       "0_AACCCAAAGACGAGCT-1                 -1.047975               -1.393238   \n",
       "0_AACGTCAGTTTACACG-1                 -0.778172               -0.275812   \n",
       "0_AACTTCTGTCCAGAAG-1                 -0.788930               -0.927254   \n",
       "0_AAGCCATAGTGATTCC-1                 -0.857209               -1.206331   \n",
       "...                                        ...                     ...   \n",
       "29_TTTATGCCAGCCCAGT-1                -0.567406                4.051761   \n",
       "29_TTTCATGGTCGACTTA-1                -0.690525               -1.695587   \n",
       "29_TTTCCTCGTTAGCGGA-1                -1.661355               -1.579366   \n",
       "29_TTTGACTCATGGGATG-1                -1.118930               -0.771188   \n",
       "29_TTTGGTTCAGTTTCGA-1                -0.954583               -1.481391   \n",
       "\n",
       "                       probability Malignant  probability Mast  \\\n",
       "0_AACCAACTCCACTAGA-1               -0.727528         -1.168746   \n",
       "0_AACCCAAAGACGAGCT-1                0.279072         -0.840426   \n",
       "0_AACGTCAGTTTACACG-1               -0.957157         -0.471111   \n",
       "0_AACTTCTGTCCAGAAG-1               -0.453907         -0.026667   \n",
       "0_AAGCCATAGTGATTCC-1               -0.683234         -0.559324   \n",
       "...                                      ...               ...   \n",
       "29_TTTATGCCAGCCCAGT-1               0.728402         -0.353512   \n",
       "29_TTTCATGGTCGACTTA-1              -2.054170         -1.055325   \n",
       "29_TTTCCTCGTTAGCGGA-1              -0.442887          0.256902   \n",
       "29_TTTGACTCATGGGATG-1              -0.980366         -0.481189   \n",
       "29_TTTGGTTCAGTTTCGA-1              -0.913094         -0.755771   \n",
       "\n",
       "                       probability Monocyte  probability NK  \\\n",
       "0_AACCAACTCCACTAGA-1              -0.569176       -1.320954   \n",
       "0_AACCCAAAGACGAGCT-1              -0.061285       10.557428   \n",
       "0_AACGTCAGTTTACACG-1              -0.729114       -1.426803   \n",
       "0_AACTTCTGTCCAGAAG-1              -0.981914       -0.534299   \n",
       "0_AAGCCATAGTGATTCC-1              -1.068178       -0.820980   \n",
       "...                                     ...             ...   \n",
       "29_TTTATGCCAGCCCAGT-1             -1.202108       -0.631813   \n",
       "29_TTTCATGGTCGACTTA-1             -0.272783       -1.668621   \n",
       "29_TTTCCTCGTTAGCGGA-1             -0.620646       -1.139950   \n",
       "29_TTTGACTCATGGGATG-1             -1.247963       -0.962314   \n",
       "29_TTTGGTTCAGTTTCGA-1             -0.679273       -1.546723   \n",
       "\n",
       "                       probability Plasma  probability Treg  \n",
       "0_AACCAACTCCACTAGA-1            -0.929309         -1.167010  \n",
       "0_AACCCAAAGACGAGCT-1            -1.192751         -1.279777  \n",
       "0_AACGTCAGTTTACACG-1            -0.044268         -0.892382  \n",
       "0_AACTTCTGTCCAGAAG-1            -1.087195         -2.054266  \n",
       "0_AAGCCATAGTGATTCC-1            -1.125220         -1.229909  \n",
       "...                                   ...               ...  \n",
       "29_TTTATGCCAGCCCAGT-1           -1.264545         -0.332282  \n",
       "29_TTTCATGGTCGACTTA-1           -0.500289          6.091204  \n",
       "29_TTTCCTCGTTAGCGGA-1           -1.634987          5.346128  \n",
       "29_TTTGACTCATGGGATG-1            9.834721         -1.319493  \n",
       "29_TTTGGTTCAGTTTCGA-1           -0.758668         -1.101037  \n",
       "\n",
       "[5748 rows x 19 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "b = sc.read_h5ad('/data/pollock/benchmarking/results/svm_linear/scRNAseq_brca.h5ad')\n",
    "b.obs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "030f26f3",
   "metadata": {},
   "source": [
    "For SVMrejection, a threshold of 0.7 was used on the posterior probabilities to assign cells as “unlabeled.” During the rejection experiments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e7ff66f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92539ccc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8526b886",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0ba0591",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dca3316",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e93af96",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "349640f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5be70090",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b08109b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61d8e077",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49e8a47f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee6836ff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb08ee8b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f691c1c2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bec3823d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdc98547",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "904015ee",
   "metadata": {},
   "outputs": [],
   "source": [
    "classes = sorted(set(a.obs['cell_type']))\n",
    "y_pred = [classes.index(l) for l in a.obs['predicted_cell_type']]\n",
    "y_score = a.obs[[c for c in a.obs.columns if 'probability ' in c]].values\n",
    "y_true = [classes.index(l) for l in a.obs['cell_type']]\n",
    "\n",
    "y_pred = label_binarize(y_pred, classes=np.arange(len(classes)))\n",
    "y_true = label_binarize(y_pred, classes=np.arange(len(classes)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "ec202f02",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Compute ROC curve and ROC area for each class\n",
    "fpr = dict()\n",
    "tpr = dict()\n",
    "roc_auc = dict()\n",
    "for i in range(len(classes)):\n",
    "    fpr[i], tpr[i], _ = roc_curve(y_true[:, i], y_score[:, i])\n",
    "    roc_auc[i] = auc(fpr[i], tpr[i])\n",
    "\n",
    "# Compute micro-average ROC curve and ROC area\n",
    "fpr[\"micro\"], tpr[\"micro\"], _ = roc_curve(y_true.ravel(), y_score.ravel())\n",
    "roc_auc[\"micro\"] = auc(fpr[\"micro\"], tpr[\"micro\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "e3af2cbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYoAAAEWCAYAAAB42tAoAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAAA8z0lEQVR4nO3dd3gU5fbA8e9JAkmAEKpIlSqJIE2qiiJFqqI/r2K5KFy9EpqKqHixYRcLRSkRQfHauNcGCCqKBbgoVUIREKIgBBGpoSUh5fz+mCEsIdksIZtNOZ/n2Sc7O+3MZHfOzPvOvK+oKsYYY0xOggIdgDHGmMLNEoUxxhivLFEYY4zxyhKFMcYYryxRGGOM8coShTHGGK8sURQTIvKziHQKdByBJiKxIvJYAa9zpog8U5Dr9BcRuU1EvsrjvMX2OygiKiINAx1HoIg9R5H/RGQ7UA1IB44CXwLDVPVoIOMqbkRkAHCXql4e4DhmAgmq+miA4xgDNFTVvxfAumZSCLa5oIiIAo1UNT7QsQSCXVH4zzWqWg5oAbQE/hXYcM6eiISUxHUHku1zUyipqr3y+QVsB7p6DL8IzPcYbg/8ABwC1gKdPMZVAt4C/gAOArM9xvUB4tz5fgCaZV0nUANIAip5jGsJ7ANKucP/ADa5y18AXOAxrQJDga3Athy271rgZzeO74HoLHH8C9joLv8tIOwstmEUsA5IAUKAh4FfgSPuMq93p40Gkjl11XbI/Xwm8Iz7vhOQAIwE/gJ2AwM91lcZ+Aw4DKwEngH+5+X/ernH/20nMMBjnZOB+W6cy4EGHvNNdKc/DKwGOnqMGwN8BLzrjr8LaAv86K5nNzAJKO0xTxPga+AAsAcYDfQATgCp7v5Y604bCcxwl7PL3cZgd9wAYCkw3l3WM+5n/3PHizvuLyDR/b80Be5213PCXddnWb/3QLAb18n/3Wqgdg77NdvfA3Apzve2tjvc3J0myh3O9ruRzbYdAn5zlzfA/V/8BdzhMf1MINbdr0eARZz5u2jovg8FXgZ2uPs/FggP9HHHr8e0QAdQHF9ZfjC1gPXARHe4JrAf6IVzRdfNHa7qjp8P/AeoCJQCrnQ/b+V+udu5P8I73PWEZrPOb4F/esTzEhDrvr8OiMc50IYAjwI/eEyr7o+lUnZffuBC4JgbdyngIXd5pT3i2ADUdpexlFMHbl+2Ic6dN9z97Eac5BcE9HPXXd0dN4AsB3bOTBRpwFNurL2A40BFd/ws91UGuAjnAJJtogDq4BxAbnGXVRlo4bHOAzgH+BDgPWCWx7x/d6cPwUlaf+ImT5xEker+X4KAcOASnINnCFAXJ6nf504fgXPQHwmEucPtPJb1bpa4ZwOvA2WB84AVwCCP/ZcGDHfXFc7piaI7zgG+Ak7SiPbY95n7OYfv/YM43/vG7rzNgcrZ7Nfcfg/P4nyfw3ES1TCPeXP7bqQBA3G+a8/gHNgn4xzor3b/n+U8tucIcIU7fiIe3wVOTxQTgLk43+8InJON5wN93PHrMS3QARTHl/uDOep+8RT4BqjgjhsFvJNl+gU4B83qQAbugSzLNFOBp7N89gunEonnj/Qu4Fv3veAcAK9wh78A7vRYRhDOwfMCd1iBzl627THgv1nm38Wps8DtQIzH+F7Ar2exDf/IZd/GAX3d9wPIPVEkASEe4//COQgH4xygG3uMy/GKAucq6dMcxs0EpmfZ5s1etuEg0Nx9PwZYnMs233dy3TiJak0O043BI1Hg1JOl4JHw3fm/89h/O7IsI3OfAp2BLe7+CsppP2f53p/8Dv5y8v+Uy7bl+Htw35fCSVbrcer65Cy+G1s9xl2M892u5vHZfk5P9p7JvRzO1erJqxkFGuL8no5x+hVjB3K4+i4uL6uj8J/rVDUC52AVBVRxP78AuFFEDp184RRpVMc5kz6gqgezWd4FwMgs89XGOaPK6iOgg4jUwDlDUmCJx3ImeizjAM6Xv6bH/Du9bFcN4PeTA6qa4U6f0/y/e8Toyzactm4RuV1E4jymb8qpfemL/aqa5jF8HOcgUBXnLNpzfd62uzZOMUdO/sxmHQCIyEgR2SQiie42RHL6NmTd5gtFZJ6I/Ckih4HnPKbPLQ5PF+AcaHd77L/Xca4ssl23J1X9FqfYazKwR0SmiUh5H9fta5zefg+oairOQbwp8Iq6R2bw6buxx+N9kru8rJ+V8xjO3Bfq3HhygDN/X1VxrkBXe6z3S/fzYssShZ+p6iKcL/rL7kc7cc6gKni8yqrqC+64SiJSIZtF7QSezTJfGVX9IJt1HgK+Am4CbgU+8PiB7cQpevBcTriq/uC5CC+b9AfOjxsAERGcg8Iuj2lqe7yv487j6zZ4HgguAN4AhuEUW1TAKdYSH+LMzV6coolaOcSd1U6gwdmuREQ64pw134RzpVgBp7xfPCbLuh1Tgc04d9mUxynrPzm9tziyLmcnzhVFFY/9XV5Vm3iZ5/QFqr6qqpfg1ItciFOklOt8ucSZdbqcfg+ISE3gCZy6rldEJNT9PLfvRl5k/v9FpBxO0dIfWabZh5NgmnjEG6nOjSvFliWKgjEB6CYiLXAqLa8Rke4iEiwiYSLSSURqqepunKKhKSJSUURKicgV7jLeAGJEpJ04yopIbxGJyGGd7wO3Aze470+KBf4lIk0ARCRSRG48i235L9BbRLqISCmcsvIUnMrIk4aKSC0RqYRzkPtPHrehLM4Baa8b60Ccs8aT9gC1RKT0WcQPgKqmA58AY0SkjIhE4eyvnLwHdBWRm0QkREQqu//P3ETgJKS9QIiIPA7kdlYegVOxfdSNa7DHuHnA+SJyn4iEikiEiLRzx+0B6opIkLuNu3FOGF4RkfIiEiQiDUTkSh/iRkTauP+rUjjFLSdvHji5rvpeZp8OPC0ijdz/dTMRqZzNdDn+HtyTkJk4lfF34tTNPO3Ol9t3Iy96icjl7vfpaWC5qp52xeVeQb8BjBeR89x11xSR7ue47kLNEkUBUNW9wL+Bx9wvXl+cA+henDOqBzn1v+iPU3a+Gac8/T53GauAf+IUBRzEqUAe4GW1c4FGwB5VXesRy6fAWGCWW6yxAeh5FtvyC07l7Gs4Z1fX4NwKfMJjsvdxDlC/ua9n8rINqroReAXnDqA9OOXMSz0m+Rbn7qs/RWSfr9vgYRhOMdCfwDvABzhJL7tYduDUPYzEKZKIw6mgzc0CnOS/BacYLhnvRVwAD+BcCR7BOSidTLSo6hGcCt9r3Li3Ale5oz90/+4XkZ/c97cDpTl1F9pHuMU6Pijvrv+gG/t+Tl0ZzwAucotfZmcz7zick4qvcJLeDJwK6dPk8nu4B6ee5TH3inggMFBEOvrw3ciL93GuXg7g3FBwWw7TjcL57i5zf0MLcSrtiy174M7kK3EeNrxLVRcGOpazJSJjgfNV9Y5Ax2IKlpSwBwjPll1RmBJLRKLcIhERkbY4xRufBjouYwobexLTlGQROMVNNXCK+V4B5gQ0ImMKISt6MsYY45UVPRljjPGqyBU9ValSRevWrRvoMIwxpkhZvXr1PlXN04OBRS5R1K1bl1WrVgU6DGOMKVJE5Pfcp8qeFT0ZY4zxyhKFMcYYryxRGGOM8coShTHGGK8sURhjjPHKEoUxxhiv/JYoRORNEflLRDbkMF5E5FURiReRdSLSyl+xGGOMyTt/XlHMxOnwPSc9cZrBboTTWftUP8ZijDEl1okT6blP5IXfHrhT1cUiUtfLJH2Bf7vtzC8TkQoiUt3tbMV3n/SGbZ+fQ6TGGFN8TVzSjunLz63AJpB1FDU5vQOXBE7vdzmTiNwtIqtEZNXevXtPjbAkYYwxXjWvvoeNe86tS+9ANuGRXd+22TZlq6rTgGkArVu31jMSRL1e8H/z/RGjMcYUKTt3JjJv3hYGD24DQCcg/qGD1K//VJ6XGchEkcDpndnX4syOzLNnScIYY06TlpbBq68u5/HHv+PYsVSaNj2Pjh0vAKBevYrntOxAJoq5wDARmQW0AxJ9qp84tPXU+5HWl4YxxixfnsCgQfNYu3YPADfcEE39+ueWHDz5LVGIyAc4Vz1VRCQBp9PyUgCqGgt8jtNZfTxwHKfj9NylHHb+1uuVzxEbY0zRcvBgEqNHf8Prr69GFerWrcCkST3p3fvCfF2PP+96uiWX8QoMzfMKrLjJGFPCPfnkImJjVxMSEsQDD3TgsceupEyZUvm+niLXH4UxxpRkaWkZhIQ4N6w++ugVbNt2iGef7UzTpuf5bZ3WhIcxxhQByclpPPnk97Rp80bmA3RVqpRhzpyb/ZokwK4ojDGm0Pvmm98YPHg+W7ceAGDBgniuuaZxga2/aCYKq8g2xpQAe/YcZeTIr3jvvfUAREdXYerU3lx5Zd0CjaNoJgqryDbGFHPvvruO4cO/4NChZMLCQnj88SsYOfJSSpcOLvBYimaiMMaYYi4jQzl0KJkePRoyeXKvfH0u4mxZojDGmELg6NET/PjjTrp1awBA//7NqFEjgi5d6iGSXYtHBcfuejLGmACbPXsz0dGTueaaD4iPdyqsRYSuXesHPEmAXVEYY0zA/P77Ie6550vmzv0FgNata5CSkhbgqM5kicIYYwpYamo6EyYsY8yYRRw/nkpERGmee64Lgwe3Jji48BX0WKIwxpgCds89XxAbuxqAm25qwvjx3alRIyLAUeXMEoUxxhSw++5rz6JFvzNuXHd69GgY6HByJU7bfEVH69qiq3YWrZiNMSWXqvLuu+v4/PN43n///zIrpzMylKCggquoFpHVqto6L/PaFYUxxvjJL7/sY/Dg+Xz33XbAueW1V69GAAWaJM6VJQpjjMlnSUmpPP/8/xg7diknTqRTuXI4r7xyNT17Fv5ipuxYojDGmHy0cOFvxMTM49dfDwJw550tGTu2K5UrlwlwZHlnicIYY/LRDz/s5NdfD9KkSVViY/tw+eV1Ah3SObNEYYwx5yA9PYP4+AM0blwFgFGjLqNKlTLcdVergDTg5w+F78kOY4wpItas2c2ll77J5Ze/xYEDSQCEhoYwZEibYpMkwBKFMcactSNHUhgx4ktat36DFSt2ERoazK+/Hgh0WH5jRU/GGOMjVeWTTzZx771fsmvXEYKChBEj2vPkk52IiAgNdHh+Y4nCGGN8dN99X/LqqysAaNOmBq+/3oeWLasHOCr/s6InY4zx0fXXRxMZGcrkyb348cc7S0SSAGvCwxhjcvS//+3gu++28dhjV2Z+dvhwCuXLF71iJmvCwxhj8tH+/ccZNWohM2asAaBLl/pcemltgCKZJM6VJQpjjHGpKv/+91oeeOBr9u07TqlSQTz88OW0bHl+oEMLKEsUxhgDbNq0l8GD57No0e8AXHVVXaZM6U1UVJUARxZ4liiMMQYYN+5HFi36napVyzBuXHduu+3iQtFfdWFgicIYU2IlJiYTGRkGwPPPd6Vs2dI8/viVVKoUHuDIChe7PdYYU+L88ccR+vX7iPbtZ3DiRDoAVaqUYcKEHpYksmGJwhhTYqSnZ/Daa8uJiprEf//7Mzt2JPLTT7sDHVahZ0VPxpgSYfXqPxg0aB6rVzuJ4dprG/Paaz2pUycywJEVfn69ohCRHiLyi4jEi8jD2YyPFJHPRGStiPwsIgP9GY8xpmQaM+Z72radzurVu6lduzyzZ/djzpybLUn4yG9XFCISDEwGugEJwEoRmauqGz0mGwpsVNVrRKQq8IuIvKeqJ/wVlzGm5KlfvyIiMHJkB8aM6US5cqUDHVKR4s+ip7ZAvKr+BiAis4C+gGeiUCBCnHvQygEHgDQ/xmSMKQF+++0gK1fuol+/pgD079+Mdu1qZnYuZM6OP4ueagI7PYYT3M88TQKigT+A9cC9qpqRdUEicreIrBKRVf4K1hhT9J04kc5zzy2hSZMp3HHHbOLjnT4iRMSSxDnw5xVFdk+qZG3NrzsQB3QGGgBfi8gSVT182kyq04Bp4DQKmP+hGmOKusWLfycmZh6bNu0D4LbbLi6R7TL5gz8TRQJQ22O4Fs6Vg6eBwAvqNGEbLyLbgChghR/jMsYUI/v2HefBB79m5sw4ABo1qsTUqb3p0qV+YAMrRvyZKFYCjUSkHrALuBm4Ncs0O4AuwBIRqQY0Bn7zY0zGmGImJmYeH3+8idDQYEaP7shDD11GWJjd+Z+f/LY3VTVNRIYBC4Bg4E1V/VlEYtzxscDTwEwRWY9TVDVKVff5KyZjTPGQkaEEBTml288+25mkpDQmTOhOo0aVAxxZ8WQdFxljiozjx1N5+ulFxMXt4fPPb7VG+86CdVxkjCn25s/fwrBhX7B9+yFEYMWKXbRrVyvQYZUIliiMMYVaQsJh7r33Sz75ZBMAzZtXIza2jyWJAmSJwhhTaE2ZspJRoxZy9OgJypYtxdNPX8Xw4e0ICbH2TAuSJQpjTKG1b99xjh49wfXXRzFxYg9q17a2mQLBEoUxptA4dCiZzZv30b69U6w0atRltG1bkx49GgY4spLNrt+MMQGnqsyatYHo6Mlce+0HHDiQBEBoaIgliULAEoUxJqDi4w/Qo8d73HLLx/z551EaNapMYmJyoMMyHqzoyRgTECkpabz44lKefXYJKSnpVKwYxosvduMf/2iZ+TCdKRx8ThQiUlZVj/kzGGNMydGv30fMmfMLALff3pyXXurGeeeVDXBUJju5Fj2JyKUishHY5A43F5Epfo/MGFOs3Xdfe6KiqvDtt7fz9tvXWZIoxHJtwkNElgN/A+aqakv3sw2q2rQA4juDNeFhTNGTkaG8+eYaNm3ayyuvdM/8PD09g+BgqyotCH5vwkNVd2ZpUyU9LyszxpQ869fvISZmPj/84PRjdvvtzWne/HwASxJFhC+JYqeIXAqoiJQG7sEthjLGmJwcO3aCJ59cxLhxP5Kerpx/fjkmTOhOs2bVAh2aOUu+JIoYYCJON6YJwFfAEH8GZYwp2j777BeGDfuCHTsSEYGhQ9vw7LOdiYwMC3RoJg98SRSNVfU2zw9E5DJgqX9CMsYUdbNnb2bHjkRatjyf11/vQ5s2NQMdkjkHviSK14BWPnxmjCmh0tIy2LXrMBdcUAGAsWO70bJldWJiWlsDfsVAjolCRDoAlwJVReR+j1HlcXqsM8YYli1LICZmHikp6axdG0Pp0sFUqVKGYcPaBjo0k0+8pfrSQDmcZBLh8TqMc7usMaYEO3gwicGD53HppTNYu3YPyclpbN9+KNBhGT/I8YpCVRcBi0Rkpqr+XoAxGWMKMVXlgw82MGLEAv766xghIUE8+OClPProFZQpUyrQ4Rk/8KWO4riIvAQ0ATJvWVDVzn6LyhhTaN122yd88MEGADp2rMPUqb1p0uS8AEdl/MmXWqb3gM1APeBJYDuw0o8xGWMKsR49GlK5cjhvvnkt338/wJJECeBLEx6rVfUSEVmnqs3czxap6pUFEmEW1oSHMQVr4cLf+PXXAwwa5LT+oKocPJhMpUrhAY7MnA1/N+GR6v7dLSK9gT8A69XcmGJuz56j3H//V7z//npCQ4Pp2rU+DRpUQkQsSZQwviSKZ0QkEhiJ8/xEeeA+fwZljAmcjAxl2rTVPPzwQhITUwgLC+Hxx6+w/qpLsFwTharOc98mAldB5pPZxphiZu3aPxk0aB7Ll+8CoGfPhkya1Iv69SsGODITSN4euAsGbsJp4+lLVd0gIn2A0UA40LJgQjTGFJSHHlrI8uW7qFEjgokTe3DDDdFkaTnalEDerihmALWBFcCrIvI70AF4WFVnF0Bsxhg/U1WOH0+lbNnSALz6ag9iY1fx5JNXUb58aICjM4VFjnc9icgGoJmqZohIGLAPaKiqfxZkgFnZXU/G5I/ffz/E8OFfcOxYKgsX9rcrh2LOX3c9nVDVDABVTRaRLYFOEsaYc5eams748ct48slFHD+eSkREabZuPcCFF1YOdGimkPKWKKJEZJ37XoAG7rAAevKZCmNM0bF06Q5iYuazYcNfAPTr14Rx47pTo0ZEgCMzhZm3RBFdYFEYY/xu+PDPmTTJaVShfv2KTJ7cix49GgY4KlMUeGsU0BoCNKYYqVq1LKVKBTFq1GWMHt2R8HBrwM/4JtcmPM5p4SI9cLpRDQamq+oL2UzTCZgAlAL25dY0iFVmG+ObzZv3sWNHIldf3QCAlJQ0tm07RFRUlQBHZgLB30145In7HMZkoBtOX9srRWSuqm70mKYCMAXooao7RMRaFzPmHCUlpfLcc0sYO3YpFSqEsXnzMCpVCic0NMSShMkTnxKFiIQDdVT1l7NYdlsgXlV/c5cxC+gLbPSY5lbgE1XdAaCqf53F8o0xWXz11a8MGTKfX389CMC11zbG7no15yrXZsZF5BogDvjSHW4hInN9WHZNYKfHcIL7macLgYoi8r2IrBaR232K2hhzmt27j3DzzR/Rvfu7/PrrQZo0qcqSJQOZPv1aKla0BvzMufHlimIMztXB9wCqGicidX2YL7vzmKyVCyHAJUAXnGZBfhSRZaq65bQFidwN3A1wibVba8wZ/u///suyZQmEh4cwZkwnRoxoT6lS1rW9yR++dFyUpqqJeVh2Ak4TICfVwmmiPOs0X6rqMVXdBywGmmddkKpOU9XWea2IMaY48rwR5YUXutCnz4Vs3DiUhx66zJKEyVe+JIoNInIrECwijUTkNeAHH+ZbCTQSkXoiUhq4GchaZDUH6CgiISJSBmgHbDqL+I0pcY4cSWHEiC8ZNGhe5mdXXlmXzz67hbp1KwQuMFNs+ZIohuP0l50CvI/T3Ph9uc2kqmnAMGABzsH/v6r6s4jEiEiMO80mnLqPdTiND05X1Q152A5jij1V5eOPNxIdPZkJE5bz1ltxbN9+KNBhmRLAl65QW6rqmgKKJ1f2HIUpibZtO8iwYV/w+edbAWjbtiaxsb1p2bJ6gCMzRYW/n6MYJyLVgQ+BWar6c15WZIw5e6rKiy8u5cknF5GUlEZkZCjPP9+Fu+++hOBgXwoEjDl3vvRwd5WInI/TidE0ESkP/EdVn/F7dMaUcCLCli37SUpK45ZbmjJuXHfOP79coMMyJcxZNeEhIhcDDwH9VLW036LywoqeTHG3b99x/vzzKE2bnpc5vGbNbrp1axDgyExRdi5FT748cBctImPcjowm4dzxZE8zGJPPVJWZM+OIiprEjTd+yIkT6QBUqVLGkoQJKF/qKN4CPgCuVtWsz0EYY/LBpk17iYmZz+LFTqPNzZufz8GDSVSrZsVMJvB8qaNoXxCBGFMSHT+eyrPPLuall34gNTWDqlXLMG5cd2677WLrmtQUGjkmChH5r6reJCLrOb3pDevhzph8oKp07vw2y5fvAmDQoEt4/vku1jaTKXS8XVHc6/7tUxCBGFPSiAhDhrTh+PFUXn+9Dx061M59JmMCwJcH7saq6qjcPisodteTKarS0zOYMmUlqakZ3H9/B8C5qkhLy7C2mYzf+fWuJ5yOh7LqmZeVGVNSrVr1B+3aTeeee75k9Ohv+OOPI4BzVWFJwhR23uooBgNDgPoiss5jVASw1N+BGVMcJCYm8+ij3zJ58kpUoXbt8rz2Wk9q1IgIdGjG+MxbHcX7wBfA88DDHp8fUdUDfo3KmCJOVfnww43cd9+X7N59lOBgYcSI9jzxRCfKlQvIs6rG5Jm3RKGqul1EhmYdISKVLFkY493rr69m9+6jtG9fi9jY3jRvfn6gQzImT3K7ougDrMa5Pdbzpm4F6vsxLmOKnJSUNA4dSqZatXKICFOm9OL777fzz39eQlCQPRNhiq6zauupMLC7nkxhtGjRdmJi5lOjRgQLF/a3h+VMoePvtp4uE5Gy7vu/i8g4EamTl5UZU9zs3XuMAQNm06nT22zevI+dOxPZs+dYoMMyJl/5cnvsVOC4iDTHaTn2d+Adv0ZlTCGXkaHMmPETUVGTefvttYSGBvPkk51Yt26wNQNuih1fGgVMU1UVkb7ARFWdISJ3+DswYworVaV793dZuPA3ALp2rc+UKb1o1KhygCMzxj98SRRHRORfQH+go4gEA6X8G5YxhZeI0LFjHdav38P48d25+eamVidhijVfmvA4H7gVWKmqS9z6iU6q+u+CCDArq8w2gTB//hZSUzO47roowLnDKSkpjQoVwgIcmTG+8Wuf2ar6p4i8B7QRkT7AikAlCWMKWkLCYe6990s++WQTVaqU4YorLqBSpXBCQ0MIDfXlgtyYos+Xu55uAlYAN+L0m71cRP7m78CMCaS0tAzGj/+R6OjJfPLJJsqWLcXo0ZdTvnxooEMzpsD5ckr0CNBGVf8CEJGqwELgI38GZkygrFixi0GD5hEX9ycA118fxcSJPahdOzLAkRkTGL4kiqCTScK1H99uqzWmyMnIUAYOnMPGjXupUyeSSZN6cs01jQMdljEB5Uui+FJEFuD0mw3QD/jcfyEZU7BUlZSUdMLCQggKEiZP7sUXX2zl8cevpGxZa8DPGJ+a8BCR/wMux2nvabGqfurvwHJidz2Z/BQff4AhQ+ZTu3Z5ZszoG+hwjPEbv9z1JCKNgJeBBsB64AFV3ZW3EI0pXFJS0hg7dinPPbeElJR0KlUK58UXj1O5cplAh2ZMoeOtruFNYB5wA04Lsq8VSETG+Nm3326jWbNYnnjie1JS0rnjjuZs3jzUkoQxOfBWRxGhqm+4738RkZ8KIiBj/CU9PYOBA+fwzjtOh42NG1cmNrYPnTrVDWxgxhRy3hJFmIi05FQ/FOGew6pqicMUKcHBQYSEBBEWFsKjj3bkgQcutYfmjPFBjpXZIvKdl/lUVTv7JyTvrDLbnI316/eQnJxGmzY1Adi//ziHDiXToEGlAEdmTMHyS2W2ql6V95CMCaxjx04wZsz3jB+/jEaNKrN2bQylSwdTuXIZq4sw5izZdbcpdubO/YXhw79gx45ERKBr13qkpqZTunRwoEMzpkjy6xPWItJDRH4RkXgRedjLdG1EJN3akDLnYseORK67bhZ9+85ix45EWrWqzooV/+S113rZg3PGnAO/XVG4/VZMBroBCcBKEZmrqhuzmW4ssMBfsZjiLz09g06dZrJt2yEiIkrzzDOdGTKkDSEh1tqMMecq10QhTo8stwH1VfUptz+K81V1RS6ztgXiVfU3dzmzgL7AxizTDQc+BtqcbfDGqCoiQnBwEGPGdOKzz7YwYUJ3atYsH+jQjCk2fDndmgJ0AG5xh4/gXCnkpiaw02M4wf0sk4jUBK4HYr0tSETuFpFVIrLKh/WaEuDgwSRiYubx3HNLMj/r378ZH354oyUJY/KZL0VP7VS1lYisAVDVgyLiS4Fvdn1DZr2vdQIwSlXTvXUlqarTgGng3B7rw7pNMaWqvP/+eu6//yv++usYERGlGTasLZGRYdYdqTF+4kuiSHXrERQy+6PI8GG+BKC2x3At4I8s07QGZrk/8CpALxFJU9XZPizflDBbtuxnyJD5fPPNNgA6dqzD1Km9iYy07kiN8SdfEsWrwKfAeSLyLPA34FEf5lsJNBKResAu4GacvrczqWq9k+9FZCYwz5KEySotLYNnnlnM88//jxMn0qlcOZyXXurGgAEt7CrCmALgS5/Z74nIaqALTnHSdaq6yYf50kRkGM7dTMHAm6r6s4jEuOO91ksYc1JwsLBkyQ5OnEjnH/9owdix3ahSxR6aM6ag5NofhXuX0xlUdYdfIsqFNeFRMuzZc5Tk5DQuuKACAFu37mf37qNcccUFgQ3MmCLKL014eJiPUz8hQBhQD/gFaJKXFRrjTUaGMm3aah5+eCGtW9fg66/7IyI0alSZRo0qBzo8Y0okX4qeLvYcFpFWwCC/RWRKrLi4P4mJmcfy5U7/WKVLB3P06AkiIkIDHJkxJdtZP5mtqj+JiD0cZ/LNkSMpPPHE90ycuJyMDKVGjQgmTuzBDTdEW2W1MYWAL09m3+8xGAS0Avb6LSJTopw4kU6rVtOIjz9AUJBw773teOqpqyhf3q4ijCksfLmiiPB4n4ZTZ/Gxf8IxJU3p0sH079+Mzz7bQmxsby65pEagQzLGZOH1rif3QbsXVPXBggvJO7vrqWhLTU1n/Phl1KkTyc03NwWcq4rgYKe9JmOMf/jlricRCXGfhWiV99CMOWXp0h3ExMxnw4a/qFq1DH36XEi5cqWtnwhjCjlvRU8rcOoj4kRkLvAhcOzkSFX9xM+xmWLiwIEkRo36munT1wBQv35FpkzpRbly1keEMUWBL3UUlYD9QGdOPU+hgCUK45Wq8s476xg58iv27TtOqVJBjBp1GaNHdyQ8vFSgwzPG+MhbojjPveNpA6cSxElWSWBylZqawfPP/499+45z5ZUXMHVqb6KjqwY6LGPMWfKWKIKBcvjWXLgxACQlpXLiRDqRkWGULh3MtGl9+O23g9x+e3N7JsKYIirHu55E5CdVLXQV2XbXU+G1YEE8Q4Z8TqdOFzBjRt9Ah2OM8eCvtp7s9M/4ZPfuI4wYsYD//OdnAMqWLcXx46mUKWP1EMYUB95uXO9SYFGYIik9PYNJk1YQFTWZ//znZ8LDQxg7tiurV99tScKYYiTHKwpVPVCQgZiiJTk5jSuueIuVK51OC/v0uZDXXutJ3boVAhuYMSbfnXWjgMYAhIWF0LTpeezefZRXX+3BdddFWWW1McWUJQrjE1Xlk082Ua1aOS6/3OnLaty47gQHizUDbkwxZ4nC5GrbtoMMG/YFn3++laioKsTFDSI0NIQKFcICHZoxpgBYojA5OnEinVde+YGnn15MUlIakZGh3HtvO0JCrPE+Y0oSSxQmW0uW/E5MzHw2bnS6Hrn11ot55ZWrOf/8cgGOzBhT0CxRmDMkJaXyt799yF9/HaNhw0pMmdKLbt0aBDosY0yAWKIwgFNZnZ6uhIQEER5einHjrmbLlv38618dCQuzr4kxJZkdAQwbN+4lJmYe3brV57HHrgTgttuaBTgqY0xhYbWSJdjx46mMHv0NzZvHsmTJDqZPX0NKSlqgwzLGFDJ2RVFCffHFVoYO/Zxt2w4BMGjQJTz/fBdCQ+0rYYw5nR0VSphjx04wYMAcPvpoIwDNmlUjNrY3HTrUDnBkxpjCyhJFCVOmTCkOHEiibNlSPPlkJ+69t709F2GM8coSRQmwatUfVKgQRsOGlRARpk+/huDgIOrUiQx0aMaYIsBOJYuxxMRkhg//nLZt3yAmZh4nO6mqV6+iJQljjM/siqIYUlX++9+fue++Bfz551GCg4VWraqTlpZBqVLBgQ7PGFPEWKIoZn799QBDh37OggW/AtChQy1iY/vQrFm1AEdmjCmqLFEUI0eOpNC69RscOpRMhQphjB3blbvuakVQkPUTYYzJO78mChHpAUwEgoHpqvpClvG3AaPcwaPAYFVd68+YirOIiFBGjGhPfPwBXn75as47r2ygQzLGFANysoIz3xcsEgxsAboBCcBK4BZV3egxzaXAJlU9KCI9gTGq2s7bclvXFl210z8xFzV79x7jwQe/pkuXevTv3xxw6iespzljTFYislpVW+dlXn/e9dQWiFfV31T1BDAL6Os5gar+oKoH3cFlQC0/xlNsZGQo06f/ROPGk3j77bU88si3pKamA1iSMMbkO38miprATo/hBPeznNwJfJHdCBG5W0RWiciqfIyvSNqw4S+uuOIt/vnPzzh4MJmuXevzzTe3291Mxhi/8WcdRXanttmWGYnIVTiJ4vLsxqvqNGAaOEVP+RVgUZKUlMqYMd8zbtwy0tIyqFatLOPHd+fmm5vaVYQxxq/8mSgSAM8GhGoBf2SdSESaAdOBnqq634/xFGlBQcLcuVtIT89gyJDWPPtsF+uz2hhTIPyZKFYCjUSkHrALuBm41XMCEakDfAL0V9UtfoylSEpIOEyZMqWoVCmc0NAQZs50qnjatbOqHGNMwfFbHYWqpgHDgAXAJuC/qvqziMSISIw72eNAZWCKiMRZHYQjLS2D8eN/JDp6Mg8++FXm5+3a1bIkYYwpcH59jkJVPwc+z/JZrMf7u4C7/BlDUbN8eQKDBs1j7do9ACQmppCWlmEtvBpjAsaezC4kDh1KZvTob4iNXYUqXHBBJJMm9aJPnwsDHZoxpoSzRFEIHDyYxEUXTeHPP48SEhLEyJEdeOyxKyhbtnSgQzPGGEsUhUHFiuH07NmQLVv2M3Vqby6+2BrwM8YUHpYoAiAlJY2xY5dy5ZUXcOWVdQGYNKkXYWEh1oCfMabQsURRwL79dhuDB89ny5b9REdXYf36wQQHB1GmTKlAh2aMMdmyRFFA/vrrGCNHfsW7764DICqqClOm9CY42O5mMsYUbpYo/OxkA36jRi3k0KFkwsJCePTRjjz44GWULm3tMxljCj9LFH6WmJjMI498y6FDyXTv3oDJk3vRoEGlQIdljDE+s0ThB8eOnSAkJIjQ0BAqVgwnNrY36enKjTdeZA34GWOKHCsgz2dz5/7CRRdN4cUXl2Z+dsMNF3HTTU0sSRhjiiRLFPlkx45ErrtuFn37zmLHjkQWLPiVjIwS2SK6MaaYsURxjlJT03n55R+Ijp7MnDm/EBFRmokTe7Bo0QB7JsIYUyxYHcU52LfvOF26/Jt165wG/G688SLGj+9OzZrlAxyZMcbkH0sU56By5XCqVClDvXoVmDSpF716NQp0SKYQSU1NJSEhgeTk5ECHYkqQsLAwatWqRalS+fcQryWKs6CqvPfeetq2rcmFF1ZGRHj33euJjAyzJ6vNGRISEoiIiKBu3bp2I4MpEKrK/v37SUhIoF69evm2XKuj8NEvv+yja9d36N//U4YMmY+qU1FdvXqEJQmTreTkZCpXrmxJwhQYEaFy5cr5fhVrVxS5SE5O4/nnl/DCC0s5cSKdypXD+fvfmwU6LFNEWJIwBc0f3zlLFF4sXPgbgwfPJz7+AAD/+EcLXnyxG5UrlwlwZMYYU3Cs6CkHe/YcpU+f94mPP8BFF1Vl8eIBzJjR15KEKVKCg4Np0aIFTZs25ZprruHQoUOZ437++Wc6d+7MhRdeSKNGjXj66aczi1QBvvjiC1q3bk10dDRRUVE88MADAdgC79asWcNddxXe3pQXL15Mq1atCAkJ4aOPPspxutWrV3PxxRfTsGFD7rnnnsz/Q0pKCv369aNhw4a0a9eO7du3A7B371569OhREJsAWKI4TUaGZv6DqlUrx1NPXcXzz3dhzZpBdOx4QYCjM+bshYeHExcXx4YNG6hUqRKTJ08GICkpiWuvvZaHH36YLVu2sHbtWn744QemTJkCwIYNGxg2bBjvvvsumzZtYsOGDdSvXz9fY0tLSzvnZTz33HMMHz68QNd5NurUqcPMmTO59dZbvU43ePBgpk2bxtatW9m6dStffvklADNmzKBixYrEx8czYsQIRo0aBUDVqlWpXr06S5cu9bbYfGNFT664uD+JiZnH0KFt6N+/OQAPPXRZgKMyxcYrfqqrGOn70/8dOnRg3Tqnmfv333+fyy67jKuvvhqAMmXKMGnSJDp16sTQoUN58cUXeeSRR4iKigIgJCSEIUOGnLHMo0ePMnz4cFatWoWI8MQTT3DDDTdQrlw5jh49CsBHH33EvHnzmDlzJgMGDKBSpUqsWbOGFi1a8OmnnxIXF0eFChUAaNiwIUuXLiUoKIiYmBh27NgBwIQJE7jsstN/j0eOHGHdunU0b+78XlesWMF9991HUlIS4eHhvPXWWzRu3JiZM2cyf/58kpOTOXbsGJ999hnDhw9n/fr1pKWlMWbMGPr27cv27dvp378/x44dA2DSpElceumlPu/f7NStWxeAoKCcz8l3797N4cOH6dChAwC33347s2fPpmfPnsyZM4cxY8YA8Le//Y1hw4ahqogI1113He+9994Z+8UfSnyiOHIkhSee+J6JE5eTkaGkpKTz9783s0pIU6ykp6fzzTffcOeddwJOsdMll1xy2jQNGjTg6NGjHD58mA0bNjBy5Mhcl/v0008TGRnJ+vXrATh48GCu82zZsoWFCxcSHBxMRkYGn376KQMHDmT58uXUrVuXatWqceuttzJixAguv/xyduzYQffu3dm0adNpy1m1ahVNmzbNHI6KimLx4sWEhISwcOFCRo8ezccffwzAjz/+yLp166hUqRKjR4+mc+fOvPnmmxw6dIi2bdvStWtXzjvvPL7++mvCwsLYunUrt9xyC6tWrToj/o4dO3LkyJEzPn/55Zfp2rVrrtuf1a5du6hVq1bmcK1atdi1a1fmuNq1awNOso6MjGT//v1UqVKF1q1b8+ijj571+vKixCYKVWX27M3cc8+XJCQcJihIuPfedjz11FWWJEz+O4sz//yUlJREixYt2L59O5dccgndunUDyDwrzc7ZfP8XLlzIrFmzMocrVqyY6zw33ngjwcFOXyz9+vXjqaeeYuDAgcyaNYt+/fplLnfjxo2Z8xw+fJgjR44QERGR+dnu3bupWrVq5nBiYiJ33HEHW7duRURITU3NHNetWzcqVXKa9//qq6+YO3cuL7/8MuDcxrxjxw5q1KjBsGHDiIuLIzg4mC1btmQb/5IlS3LdxrPhWS900sn/gbdx5513Hn/88Ue+xpKTEpko9u07zsCBc5g3z/kitG5dg9df70OrVtUDHJkx+etkHUViYiJ9+vRh8uTJ3HPPPTRp0oTFixefNu1vv/1GuXLliIiIoEmTJqxevTqzWCcnOSUcz8+y3tNftmzZzPcdOnQgPj6evXv3Mnv27Mwz5IyMDH788UfCw8O9bpvnsh977DGuuuoqPv30U7Zv306nTp2yXaeq8vHHH9O4cePTljdmzBiqVavG2rVrycjIICwsLNv15vcVRa1atUhISMgcTkhIoEaNGpnjdu7cSa1atUhLSyMxMTEz4SUnJ3vdP/mpRFZmR0SUJj7+AOXLhzJpUk+WLbvTkoQp1iIjI3n11Vd5+eWXSU1N5bbbbuN///sfCxcuBJwrj3vuuYeHHnoIgAcffJDnnnsu86w6IyODcePGnbHcq6++mkmTJmUOnyx6qlatGps2bcosWsqJiHD99ddz//33Ex0dTeXKlbNdblxc3BnzRkdHEx8fnzmcmJhIzZo1AZg5c2aO6+zevTuvvfZa5tn6mjVrMuevXr06QUFBvPPOO6Snp2c7/5IlS4iLizvjlZckAVC9enUiIiJYtmwZqsq///1v+vbtC8C1117L22+/DTh1PZ07d85Mwlu2bDmt6M2fSkyiWLp0B/v3HwcgNDSEWbNuYPPmoQwd2tb6rTYlQsuWLWnevDmzZs0iPDycOXPm8Mwzz9C4cWMuvvhi2rRpw7BhwwBo1qwZEyZM4JZbbiE6OpqmTZuye/fuM5b56KOPcvDgQZo2bUrz5s357rvvAHjhhRfo06cPnTt3pnp17ydh/fr14913380sdgJ49dVXWbVqFc2aNeOiiy4iNjb2jPmioqJITEzMPLt/6KGH+Ne//sVll12W40EenCuP1NRUmjVrRtOmTXnssccAGDJkCG+//Tbt27dny5Ytp12F5NXKlSupVasWH374IYMGDaJJkyaZ41q0aJH5furUqdx11100bNiQBg0a0LNnTwDuvPNO9u/fT8OGDRk3bhwvvPBC5jzfffcdvXv3PucYfSHZlYEVZq1ri67a6XvM+/cf5+GHFzJ9+hruvLMl06df68fojDll06ZNREdHBzqMYm38+PFEREQU6mcp/OWKK65gzpw52dYLZffdE5HVqto6L+sqtqfSqsrbb8cRFTWZ6dPXUKpUEDVqRGRbOWSMKZoGDx5MaGhooMMocHv37uX+++/36eaB/FAsK7M3b95HTMw8Fi36HYBOneoydWpvoqKqBDgyY0x+CgsLo3///oEOo8BVrVqV6667rsDWV+wSRULCYZo3j+XEiXSqVCnDK69cTf/+9lyECQxvt6Ea4w/+KDUpdomiVq3y9O/fjKAg4YUXulKpUsHcPmZMVmFhYezfv9+aGjcF5mR/FDnd2ptXRb4ye/fuI4wYsYCYmNZ06lQXcNpssv6qTaBZD3cmEHLq4e5cKrOL7BVFenoGU6eu4pFHvuXw4RTi4w+wcuU/ERFLEqZQKFWqVL72MmZMoPj1ricR6SEiv4hIvIg8nM14EZFX3fHrRKSVL8v96afdtG8/g+HDv+Dw4RSuueZCPv74Jru8N8YYP/DbFYWIBAOTgW5AArBSROaq6kaPyXoCjdxXO2Cq+zdHOw+Vp02bN8jIUGrVKs9rr/Wkb9/GliSMMcZP/HlF0RaIV9XfVPUEMAvom2WavsC/1bEMqCAiXh/jPHA8HBG4//72bNo0lOuui7IkYYwxfuTPOoqawE6P4QTOvFrIbpqawGltBYjI3cDd7mAKPLFh3DjIpumZkqYKsC/QQRQSti9OsX1xiu2LUxrnPkn2/JkosjvNz3qLlS/ToKrTgGkAIrIqrzX3xY3ti1NsX5xi++IU2xeniMiZnWv4yJ9FTwlAbY/hWkDWxtN9mcYYY0wA+TNRrAQaiUg9ESkN3AzMzTLNXOB29+6n9kCiqp7ZRKUxxpiA8VvRk6qmicgwYAEQDLypqj+LSIw7Phb4HOgFxAPHgYE+LHqan0IuimxfnGL74hTbF6fYvjglz/uiyD2ZbYwxpmAV22bGjTHG5A9LFMYYY7wqtInCX81/FEU+7Ivb3H2wTkR+EJHmgYizIOS2LzymayMi6SLyt4KMryD5si9EpJOIxInIzyKyqKBjLCg+/EYiReQzEVnr7gtf6kOLHBF5U0T+EpENOYzP23FTVQvdC6fy+1egPlAaWAtclGWaXsAXOM9itAeWBzruAO6LS4GK7vueJXlfeEz3Lc7NEn8LdNwB/F5UADYCddzh8wIddwD3xWhgrPu+KnAAKB3o2P2wL64AWgEbchifp+NmYb2i8EvzH0VUrvtCVX9Q1YPu4DKc51GKI1++FwDDgY+BvwoyuALmy764FfhEVXcAqGpx3R++7AsFIsRp76ccTqJIK9gw/U9VF+NsW07ydNwsrIkip6Y9znaa4uBst/NOnDOG4ijXfSEiNYHrgdgCjCsQfPleXAhUFJHvRWS1iNxeYNEVLF/2xSQgGueB3vXAvaqaUTDhFSp5Om4W1v4o8q35j2LA5+0UkatwEsXlfo0ocHzZFxOAUaqaXswbi/RlX4QAlwBdgHDgRxFZpqpb/B1cAfNlX3QH4oDOQAPgaxFZoqqH/RxbYZOn42ZhTRTW/McpPm2niDQDpgM9VXV/AcVW0HzZF62BWW6SqAL0EpE0VZ1dIBEWHF9/I/tU9RhwTEQWA82B4pYofNkXA4EX1CmojxeRbUAUsKJgQiw08nTcLKxFT9b8xym57gsRqQN8AvQvhmeLnnLdF6paT1Xrqmpd4CNgSDFMEuDbb2QO0FFEQkSkDE7rzZsKOM6C4Mu+2IFzZYWIVMNpSfW3Ao2ycMjTcbNQXlGo/5r/KHJ83BePA5WBKe6ZdJoWwxYzfdwXJYIv+0JVN4nIl8A6IAOYrqrZ3jZZlPn4vXgamCki63GKX0aparFrflxEPgA6AVVEJAF4AigF53bctCY8jDHGeFVYi56MMcYUEpYojDHGeGWJwhhjjFeWKIwxxnhlicIYY4xXlihMoeS2/Brn8arrZdqj+bC+mSKyzV3XTyLSIQ/LmC4iF7nvR2cZ98O5xugu5+R+2eC2hlohl+lbiEiv/Fi3Kbns9lhTKInIUVUtl9/TelnGTGCeqn4kIlcDL6tqs3NY3jnHlNtyReRtYIuqPutl+gFAa1Udlt+xmJLDrihMkSAi5UTkG/dsf72InNFqrIhUF5HFHmfcHd3PrxaRH915PxSR3A7gi4GG7rz3u8vaICL3uZ+VFZH5bt8GG0Skn/v59yLSWkReAMLdON5zxx11//7H8wzfvZK5QUSCReQlEVkpTj8Bg3zYLT/iNugmIm3F6Ytkjfu3sfuU8lNAPzeWfm7sb7rrWZPdfjTmDIFuP91e9sruBaTjNOIWB3yK04pAeXdcFZwnS09eER91/44EHnHfBwMR7rSLgbLu56OAx7NZ30zcviuAG4HlOA3qrQfK4jRN/TPQErgBeMNj3kj37/c4Z++ZMXlMczLG64G33felcVryDAfuBh51Pw8FVgH1sonzqMf2fQj0cIfLAyHu+67Ax+77AcAkj/mfA/7uvq+A0+5T2UD/v+1VuF+FsgkPY4AkVW1xckBESgHPicgVOM1R1ASqAX96zLMSeNOddraqxonIlcBFwFK3eZPSOGfi2XlJRB4F9uK0wtsF+FSdRvUQkU+AjsCXwMsiMhanuGrJWWzXF8CrIhIK9AAWq2qSW9zVTE71yBcJNAK2ZZk/XETigLrAauBrj+nfFpFGOK2Blsph/VcD14rIA+5wGFCH4tkGlMknlihMUXEbTs9kl6hqqohsxznIZVLVxW4i6Q28IyIvAQeBr1X1Fh/W8aCqfnRyQES6ZjeRqm4RkUtw2sx5XkS+UtWnfNkIVU0Wke9xmr3uB3xwcnXAcFVdkMsiklS1hYhEAvOAocCrOG0Zfaeq17sV/9/nML8AN6jqL77EawxYHYUpOiKBv9wkcRVwQdYJROQCd5o3gBk4XUIuAy4TkZN1DmVE5EIf17kYuM6dpyxOsdESEakBHFfVd4GX3fVklepe2WRnFk5jbB1xGrLD/Tv45DwicqG7zmypaiJwD/CAO08ksMsdPcBj0iM4RXAnLQCGi3t5JSItc1qHMSdZojBFxXtAaxFZhXN1sTmbaToBcSKyBqceYaKq7sU5cH4gIutwEkeULytU1Z9w6i5W4NRZTFfVNcDFwAq3COgR4JlsZp8GrDtZmZ3FVzh9Gy9Up+tOcPoS2Qj8JCIbgNfJ5YrfjWUtTrPaL+Jc3SzFqb846TvgopOV2ThXHqXc2Da4w8Z4ZbfHGmOM8cquKIwxxnhlicIYY4xXliiMMcZ4ZYnCGGOMV5YojDHGeGWJwhhjjFeWKIwxxnj1/+4WeUErrenSAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "plt.figure()\n",
    "lw = 2\n",
    "plt.plot(\n",
    "    fpr[2],\n",
    "    tpr[2],\n",
    "    color=\"darkorange\",\n",
    "    lw=lw,\n",
    "    label=\"ROC curve (area = %0.2f)\" % roc_auc[2],\n",
    ")\n",
    "plt.plot([0, 1], [0, 1], color=\"navy\", lw=lw, linestyle=\"--\")\n",
    "plt.xlim([0.0, 1.0])\n",
    "plt.ylim([0.0, 1.05])\n",
    "plt.xlabel(\"False Positive Rate\")\n",
    "plt.ylabel(\"True Positive Rate\")\n",
    "plt.title(\"Receiver operating characteristic example\")\n",
    "plt.legend(loc=\"lower right\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "1d7af004",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAaAAAAEWCAYAAAAgpUMxAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjUuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/YYfK9AAAACXBIWXMAAAsTAAALEwEAmpwYAACPSElEQVR4nOydd3hUxdfHv7PZVNIT0kmH9GwqTZASQKQjVREIoggoqAgCKkVsvAhWxK6oICAoTRH8UaUpCSW0QEhIJb1t6mazu+f9Y5NlN9k0SLIJzOd57gN379y5585u7rkzc+Z8GRGBw+FwOJz2RqBrAzgcDofzcMIdEIfD4XB0AndAHA6Hw9EJ3AFxOBwORydwB8ThcDgcncAdEIfD4XB0glDXBjxsMMb+ArCdiH5s5+u+A2AuABkROTT3vPPnz9sJhcJvAQSiFV9YqqqqjMRisY2dnd2d1qqzs5Kdne1qa2ubKRQKZe11zaysLLeuXbveac9rthUFBQX2xsbG5SYmJmUtOe9B+A1WVVUZFRcX29rb22fUfKQAcFUmkz0bHh6eyxhzBXAdgAURyRuqhzE2EMAWInJpc6PVuGcHxBjrB2AdgAAAcgDxAF4mophWsq1dYYylAHiWiA635XWI6PG2rF8bjLFuAF4F4EZEuVqODwRwFEAFAAKQCWAtEf0gFAq/dXBw8OvatWuRQCBo7UVjGU0XebCIj4/3sbKyKnBwcMiv/UwkEuU3dk5bUF1d7ebv719kbGxc1d7XVicuLi7I1dU1xcrKqvQ+qmlW+8XGxoYHBARcrXPPLf4NpqenO1VVVRl6e3snt/Tc1qa4uNgsJSXFuvY3dOnSpSALCwvDkpKSbwGMIaI0AKa6tbJh7umNljFmDuAPAJ8BsAbgDOAtADr9MXMaxA1AgTbno0YmEZkCMAfwCoBvGGM+AAK7du1a0gbOp01QKBQP5bV1hS7vmYjAF9JrwhiDhYVFGZQjFh2ee+0B9QAAItpWs18J4O/ag4wxAYDXATwHwBjAQQALiEjMGHMHkAzgGQBroPTOywGcB/AdAFcou4IvqtX3DIAlABwAnAMwh4hStRnGGOsN4EMA/gBSAbxERMcZY30B7AMQSkTpjDERgBMAegN4o+a6+xljcgBriGhdQ3XVXOc4gJMABgMIBnAWwFNElM8YMwLwLYDHAegBuAVgFBHl1Jy3hYi+bWY7RQN4G4AJgI+I6N0G7tsCyheCx6HsyXwD4L0a+/YDMGSMlQHYRUTR2uoAAFL+RR9gjBXW3JeAMUYZGRkOhYWFtnK5XGhqalri7u6eqq+vLwcAsVhseufOHReJRGIkEAgUjo6Od+zt7QsUCgVLS0tzFovFVgqFQmBhYVHk5uaWrqenRzVvbh4hISGXMzIyHCoqKkx69Ohxu9aO5OTkbgDg4eGRLpPJ9FJTU11KS0stAMDa2jq/W7dumYwx5OTk2BQUFHQ1MTEpLyoqsrGxscl1dXXNVL+nGjtciouLrQDA0tKyyNXVNUMgEKjssLW1zc3Ly7Ovtd/Ozq5Q7dxG78HW1jY3Pz/f3tTUtMTNzS3t9u3bHhUVFV2IiHXp0qXM3d091dDQsDotLc25vLzctKKioktmZmY3KyurAg8PjzT1N/PExER3PT09hVQqNSgvLzczNDSs9PT0TK59ay8qKjJPT093lclkQktLy0KJRGJsbW2t0aNS+y5x584dh8LCQluZTKZvYGAg8fb2TjQyMqqu+d7Mb926ZS+TyYRWVlaF7u7uaYwxVFZWGqakpLhJJBITAGRmZlbi7u6eJhQK5YCy12JjY5NXVFRkLZVKjUJDQy9kZmaqrqOvry91dHS8Y2trW1xrS3Z2tm1eXp59dXW1gb6+vtTd3T05JyfHvrq62uD27dvdGWNkb2+f6ezsnFNSUtIlIyOjm0QiMTIwMJC6uLikW1palgLKHmSXLl3KysrKzCorK038/f2vpaSkuNf2KisrKw2Tk5PdJRKJMWOMTE1NS7t37377+vXrPgBw/fp1f8YYunXrlqKvry+r/Q0CgEQi0U9LS3MtLy83BcAsLS0LPTw80tTbtLCw0Dw3N9cBAC5cuGBpYGBQFRgYeL2qqko/JSXFraKiwlRPT09mZ2eXre07AYDExER3gUCgkEqlhhUVFaZGRkaV3t7eiZmZmY6FhYU2QqGw2tPT87apqWklUL/nlpiY6G5gYCCt+ztPTEz0qK6uNkhOTu6en5+PkJCQ1wD8CuVzRJ+IZIwxawAbADwG5TPnBBGNq2sjY2wZlM8lOwDpAN4got01x7yhfFaHAKgGcISIpjDGGJTPy2kADKF8Zj5FRFe1tQNw7w4oAYCcMfYjgO0A/iWiIrXj0TXbIAC5AH4CsBHAdLUyvQB0B/AolI7hIIAhAPQBXGSM7SSiE4yxcVA+pEdD+SBfBmAbgL51jWKMOQP4s+Y6BwFEAfiNMeZLRGcYY18B+JExNhLAzwDeJKIbAKYzxvpDbQiuibryai75FJQP/HQAfwFYXGPfTAAWALpB2SsMgdJJ16U57dQPgA+UTv8cY+x3IorXUtdnNdf0BGAD5QtBFhF9xxh7HM0c361xiqMA2AJIBICsrCw7sVhs6ePjc7Pmj9Y1JSXFtXv37skSicQgKSmpe7du3VJtbGyK5HK5oKqqygAA0tLSXKqqqgz9/f2vM8YoKSnJMyMjw8nNzU1jzN3W1rbw2rVrjjKZTCAUChVEBLFYbOXp6ZkEAElJSe76+vqyoKCgq3K5XHDr1q3uOTk50to/8IqKii5WVlaFIpHoEhGxuveUkZHhWF5e3sXf3/86ACQmJnpnZGQ41v4By2QyfZlMJhSJRJdLSkq6JCUldTc1NS03MTGpauoeZDKZvlwu1wsODr5MRFAoFAIbG5t8b2/v20SE27dvu6ekpLj6+Pgkubq63ikvLzetOwRXl+LiYmsvL68EU1PTitu3b3tkZGQ4d+/e/XZ1dbUwOTnZy9XVNdna2ro4OzvbrqioyBZAgbZ6MjMz7YuKiqy9vb1vGRsbV5WXlxvr6empuixisdjC398/XiaT6cXHx/sVFRUVW1tblwCAg4NDtoWFRalMJtNLTEz0ysjIcHJ3d09Xt7F79+639PX1ZQKBAEZGRlU+Pj43DQwMqvPz861SU1M9zMzMrhoaGlbn5+dbZWdnO3l5eSWamppWSCQSQ8YYeXt7J8fFxZmqD8FVVVXpJyUldXdzc0u2srISFxcXmycnJ3sFBARcNTAwkAGA2j1JAGh83xkZGU7m5uZiPz+/mwqFgpWVlXUBAH9//5uxsbHh/v7+12sf5MXFxWa15xEREhMTu5uampZ6eXldAUC156pjbW1dUl5enl13CC4pKcnTyMioMjg4OK6ystLo1q1bPYyMjKpqHWddxGKxlbe39y0TE5PKhISE7vHx8X6Ojo6Zbm5u6enp6U7p6end/Pz8Ehr6jWhDvT2FQqFhzUu0e51iPwMog3LqpAxanqO1twSgP4BsAJMAbGGMeRNRFpQvxH9D+dwyABBRc84wKJ/nPQCIAfgCKG7M5nsagiOiEigfjATlm3YeY2wfY8y+psg0AB8S0W0iKoOyhzOVMabu8N4mIgkR/Q2gHMA2IsolojtQ9ixCa8o9D+B9IoonIhmUb/UhjDE3LaY9DeAAER0gIgUR/Q9ALIARNcdXQ/mQPgflPMfnjdxmU3UBwA9ElEBElVC+aYTUfF4NpRPwJiI5EZ2vabO6NKed3iKiSiKKAxAHQFS3EsaYHoApAJYTUSkRpUD5ljO9btlGcGKMFUPpKHcDWEREFwGgoKCgq7Oz8x1DQ8NqgUBAzs7OmSUlJVYKhQL5+fnWpqamJV27di0UCASkr68vNzU1rSQiFBQU2Lq6uqbr6+vLhUKhwtHRMau4uNi67oWNjIykxsbGFYWFhVYAUFxcbC4QCBTm5ublUqlUWFZWZuHm5pamp6enMDAwkNnZ2eUUFRWp6tHX15cePXq06s0333TQ09OrNyZTXFxs7ejomGVgYCAzMDCQOTo6ZhYVFdmol+nWrVumQCAgS0vLMnNzc3FBQYF1c+6BMUYuLi6ZAoGA9PT0SF9fX25ra1usp6enEAqFCicnp6zy8nKzujY1hrm5eZG5uXmFQCCAtbV1YWVlpTEAFBUVWRgaGlba2toWCwQCODo65urp6WkEEXz66ac2VlZWIl9fX/9HHnnEefv27eUmJiZVjDGYmppWfvLJJ1YeHh4BEydOxMSJEw2PHDlibGRkJDU1NS0Vi8Um8+fPd/b19e3es2fPbiKRyHfPnj0m9vb2OWVlZRr30LVr1xwjI6Pq2va2tbUtMjQ0rGaMoWvXrkWGhoZVtQ/w/Px8Wzs7u2wzM7MKxhiMjY2rjIyMpNruPS8vz8bMzExsbW0tZozBysqqxNjYuLyoqMiitsz69eur//nnH6FAIEDdoWHGGEmlUkOpVKqvp6dHNcNRTVJaWtqlurpa39XVNV1PT0/RknMlEol+RUWFqaura4aenh4tX77c+vHHHxc4OTn1aOgcc3Pz4vfee8/cw8Mj4PHHHzc+c+YM7OzsChhjSEhIqBw9erSZq6trYHR0dLfaYcb33nuv6yeffGLTUJ1NwRhzhPKFeS4RFRFRNRGd0FaWiHYSUWbNs28HlC//PWsOV0M5rO9U8ww/pfa5GZSOh9U8s7Mas+megxBq3sKja27MF8AWAB8DeBKAE5Tdr1pSa65lr/ZZjtr/K7Xs106cuQH4hDG2Qe04g3Leqe4wnBuASYyx0Wqf6QM4VmNzNWNsM4BPoXzANjaA3GhdNWSr/b9Czeafoez9bGeMWULZNm8QUXWdazSnnRq6hjq2UL6J1K3LWUvZhsgkIhfGmCGAtVAO3X0MAFKp1OD27dveACjssTN6hgDCAQBnw6X/jcgzMDCoN/cn33jePnzNvwLgbCAA5D9uR2kveytQ5421Fisrq8LCwkJrOzu7gsLCQmtLS8tCAJBIJAZExOLi4tQdL9PX11c9wPT19aunTZsmhvKtqx7V1dUGhoaGVWrlpVKpVL92X09PT6beM9DX15dWV1frV1dXC4lIEB8f71enSlbnXNXvSC6XC1JTU7uVlpaay+VyIQAoFAoBEUE5QgFUV9f9GWiir6+vKiAQCBQKhUIPAKRSqb6BgYHqvhljUG+HWkaPHl30008/pR05ciR04sSJVrNmzcr09vau3rZtm8UPP/zQ9cyZMzfv3LkjKi0tzXjqqac8/vvvv3jGmOLdd9+1KCwsrLp8+fLN3NzcbqmpqaaxsbHe3t7eJBAINCKoDAwMNG4iNzfXJjc3114qlRrU3LOeTCYT1ra/kZFRFQDIZDIIhQ0/dqRSqYFYLLa6cOGCyuEQETM1NS0FgOLiYsTFxen//PPPWp2Dq6trRnp6unN8fLyfnp6e3M7OLtva2rpAX19fW3EVVVVVBvr6+lKBoOXv5FKp1EBPT08mFAoVADBu3LjiWbNmyfv169fg319iYiL9/vvv1jdv3rx28eJF+8mTJzs999xzEAqFeOWVVxxff/11REdHXx04cGD3M2fOIDAwEAsWLCjo2bOn78iRIytabKSSbgAK64xWaYUxNgPAIgDuNR+ZQvmcAYDXoOwFnWOMFQHYQETfE9FRxthGKF/sXRljuwEsbuDlG0ArhdXWDGNtxt2Jr0woH+C1uAKQQdPJNJd0AM8TkaXaZkxEZxoo+3Odsl2IaC2gGlZbBeAHABtqHraq22hJXY1R82bxFhH5Q9nFHQVghpairdVO+bj7VqJeV4vDS4moCsBSAEE1w5/Q19ev9vLySggLC7tUt7yBgYFUKpUa1v1cj2k+sGyNLArCwsIuhYWFXdR2XRsbm6Ly8nKzqqoqfbFYbGljY1MIAGlpaZg4cSLWrl1bNGXKlOrVq1eL09LSEmfOnCl3c3ML/O+//wwA0KeffmozY8YMVwBIT08XDh061MvHx8ffx8fH/+rVq9Xx8fFdPD09A55++mnXiIgI7/z8/Ornn3/eJSIiwn3SpEnCr7/+WtWrqZmnqNbX15ddu3ZN8cwzz1RNmzZNOnv2bImenl58WFjYxeDgYN/Lly8b1J7Ts2dPn5MnT5rcunXLYdGiRZbTpk2TT5s2rSouLu4OoOyZPP74457PP/+80YQJE+zFYrGgT58+Pfz9/f2mTp2KHTt2qHoY69evN/Xw8Ajo27dv92nTpjlu3rxZAACpqak0c+ZM84CAAL/w8HCfixcvGlVXV6tsqEvXrl2ru3XrJktPT9evqdfh/fffz3B0dJQBQO/evSWTJ08u2LBhg11FRQV27drV5dtvv03Lz893AkADBgy4tmzZsgtubm4a0V5XrlxhgwYN6ubj4+MfFBTkl52dbbRp0ya3tWvXVoWGhl4KCwu79MorryiOHTtmBAC9evUyevPNN+2Cg4N9ly9f7jhixAjP2rrOnTvHxo8f7wwAv//+u/n48eMtnn76afkbb7xR4uXldTksLOxSeHj4RRcXl2wAOHjwoP6jjz6qGs5evHix48SJE0369evn9OSTT7oJhUKZl5dX6pw5c6o2bdpU/vjjj7u/9dZbzidPnjSZM2cOwsPDvfv169c9NTVVHwB+++03QWBgoF+/fv2cFy5caCwWi5vzTNR4VhgYGEjlcrlQJpMJACAqKqrcxsZGr7EKDh48aPTEE08UGhsbk6enp8zV1VVx/PjxLqmpqfplZWWC4OBgCAQCTJs2reDEiRMkl8sFZmZmChcXl6rY2FijZtiojXQA1jUvxQ1SM7r0DYAXAdgQkSWAq6h58SKibCJ6joicoByh2lQzLwQi+pSIwqEc4usB5dx9g9xrFJwvY+xVxphLzX43KHs+/9YU2QbgFcaYB2PMFMphsx01Q2gt5UsAyxljATXXsmCMTWqg7BYAoxljjzHG9BhjRoyxgYwxl5oJss1QTp7NBlA7lllLDpTzJ03W1ZTBjLFBjLGgmqGxEiidg7YY/FZpJ1LG9/8K4F3GmFnND2hRzT20GCKSQjmEtxIAbG1tczMzM10kEkm9h52trW1hWVmZeV5enpVCoUB1dbVeWVmZMdPSz6mqqtIvKioy13ZNAwMDWZcuXUpv377tbmBgIO3SpYsEAPT19WUZGRmYMWOG4OrVq/FJSUlGW7Zs6Xr48OGMd999N+Pjjz+2rFvX3LlzXfv371968+bN69euXbseFhZWWFhYaJeSkmI0Y8aM4t9//706MTGx/MqVK8b//vtvyueff46VK1e6JScn64vFYtOSkhILGxubIsYYQkJCCrZt21YVFxeXsGrVqjtLlizpVlRUZD5hwoTCnTt3mgFAamqqfm5urn7//v0rPvzwQ/NevXpVx8XFxR89ejRx9erVzpWVyuflhQsXTNevX1+2c+fOYhMTE8Wff/6ZeP369fgvv/wSK1eudFQoFLh06ZLw0KFDxleuXLn+559/Jl25ckX1sHn11VctX3vtNfbPP/9krVu3LuP555/3ru1laaO8vLywqqrKICgoSFEzx2Hcq1cviXqZyMjIivj4eOPk5GQ9BwcHubW1tUKhUOgJBAKFUCiUV1VV6efk5KjWjUkkErZ06VL99957L/fmzZvXT5w4cdPISGkiY0wBADk5OTZQe7ZUVlbCzc2ty+nTp9PefffdrIsXL5rm5eUZAsDff//NRo8eXZmVlSV87733HA8fPpywbds2BAUFydesWWMvl8tZcXGxWVVVlX5NG+qJRCJVb3bJkiW5u3btqjh16lRmZWWl4IcffnCuLVtSUsK++eYbWrJkSd7ChQtd169fLzt9+nT6zJkz8xcvXuwMAFFRUYqrV6/G37hx46qXl5d83bp13nK5XCCXy5lYLO4CAPv37zfz9fX1r92GDBnSdeTIkZahoaG+AGBkZFRtYmJSlp6e7iKXy1lZWZlxYWGhLeq/1KrIycnR69atm6r3am9vr0hPTzdITU3Vd3BwUP39u7m5SfPz8xW1Q8JBQUHyf//916SheoVCYXVVVVW9F0IAqBkO+wtKh2HFGNNnjD2qpWiXGtvzar7XWVCLqmOMTVJ7DhbVlJUzxiIZY70YY/pQTqtIoP25d9fexg42QimUQQSLarxpMZRh2bXe7nsoh5f+AWAE4BCABfdyISLaXfNw3l7zYBUD+B+AnVrKpjPGxkK5PmkblDd/DsA8AAuhHNpaQURU06hxjLH9RHQSwPsAPmOMrQPwDhGtb6SupnCA0nG6QDnRtwPanUGrtVPNeZ8BuA3lF/9NTf33yvcAVkskEoWjo2MaAJaQkNAjuE4hIyMjqZeX162MjAyX9PR0d4FAIHd0dLxTd5wwXyK2yUpIMLWxscmzsrLS2iW3trYuSEtL83B0dNRYm+Hs7Fzl7+8vj4+PD3R3dxeGh4cL5XK5flhYWPnbb78tBKAxDHXmzBmzXbt2JQOAUCiEv79/ZkFBgdDR0dHEzs7O3djYuOjixYuYPHmyWCgUwt7evjoyMrJ6z549AYMHD5Z369Yt1cTERAIAxsbG2XPnzu2RmpoazBhjMpmMKisrS2bMmFE0ZMgQ+9mzZ+Onn36yGj16dBEAnD59WnD06FHDzZs3hwEgqVSqyM5WjqL279+/xNfXNy8lJcUjOzu768cffyw9f/48ACA3N1c/IyNDGBMTI4yKipKYmpoSAIqKiioDYC4WiwWXLl3qsnTpUikATyJi1dXVciMjowqBQKARC71//34rb29vs5SUFKO33367+M6dO93T0tKEAAQKhULjpVN9aLAWJyenzOTkZI+LFy+GGhgYVFlZWRXk5eXZA8Dly5eNbG1tqU+fPpKa70wBQGJkZFQilUqtLl26ZG5lZVUAtQePnp4epk2bdic5OdlTJpPp9+nTh3bu3Gn53HPP5Zw6dQqvvPKK+Y4dO6wTExPRv39/LwByqVRqHRQUxC5fvmxnYmJS7ubmlgYA+fn5rGvXrqr7/euvv8w2bNhgIpFIDMViMTw9Pavi4+Nt5HK5/uDBg6XOzs5pCQkJglu3bhnPmzdPDsBbLpczOzu7SgAFiYmJ7NVXX/UpLS3VKy8vp759+xpevnw5GABZWloWWlhYlI8ePbp09OjR12uvWV1drXfr1i1viURifPXqVb/AwMB4Ly+v2ykpKW6XL18W6enpyRwdHTOh/PvXirbRf8aY1lkBPT29ypKSEsuLFy/aWVhYVCYlJTW43MXBwSE7IyOjW3Z2tjAkJGQxgF11ikwH8BGAG1AO2x+D8vmjbtv1mimPs1AubP0JwGm1IpEAPmbKyNscKKODkxljnjV1e0L5DDoEYH1DtgL36IBqAgUmN3JcAWWI9Rotx1JQZx6gbnQWET1dZ/9nKOdVmmPbfwAGaDn0Sc1WWy4TQFe1/b0A9jazLhDRwDr7m6HsYdWGp2+rd1Kd8+6hnQbWLad2rAjKwAltx46j8T+GeseJqAKAbVxcXApjDE5OTjlOTk45yKvrggALC4syCwuLGxofvmAPvBCmivSyxd0BZACwtLQsrQ1/rcXOzq6wNvxZHQMDA/L09EwDkGZkZOTu4OAg7tq1a1FhYaGBQqFQ+Pv73zx8+HCDk7N6enrk6OiYZWJiYhoSEnKt5uNudcpI7OzsMg8cOGDw448/OgBwOHjw4K3ly5c7DR48OPfNN9/MvXnzpsHgwYN9nJyccmvuQVpVVZX2+++/u3311Ve182/022+/XVd/SweUQ3AmJiYKc3Pz8uDg4KuffvqpTUlJicWVK1eSDQ0NydnZOaiyslJgYmJSKJVKhWp2Sc3NzbPkcjnMzMxkN2/eVIW0EhHi4uKC687H1M4BHT58uMuECRO6T5s27aqrq6vM29vbJyYmxmDMmDFVERER5wHg/PnzNr6+vpWDBw/Ozs7ODi4qKhJYWVlJAgMDNSItnZ2dc2qvKRQKy+suHDU3Ny/U19eXhoaGpgGAXC43MTU1Fdd8fwoXF5c8FxeXPACYOXOm2eeff27n4OBQGRwcXNa3b9+klJQUi/79+1vv37+/0cWdJiYmpUKhsAQAKioq2Kuvvur233//XfX29q5etGiREwCEhITc1NPT8+nevXu6g4NDRVpamrG3t3flpUuXbtStb9WqVYpdu3al9enTp/LTTz+1OXHihFloaGiKepn9+/ebLVmypFvdc42NjSUXL168AQCGhobVPj4+iXWKaP2b8/b2TnF3d3dIT083AAAHB4f8vLw8KxcXl2p3d/fq7OxsvdrvJzU11cDR0VESFBSUCgB//fWXnaWlpbA2grPu35GNjU2xjY1NMQBbIqp9+KueI0RUCGWUrgZ1nwFE9AaUy1PqQUSvQTkPVPfzI1Au3Wg2PBcc54HjkUceKf3ggw+6AspJ78LCwnq/8wEDBpTu2rXLuuY4zp07Z9q/f//y5cuX5924ceP6jRs3rru7u1eXlJToubi4SAHgq6++UvehmDhxYuF7773nUFpaqtezZ89KABg0aFDJhg0b7GsXaJ4+fdpYm41isVjP1ta22tDQkPbv32+WmZlpAAADBw4sO3TokEVFRQUTi8WCw4cPWwLKnoaLi4v0iy++cJLJZHoymYwdOHCgGwCYmZlpnZAfMmRI+RNPPFHwf//3f/YAsGjRouzXX3/dJTs7Ww8Azpw5Y7xjxw6bRYsW5ZmZmSmmTp2a/9xzz7lKJBIGKIcWN23apBG1KBKJJDk5OQYnTpwwAYCioiJBdXU1vLy8pNeuXTORy+VITEzUv3z5cr0Q5lpGjhxZeu3aNZNvvvnGdtKkSYU1910eGxtrevXqVUMAKC0tFVy+fLneUJKPj48kISHBEAAqKioEAODg4CATi8WC/fv3W2m7XnBwsKSwsFB4+PDhLgBQVVXFaudRKioqBK6urtVVVVVs+/bt9SI0AWD06NGltb8J9a3W+dwLEyZMKP7999+tKysr2Y0bNwxSUlKMBg4cWO7m5lbdpUsXxZEjR7ooFAps3brVZuzYscW15yUkJBgGBgZqW9LRKeEOiPPA8cUXX6SdOHHCrEePHv6BgYH+Fy5cqOcEpk+fXhwQEFDZu3dv9zlz5gjfeuutDFdX13pzb0uXLs1evXq1S1hYmK9crjmc/fTTTxf9+eef1mPHjlX12tauXZspk8mYr6+vf/fu3QPefPNNrZFQzz77bGFcXFyXwMBAvy1btlh7eHhIAGDAgAEVw4cPF/v7+weMGDHCKzg4uNzCwkIOANu2bbu9ZcsWa39//5AePXqE/fnnn5aenp5J2kLPa1m1alX2jh07bIuKigTTpk0TT58+Pb93795+Hh4eAXPmzHH//vvvk93c3KoB4OOPP75ja2sr69GjR0D37t0DRo8e7WVvb6/RJkZGRrR169akhQsXuvr4+PgPHDiwR0VFhWDo0KFl3bp1q/Lx8Ql46aWXuvn7+zcYqSUUChEVFSU+ceKExZQpU8QA4OTkJPvqq69Spk6d6tmjRw//8PBwX/X5r1rGjBkjPnHihBkA2NrayqdNm5bn7+8f8Pjjj3uLRKJybdczMjKi7du3Jy1btszFx8fHPyAgwP/EiROmALBs2bLMnj17+vXv379H9+7dJdrObylz5851sbe3D5ZIJAJ7e/vg2p7Z1q1bLV5++WUnAIiIiJCMGzeusEePHgHDhw/v8eGHH6bWRgdu2rQpde7cue5ubm6B7u7uVZMmTVJFd8bExJiOHj36ftIWdShY45HInIeduLi4FF3kKnuYEYvFAgsLC0VpaamgT58+Pl9++WVqv3797jX09oEjPDzc59ChQ4m2traNTnA/aJw+fdr4gw8+cNizZ0+TOeji4uJsRSKRezuYdV/wbNgcTgfj6aefdrt165ZxVVUVmzp1agF3Ppp88MEHGUlJSQa2trYPzFBUc8jNzdX/v//7v06buVsbna4HZGtrS+7u7ro246Fh3bp1cHBotnoDh8PpAGRnZ+O11zTjBM6fP59PRF0bOEUndLoekLu7O2JjY3VtxkNDfHw8/PzqJgLgcDgdGcZYveckY0xrAmddwoMQOBwOh6MTuAPicDgcjk7gDojD4XA4OoE7IA7nASUlJQXGxsYICQmBv78/ZsyYoZGJ+9SpU+jZsyd8fX3h6+uLr7/+WuP8n376CYGBgQgICIC/vz/Wr280q4pO2LNnD9asqZdIpMOwc+dOBAQEQCAQNDp3ffDgQfj4+MDb2xtr197Nd1xYWIihQ4eie/fuGDp0KIqKlImsr1y5gujo6LY2v83hDojTIhh7S2NriK+/Pq9Rbs6c/e1oZcuou8C0PSGliF2b1e/l5YVLly7hypUryMjIwK+//gpAGSX11FNP4csvv8SNGzdw6tQpfPXVV/jzzz8BAH/99Rc+/vhj/P3337h27RouXLgACwuLxi7VYmSye8lNrMm6deswf/78dr1mSwgMDMTvv/+ORx/VlvNTiVwuxwsvvIC//voL169fx7Zt23D9ujL13Nq1axEVFYVbt24hKipK5ZyCgoKQkZGBtLS0BuvtDHAHxOnQpKSkwNfXF88++ywCAwMxbdo0HD58GI888gi6d++Oc+fOAQDOnTuHvn37IjQ0FH379sXNmzcBKP+4Fy9ejKCgIAQHB+Ozzz4DoIymXLNmDfr164edO3eiJgMzAgMDsXTpUq22lJWVISoqCmFhYQgKCsLevcrUgUuXLsWmTZtU5VavXo0NG5TyVR988AEiIyMRHByMVatWqe7Jz88P8+fPR1hYGNLT0zFv3jxEREQgICBAVQ4ADhw4AF9fX/Tr1w8LFy7EqFGjAADl5eV45plnEBkZidDQUJUtDaGnp4eePXvizh3lMpLPP/8c0dHRCAsLAwDY2tpi3bp1qgfc+++/j/Xr18PJyQkAYGRkhOeee65evTk5ORg/fjxEIhFEIhHOnDmDlJQUBAaqkidj/fr1WL16NQBg4MCBeP311zFgwAC8++67cHd3VzngiooKdOvWDdXV1UhKSsLw4cMRHh6O/v3748aN+llvEhISYGhoCFtbZYak/fv3o1evXggNDcWQIUOQk5Oj+j7mzJmDYcOGYcaMGcjLy8OECRMQGRmJyMhInD6tzLPZ0G/ofvDz84OPj0+jZc6dOwdvb294enrCwMAAU6dOVX2fe/fuxcyZytRtM2fOxJ49e1TnjR49Gtu3b79vG3VJm4VhM8a+h1IHJ5eIArUcZ1AmBx0BpdBaNBFduJdrJSQUICenDAoFQSZTICDADg4O9XXbiAgHD2rmC3z88e5a60xNLcb163mqfVdXCwQE2Gkte+pUGkpL7+aefOQRV5ib18+IXlwswdmzKlVjWFgYoW/fejkOAQBXruQgI+Nu0ujAQDt061b/DbSt78nevhpisQSmpgbQ06v/viKTKVBefjcZtZ6eAKam2iVqKiqqUV19t7dhbCyEgUH9nyARoaRE2Z4lJVVITEzEzp078fXXXyMyMhK//PILTp06hX379uGdd97Fzz/vgKOjO/bv/xsmJoY4ffoEXn/9dfz222/4+uuvkZycjIsXL6KyUo6CggKIxRIoFAQDA0OcOnUKmZmZ6N27N86fPw8rKysMHToMW7f+ilGjxmjck5GREXbv3g1zc3Pk5+ejd+/eGDLkcYwcOR7Lly/BtGnPwNhYiF9//RUHDx7E33//jVu3buHcuXNQKBQYOXI0Dhw4DBeXbrh58yZ++OEHleN69913YW1tDblcjkGDBmPYsBh4e3fHnDlz8L//HYWfXw88+eSTqjZ69913MXjwYHzyyRcoLCxCVFR/REb2g729ldbvqaysAqdPn8XateshFksQF3cFs2fP0igTERGBa9euoaKiGleuXIW3dwDEYkmj39O8eS+gZ8++2Lx5G+RyOfT0ZKphorq/E7FYAplMgdzcAhw8eBjGxvq4cOECTpw4gUGDBmH//v147LHHIJEo8Mwzz+Kjjz6Dl5c3rl27iPnz5+Po0aMadf7zz0kEBARDLFZm0AkJ6Yl///0XjDF8++23WLdunepFICYmFn/9dQTGxsZ49tmZePHFBRg0aCDS0tLw2GOPIT4+Hr6+vjhx4gQqKuQ4fvwolixZip9/3g4Li7sZgUpLS9G/f3/V/at3Xn/88SeEhdUTLAYAyOUKlJVVqWxV/3u6c+cOunXrpmona2t7nD9/DmKxBDk5OXB0dAQAODo6Ijc3V1VnYGAIPvhgHZ5/fiGAxv+e6j4jOgptuQ5oM4CNUKby1sbjALrXbL0AfFHzb4t5N/gb/FR11wFs2zYBU6fW83lQ/HgVI2b9rtoXMECuWFWvHAD8OWw7Xki4qws3b1IANv06sX7BuFy8OPAnxKkN41y8+DxCQuov3rw57yBGbI9T7ff0tMJ/SQu1Xv/jyB/wvdo9ffvtaMyeHdbu9/TXX8OgUBTC378rTEzqP9gkaWLcKry7IL2LoR78guzrlQOA3Ph85KstfHZzs0DXrlp+ggWVuJVSDADIzCyGk1M3BAUFAQACAgIQFRUFxhiCgoJwOyEJt24VIjv7DjZsWInMOykwMBSq5joOHz6MuXPnQiglpN0sQCWA/PxCyGQKjBs3AQAQExODgQMHomtX5Rq9iY+Nw4EDh+Hj00/jnogIr7/+Ov755x8IBALcuXMH10/Fw8TGFZmZ2Thz5jqMjKSwsrKCq6srPv30U/z9998IDQ0FZAoUiEtx9mwcIiPN4Ojogt69e6tu+ddff8XXX38NmUyGzIxMnDgRi5SUItjZdYOJgfIN/8knn1TN0/z999/Yt2cv3lnzPhSodTBXMXJkX43vKSkpCSEhIbiVcAuDBo+AoaEzbt0qREW5tJ4MA6BcP5Ibnw+FgnD7dhFMTeWNfk/Hjh3D4sUf4NYtZTq8iAgnrQ6oqrASt24VorJShsjIYcjNLIWblzWmTJmCHTt2YNCgQdi+fTvmPzMHNy6m4b///sXUqVMAAEZGQlRX11fwTr+RAqCL6tp3UhMwe/Z0ZGVlQSqVwsPDQ1V2YM/ByMioBFCJI0eOID4+HkKhsp1KSkpQWloKsViMmU8+jcs3EsAYg0xWjVu3ChER4aSqx8zMDJcuXQIA5F7NRZrk7pCejUW91HVKKqohqZAhLa0EpqZKW9X/ntSTAUjSxMjJKYNYLMWtW4VQKBpOFCAoESI1NV11/w19T8pnxK8N1qNL2swBEdE/jDH3RoqMBfBTjQDGv4wxS8aYY1Ma4sg5j08/244resNwaKQp0t0MAMkyjSJP1mz1iA5SbjU0qA8NADfnaux+UbPVQ2QHyN7U+Ci0oTq3jVNuNZxr7Pp17unZmq0ebX1P8fGAnxNqxVBiSNO53QAAz7tJiMsBxAIImxOOmDnhGmXrJpRLRX1NdQCArYlyA4AUKfTNu6B2+rZQIECGoSFiAWQKBKgUAohwwpfRryN8/OP4YOFCZKakYO7AgYgFUEiEBMZgZaIPqD1IYKCHDFtLlAG4RYSCGrsBILtrF8DeFFfl6Xjv+ecBAM+vWYOSwkLczMvD1+fPQ6ivjzHu7sjysYCTuxMGPz0VR26dQkF2NvpOnYpYAFlEeGr5cjxRU0ctmSkpMLIyV13vTnIy3l2/Hj/GxMDcygqro6NR5WQM+HcFzAyQ52WNPAC3oBTDigVQToS3d/8O9zrDO9fVrwPAycsL3166hPysLDw/cCBOZMZiwJgxcOsThn2xsXAaM0ZVPub8ebj6+yM/3BGewYGIV2QiMsK38e9JKADCnQBDZY8/FkCOUIgKhUJ1f0kSCeR2XZTtb2YA43A31T25jBmDvcuXY1JhIc6eP49lu4ahvLwcplaW+CXhmsbl6k7hlzpbQyoWq77XNYufwlOLFmHAmDE4f/w4vl69Wvk7AWDc3UFVTiEAvoz5D0bGd3PU3gSwesEC+D42BG/8uU/1G0KEk8Z1y0tL8VxND6gu7/zyC/KgJaG2iT4UZgbK77PGBvXvqdTFBdfS05XX8bRCjn45bEO6AxFOsHJ0wMGsLNg6OiI/KwvmdnYqe3K8zWFoY66qU/17ygfgX3sB8VEgcI9S07SDocs5IGcoJWJryaj5rB6MsTmMsVjGWGwu88D6MU/g2/nWSufD4QAoF4th56z8+fyxebPq897DhuH3L79UTT6LC+vJDSGwVy9cOHECxfn5kMvlOLRtG8IGDEBgr1745dIl/HLpEgaMGYMysRjWdnYQ6usj9tgxZKXefSwPmzoVf2/fjqO7diFqorJX2eexx7Dv++9RUaZUS8i9cweFakMoKttLSmDcpQtMLSxQkJODs3/9BQBw8/XFndu3kZmSAgD4344dd+/rscfw62efqd6eb17UqnSuwtbRES+uXYsf338fADDphRfwx+bNuFnzNl9cUIDPli7F9Jr0LdHLl+Oz115Dfo2YnrSqCts//bRevZFRUdj1hfI1Ri6Xo6ykBDb29ijMzUVxQQGkVVU49ccfDdplYmqKgJ49seGll9Bv1Cjo6enB1NwcTh4eOLxTqTlJREiIi6t3rrufH9IT7w4tlan/Bn78scFr9h42DDs3blTt17ZBQ78hdbqYmal+E3U3T39/rec0hX9kJNJu3cKd5GRUS6X43/bteLTmxeDRMWNU9/LHjz9iwNixqvPSEhLgFVh/pKceIhFw/XrT5XSALlPxaHtR19rfJKKvAXwNAAbB4VSt5ni6pUoxf9vf+HNFHARQDkEt2P4EnniifvoY+eYrGKUWjSVwMcWft7UPgf0h+g6fx98drhr5ai+8+H5U/YJxuXih9/e4La8ZDDYzwMZzz8DLq/6b0I3oP/DKL3c12HwGueHjQ9O0Xv8j0w/wt/RuyOzL+yfjsce82/2ejPcNhnlVNlx9bWFkVP/nUplcjHS1ITgjMwO49tCuDZdzIRtiteEGe28rjfH1Wii/ArdSlRnoSzJzoSeRIaLmmC0ALwARAFIAGFTJYX4+C3PHPINli17GrrUf4LHhQ2FQUybk2WfxWkICngkKAskFmDTuKTw95RkIpHL4SmRKBS5HR2x4/328MmgQiAjDHhmM0S49gfNZGvfkPm0aRo8ejbkREQgJCYGvry8cr+bBvMAQobBGVW4RXB0dMbxmzD5i2DDI4+PxQp8+gFwBodAYH7z9GUwFehr3FCES4e/QUMwMCICnpyf6BkXAOKUYdteL8daid/DqkGGwd7BDz549kVNzX5tWrMDLLy7E9O7+UBDB2akbvvrsZ43vKQWAcU35yuRimHXrhe8KSpDw7R70e7Q/ft2yBa8+9xxKS0tBRHj55Zcxb/Ro5FzIxuP2oagYOQ0LHxkIPQMB9PQEeOaZZ1Q2135Pq595HSveXYI/N34FgVAP3//wDXr36YM1K1dibq9e8PDwQLivL6xzy2F+PgvCUim6xOfD1aEEdi5Kdfa5U6Zg0qRJOH78OCIqqpEan4+Pln2E1WuXYfMbq8H0CE8+ORVPiTTnVzxcg/HZ2RiYxWaCMYYlzy3CqkmT4OzsjN69e6M0ORkRUEo1U0YpzGu+z9WzX8e6z1fjmeBgyGQyPProo5j25Zd477XXMPPpGdi+Zi16R/aDQCqHxZUcdG9gSLn4Wh5y1YbgLO27qO6plt27d2PBiwuQl5eHRY89Dr8eAfjui+0Qmkvx4otzceDAAUAoxDcbN+Llxx6DrKoaY0dOQajEGjifhcUT52DR63Px5HffwdXVFTt37lT1sTb9+ieG9h6guq/avyepVIpLpaWqh6k8xQaDBa/gH8UGrfehS9o0GWnNENwfDQQhfAXgeI16KBhjNwEMbGoIjkVEEGri6Z/dVIggV30sHGXW6rZzlPBccLqlrKwMpqamICK88MIL6N69O1555RVdm9VheOmllzB69GgMGTJE16a0K1VVVRgwYABOnTqFWh0hIkJOTg4yMzORm5uLLl26qAImAIAxdp6IIhqqUxfocghuH4AZTElvAOIm539qsM2RoVuqFN84TOfOh/NA88033yAkJAQBAQEQi8V4vs580sPO66+/joqKh0+tIi0tDWvXrlU5n7KyMly/fh0ZGRlQKBQwMTGBp6enjq1smjbrATHGtgEYCOWoSQ6AVQD0AYCIvqwJw94IYDiUYdiziKjJNNe1PaBPPtuOhQumtontnLvwHhCH03GRyWS4c+cO8vKUyysMDAzg6uqKrKysen+3HbEH1JZRcFoD0dSOE4AX7rX+hdInAXAHxOFwHl4yMzORl5cHxhjs7e3h6OgIPT09ZGU1azBJ53Q6PSBAGXjA4XA4DyNEpFrD5ejoCKlUCmdnZxirhZV3FjplKp7H/izTtQkcDofTrigUCmRmZiI+Pl6VvkhfXx/e3t6d0vkAnbQHBADwGKFrCzgcDqddKCkpQWpqKqqqqlT7lpaWujWqFeiUPaAg+d/AE3/q2gxOO6Gnp4eQkBAEBgZi9OjRKC4uVh27du0aBg8ejB49eqB79+54++23NVKb/PXXX4iIiICfnx98fX2xePFiHdzBvfHkk08iODgYH330UbPKm5rWz3/YGhARFi5cCG9vbwQHB+PCBe0pG4kIgwcPRklJidbjHYHhw4fD0tJSldRVG1VVVZgyZQq8vb3Rq1cvpNQsBAaAH3/8Ed27d0f37t3xo9pi16lTp+LWrVutbm91dTVu376NhIQEVFVVwcjICD4+Pg+E8wGg/NF0pg3h4UTrQZz24fr167o2gbp06aL6/4wZM+idd94hIqKKigry9PSkQ4cOERFReXk5DR8+nDZu3EhERFeuXCFPT0+Kj48nIqLq6mr6/PPPW9W26urqVq2vlqysLHJ1dW3ROert1Jr8+eefNHz4cFIoFHT27Fnq2bOn1nJ//PEHvfzyyy2qWyaTtYaJzebw4cO0b98+GjlyZINlPv/8c3r++eeJiGjbtm00efJkIiIqKCggDw8PKigooMLCQvLw8KDCwkIiIjp+/Dg9++yzrWprfn4+XbhwgWJiYig2NpYyMzNJLpc361xtf7cAYqkDPMPVt07ZA+LDb7qBtdHWEvr06aOSFPjll1/wyCOPYNiwYQAAExMTbNy4USUpsG7dOrzxxhvw9fUFAAiFQq3aMWVlZZg1a5ZKsuG3334DoNmj2LVrl0oALDo6GosWLcKgQYOwZMkSuLu7a/TKvL29kZOT02Daf3UkEonq2qGhoTh27BgAYNiwYcjNzUVISAhOnjypcY42CYS696NNNqK8vBwjR46ESCRCYGAgdtSk9lm2bBn8/f0RHBystYe4d+9ezJgxA4wx9O7dG8XFxVqjrLZu3Yqxaqlixo0bh/DwcAQEBGiI3ZmammLlypXo1asXzp49iy1btqBnz54ICQnB888/r9Jnakii4n6IioqCmVnjawfVJRAmTpyII0eOgIhw6NAhDB06FNbW1jVZ04fi4MGDAID+/fvj8OHDrao3RESQy+UwNzdHYGAgHB0dIRB0zkd2Q3TOOaDPZwBP6NoITnsjl8tx5MgRzJ49G4By+C08XDPhqZeXF8rKylBSUoKrV6/i1VdfbbLet99+GxYWFrhy5QoAaM3mXJeEhAQcPnwYenp6UCgU2L17N2bNmoX//vsP7u7usLe3x1NPPYVXXnkF/fr100j7r87nn38OQKlweePGDQwbNgwJCQnYt28fRo0apcq8rM7ChQsxYMAA7N69W5l/rUwzKEebbMSYMWNw8OBBODk5qUTnxGIxCgsLsXv3bty4cQOMMQ1HWou6XAAAuLi44M6dOyqZgFpOnz6Nr776SrX//fffw9raGpWVlYiMjMSECRNgY2OD8vJyBAYGYs2aNYiPj8f//d//4fTp09DX18f8+fOxdetWzJgxQ0OiIioqCpcvX0ZwcLDGNT/44ANs3bq1ns2PPvooPtWSu645qN+vUCiEhYUFCgoKGmwHABAIBPD29kZcXFy932RzkcvlKC8vh7m5Mp2PjY0NDAwMYGZmpjVz+YNA53RAl/OaLsNpddouaVPjVFZWIiQkBCkpKQgPD8fQoUOV9qiFo9alJX+whw8f1hD2srKyaqS0kkmTJkFPTw8AMGXKFKxZswazZs3C9u3bMWXKFFW919WSQNam/Vd/Az916hQWLFgAAPD19YWbmxsSEhJUDyFtHD16FD/9pFQ50dPTq6dUSlpkI3JychAUFITFixdj6dKlGDVqFPr37w+ZTAYjIyM8++yzGDlypNa5EdKyWF1b+xYWFmrc26effordu3cDANLT03Hr1i3Y2NhAT08PEyYopTCOHDmC8+fPIzIyEoDyu7azU2pUqUtUZGVl4fr16/Uc0JIlS7BkyZIG2+peaOh+m2oHOzs7ZGZm3pMDKioqQnp6OqqrqxEQEAAjIyMwxhr9HTwIPFj9Oc4DibGxMS5duoTU1FRIpVJVryEgIACxsZrJM27fvg1TU1OYmZkhICAA58+fb7L+hhyZ+mcSiUTjWJcuXVT/79OnDxITE5GXl4c9e/bgiSeU3XOFQoGzZ8/i0qVLuHTpEu7cuVNv+EfbQ+1+2bp1K/Ly8nD+/HlcunQJ9vb2kEgk6NGjB86fP4+goCAsX74ca9asgVAoxLlz5zBhwgTs2bMHw4cPr1efi4sL0tPvJq7PyMhQKaWqIxQKVeHBx48fx+HDh3H27FnExcUhNDRU1YZGRkYq501EmDlzpqqNbt68idWrVyM5ORnr16/HkSNHcPnyZYwcObLedwAoe0AhISH1toULtSfkbQ7q9yuTySAWi2Ftbd1kO0gkkhaHQ1dVKQUXk5KSIJVKYWxs3Ca/iY4Kd0CcToOFhQU+/fRTrF+/HtXV1Zg2bRpOnTqFw4cPA1C+PS9cuBCv1UgKLFmyBO+99x4SEhIAKB3Chx9+WK/eYcOGYaNaev7aITh7e3vVmovaN3ltMMYwfvx4LFq0CH5+frCxsdFar7bhtEcffVQ1hJSQkIC0tLQmJZyjoqLwhZoEQt2oM7FYDDs7O+jr6+PYsWNIrZGNyMzMhImJCZ5++mksXrwYFy5cQFlZGcRiMUaMGIGPP/5Yq41jxozBTz/9BCLCv//+CwsLi3rDbwDg4+OD27dvq2ywsrKCiYkJbty4gX///bfBe9m1a5dK6bOwsBCpqakoKSlBly5dYGFhgZycHPxVI1FRlyVLlqicl/p2r8NvtfdbG+G2a9cuDB48GIwxPPbYY/j7779RVFSEoqIi/P3333jsscdU5yUkJCAgIKBZ11AoFMjOzsa1a9dQXFwMgUAAV1dX+Pn5ddo1PfeErqMgWrohPJzoUk4DsR+c1qajRcEREY0aNYp++uknIiK6fPkyDRgwgHr06EFeXl60evVqUigUqrL79++nsLAw8vX1JT8/P1q8eHG9+ktLS2nGjBkUEBBAwcHB9NtvvxER0c6dO8nT05MGDBhAL7zwAs2cOZOIiGbOnEk7d+7UqCMmJoYA0ObNm1Wf5eXl0eTJkykoKIj8/PxUkVXqVFZW0syZMykwMJBCQkLo6NGjRESUnJxMAQEBWtsjOzubxowZQ4GBgSQSiejMmTMa7ZSXl0e9e/em8PBwmj17Nvn6+lJycjIdPHiQgoKCSCQSUUREBMXExFBmZiZFRkZSUFAQBQYGathfi0KhoPnz55OnpycFBgZSTEyMVrvWrFlD33zzDRERSSQSGj58OAUFBdHEiRNpwIABdOzYMQ07a9m+fTuJRCIKCgqisLAwOnv2rKqdfX19acSIETR+/Hj64YcftF63JfTr149sbW3JyMiInJ2d6eDBg0REtGLFCtq7dy8RKb+TiRMnkpeXF0VGRlJSUpLq/O+++468vLzIy8uLvv/+e9Xn2dnZFBkZ2Ww7UlJSKCYmhmJiYigxMZGqqqru+97U6SxRcG0qx9AWsIgIotgmc5ZyWgmejJTTXLKysjBjxgz873//07Up7c5HH30Ec3NzVYBMU0gkEiQmJqJbt2715vBaA21/tx0xGSkfguNwOK2Co6MjnnvuuQ69ELWtsLS0VIVu14WIUFBQgNu3b6vmd4yMjBAQENAmzqcz0Tmj4DgcTodk8uTJujZBJ8yaNUvr5xKJBKmpqSgtLQWgDK2udToPamh1S+AOiMPhcFoZhUKBrKwsZGdng4ggFArh4uLywIdVtxTugDgcDqcVqZs41NbWFi4uLir1Us5dOmeLxOUCIjtdW8HhcDj1KCsrQ1VVFYyNjeHq6tpk6p+Hmc7pgIb8CuS9qGsrOBwOB0SkylQNAA4ODhAKhbC1tX3gcre1Nrx1OB0eLsegWzmGGzduoE+fPjA0NMT69esbLEf08MkxVFRU4MaNGxg7dqwqz59AIICdnR13Ps2AtxCnw1Obiufq1auwtrZWpeKprKzEmDFjsGzZMiQkJCAuLg5nzpzBpk2bAABXr17Fiy++iC1btiA+Ph5Xr16Fp6dnq9rWmtmP1cnOzsaZM2dw+fJlvPLKK21yjeZibW2NTz/9tEnnfeDAAYhEohZNtNdmvm4vlixZgp9//rnRMt999x2srKyQmJiIV155BUuXLgWgzNLw1ltv4b///sPZs2exYsUK/PvvvygvL8fkyZOxbt269riFB4rO6YCCu+ragoeTDaxtthbA5RjaX47Bzs4OkZGR0NfX1/aVqHgY5BiGDBkCxhju3LmDiIgInD17Fvb29pgxYwaOHz/eZi8kDyqdcw7oyBRdW8DRAVyOQUl7yzE0l4dBjsHc3BxJSUkAlElLGWOqsvcrx/Aw0jkdEEc3vKqbtE1cjkGT9pZjaC4PgxyDsbEx9PT04OzsDFtbWxgaGqrK3I8cw8NK5xyC4zxUcDmGltHacgzN5UGUY6jNfF0rx5Cbm4vg4GDY2dnhzp079y3H8LDDHRCn08DlGJS0txxDc3mQ5Bh++OEHpKSk4IsvvkB4eDjKy8tVcgwlJSX3LcfAqUHX6bhbuiE8XEvycU5bweUYuBxDVlYWOTs7k5mZGVlYWJCzszOJxeJ65R4EOQaFQkHp6ek0ZMgQcnFxIX9/fzp9+jTJ5XIiaj05hraGyzG0EVyOoX3hcgyc5tLZ5RgqKyuRlpamShxqZmYGV1fXZg2rtVSOoa3hcgxtyaKjuraAw+HUobPLMeTk5KC0tBRCoRAeHh7o0aNHs+d0GpNj4DRM54yC+/k68OFgXVvB4XDq0NnkGGQymSpJqLOzMwQCAZycnFqcOLQhOQZO43TOHhCHw+HcB1KpFElJSbhx44Yqck9fXx+urq48a3U7wluaw+E8NBARcnNzcefOHSgUCggEAlRUVLRZHj1O43ROB7RhoK4t4HA4nYzy8nKkpqaioqICgHLeplu3bhqLSTntS5sOwTHGhjPGbjLGEhljy7Qct2CM7WeMxTHGrjHGmjeQOiOw1W3lcDgPLpmZmYiPj0dFRQUMDAzg7e0Nb29v7nx0TJs5IMaYHoDPATwOwB/Ak4wx/zrFXgBwnYhEAAYC2MAYM2grmzidEy7HoFs5hq1btyI4OBjBwcHo27cv4uLitJajDizHYGCgfKwsXrwYjz76KJ5++ukGy7ZEjqGWqVOn4tatW21m/wNLWy0wAtAHwCG1/eUAltcpsxzAJgAMgAeARACCRuvlC1HblY62EHXGjBn0zjvvEBFRRUUFeXp60qFDh4iIqLy8nIYPH04bN24kIqIrV66Qp6cnxcfHExFRdXU1ff75561qW3V1davWV0tWVha5urq26Jy6Czxbi9OnT1NhYSERER04cIB69uyptdwff/xBL7/8covqlslk922fNiQSCRUUFKj2FQoFVVZW0uHDh2nfvn00cuTIBs/9/PPPVYuGt23bRpMnTyYiooKCAvLw8KCCggIqLCwkDw8PVbscP36cnn322Ta5l3uhsyxEbcs5IGcA6Wr7GQB61SmzEcA+AJkAzABMISJF3YoYY3MAzAEA8ER/OuO5TYVtUu83862bXbZPnz64fPkygIblGAYOHIgXXnihRXIMCxYsQGxsLBhjWLVqFSZMmABTU1NVpuldu3bhjz/+wObNmxEdHQ1ra2tcvHgRISEh2L17Ny5dugRLS0sAyqzIp0+fhkAgwNy5c5GWlgYA+Pjjj/HII49oXFsikWDevHmIjY2FUCjEhx9+iEGDBmnIMXz22Wfo37+/6pycnBzMnTtXlfbmiy++QN++fTXuZ+zYsSgqKkJ1dTXeeecdjB07VqVbk5GRAblcjhUrVmDKlClYtmwZ9u3bB6FQiGHDhtUTnVOvu3fv3sjIyND63WzduhVz5sxR7Y8bNw7p6emQSCR46aWXVMdMTU2xaNEiHDp0CBs2bEBKSgo+/fRTSKVS9OrVC5s2bYKenh7mzZuHmJgYVFZWYuLEiXjrrbe0XlcdhUKBnJwcZGZmAlD+JoyMjMAYg5GREaKionD8+PFG69i7dy9Wr14NQCnH8OKLL4JIKccwdOhQWFsrf69Dhw7FwYMH8eSTT6J///6Ijo7WCOvmNE1btpS2dMR10y48BuASgMEAvAD8jzF2kog0+vBE9DWArwFlJoTWN5XTGeByDEp0Kcfw3Xff4fHHH9d6TNdyDAqFAlKpFAqFAmFhYXjvvfdUSU9bQmNyDLWfA8qkpbXaVAKBgMsx3ANt6YAyAHRT23eBsqejziwAa2u6h4mMsWQAvgDOtaFdnHukJT2V1oTLMWiiKzmGY8eO4bvvvsOpU6e0HteVHMPLL7+MjIwMFBQUAAAMDQ3h5ubWImXWuu1Xl1o5Bm2f18LlGFpOW0bBxQDozhjzqAksmArlcJs6aQCiAIAxZg/AB8DtJmvuurHJIpwHBy7H0DLaQo7h8uXLePbZZ7F3715Vtu+66EqOQSQS4bHHHsO0adMwc+ZMTJ8+HW+++eY9t5+6HINMJoNYLFbJMdR+DgAZGRlcjuE+aTMHREQyAC8COAQgHsCvRHSNMTaXMTa3ptjbAPoyxq4AOAJgKRHlt5VNnM4Nl2NQ0t5yDGlpaXjiiSfw888/o0ePHg3a1Z5yDLWOe8mSJbhw4QL+/PNPxMXF4dq1a60ix1Ab4bZr1y4MHjwYjDGVHEOtRhCXY2gFdB0F0dIN4eFEtp/VD/vgtAkdLQqOiMsxtLccw+zZs8nS0pJEIhGJRCIKbyAStT3kGGQyGaWnp9PNmzc1vufm0pQcA5HyO5k4cSJ5eXlRZGQkJSUlqc7ncgytu3VOOYbUaCDvRV2b8lDA5Rg4zaWt5RiKi4uRlpYGqVQKQDln1lFS6HA5hnujc8YLcufD4XQ41OUY7jUAQBtSqRRpaWmqCD1jY2O4ubl1GOcDKNP6TJ8+XddmdDo6pwPicDgdktaWY8jNzUVGRoYqcaizszPs7OxaFOXYHnA5hnuDOyAOh9NhkclkUCgUsLS0hKurqyqlDufBgDsgDofTYZDJZJBIJKrhNQcHB1U0HOfBgzsgDoejc4gIRUVFSE9PBxEhMDAQQqEQAoGAO58HGO6AOByOTpFIJEhLS1OtZzI1NYVcLuc51R4COqck909XdW0Bpx3hcgy6lWPYu3cvgoODERISgoiIiAZT8RC1TI5BoVAgMzMT165dQ0lJCfT09ODm5gYfH5820+kZPnw4LC0tG005xOUY2pHmLhgC0EXXi5aIL0RtdzraQlQux9AwbSXHUFpaqlr0GRcXRz4+PlrLtVSO4datW/Tvv/9STEwM3b59m6RSaavY2xhcjkH3z3D1rckeEGOsL2PsOpTpdMAYEzHGNrWtW+R0RFgbbS2hT58+qgzEDckxrF27FgBaJMcwa9YsBAUFITg4GL/99hsAzR7Frl27EB0dDQCIjo7GokWLMGjQICxZsgTu7u4avTJvb2/k5OQgLy8PEyZMQGRkJCIjI3H69Ol615ZIJKprh4aG4tixYwCgIcdw8uRJjXNycnIwfvx4iEQiiEQinDlzpt79REVFISwsDEFBQdi7dy8ApST1yJEjIRKJEBgYiB07dgAAli1bBn9/fwQHB2vtIZqamqrCnsvLyxsMgd66dSvGjh2r2h83bhzCw8MREBCAr7/+WqO+lStXYurUqbh58yYuXryIKVOmIDIyEs8//zzkcjkAYN68eYiIiEBAQABWrVql9ZotJSoqql4+vrrs3bsXM2fOBKCUYzhy5AiINOUYrKysVHIMANC/f38cPnwYMpmsVex8WGjOIOtHUMom7AMAIopjjD3aplZxOFrgcgxKdCHHsHv3bixfvhy5ubmq8+vSlBzDgAEDYGJioiHHcP36dSxdurRV5BjUefTRR+85HxyXY2g/mjXLR0Tpdd565G1jTjOZXlfZm9Me6CppE5dj0EQXcgzjx4/H+PHj8c8//2DFihWqBLDqNCTHoFAokJqaijNnziAoKEhDjuHo0aP3LcewZMmSBtvqXiAt6cm4HEPb0JwghHTGWF8AxBgzYIwtRs1wnM74cLBOL89pX7gcQ8toCzmGWh599FEkJSUhP79+0vq6cgz/+9//sGvXLvzwww/o0aMH5HI5PD09W12OISQkpN62cOHCe24/LsfQfjTHAc0F8AKUEtsZAEIA1B9I53DaGC7HoKS95RgSExNVjvLChQuQSqVaNYHU5RgyMzMhFAohFouRkpKCa9euwcPDQyVnrX4v9yLHUMuSJUtUzkt943IMnYSmohQAPNKcz9prQwOp4DltQ0eLgiPicgztLcewdu1a8vf3J5FIRL1796aTJ09qtUtdjuHGjRvUp08f6tGjB40bN67V5BjuFy7HoPvIN/WtSTkGxtgFIgpr6rP2gkVEENUZduG0HVyOgdMciAipqal47rnn8L///Q/V1dUoKipC165dO1zi0LaAyzHcGw0GITDG+gDoC6ArY2yR2iFzAHptbRiHw+kclJWVITU1FUSE2bNnq+QYaoMJHga4HMO90VgUnAEA05oy6jOnJQAmtqVRHA6n4yOTyXDnzh3k5eUBAAwMDDBu3DgYGRnp2LL2h8sx3BsNOiAiOgHgBGNsMxGltqNNTRO1AzgyRddWcDgPJUSEwsJCpKenQyaTgTEGe3t7ODo6qqLbOJzm0Jx1QBWMsQ8ABABQvdoQke5ioS/n6ezSHM7DTnJyMgoLCwEosxq4ubnx8GPOPdGcMOytAG4A8ADwFoAUADFtaBOHw+nAmJubQygUwt3dHT4+Ptz5cO6Z5vSAbIjoO8bYS2rDcifa2jAOh9MxKCkpQVVVFbp27QoAsLGxgaWlJZdL4Nw3zekBVdf8m8UYG8kYCwXg0oY2Nc3h1tWd53RsuByDbuQYqqurcfv2bdUC2VOnTkFPTw+//fabVudD1DI5Bl3A5Rg6GE0tFAIwCoAFgEAAxwCcBzBaVwuX+ELU9qWjLUTlcgwN01pyDAqFgnJycujChQsUExNDsbGxlJ6eToMGDaLHH3+83iLcWloqx0BEJJPJWsPkZsPlGHS/+FR9a7IHRER/EJGYiK4S0SAiCgdQ2IY+kdNR2cDaZmsBXI6hbeUYXnrpJdy4cQNpaWmQy+WwsLBAYGAgdu3ahQkTJjS6tqelcgy9evXC2bNnsWXLFvTs2RMhISFcjuEho7GFqHoAJkOZA+4gEV1ljI0C8DoAYwCh7WMih6OEyzEoaUs5htjYWJSXl0NfXx/dunWDlZUVMjMzsXv3bhw9ehQxMQ3HHzUlxzBhwgTY2NhoyDHEx8fj//7v/7gcw0NKY7OI3wHoBuAcgE8ZY6kA+gBYRkR72sE2TkfjVd0IMnA5Bk1aU46hurpaQ45hyJAhEIvFcHJyUt3fyy+/jP/7v/9rco1PQ3IMAJCeno5bt27BxsZGQ47hyJEjXI7hIaYxBxQBIJiIFIwxIwD5ALyJKLt9TONwlNTKMYjFYowaNQqff/45Fi5ciICAAPzzzz8aZbXJMYhEokbrb8iR3ascw5tvvgngrhxDY2HK2h5q94u6HIO+vj7c3d015BgOHDiA5cuXY/DgwZg+fTq2bNmCtLQ07NixAxs3bsTRo0c16ouNjcXUqVMBAPn5+Thw4ACEQiHGjRunUa5WjkEgEOD48eM4fPgwzp49CxMTEwwcOFDVhtrkGN5//32NumrlGGJiYmBlZYXo6OgG5RhauwdUK7vg4uJST47h+PHjqnIZGRkYOHCgap/LMbScxuaApESkAAAikgBI6DDOJy5X1xZwdACXY1Byv3IMTz31FGbPno1//vkHWVlZyMnJQVRUVINyDMnJyUhJSUFKSgomTpyITZs21XM+gKYcg1gshpWVFUxMTHDjxg38+++/Dd4Ll2N4eGnMAfkyxi7XbFfU9q8wxi63l4FaGfKrTi/P0R2hoaEQiUTYvn07jI2NsXfvXrzzzjvw8fFBUFAQIiMj8eKLLwIAgoOD8fHHH+PJJ5+En58fAgMDkZWVVa/ON998E0VFRQgMDIRIJFIFAqxduxajRo3C4MGD4ejo2KhdU6ZMwZYtW1TDb4ByCCo2NhbBwcHw9/fHl19+We+8+fPnQy6XIygoCFOmTMHmzZthaGjY6LU++eQTHDt2DEFBQQgPD8e1a9c0jk+bNg2xsbGIiIjA1q1bVUEYV65cQUREBPz9/fHBBx/gmWeegb6+PpYvX46ePXtiwIABzQ751sbIkSNVPYThw4dDJpMhODgYK1asQO/evbWe4+/vj3feeQfDhg1DcHAwhg4diqysLIhEIoSGhiIgIADPPPMMHnnkkXu2S53+/ftj0qRJOHLkCFxcXHDo0CEAwMqVK7Fv3z4AwOzZs1FQUABvb298+OGHqqAWa2trrFixQhVUsnLlSpW+UU5ODoyNjZv8nXA0aVCOgTHm1tiJpKP8cCwigig1Gsh7UReXf+jgcgwPDmlpaaqehqGhIVxdXevNH90PWVlZmDFjBv73v/+1Wp2dBS7HcG80loy0YyUg5XA494VQKARjDA4ODnB0dIRA0Jx16M3H0dERzz33nEqO4WGCyzHcG637C6wDY2w4Y+wmYyyRMbasgTIDGWOXGGPXmp3iJ7hrq9rJ4TyIVFZWQiwWq/YdHBzg7+8PZ2fnVnc+tUyePPmhcz6AUo6BpyZqOW3WYjXriD4HMBRABoAYxtg+IrquVsYSwCYAw4kojTHWPAUrLsXA4TSIQqFAVlYWsrOzoaenh8DAQAiFQggEAh6lxelQNMsBMcaMAbgS0c0W1N0TQCIR3a6pYzuAsQCuq5V5CsDvRJQGAETEw9s4nPtALBYjLS0NVVVVAJRDQxxOR6XJfjhjbDSASwAO1uyHMMb2NaNuZwDpavsZNZ+p0wOAFWPsOGPsPGNsRrOs5nA4GkilUiQlJeHWrVuoqqqCsbExfHx84O7uzoeGOB2W5vwyV0PZmzkOAER0iTHm3ozztC1FrxtyJwQQDiAKyvQ+Zxlj/xJRgkZFjM0BMAcAwFcZczj1SEpKQnl5OQQCAZycnGBnZ9dm8zwcTmvRnF+ojIjETRerRwaUqXxqcQGQqaXMQSIqJ6J8AP8AqLdsnYi+JqKIjhZCyGkfuByD9rU56vfp4uKCAQMGICAgAA4ODq3qfI4fPw4LCwuEhIQgJCQEa9asadAeLsfAaRFNpcuGMifcUwAuA+gO4DMAXzbjPCGA21AqqRoAiAMQUKeMH4AjNWVNAFwFENhovVyOoV3hcgyNows5BplMRmlpaZScnKzxeWvJMdTl2LFjjcoX1MLlGLgcQ0u35gzBLQDwBoAqAL8AOATgnWY4Nhlj7MWa8noAvieia4yxuTXHvySieMbYwRrnpgDwLRFdbdKiRUeBDwc3w3ROa/LcprZR4fhmvnWzy/bp0weXLysTcTQkxzBw4EC88MILLZJjWLBgAWJjY8EYw6pVqzBhwgSYmpqqMk3v2rULf/zxBzZv3ozo6GhYW1vj4sWLCAkJwe7du3Hp0iXVhL+3tzdOnz4NgUCAuXPnIi0tDQDw8ccf11vRL5FIMG/ePMTGxkIoFOLDDz/EoEGDNOQYPvvsM/Tv3x9EhOLiYly8eBFvv/027ty5AyMjI3z55Zfo27evxv2MHTsWRUVFqK6uxjvvvIOxY8eivLwckydPRkZGBuRyOVasWIEpU6Zg2bJl2LdvH4RCIYYNG4b169c3+/tQZ+vWrZgzZ45qf9y4cUhPT4dEIsFLL72kOmZqaopFixbh0KFD2LBhA1JSUvDpp59CKpWiV69e2LRpE/T09DBv3jzExMSgsrISEydOxFtvvXVPdqkTFRWlkc9NG3v37sXq1asBKOUYXnzxxXpyDABUcgxPPvkk+vfvj+joaMhkMj7n1gKa01I+RPQGlE6oRRDRAQAH6nz2ZZ39DwB80KKKf77OHdBDyMMsx1BVVYW0tDSIxWK8//776N27N1asWAFDQ8NWlWNQH95U5+zZsxCJRHBycsL69eu15jzjcgxcjqGlNMcBfcgYcwSwE8B2IrrW1AmcB5OW9FRak4dZjoGIkJ2djaysLCgUCujp6eHixYvYv38/jIyMAOC+5BhkMpmGHIO2uZGwsDCkpqbC1NQUBw4cwLhx47TOd3A5Bi7H0FKao4g6CMBAAHkAvq5JRvpmWxvG4dRSK8eQmpoKqVSq6jUEBAQgNjZWo6w2OYamaMiR3ascwxNPPAHgrhxDbYbmO3fu1FPj1PZQq2tDVVUVFAoFrK2tERgYCMZYow5WXY7h0qVLsLe315BjCAoKwvLly7FmzRoIhUKcO3cOEyZMwJ49ezB8+PB69Zmbm6vUYUeMGIHq6mrk5+fXK1crxwBAQ44hLi4OoaGhjcox1LbRzZs3sXr1apUcw5EjR3D58mWMHDmyQTmG2uAI9W3hwoWNtmtj1MoxAKgnx1D7OaCUY3ByclLtczmGltOsUBkiyiaiTwHMhXJN0Mq2NKpJNgzU6eU5uuFhkWPw9PTUeNg6OzujR48e8PT0hL6+/n3LMTz99NNYvHgxLly4gLKyMojFYowYMaJBOYbs7GyVozx37hwUCoXqHtXhcgxcjqHFNBWlAGWk2mooI9ROAJgHwE5XURM8Cq596WhRcEREo0aNop9++omIiC5fvkwDBgygHj16kJeXF61evZoUCoWq7P79+yksLIx8fX3Jz8+PFi9eXK/+0tJSmjFjBgUEBFBwcDD99ttvRES0c+dO8vT0pAEDBtALL7xAM2fOJCKimTNn0s6dOzXqiImJIQC0efNm1Wd5eXk0efJkCgoKIj8/P1VklTqVlZU0c+ZMCgwMpJCQENq9ezddvHiRDh48SAEBAVrbIzs7m8aMGUOBgYEkEonozJkzGu2Ul5dHvXv3pvDwcJo9ezb5+vpScnIyHTx4kIKCgkgkElFERATFxMRQZmYmRUZGUlBQEAUGBmrYX8tnn31G/v7+FBwcTL169aLTp09rtWvNmjX0zTffEBGRRCKh4cOHU1BQEE2cOJEGDBhAx44d07Czlu3bt5NIJKKgoCAKCwujs2fPqtrZ19eXRowYQePHj6cffvhB63VbQr9+/cjW1paMjIzI2dmZDh48SEREK1asoL179xKR8juZOHEieXl5UWRkJCUlJanO/+6778jLy4u8vLzo+++/V32enZ1NkZGR921fa9FZouAalGOohTH2L4BtAHYSUd11PO0Oi4ggqjPswmk7uBxD+1BZWYnU1FRVQIGZmZmqx9NZ4HIMXI6hpTQZhEBE2pWkOBzOfSOXy5Gdna0a5hIKhejWrRusra1bFEjREeByDFyOoaU06IAYY78S0eQaNVT1bhIDQEQU3MCpHA6nGRAREhISUF5eDgDo2rUrnJ2dO/U6ksmTJ+vaBJ0wa9YsXZvQKWnsl/5Szb8N56zgcDj3DGMMXbt2hUKhgJubmyrSjMN5WGgwCo6Ismr+O5+IUtU3APWXk3M4nEYhIuTk5CA7O1v1mY2NDfz8/Ljz4TyUNCcMe6iWzx5vbUNaRNeNTZfhcDoQ5eXliI+PR3p6Ou7cuQOpVApA2QviWas5DyuNzQHNg7Kn48kYu6x2yAzA6bY2jMN5EJDJZMjMzFStczEwMICrqysMDAx0bBmHo3sae/X6BcBoAPtq/q3dwono6XawjcMB0DnlGIgIhYWFuHbtmsr52NvbIyAgoNkqpU3JMdSlLYfxjh8/jpCQEAQEBGDAgAFayxBxOQZOC2logRAA85p/rbVtulq4hPBwItvPGlh+xWltOtpC1M4ix6BQKOjGjRsUExND169fp/Ly8hbV25gcQ0O0lRxDUVER+fn5UWpqKhER5eTkaC3H5Ri4HENLt8Yc0B81/yZDqeuTrLbd1pnBPBNCu6L+Q26rL7Up1B+sX3zxBc2bN4+IiL799luaPn26RtnExERycXEhIqLp06fTd99912T9paWlFB0dTYGBgRQUFES7du2qd92dO3dqZEJ45ZVXaODAgfTyyy+Tm5sbFRUVkVwuJ6lUSl5eXpSdnU1paWk0cuRIioiIoIiICDp16lS9a1dWVqquHRISQkePHiUioqCgIDIyMiKRSET//POPxjnZ2dk0btw4Cg4OpuDgYFVmglp7S0tLafDgwRQaGkqBgYG0Z88eIiIqKyujESNGUHBwMAUEBND27duJiGjp0qXk5+dHQUFB9Oqrr9az8fPPP6c33nijyXZ88sknVdkOiIjGjh1LYWFh5O/vT1999ZXq8y5dutCKFSuoZ8+edPLkSfr5558pMjKSRCIRzZkzR+WU5s6dS+Hh4eTv708rV65s8vrNpSl9o2HDhqmyS1RXV5ONjQ0pFAr65ZdfaM6cOapyc+bMoV9++YWIiORyObm7u7eZPlRL6SwOqME5ICIaVfOvR9v0vTicltGR5Ri2bduG/v374/r16xpyDMuWLWsVOQZ1Fi5ciAEDBmD37t2Qy+VtLseQkJCA6upqDBw4EKWlpXjppZcwY8aMeuW4HAOXY2gpTa54Y4w9AuASEZUzxp4GEAbgYyJKa3PrOB2KxpM2tR0dWY6huroa/fv3x4YNGxAZGYn9+/dj4sSJqnrvV45BG0ePHsVPP/0EQDk/1tZyDDKZDOfPn8eRI0dQWVmJPn36oHfv3ujRo4dGOS7HwOUYWkpz4j+/AFDBGBMBeA1AKoCf29QqDkeNjijHYGJigry8PFy7dg1ubm7IyMiAoaEhTp06hUmTJgFoHTmGe6G15RhcXFwwfPhwdOnSBba2tnj00UcRFxdXrxyXY+ByDC2lOQ5IVjN+OBbAJ0T0CZSh2BxOu9KR5BiysrKQmpoKmUwGCwsLTJw4EevWrWsVOQYfH59G26G95RjGjh2LkydPQiaToaKiAv/995/WBLVcjoHLMbSYpiaJoJRgWA7gFgAHAHoAruhq0ooHIbQvHS0KjqhjyDF89dVXdOnSJcrPzyeFQtFqcgy1QQjJyckdRo6BiGjdunXk5+dHAQEB9NFHH2ktw+UYuBxDS7fmyDE4AHgKQAwRnWSMuQIYSEQ/taFfbNieiAiihZuBGYG6uPxDB5djUFJcXAwiUs0PKRQKKBSKTp04tLXhcgxcjqGlNEeSOxvAVgAWjLFRACS6cj4qXj2u08tzHh6kUikSExORmJioGnIDlFFP3Plooi7H8LBhaWmJmTNn6tqMTkdzouAmA/gAwHEopRg+Y4wtIaJdbWwbh6MziAi5ubm4c+cOFAoFBAIBHB0dVZPnHO1wOQZOS2jOK9wbACKJKBcAGGNdARwGwB0Q54GkvLwcqampqKioAKB8u+X52zic1qc5DkhQ63xqKEDzoufajun+Or0858GFiJCcnAyJRKJKHNrc3G0cDqdlNMcBHWSMHQKwrWZ/CoADbWdSM/hwsE4vz3mwqI3IEQgEYIzBzc0NYrGYD7lxOG1Mkw6IiJYwxp4A0A/KOaCviWh3E6dxOJ0CiUSCtLQ0GBgYwN3dHQBgZmZWb8Eoh8NpfRocSmOMdWeM7WWMXQUwCcAGInqFOx9Oe9MWcgwKhQKZmZm4du0aSkpKUFxcrIpw6yh0FDkG9WwDgYGB0NPTQ2FhYb1yRFyOgdNCGlogBOAkgOcA+ABYDOB3XS9aIr4Qtd3paAtRW0OOQSwW05UrVygmJoZiYmLo9u3bJJVK78m2tsp+3JHkGNTZt28fDRo0SOsxLsfA5RhaujUWTGBGRN8Q0U0iWg/Ava2dIaeDs4G1zdYC+vTpo8pA/Msvv+CRRx7BsGHDACjzs23cuBFr164FAKxbtw5vvPEGfH19ASh7Uo8//jgSEhIgkUhgZGQEHx8fdO3aFXPmzEFQUBCCg4Px22+/AdDsUezatQvR0dEAgOjoaCxatAiDBg3CkiVL4O7urtEr8/b2Rk5ODvLy8jBhwgRERkYiMjISp0/XFxKWSCSYNWsWgoKCEBoaimPHjgFQpvHJzc1FSEgITp48qXFOTk4Oxo8fD5FIBJFIhDNnzmgcLysrQ1RUFMLCwhAUFIS9e/cCUEb3jRw5EiKRCIGBgdixYwcAYNmyZfD390dwcHCTgn3btm3Dk08+qfXY1q1bMXbsWNX+uHHjEB4ejoCAAHz99deqz01NTbFy5Ur06tULZ8+exZYtW9CzZ0+EhITg+eefh1wuBwDMmzcPERERCAgIwKpVqxq1q7lERUU1Oby6d+9e1ZqeiRMn4siRIyAiHDp0CEOHDoW1tTWsrKwwdOhQHDx4EADQv39/HD58uMP1ojs6jc0BGTHGQqGc9wEAY/V9IrrQ1sZxOOrcrxwDY0y1OTk5wd7eHgKBAEuXLr1vOYbdu3dj1qxZ+O+//zTkGF555ZVOL8dQS0VFBQ4ePKiR304dLsfA5RhaSmMOKAuAeubGbLV9AqC7ULSoHcCRKTq7/EPLq7oRZLgfOYbKykqUl5ejS5cuAJQPDQcHBxgZGanK3o8cAwBMmTIFa9aswaxZs7B9+3ZMmTJFVe+DIMdQy/79+/HII4/A2tpa63Eux8DlGFpKY4J0g9rTkBZxOU/XFnDakVo5BrFYjFGjRuHzzz/HwoULERAQgH/++UejbK0cg4mJCdzd3XHgwAGYmprC399flT6nbgqdhhxZY3IMtQ4NUA4LJiYmIi8vD3v27MGbb74J4K4cQ2Mp+rU91O4XdTkGfX19uLu7a8gxHDhwAMuXL8ewYcOwcuVKnDt3DkeOHMH27duxceNGHD16VGu927dvb3D4DbgrxyAQCDTkGExMTDBw4MBG5Rjef/99jbpq5RhiYmJgZWWF6OjoBuUYWrsHVCu74OLiUk+O4fjx46pyGRkZGDhwoGqfyzG0HN0uKOVwWkBz5RgWLFiAa9euYdKkSfjhhx+Qn58PImpVOQZ1GGMYP348Fi1a9EDKMdTWeeLECY05nrpwOQYux9BS2tQBMcaGM8ZuMsYSGWPLGikXyRiTM8YmtqU9nM5PaGgoRCIRtm/fDmNjY+zduxfvvPMOfHx8EBgYCC8vLwwaNAhSqRQikQgfffQRXn31VQQGBiIwMBBZWVn16nzzzTdRVFSEwMBAiEQiVSDA2rVrMWrUKAwePBiOjo6N2jVlyhRs2bJFNfwGKIegYmNjERwcDH9/f3z55Zf1zps/fz7kcjmCgoIwZcoUbN68GYaGho1e65NPPsGxY8cQFBSE8PBwXLt2TeP4tGnTEBsbi4iICGzdulUVhHHlyhXVZP+7776LN998E6WlpRg1ahSCg4MxYMCABkO+d+/ejWHDhmn0/OoycuRIVQ9h+PDhkMlkCA4OxooVK9C7d2+t5/j7++Odd97BsGHDEBwcjKFDhyIrKwsikQihoaEICAjAM888g0ceeaTRNmku/fv3x6RJk3DkyBG4uLjg0KFDAICVK1di3759AIDZs2ejoKAA3t7e+PDDD1VBLdbW1lixYoUqqGTlypWq4cicnBwYGxs3+TvhaNKkHMM9V8yYHoAEAEMBZACIAfAkEV3XUu5/ACQAvqcmkpyyiAii7w4AIrs2sZujSWeRYyAiXL16FVVVVRAIBHB2doadnV2LpLk59weXY+ByDC2lyR4QU/I0Y2xlzb4rY6xnM+ruCSCRiG4TkRTAdihVVeuyAMBvAHK1HNMOdz6cGmpfoBhjcHR0hJWVFQIDA2Fvb8+dTzvD5Ri4HENLac4Q3CYAfQDUzj6WAvi8Gec5A0hX28+o+UwFY8wZwHgA9ccmNMvNYYzFMsZim3FdzkOATCZDamoqsrOzVZ/Z2NjAy8uLZ63WIZMnT240gu9BZdasWVwf6h5oTov1IqIwxthFACCiIsZYc/7Ctb1+1h3v+xjAUiKSN/a2SkRfA/gaUA7BNePanAcUIkJhYSHS09Mhk8kgEAjQtWtXCIVC3uPhcDoZzXFA1TXzNASo9IAUzTgvA0A3tX0XAJl1ykQA2F7z4LAFMIIxJiOiPc2on/OQIZFIkJqaitLSUgDKFfVubm78zZPD6aQ05y/3UwC7Adgxxt4FMBHAm804LwZAd8aYB4A7AKYCeEq9ABF51P6fMbYZwB/c+XDqQkTIyspCVlYWiAhCoRAuLi6wsbHhvR4OpxPTHDmGrYyx8wCioBxWG0dE8U2cBiKSMcZeBHAIgB6UEW7XGGNza443Ou/TKHG5PBDhIaO0tBREBFtbWzg7O0NfX1/XJnE4nPukOVFwrgAqAOwHsA9Aec1nTUJEB4ioBxF5EdG7NZ99qc35EFF0UyHYKob82qxinM5LdXU1qqqqAChX2E+ePBkzZszAggULUF5erip3r3IMnYGOIscgFosxevRoiEQiBAQE4IcfftBajrgcQ1ua/mDSVLpsAFcAXK759xYAGYBrukrfjfBwItvPiNM+tLccg0KhoJycHLpw4QLduHGDFApFq8sxtCYPgxzDu+++S6+99hoREeXm5pKVlRVVVVXVK8flGLgcQ0u35gzBBanvM8bCADzf6p6Q0+F5blN9EbLW4Jv5ytXkFRUVSE1NVfVwBAIBFArNeJc+ffrg8uXLABqWYxg4cCBeeOGFenIMQqEQ8+fPr3f9srIyLFiwALGxsWCMYdWqVZgwYQJMTU1VmaZ37dqFP/74A5s3b0Z0dDSsra1x8eJFhISEYPfu3bh06RIsLS0BKOUYTp8+DYFAgLlz5yItLQ0A8PHHH9db0S+RSDBv3jzExsZCKBTiww8/xKBBgzTkGD777DP0799fdU5OTg7mzp2rSnvzxRdfoG/fvhr3M3bsWBQVFaG6uhrvvPMOxo4di/LyckyePBkZGRmQy+VYsWIFpkyZgmXLlmHfvn0QCoUYNmwY1q9fr2EjY0w1BFpWVgZra2utgR9bt27FnDlzVPvjxo1Deno6JBIJXnrpJdUxU1NTLFq0CIcOHcKGDRuQkpKCTz/9FFKpFL169cKmTZugp6eHefPmISYmBpWVlZg4cSLeeuutetdsKVFRURr53LSxd+9erF69GoBSjuHFF1+sJ8cAQCXH8OSTT6J///6Ijo6GTCbjQTEtoMUtRUQXGGORbWFMswnuqtPLc1oXuVyOzMxM5OTkAAD09fXRrVs3WFlZaQQZ3K8cQ0O8/fbbXI6hETmGF198EWPGjIGTkxNKS0uxY8cOCAT1R++5HAOXY2gpTTogxtgitV0BgDAAuk1HzaUYdEJtT6U1USgUuHbtmmq+x87ODs7OzqpsycD9yTE0By7H0Lgcw6FDhxASEoKjR48iKSkJQ4cORf/+/evZyOUYuBxDS2lOJgQztc0QwJ/QnlKHw2kxAoEANjY2MDExgZ+fH1xdXTWcD3BXjiE1NRVSqVTVawgICEBsrGZyjFo5BjMzMwQEBOD8+fNN2tCQI7tXOYYnnngCwF05htoMzXfu3KmnxqntoXa/qMsxXLp0Cfb29hpyDEFBQVi+fDnWrFkDoVCIc+fOYcKECdizZw+GDx9er74ffvgBTzzxBBhj8Pb2hoeHB27cuFGvXK0cAwANOYa4uDiEhoY2KsdQ20Y3b97E6tWrVXIMR44cweXLlzFy5MgG5RhCQkLqbQsXLrzn9quVYwBQT46h9nNAKcfg5OSk2udyDC2nUQdUswDVlIjeqtneJaKtRFT/l8DhNAOFQoHs7GwUFt6dT3JwcICfn1+jmZaB5ssxvPbaawCUb8fvvfceEhISVNfmcgwtl2NwdXXFkSNHACjnn27evAlPT8965bgcA5djaDENRScAENb8e0TXkRIadoWHa4364LQNrRkFV1paSlevXqWYmBi6ePFisyOg6kZ3jRo1in766SciIrp8+TINGDCAevToQV5eXrR69WpSKBSqsvv376ewsDDy9fUlPz8/Wrx4sVa7ZsyYQQEBARQcHEy//fYbERHt3LmTPD09acCAAfTCCy/QzJkziYho5syZtHPnTo06YmJiCABt3rxZ9VleXh5NnjyZgoKCyM/PTxVZpU5lZSXNnDmTAgMDKSQkhI4ePUpERMnJyRQQEKC1PbKzs2nMmDEUGBhIIpGIzpw5o9FOeXl51Lt3bwoPD6fZs2eTr68vJScn08GDBykoKIhEIhFFRERQTEwMZWZmUmRkJAUFBVFgYKCG/bXcuXOHhg4dSoGBgRQQEEA///yzVrvWrFlD33zzDRERSSQSGj58OAUFBdHEiRNpwIABdOzYMQ07a9m+fTuJRCIKCgqisLAwOnv2rKqdfX19acSIETR+/Hj64YcftF63JfTr149sbW3JyMiInJ2d6eDBg0REtGLFCtq7dy8RKb+TiRMnkpeXF0VGRlJSUpLq/O+++468vLzIy8uLvv/+e9Xn2dnZFBkZed/2tRadJQquQTkGxtgFUuaA2wCgO4CdAFQLMIjo97Z2jlrtioggiuU5SduL1pBjkMlkyMjIQH5+PgDA0NAQrq6u9eYuOJ0bLsfA5RhaSnOi4KwBFAAYDGU+OFbzr04cEKfzQEQoKChARkYGZDIZGGNwcHCAo6Oj1igqTudGXY7hYcuIbWlpienTp+vajE5HYw7IriYC7iruOp5adJuRetFR4MPBOjWB0zREhOzsbMhkMpiZmcHV1ZVP0j7gTJ48Wdcm6IRZs2bp2oROSWMOSA+AKZonq9C+/HydO6AOikKhgEKhgFAohEAggJubG6qqqnjiUA6HU4/GHFAWEa1pN0s4nR6xWIy0tDSYmZnB3d0dAGBmZlYv9JjD4XCAxh0Qf13lNAupVIr09HRV+LJAIIBcLq+3nofD4XDUacwBRbWbFS1lw0BdW8CBco4nLy8Pd+7cgVwuh0AggJOTE+zs7HiQAYfDaZIGnxJE1DaZJ1uDGYG6tuChR6FQ4MaNG0hLS4NcLoeFhQUCAgLg4ODQ6s5HT08PISEhCAwMxOjRozXylXE5hru0lRxDUVERxo8fj+DgYPTs2RNXr17VWo6IyzFwWoiuFyK1dOMLUduXxhaiJicnU1xcHBUWFmos/mxtuBxD82grOYbFixfT6tWriYgoPj6eBg8erLUcl2Pgcgwt3XRuQIsN5g6oXan9ISsUijb7UptC/cH6xRdf0Lx584iI6Ntvv6Xp06drlE1MTCQXFxciIpo+fTp99913TdZfWlpK0dHRFBgYSEFBQbRr16561925c6dGJoRXXnmFBg4cSC+//DK5ublRUVGRqqyXlxdlZ2dTbm4uPfHEExQREUERERF06tSpeteurKxUXVs9E0JQUBAZGRmRSCSif/75R+Oc7OxsGjduHAUHB1NwcDCdPn1aw97S0lIaPHgwhYaGUmBgIO3Zs4eIiMrKymjEiBEUHBxMAQEBtH37diIiWrp0Kfn5+VFQUBC9+uqr9WwcMWIEnTx5UrXv6elJ2dnZ9co9+eSTqmwHRERjx46lsLAw8vf3p6+++kr1eZcuXWjFihXUs2dPOnnyJP38888UGRlJIpGI5syZo3JKc+fOpfDwcPL396eVK1fWu969cuzYsUYd0LBhw1TZJaqrq8nGxoYUCgX98ssvNGfOHFW5OXPm0C+//EJERHK5nNzd3dvshaSldBYHxIUrOE1SVVWl1LTp3l2ndnA5BiXtLccgEonw+++/o1+/fjh37hxSU1ORkZEBe3t7jXJcjoHLMbQU7oA4DSKVSiEWi3Ht2jUoFApcuHgRzs7O6Nq1a7uu6eFyDJq0txzDsmXL8NJLLyEkJARBQUEIDQ3VKrrG5Ri4HENL4aFKHK2cPHkSoaGhKC4uhkKhgLW1NQIDA2FnZ9fuC0q5HEPLaG05BnNzc/zwww+4dOkSfvrpJ+Tl5cHDw6NeOS7HwOUYWkrndEBdNzZdhnPP1EogX79+HUKhED169ICnpyf09fV1aheXY1DS3nIMxcXFkEqlAIBvv/0Wjz76qNYeGpdj4HIMLUbXk1At3RAeTmT7WWPzb5x7QKFQaEygbtmyhVauXEnXrl3ToVVKuByDJu0tx3DmzBny9vYmHx8fGj9+vCryqy5cjoHLMbR0a1COoaPCIiKIUqOBvBd1bcoDw/Xr1zF37lwMHToUK1as0DjWGnIMnIcDLsfA5RhaSuccguO0ChUVFXj99dchEolw8uRJfPvtt6iqqtK1WZxOirocw8OGpaUlZs6cqWszOh2d0wHx3s9989dffyEwMBDvv/8+ZDIZnn/+eVy6dAmGhoa6No3TiZk8efJDpwUEKOUYtEUGchqHt9hDRnl5OaKjo7Fr1y4AQHBwML788kv06dNHx5ZxOJyHjc7ZA+LcMyYmJigsLESXLl2wfv16nD9/njsfDoejE3gP6CEgNjYWlpaW8Pb2BmMM3377LfT09ODq6qpr0zgczkMM7wE9wIjFYixYsAA9e/bE3LlzVYsePTw8uPPhcDg6hzugBxAiwo4dO+Dr64uNGzdCIBAgLCwMMplM16bdE1yOQbdyDDdu3ECfPn1gaGiI9evXaxw7ePAgfHx84O3tjbVr1zZYx8cff6xKH9QR2bhxo2qEID8/v8FyDckxJCcno1evXujevTumTJmiWrj7xx9/YNWqVW1uf6dF1wuRWrohPJzoxyuNr8J6iElMTKTHHnuMABAA6tOnD8XFxd1zfY3JMbQXXI6hebSVHENOTg6dO3eOXn/9dfrggw9Un8tkMvL09KSkpCSqqqqi4OBgrQuXq6urKSgoqEVt1d5ZpS9cuEDJycnk5uZGeXl5Wss0JscwadIk2rZtGxERPf/887Rp0yYiUi7wDgkJofLy8va5kRo6y0LUztkDevW4ri3okJSWliIiIgKHDh2CpaUlvvrqK5w6dapeAsd7ZgNrm60F9OnTR5WB+JdffsEjjzyCYcOGAVAGWGzcuFH1Jr5u3Tq88cYb8PX1BaDMVTZ//vx6dZaVlWHWrFkICgpCcHAwfvvtNwCaPYpdu3YhOjoaABAdHY1FixZh0KBBWLJkCdzd3TV6Zd7e3sjJyUFeXh4mTJiAyMhIREZG4vTp0/WuLZFIVNcODQ3FsWPHACjT+OTm5iIkJAQnT57UOCcnJwfjx4+HSCSCSCTCmTNn6t1PVFQUwsLCEBQUhL179wJQRkCOHDkSIpEIgYGB2LFjBwBlslF/f38EBwdr7SHa2dkhMjKyXiqmc+fOwdvbG56enjAwMMDUqVNV11Ln6NGjCAsLU4Upf/PNN4iMjIRIJMKECRNQUVFRr12XLl2KpKQkDB8+HOHh4ejfvz9u3LgBANi/fz969eqF0NBQDBkyBDk5OfWu2VJCQ0Ph7u7eaJlDhw5h6NChsLa2hpWVFYYOHYqDBw+CiHD06FFMnDgRADBz5kzs2bMHgDJN08CBA/HHH3/ct40PIm0ahMAYGw7gEwB6AL4lorV1jk8DsLRmtwzAPCKKa0ubHmTMzMzwyiuvIDExEevXr1dlFX5Q4HIMStpbjqEhtMkT/Pfff/XKnT59WuN7euKJJ/Dcc88BAN5880189913qozg6u0aFRWFL7/8Et27d8d///2H+fPn4+jRo+jXrx/+/fdfVUDNunXrsGHDBo1r3rx5U5WVvC7Hjx+HpaVls++zsfu9c+cOCgoKYGlpqXKw6jINABAREYGTJ09i8uTJLb7mg06bOSDGmB6AzwEMBZABIIYxto+IrqsVSwYwgIiKGGOPA/gaQK+2sulBIy8vD0uWLEFUVBSmT58OAFixYkXbZat+VTdpm7gcgybtLcfQEKQljZe2ds/KytJIC3P16lW8+eabKC4uRllZmUZCz9p2LSsrw5kzZzBp0iTVsdosHRkZGZgyZQqysrIglUq1Zub28fHR6rzvh4but6l2qJVp4NSnLYfgegJIJKLbRCQFsB3AWPUCRHSGiGpfN/8F4NKsmqf7t6adnQ6FQoFvv/0WPj4++PHHH/HGG2+guroaQMsevJ0FLsfQMlpbjqEhmpInqMXY2Fij/aKjo7Fx40ZcuXIFq1at0jhW264KhQKWlpYaGa5re48LFizAiy++iCtXruCrr77SKtNw8+ZNrTINISEhLerlNed+bW1tUVxcrAry4TINzactHZAzgHS1/YyazxpiNgCtOdcZY3MYY7GMMeXT5sPBrWVjp+Pq1at49NFH8dxzz6GoqAhDhgzBkSNHdC6V0B5wOQYl7S3H0BCRkZG4desWkpOTIZVKsX37dowZM6ZeOT8/PyQmJqr2S0tL4ejoiOrqaq1qpoBSg8jDwwM7d+4EoHTUcXFxqvtzdlY+StQj0dSp7QFp2+5l+A1Ag3IMjDEMGjRIlV3kxx9/xNixd9+1ExISEBgYeE/XfOBpq+gGAJOgnPep3Z8O4LMGyg4CEA/Apsl6w8MbCvx4oKmoqKDXXnuNhEIhASB7e3v65ZdfNKQH2oKOFgVHxOUY2luOISsri5ydncnMzIwsLCzI2dmZxGIxERH9+eef1L17d/L09FRFJ9YlJSWF+vfvr9rftGkTubu704ABA+jFF19ssF1v375Njz32GAUHB5Ofnx+99dZbRES0Z88e8vDwoH79+tHixYtpwIABWq/bEj755BNydnYmPT09cnR0pNmzZxOR8nut/T9Rw3IMSUlJFBkZSV5eXjRx4kSSSCSqYyNHjqTLly/ft40tobNEwbWlA+oD4JDa/nIAy7WUCwaQBKBHs+p9SB2QRCIhX19fYozR/PnzqaioqF2u2xEcEKfzM27cOEpISNC1Ge1OdnY2DR48uN2v21kcUFtGwcUA6M4Y8wBwB8BUAE+pF2CMuQL4HcB0IkpoQ1s6JRkZGTAxMYG1tTUMDQ2xefNmAECvXjxOg9O5WLt2LbKystC9e3ddm9KupKWl1YvQ49ylzeaAiEgG4EUAh6AcXvuViK4xxuYyxubWFFsJwAbAJsbYJdUcz0OOTCbDRx99BD8/PyxZskT1ea9evbjz4XRKfHx88Oijj+rajHYnMjISISEhujajw9Km64CI6ACAA3U++1Lt/88CeLYtbehs/Pfff3j++ec1JlxlMhnXGuFwOA8cnTMTQtQOXVvQ6hQXF2P+/Pno06cP4uLi4Obmhv3792PXrl3c+XA4nAeSzvlku5ynawtalaKiIvj7+yM7OxtCoRCvvvoqVqxYobHWhMPhcB40OqcDesCwsrLC448/joSEBHzxxRcICgrStUkcDofT5nTOIbhOTlVVFdasWYMTJ06oPtu4cSP++ecf7ny0wOUYOq4cwzPPPAM7O7smF1pyOQaOVnQdB97SDeHhRJdyGguB79AcOXKEevToQQDIz8+PZDKZrk1qlI6wDojLMTSP9pZjICI6ceIEnT9/vsFFs0RcjoHLMehmHVDbIep8WZ5zc3Px6quvYsuWLQCUiSc3bdqkSmjZGXhuU2Gb1PvNfOtml+3Tpw8uX74MoGE5hoEDB+KFF15okRzDggULEBsbC8YYVq1ahQkTJsDU1FSVaXrXrl34448/sHnzZkRHR8Pa2hoXL15ESEgIdu/erZHixdvbG6dPn4ZAIMDcuXORlpYGQNkLeOSRRzSuLZFIMG/ePMTGxkIoFOLDDz/EoEGDNOQYPvvsM/Tv3191Tk5ODubOnYvbt28DAL744gv07dtX437Gjh2LoqIiVFdX45133sHYsWNRXl6OyZMnIyMjA3K5HCtWrMCUKVOwbNky7Nu3D0KhEMOGDavXy7Gzs4OdnZ0qi7Y6jz76KFJSUhr9zrTJMXz99deQSqXw9vbGzz//DBMTE412DQsLw/z58/HCCy8gLy8PJiYm+Oabb+Dr64v9+/fjnXfegVQqhY2NDbZu3Qp7e/tGbWiK0NDQJsuoyzEAUMkxTJ06FUePHsUvv/wCQCnHsHr1asybN09DjoFnw65P53RAnYjaxKFLly5FcXExjIyM8Oabb2LJkiUwMDDQtXmdCi7HoKSjyDE0Fy7HwOUYGoI7oDZGLBbjjTfeQHFxMR577DF8/vnn8PLy0rVZ90RLeiqtCZdj0KSjyDE0Fy7HwOUYGoIHIbQB5eXlqj8WKysrfPnll9ixYwf++uuvTut8dAmXY2gZ7SXH0Fy4HAOXY2gI7oBamX379sHf3x/r1q1TfTZhwgRMnjz5gdTqaU+4HIOSjiLH0Fy4HAOXY2gQXUdBtHTrqFFwqampNHbsWAJAAOiRRx4huVyua7Pum44WBUfE5Rg6khzD1KlTycHBgYRCITk7O9O3335b73wux8DlGBradG5Aiw0ODyey/awZX0H7IJVK6YMPPiATExMCQGZmZvTJJ590+PDq5tIRHBCn88PlGNqXzuKAeBDCfZCfn4+oqChVWPCkSZPw0UcfqYYHOByOEi7HwNEGd0D3gY2NDWxtbeHh4YGNGzdixIgRujaJw+mQ+Pj4NDm39SASGRmpaxM6NJ3TAQV31clliQhbt25Fz5490aNHDzDGsGXLFlhYWMDExEQnNnE4HE5npXNGwR3RvsCsLbl58yaGDBmC6dOnY/78+cr5KACOjo7c+XA4HM490DkdUDsikUiwatUqBAcH4+jRo7CxscHTTz+ta7M4HA6n09M5h+DaicOHD2PevHmqNQzPPPMM1q1bp1rnweFwOJx7h/eAGiAnJwejRo1CYmIi/P398c8//+C7777jzkcHcDmGjinHkJ6ejkGDBsHPzw8BAQH45JNPGqyDyzFwtKLrOPCWbggPbyoE/p6Ry+Uaixj/7//+j95//32qqqpqs2t2dDrCOiAux9A82luOITMzk86fP09ERCUlJdS9e3e6du1avfO5HAOXY2ho4z2gGi5duoS+ffuq5BIA4LXXXsOyZct41uoaWBttLaFPnz6qTMMNyTGsXbsWAFokxzBr1iwEBQUhODgYv/32GwDNHsWuXbsQHR0NQJnLbNGiRRg0aBCWLFkCd3d3jV6Zt7c3cnJykJeXhwkTJiAyMhKRkZE4ffp0vWtLJBLVtUNDQ3Hs2DEA0JBjOHnypMY5OTk5GD9+PEQiEUQiEc6cOVPvfqKiohAWFoagoCDs3bsXgDJH4ciRIyESiRAYGIgdO3YAAJYtWwZ/f38EBwdr7SHa2dkhMjIS+vr6Gp87OjoiLCwMAGBmZgY/Pz+NLNC1aJNjiIyMhEgkwoQJE1BRUVGvXZcuXYqkpCQMHz4c4eHh6N+/P27cuAEA2L9/P3r16oXQ0FAMGTIEOTk59a7ZUkJDQ+Hu7t5oGXU5BisrK5UcAxHh6NGjmDhxIgClHMOePXsAQEOOgVOfzjkHtOgo8OHgVqmqtLQUq1atwieffAKFQoGqqio8/fTTPG9bB4TLMSjpiHIMKSkpuHjxInr16lXvGJdj4HIMDdE5HdDP1+/bARER9uzZg4ULFyIjIwMCgQAvvfQS1qxZw51PA7R+3ubmweUYNOlocgxlZWWYMGECPv74Y612czkGLsfQEA/lEFx+fj7GjBmDJ554AhkZGYiIiEBMTEyDf0Ac3cLlGFpGe8oxVFdXY8KECZg2bZrqvuvC5Ri4HENDPJQOyMzMDImJiTA3N8fGjRvx77//qsayOR0XLsegpKPIMRARZs+eDT8/PyxatKjBclyOgcsxNIiuoyBauiE8nOjHKw1GfzTEqVOnKD8/X7V/6dIlyszMbHE9DxsdLQqOiMsxdBQ5hpMnTxIAVZ0ikYj+/PPPeudzOQYux9DQpnMDWmxwC8Ow8/Pz6dlnnyUAGj8kTvPoCA6I0/nhcgztS2dxQA/sEBwR4ccff4Svry++/fZb6Ovrw8nJSenEOBxOu1Irx/CwweUYGqdzRsE1wY0bNzB37lycOHECADBw4EB88cUXqvUgHA6nfeFyDBxtPHAOKCMjAyKRCFKpFLa2ttiwYQOmT5/OQ6vvA6KGw505HE7HojON8jxwDsjFxQXTp0+HQCDA2rVrYW1trWuTOjVGRkYoKCiAjY0Nd0IcTgeHiFBQUAAjIyNdm9IsWGfylgDAIiKI1NZ+ZGVl4ZVXXsHcuXMxcOBAAMpwW4HggZ3ealeqq6uRkZGhda0Fh8PpeBgZGcHFxaVe6iTG2HkiitCRWVrpnD2grhshz56HL774Am+88QZKSkqQmJiImJgYMMa482lF9PX1ta4053A4nPulTZ/UjLHhjLGbjLFExtgyLccZY+zTmuOXGWPNWg16QZaO3r17Y8GCBSgpKcHo0aPx22+/8SEiDofD6US02RAcY0wPQAKAoQAyAMQAeJKIrquVGQFgAYARAHoB+ISI6mczVK/X3p4EuXlQgODi4oLPPvsMY8eO5c6Hw+FwGqEjDsG1ZQ+oJ4BEIrpNRFIA2wGMrVNmLICfatZJ/QvAkjHm2GithYVgYFi0aBHi4+Mxbtw47nw4HA6nE9KWc0DOANLV9jOg7OU0VcYZgMaKNcbYHABzanar5MDVDz/8UGter4cMWwANyzc+XPC2uAtvi7vwtrhLh1uI1ZYOSFu3pO54X3PKgIi+BvA1ADDGYjtaN1JX8La4C2+Lu/C2uAtvi7swxmKbLtW+tOUQXAaAbmr7LgDqimI0pwyHw+FwHkDa0gHFAOjOGPNgjBkAmApgX50y+wDMqImG6w1ATEQPX8IoDofDeQhpsyE4IpIxxl4EcAiAHoDviegaY2xuzfEvARyAMgIuEUAFgFnNqPrrNjK5M8Lb4i68Le7C2+IuvC3u0uHaotNlQuBwOBzOgwFPGcDhcDgcncAdEIfD4XB0Qod1QG2Vxqcz0oy2mFbTBpcZY2cYYyJd2NkeNNUWauUiGWNyxtjE9rSvPWlOWzDGBjLGLjHGrjHGTrS3je1FM/5GLBhj+xljcTVt0Zz55k4HY+x7xlguY+xqA8c71nNT15Ks2jYogxaSAHgCMAAQB8C/TpkRAP6Cci1RbwD/6dpuHbZFXwBWNf9//GFuC7VyR6EMcpmoa7t1+LuwBPD/7Z17iF3VFcZ/HybRONFRmioqxIg1tQFjNKEgNalWSdVCaQgSqlUigqL1hQ8CKha09dHkj1aktDWVhFaiGKOC4iOKwwRNGjUZk/EVxIgoiE+0oxGMfv6x9zWX8U7uSTP3nDPX9YPDee199jpr7ux11z7nfvtlYEreP6hquyv0xbXAbXn7h8BHwISqbe+AL+YCxwODI5yvVb9Z1wyoMzI+Y5O2vrD9rO2P8+560u+pupEinwtI+oL3A++VaVzJFPHFWcBq228B2O5WfxTxhYH9lHS7JpEC0I5yzew8tvtJ9zYSteo36xqARpLo2d0y3cDu3uf5pG843UhbX0g6DJgP/L1Eu6qgyOdiGnCgpD5JL0g6tzTryqWIL+4AfkL6ofsW4HLbX5djXq2oVb9Z1/mARk3GpwsofJ+STiYFoBM7alF1FPHFX4DFtr/qcpHaIr4YB8wCTgEmAuskrbe9tdPGlUwRX/wSGAB+ARwJrJG01vanHbatbtSq36xrAAoZn50Uuk9JM4BlwOm2PyzJtrIp4ovZwD05+EwGzpC0w/aDpVhYHkX/Rz6w/RnwmaR+4FjSNCndRBFfnAfc6vQg5HVJ24CjgQ3lmFgbatVv1nUILmR8dtLWF5KmAKuBc7rw220zbX1h+wjbU21PBVYBF3dh8IFi/yMPAXMkjZO0L0mN/pWS7SyDIr54i5QJIulgkjL0G6VaWQ9q1W/WMgNy52R8xhwFfXED8APgb/mb/w53oQJwQV98LyjiC9uvSHoM2Ax8DSyz3fL13LFMwc/FTcBySVtIw1CLbXfdNA2SVgInAZMlvQ38ARgP9ew3Q4onCIIgqIS6DsEFQRAEXU4EoCAIgqASIgAFQRAElRABKAiCIKiECEBBEARBJUQACmpJVrIeaFqm7qLs0Ci0t1zSttzWRkkn/B/XWCZpet6+dti5Z/fUxnydhl8Gs7rzAW3Kz5R0xmi0HQSjTbyGHdQSSUO2J4122V1cYznwsO1VkuYBS23P2IPr7bFN7a4raQWw1fafdlF+ETDb9iWjbUsQ7CmRAQVjAkmTJD2Vs5Mtkr6jgi3pEEn9TRnCnHx8nqR1ue59ktoFhn7gR7nulflag5KuyMd6JD2S55YZlLQwH++TNFvSrcDEbMfd+dxQXt/bnJHkzGuBpL0kLZH0nNI8LRcWcMs6spCkpJ8qzQW1Ka9/nFUBbgQWZlsWZtvvyu1sauXHICiNKueCiCWWkRbgK5J45ADwAEm1Y/98bjLpl9yNDH4or68CrsvbewH75bL9QE8+vhi4oUV7y8lzBwFnAv8lCXluAXpIEv4vAccBC4A7m+r25nUfKdv41qamMg0b5wMr8vYEkjLxROAC4Pp8fG/geeCIFnYONd3ffcBpeX9/YFzePhW4P28vAu5oqn8z8Lu8fQBJF66n6r93LN/PpZZSPEEAbLc9s7EjaTxws6S5JFmZw4CDgXeb6jwH3JXLPmh7QNLPgenAM1mmaAIpc2jFEknXA++TVMVPAR5wEvNE0mpgDvAYsFTSbaRhu7W7cV+PArdL2hs4Dei3vT0P+83Qzhlce4GjgG3D6k+UNABMBV4A1jSVXyHpKJK68fgR2p8H/FrS1Xl/H2AK3akRF9ScCEDBWOFs0kyWs2x/KelNUuf5Lbb7c4D6FfBvSUuAj4E1tn9boI1rbK9q7Eg6tVUh21slzSJpat0i6QnbNxa5CdtfSOojTQ+wEFjZaA641PbjbS6x3fZMSb3Aw8DvgdtJWmdP256fX9joG6G+gAW2XytibxB0kngGFIwVeoH3cvA5GTh8eAFJh+cydwL/Ik1NvB74maTGM519JU0r2GY/8Jtcp4c0fLZW0qHA57b/AyzN7Qzny5yJteIekgjkHJKAJnl9UaOOpGm5zZbY/gS4DLg61+kF3smnFzUV/R9pKLLB48ClyumgpONGaiMIOk0EoGCscDcwW9LzpGzo1RZlTgIGJG0iPaf5q+33SR3ySkmbSQHp6CIN2t5Ieja0gfRMaJntTcAxwIY8FHYd8McW1f8JbG68hDCMJ4C5wJNOU0hDmsvpZWCjpEHgH7QZoci2vEiafuDPpGzsGdLzoQZPA9MbLyGQMqXx2bbBvB8ElRCvYQdBEASVEBlQEARBUAkRgIIgCIJKiAAUBEEQVEIEoCAIgqASIgAFQRAElRABKAiCIKiECEBBEARBJXwDPOTPShRtL80AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "from itertools import cycle\n",
    "\n",
    "n_classes = len(classes)\n",
    "# First aggregate all false positive rates\n",
    "all_fpr = np.unique(np.concatenate([fpr[i] for i in range(n_classes)]))\n",
    "\n",
    "# Then interpolate all ROC curves at this points\n",
    "mean_tpr = np.zeros_like(all_fpr)\n",
    "for i in range(n_classes):\n",
    "    mean_tpr += np.interp(all_fpr, fpr[i], tpr[i])\n",
    "\n",
    "# Finally average it and compute AUC\n",
    "mean_tpr /= n_classes\n",
    "\n",
    "fpr[\"macro\"] = all_fpr\n",
    "tpr[\"macro\"] = mean_tpr\n",
    "roc_auc[\"macro\"] = auc(fpr[\"macro\"], tpr[\"macro\"])\n",
    "\n",
    "# Plot all ROC curves\n",
    "plt.figure()\n",
    "plt.plot(\n",
    "    fpr[\"micro\"],\n",
    "    tpr[\"micro\"],\n",
    "    label=\"micro-average ROC curve (area = {0:0.2f})\".format(roc_auc[\"micro\"]),\n",
    "    color=\"deeppink\",\n",
    "    linestyle=\":\",\n",
    "    linewidth=4,\n",
    ")\n",
    "\n",
    "plt.plot(\n",
    "    fpr[\"macro\"],\n",
    "    tpr[\"macro\"],\n",
    "    label=\"macro-average ROC curve (area = {0:0.2f})\".format(roc_auc[\"macro\"]),\n",
    "    color=\"navy\",\n",
    "    linestyle=\":\",\n",
    "    linewidth=4,\n",
    ")\n",
    "\n",
    "colors = cycle([\"aqua\", \"darkorange\", \"cornflowerblue\"])\n",
    "for i, color in zip(range(n_classes), colors):\n",
    "    plt.plot(\n",
    "        fpr[i],\n",
    "        tpr[i],\n",
    "        color=color,\n",
    "        lw=lw,\n",
    "        label=\"ROC curve of class {0} (area = {1:0.2f})\".format(i, roc_auc[i]),\n",
    "    )\n",
    "\n",
    "plt.plot([0, 1], [0, 1], \"k--\", lw=lw)\n",
    "plt.xlim([0.0, 1.0])\n",
    "plt.ylim([0.0, 1.05])\n",
    "plt.xlabel(\"False Positive Rate\")\n",
    "plt.ylabel(\"True Positive Rate\")\n",
    "plt.title(\"Some extension of Receiver operating characteristic to multiclass\")\n",
    "plt.legend(loc=\"lower right\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c62c518a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "b54e096d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "One-vs-One ROC AUC scores:\n",
      "0.999913 (macro),\n",
      "0.999904 (weighted by prevalence)\n",
      "One-vs-Rest ROC AUC scores:\n",
      "0.992141 (macro),\n",
      "0.999904 (weighted by prevalence)\n"
     ]
    }
   ],
   "source": [
    "macro_roc_auc_ovo = roc_auc_score(y_true, y_score, multi_class=\"ovo\", average=\"macro\")\n",
    "weighted_roc_auc_ovo = roc_auc_score(\n",
    "    y_true, y_score, multi_class=\"ovo\", average=\"weighted\"\n",
    ")\n",
    "macro_roc_auc_ovr = roc_auc_score(y_test, y_score, multi_class=\"ovr\", average=\"macro\")\n",
    "weighted_roc_auc_ovr = roc_auc_score(\n",
    "    y_true, y_score, multi_class=\"ovr\", average=\"weighted\"\n",
    ")\n",
    "print(\n",
    "    \"One-vs-One ROC AUC scores:\\n{:.6f} (macro),\\n{:.6f} \"\n",
    "    \"(weighted by prevalence)\".format(macro_roc_auc_ovo, weighted_roc_auc_ovo)\n",
    ")\n",
    "print(\n",
    "    \"One-vs-Rest ROC AUC scores:\\n{:.6f} (macro),\\n{:.6f} \"\n",
    "    \"(weighted by prevalence)\".format(macro_roc_auc_ovr, weighted_roc_auc_ovr)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43ea82fa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "824d7f4f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "dcbc59ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cell_type</th>\n",
       "      <th>barcode</th>\n",
       "      <th>sample</th>\n",
       "      <th>n_counts</th>\n",
       "      <th>size_factors</th>\n",
       "      <th>predicted_cell_type</th>\n",
       "      <th>probability B cell</th>\n",
       "      <th>probability CD4 T cell</th>\n",
       "      <th>probability CD8 T cell</th>\n",
       "      <th>probability Dendritic</th>\n",
       "      <th>probability Endothelial</th>\n",
       "      <th>probability Erythrocyte</th>\n",
       "      <th>probability Fibroblast</th>\n",
       "      <th>probability Malignant</th>\n",
       "      <th>probability Mast</th>\n",
       "      <th>probability Monocyte</th>\n",
       "      <th>probability NK</th>\n",
       "      <th>probability Plasma</th>\n",
       "      <th>probability Treg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_AACCAACTCCACTAGA-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACCAACTCCACTAGA-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4712.0</td>\n",
       "      <td>1.213182</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-1.645374</td>\n",
       "      <td>-0.798354</td>\n",
       "      <td>-1.320047</td>\n",
       "      <td>-0.613273</td>\n",
       "      <td>9.651504</td>\n",
       "      <td>0.046297</td>\n",
       "      <td>-3.343092</td>\n",
       "      <td>-0.727528</td>\n",
       "      <td>-1.168746</td>\n",
       "      <td>-0.569176</td>\n",
       "      <td>-1.320954</td>\n",
       "      <td>-0.929309</td>\n",
       "      <td>-1.167010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACCCAAAGACGAGCT-1</th>\n",
       "      <td>NK</td>\n",
       "      <td>AACCCAAAGACGAGCT-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4732.0</td>\n",
       "      <td>1.218332</td>\n",
       "      <td>NK</td>\n",
       "      <td>-0.844792</td>\n",
       "      <td>-1.667611</td>\n",
       "      <td>-1.252446</td>\n",
       "      <td>-0.681986</td>\n",
       "      <td>-0.991126</td>\n",
       "      <td>-1.047975</td>\n",
       "      <td>-1.393238</td>\n",
       "      <td>0.279072</td>\n",
       "      <td>-0.840426</td>\n",
       "      <td>-0.061285</td>\n",
       "      <td>10.557428</td>\n",
       "      <td>-1.192751</td>\n",
       "      <td>-1.279777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACGTCAGTTTACACG-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AACGTCAGTTTACACG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3296.0</td>\n",
       "      <td>0.848610</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-0.874611</td>\n",
       "      <td>-1.563723</td>\n",
       "      <td>-0.647155</td>\n",
       "      <td>-1.483668</td>\n",
       "      <td>7.808266</td>\n",
       "      <td>-0.778172</td>\n",
       "      <td>-0.275812</td>\n",
       "      <td>-0.957157</td>\n",
       "      <td>-0.471111</td>\n",
       "      <td>-0.729114</td>\n",
       "      <td>-1.426803</td>\n",
       "      <td>-0.044268</td>\n",
       "      <td>-0.892382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AACTTCTGTCCAGAAG-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>AACTTCTGTCCAGAAG-1</td>\n",
       "      <td>0</td>\n",
       "      <td>3202.0</td>\n",
       "      <td>0.824408</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>-1.695771</td>\n",
       "      <td>-0.758672</td>\n",
       "      <td>1.230153</td>\n",
       "      <td>-1.391558</td>\n",
       "      <td>-1.419928</td>\n",
       "      <td>-0.788930</td>\n",
       "      <td>-0.927254</td>\n",
       "      <td>-0.453907</td>\n",
       "      <td>-0.026667</td>\n",
       "      <td>-0.981914</td>\n",
       "      <td>-0.534299</td>\n",
       "      <td>-1.087195</td>\n",
       "      <td>-2.054266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0_AAGCCATAGTGATTCC-1</th>\n",
       "      <td>Endothelial</td>\n",
       "      <td>AAGCCATAGTGATTCC-1</td>\n",
       "      <td>0</td>\n",
       "      <td>4088.0</td>\n",
       "      <td>1.052523</td>\n",
       "      <td>Endothelial</td>\n",
       "      <td>-0.334221</td>\n",
       "      <td>-1.272656</td>\n",
       "      <td>-0.656085</td>\n",
       "      <td>-1.552842</td>\n",
       "      <td>14.972706</td>\n",
       "      <td>-0.857209</td>\n",
       "      <td>-1.206331</td>\n",
       "      <td>-0.683234</td>\n",
       "      <td>-0.559324</td>\n",
       "      <td>-1.068178</td>\n",
       "      <td>-0.820980</td>\n",
       "      <td>-1.125220</td>\n",
       "      <td>-1.229909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTATGCCAGCCCAGT-1</th>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>TTTATGCCAGCCCAGT-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4480.0</td>\n",
       "      <td>1.153450</td>\n",
       "      <td>Fibroblast</td>\n",
       "      <td>-1.617480</td>\n",
       "      <td>-0.865152</td>\n",
       "      <td>-1.466225</td>\n",
       "      <td>-0.761909</td>\n",
       "      <td>-1.233621</td>\n",
       "      <td>-0.567406</td>\n",
       "      <td>4.051761</td>\n",
       "      <td>0.728402</td>\n",
       "      <td>-0.353512</td>\n",
       "      <td>-1.202108</td>\n",
       "      <td>-0.631813</td>\n",
       "      <td>-1.264545</td>\n",
       "      <td>-0.332282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCATGGTCGACTTA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCATGGTCGACTTA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4580.0</td>\n",
       "      <td>1.179197</td>\n",
       "      <td>Treg</td>\n",
       "      <td>-1.489965</td>\n",
       "      <td>-0.297774</td>\n",
       "      <td>-0.895083</td>\n",
       "      <td>-1.135407</td>\n",
       "      <td>-0.943314</td>\n",
       "      <td>-0.690525</td>\n",
       "      <td>-1.695587</td>\n",
       "      <td>-2.054170</td>\n",
       "      <td>-1.055325</td>\n",
       "      <td>-0.272783</td>\n",
       "      <td>-1.668621</td>\n",
       "      <td>-0.500289</td>\n",
       "      <td>6.091204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTCCTCGTTAGCGGA-1</th>\n",
       "      <td>Treg</td>\n",
       "      <td>TTTCCTCGTTAGCGGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4526.0</td>\n",
       "      <td>1.165293</td>\n",
       "      <td>Treg</td>\n",
       "      <td>-1.554138</td>\n",
       "      <td>-1.126189</td>\n",
       "      <td>0.067056</td>\n",
       "      <td>-1.275277</td>\n",
       "      <td>-1.274302</td>\n",
       "      <td>-1.661355</td>\n",
       "      <td>-1.579366</td>\n",
       "      <td>-0.442887</td>\n",
       "      <td>0.256902</td>\n",
       "      <td>-0.620646</td>\n",
       "      <td>-1.139950</td>\n",
       "      <td>-1.634987</td>\n",
       "      <td>5.346128</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGACTCATGGGATG-1</th>\n",
       "      <td>Plasma</td>\n",
       "      <td>TTTGACTCATGGGATG-1</td>\n",
       "      <td>29</td>\n",
       "      <td>4452.0</td>\n",
       "      <td>1.146241</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>-1.184166</td>\n",
       "      <td>-1.148166</td>\n",
       "      <td>-1.161676</td>\n",
       "      <td>-0.978821</td>\n",
       "      <td>-1.056848</td>\n",
       "      <td>-1.118930</td>\n",
       "      <td>-0.771188</td>\n",
       "      <td>-0.980366</td>\n",
       "      <td>-0.481189</td>\n",
       "      <td>-1.247963</td>\n",
       "      <td>-0.962314</td>\n",
       "      <td>9.834721</td>\n",
       "      <td>-1.319493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29_TTTGGTTCAGTTTCGA-1</th>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>TTTGGTTCAGTTTCGA-1</td>\n",
       "      <td>29</td>\n",
       "      <td>3414.0</td>\n",
       "      <td>0.878991</td>\n",
       "      <td>CD8 T cell</td>\n",
       "      <td>-0.817018</td>\n",
       "      <td>-1.149525</td>\n",
       "      <td>1.255636</td>\n",
       "      <td>-1.025783</td>\n",
       "      <td>-0.896883</td>\n",
       "      <td>-0.954583</td>\n",
       "      <td>-1.481391</td>\n",
       "      <td>-0.913094</td>\n",
       "      <td>-0.755771</td>\n",
       "      <td>-0.679273</td>\n",
       "      <td>-1.546723</td>\n",
       "      <td>-0.758668</td>\n",
       "      <td>-1.101037</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5748 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         cell_type             barcode  sample  n_counts  \\\n",
       "0_AACCAACTCCACTAGA-1   Endothelial  AACCAACTCCACTAGA-1       0    4712.0   \n",
       "0_AACCCAAAGACGAGCT-1            NK  AACCCAAAGACGAGCT-1       0    4732.0   \n",
       "0_AACGTCAGTTTACACG-1   Endothelial  AACGTCAGTTTACACG-1       0    3296.0   \n",
       "0_AACTTCTGTCCAGAAG-1    CD8 T cell  AACTTCTGTCCAGAAG-1       0    3202.0   \n",
       "0_AAGCCATAGTGATTCC-1   Endothelial  AAGCCATAGTGATTCC-1       0    4088.0   \n",
       "...                            ...                 ...     ...       ...   \n",
       "29_TTTATGCCAGCCCAGT-1   Fibroblast  TTTATGCCAGCCCAGT-1      29    4480.0   \n",
       "29_TTTCATGGTCGACTTA-1         Treg  TTTCATGGTCGACTTA-1      29    4580.0   \n",
       "29_TTTCCTCGTTAGCGGA-1         Treg  TTTCCTCGTTAGCGGA-1      29    4526.0   \n",
       "29_TTTGACTCATGGGATG-1       Plasma  TTTGACTCATGGGATG-1      29    4452.0   \n",
       "29_TTTGGTTCAGTTTCGA-1   CD8 T cell  TTTGGTTCAGTTTCGA-1      29    3414.0   \n",
       "\n",
       "                       size_factors predicted_cell_type  probability B cell  \\\n",
       "0_AACCAACTCCACTAGA-1       1.213182         Endothelial           -1.645374   \n",
       "0_AACCCAAAGACGAGCT-1       1.218332                  NK           -0.844792   \n",
       "0_AACGTCAGTTTACACG-1       0.848610         Endothelial           -0.874611   \n",
       "0_AACTTCTGTCCAGAAG-1       0.824408          CD8 T cell           -1.695771   \n",
       "0_AAGCCATAGTGATTCC-1       1.052523         Endothelial           -0.334221   \n",
       "...                             ...                 ...                 ...   \n",
       "29_TTTATGCCAGCCCAGT-1      1.153450          Fibroblast           -1.617480   \n",
       "29_TTTCATGGTCGACTTA-1      1.179197                Treg           -1.489965   \n",
       "29_TTTCCTCGTTAGCGGA-1      1.165293                Treg           -1.554138   \n",
       "29_TTTGACTCATGGGATG-1      1.146241              Plasma           -1.184166   \n",
       "29_TTTGGTTCAGTTTCGA-1      0.878991          CD8 T cell           -0.817018   \n",
       "\n",
       "                       probability CD4 T cell  probability CD8 T cell  \\\n",
       "0_AACCAACTCCACTAGA-1                -0.798354               -1.320047   \n",
       "0_AACCCAAAGACGAGCT-1                -1.667611               -1.252446   \n",
       "0_AACGTCAGTTTACACG-1                -1.563723               -0.647155   \n",
       "0_AACTTCTGTCCAGAAG-1                -0.758672                1.230153   \n",
       "0_AAGCCATAGTGATTCC-1                -1.272656               -0.656085   \n",
       "...                                       ...                     ...   \n",
       "29_TTTATGCCAGCCCAGT-1               -0.865152               -1.466225   \n",
       "29_TTTCATGGTCGACTTA-1               -0.297774               -0.895083   \n",
       "29_TTTCCTCGTTAGCGGA-1               -1.126189                0.067056   \n",
       "29_TTTGACTCATGGGATG-1               -1.148166               -1.161676   \n",
       "29_TTTGGTTCAGTTTCGA-1               -1.149525                1.255636   \n",
       "\n",
       "                       probability Dendritic  probability Endothelial  \\\n",
       "0_AACCAACTCCACTAGA-1               -0.613273                 9.651504   \n",
       "0_AACCCAAAGACGAGCT-1               -0.681986                -0.991126   \n",
       "0_AACGTCAGTTTACACG-1               -1.483668                 7.808266   \n",
       "0_AACTTCTGTCCAGAAG-1               -1.391558                -1.419928   \n",
       "0_AAGCCATAGTGATTCC-1               -1.552842                14.972706   \n",
       "...                                      ...                      ...   \n",
       "29_TTTATGCCAGCCCAGT-1              -0.761909                -1.233621   \n",
       "29_TTTCATGGTCGACTTA-1              -1.135407                -0.943314   \n",
       "29_TTTCCTCGTTAGCGGA-1              -1.275277                -1.274302   \n",
       "29_TTTGACTCATGGGATG-1              -0.978821                -1.056848   \n",
       "29_TTTGGTTCAGTTTCGA-1              -1.025783                -0.896883   \n",
       "\n",
       "                       probability Erythrocyte  probability Fibroblast  \\\n",
       "0_AACCAACTCCACTAGA-1                  0.046297               -3.343092   \n",
       "0_AACCCAAAGACGAGCT-1                 -1.047975               -1.393238   \n",
       "0_AACGTCAGTTTACACG-1                 -0.778172               -0.275812   \n",
       "0_AACTTCTGTCCAGAAG-1                 -0.788930               -0.927254   \n",
       "0_AAGCCATAGTGATTCC-1                 -0.857209               -1.206331   \n",
       "...                                        ...                     ...   \n",
       "29_TTTATGCCAGCCCAGT-1                -0.567406                4.051761   \n",
       "29_TTTCATGGTCGACTTA-1                -0.690525               -1.695587   \n",
       "29_TTTCCTCGTTAGCGGA-1                -1.661355               -1.579366   \n",
       "29_TTTGACTCATGGGATG-1                -1.118930               -0.771188   \n",
       "29_TTTGGTTCAGTTTCGA-1                -0.954583               -1.481391   \n",
       "\n",
       "                       probability Malignant  probability Mast  \\\n",
       "0_AACCAACTCCACTAGA-1               -0.727528         -1.168746   \n",
       "0_AACCCAAAGACGAGCT-1                0.279072         -0.840426   \n",
       "0_AACGTCAGTTTACACG-1               -0.957157         -0.471111   \n",
       "0_AACTTCTGTCCAGAAG-1               -0.453907         -0.026667   \n",
       "0_AAGCCATAGTGATTCC-1               -0.683234         -0.559324   \n",
       "...                                      ...               ...   \n",
       "29_TTTATGCCAGCCCAGT-1               0.728402         -0.353512   \n",
       "29_TTTCATGGTCGACTTA-1              -2.054170         -1.055325   \n",
       "29_TTTCCTCGTTAGCGGA-1              -0.442887          0.256902   \n",
       "29_TTTGACTCATGGGATG-1              -0.980366         -0.481189   \n",
       "29_TTTGGTTCAGTTTCGA-1              -0.913094         -0.755771   \n",
       "\n",
       "                       probability Monocyte  probability NK  \\\n",
       "0_AACCAACTCCACTAGA-1              -0.569176       -1.320954   \n",
       "0_AACCCAAAGACGAGCT-1              -0.061285       10.557428   \n",
       "0_AACGTCAGTTTACACG-1              -0.729114       -1.426803   \n",
       "0_AACTTCTGTCCAGAAG-1              -0.981914       -0.534299   \n",
       "0_AAGCCATAGTGATTCC-1              -1.068178       -0.820980   \n",
       "...                                     ...             ...   \n",
       "29_TTTATGCCAGCCCAGT-1             -1.202108       -0.631813   \n",
       "29_TTTCATGGTCGACTTA-1             -0.272783       -1.668621   \n",
       "29_TTTCCTCGTTAGCGGA-1             -0.620646       -1.139950   \n",
       "29_TTTGACTCATGGGATG-1             -1.247963       -0.962314   \n",
       "29_TTTGGTTCAGTTTCGA-1             -0.679273       -1.546723   \n",
       "\n",
       "                       probability Plasma  probability Treg  \n",
       "0_AACCAACTCCACTAGA-1            -0.929309         -1.167010  \n",
       "0_AACCCAAAGACGAGCT-1            -1.192751         -1.279777  \n",
       "0_AACGTCAGTTTACACG-1            -0.044268         -0.892382  \n",
       "0_AACTTCTGTCCAGAAG-1            -1.087195         -2.054266  \n",
       "0_AAGCCATAGTGATTCC-1            -1.125220         -1.229909  \n",
       "...                                   ...               ...  \n",
       "29_TTTATGCCAGCCCAGT-1           -1.264545         -0.332282  \n",
       "29_TTTCATGGTCGACTTA-1           -0.500289          6.091204  \n",
       "29_TTTCCTCGTTAGCGGA-1           -1.634987          5.346128  \n",
       "29_TTTGACTCATGGGATG-1            9.834721         -1.319493  \n",
       "29_TTTGGTTCAGTTTCGA-1           -0.758668         -1.101037  \n",
       "\n",
       "[5748 rows x 19 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = sc.read_h5ad('/data/pollock/benchmarking/results/svm_linear/scRNAseq_brca.h5ad')\n",
    "a.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "c51f666a",
   "metadata": {},
   "outputs": [],
   "source": [
    "classes = sorted(set(a.obs['cell_type']))\n",
    "y_pred = [classes.index(l) for l in a.obs['predicted_cell_type']]\n",
    "y_score = a.obs[[c for c in a.obs.columns if 'probability ' in c]].values\n",
    "y_true = [classes.index(l) for l in a.obs['cell_type']]\n",
    "\n",
    "y_pred = label_binarize(y_pred, classes=np.arange(len(classes)))\n",
    "y_true = label_binarize(y_pred, classes=np.arange(len(classes)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "a9097894",
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Target scores need to be probabilities for multiclass roc_auc, i.e. they should sum up to 1.0 over classes",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_11826/3346692495.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_score\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmulti_class\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"ovo\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maverage\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"weighted\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m )\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0mmacro_roc_auc_ovr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mroc_auc_score\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my_test\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_score\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmulti_class\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"ovr\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maverage\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"macro\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m weighted_roc_auc_ovr = roc_auc_score(\n\u001b[1;32m      7\u001b[0m     \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_score\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmulti_class\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"ovr\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maverage\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"weighted\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_ranking.py\u001b[0m in \u001b[0;36mroc_auc_score\u001b[0;34m(y_true, y_score, average, sample_weight, max_fpr, multi_class, labels)\u001b[0m\n\u001b[1;32m    559\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmulti_class\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"raise\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    560\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"multi_class must be in ('ovo', 'ovr')\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 561\u001b[0;31m         return _multiclass_roc_auc_score(\n\u001b[0m\u001b[1;32m    562\u001b[0m             \u001b[0my_true\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_score\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmulti_class\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maverage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msample_weight\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    563\u001b[0m         )\n",
      "\u001b[0;32m/data/estorrs/miniconda3/envs/pollock_new/lib/python3.8/site-packages/sklearn/metrics/_ranking.py\u001b[0m in \u001b[0;36m_multiclass_roc_auc_score\u001b[0;34m(y_true, y_score, labels, multi_class, average, sample_weight)\u001b[0m\n\u001b[1;32m    626\u001b[0m     \u001b[0;31m# validation of the input y_score\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    627\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mallclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my_score\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 628\u001b[0;31m         raise ValueError(\n\u001b[0m\u001b[1;32m    629\u001b[0m             \u001b[0;34m\"Target scores need to be probabilities for multiclass \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    630\u001b[0m             \u001b[0;34m\"roc_auc, i.e. they should sum up to 1.0 over classes\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Target scores need to be probabilities for multiclass roc_auc, i.e. they should sum up to 1.0 over classes"
     ]
    }
   ],
   "source": [
    "macro_roc_auc_ovo = roc_auc_score(y_true, y_score, multi_class=\"ovo\", average=\"macro\")\n",
    "weighted_roc_auc_ovo = roc_auc_score(\n",
    "    y_true, y_score, multi_class=\"ovo\", average=\"weighted\"\n",
    ")\n",
    "macro_roc_auc_ovr = roc_auc_score(y_test, y_score, multi_class=\"ovr\", average=\"macro\")\n",
    "weighted_roc_auc_ovr = roc_auc_score(\n",
    "    y_true, y_score, multi_class=\"ovr\", average=\"weighted\"\n",
    ")\n",
    "print(\n",
    "    \"One-vs-One ROC AUC scores:\\n{:.6f} (macro),\\n{:.6f} \"\n",
    "    \"(weighted by prevalence)\".format(macro_roc_auc_ovo, weighted_roc_auc_ovo)\n",
    ")\n",
    "print(\n",
    "    \"One-vs-Rest ROC AUC scores:\\n{:.6f} (macro),\\n{:.6f} \"\n",
    "    \"(weighted by prevalence)\".format(macro_roc_auc_ovr, weighted_roc_auc_ovr)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2b86cac",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
