{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a5303eeb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Attaching SeuratObject\n",
      "\n",
      "Registered S3 method overwritten by 'SeuratDisk':\n",
      "  method            from  \n",
      "  as.sparse.H5Group Seurat\n",
      "\n",
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: dplyr\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Loading required package: pheatmap\n",
      "\n",
      "Loading required package: reshape2\n",
      "\n",
      "Loading required package: cowplot\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(Seurat)\n",
    "library(SeuratDisk)\n",
    "library(singleCellNet)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "49fb2126",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "addf418a",
   "metadata": {},
   "source": [
    "cross fold "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "93da6b48",
   "metadata": {},
   "outputs": [],
   "source": [
    "fmap = list(\n",
    "    list(\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/cesc_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/hnscc_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/melanoma_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/myeloma_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/pdac_fold4_val.rds')\n",
    "        )\n",
    "    ),\n",
    "    list(\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/brca_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/ccrcc_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snRNAseq/gbm_fold4_val.rds')\n",
    "        )\n",
    "    ),\n",
    "    list(\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/brca_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/ccrcc_fold4_val.rds')\n",
    "        ),\n",
    "        list(\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold0_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold0_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold1_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold1_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold2_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold2_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold3_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold3_val.rds'),\n",
    "            list('/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold4_train.rds',\n",
    "             '/data/pollock/benchmarking/pollock_datasets_with_folds/snATACseq/gbm_fold4_val.rds')\n",
    "        )\n",
    "    )\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "62095afe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  207  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 323 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  204  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 323 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  210  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 324 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  209  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 321 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  209  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 325 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"cesc_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  196  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"cesc_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  198  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"brca_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  210  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 324 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"cesc_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  197  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"cesc_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  195  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"hnscc_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  188  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"hnscc_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  186  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"hnscc_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  189  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"hnscc_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  188  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"hnscc_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  185  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 275 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"melanoma_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  162  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  NK \n",
      "There are 250 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"melanoma_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  164  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  NK \n",
      "There are 249 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"melanoma_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  164  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  NK \n",
      "There are 250 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"melanoma_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  163  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  NK \n",
      "There are 249 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"melanoma_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  163  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  NK \n",
      "There are 248 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"myeloma_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  177  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Platlete \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Plasma \n",
      "There are 273 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"myeloma_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  182  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Platlete \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Plasma \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"myeloma_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  178  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Platlete \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Plasma \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"myeloma_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  179  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Platlete \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Plasma \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"myeloma_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  173  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Platlete \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Plasma \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"pdac_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  250  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Tuft \n",
      "nPairs =  190  for  Acinar \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Islet \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Mast \n",
      "There are 419 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"pdac_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  249  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Tuft \n",
      "nPairs =  190  for  Acinar \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Islet \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Mast \n",
      "There are 416 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"pdac_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  248  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Tuft \n",
      "nPairs =  190  for  Acinar \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Islet \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Mast \n",
      "There are 415 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"pdac_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  249  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Tuft \n",
      "nPairs =  190  for  Acinar \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Islet \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Mast \n",
      "There are 420 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"scRNAseq\"\n",
      "[1] \"pdac_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  249  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Tuft \n",
      "nPairs =  190  for  Acinar \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Erythrocyte \n",
      "nPairs =  190  for  Islet \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Mast \n",
      "There are 420 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"brca_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  196  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Adipocyte \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Plasma \n",
      "There are 323 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"brca_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  189  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Adipocyte \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Plasma \n",
      "There are 321 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"brca_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  193  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Adipocyte \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Plasma \n",
      "There are 320 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"brca_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  193  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Adipocyte \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Plasma \n",
      "There are 321 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"brca_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  193  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Adipocyte \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Plasma \n",
      "There are 321 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"ccrcc_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  171  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Epithelial \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"ccrcc_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  171  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Epithelial \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"ccrcc_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  173  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Epithelial \n",
      "There are 273 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"ccrcc_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  171  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Epithelial \n",
      "There are 274 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"ccrcc_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  172  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Epithelial \n",
      "There are 273 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"gbm_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  162  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Neuron \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Monocyte \n",
      "There are 225 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"gbm_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  160  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Neuron \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Monocyte \n",
      "There are 225 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"gbm_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  158  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Neuron \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Monocyte \n",
      "There are 225 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"gbm_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  164  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Neuron \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Monocyte \n",
      "There are 225 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snRNAseq\"\n",
      "[1] \"gbm_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  160  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Neuron \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Monocyte \n",
      "There are 225 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"brca_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  146  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Mast \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"brca_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  147  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Mast \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"brca_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  149  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Mast \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"brca_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  144  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Mast \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"brca_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  147  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Mast \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"ccrcc_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  111  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "There are 150 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"ccrcc_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  112  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "There are 150 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"ccrcc_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  115  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "There are 150 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"ccrcc_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  109  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "There are 150 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"ccrcc_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  115  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Epithelial \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "There are 150 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"gbm_fold0\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  150  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Neuron \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"gbm_fold1\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  151  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Neuron \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"gbm_fold2\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  151  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Neuron \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"gbm_fold3\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  150  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Neuron \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n",
      "[1] \"snATACseq\"\n",
      "[1] \"gbm_fold4\"\n",
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  151  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  Microglia \n",
      "nPairs =  190  for  Oligodendrocytes \n",
      "nPairs =  190  for  T cells \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Neuron \n",
      "There are 200 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n",
      "Loaded in the cnProc\n",
      "All Done\n"
     ]
    }
   ],
   "source": [
    "for (dtype_list in fmap) {\n",
    "    for (d1 in dtype_list) {\n",
    "        for (disease_list in d1) {\n",
    "            train_fp = disease_list[[1]]\n",
    "            val_fp = disease_list[[2]]\n",
    "\n",
    "            dtype = strsplit(strsplit(train_fp, \"_train\")[[1]][[1]], \"/\")[[1]][[6]]\n",
    "            disease = strsplit(strsplit(train_fp, \"_train\")[[1]][[1]], \"/\")[[1]][[7]]\n",
    "\n",
    "            print(dtype)\n",
    "            print(disease)\n",
    "\n",
    "            train <- readRDS(train_fp)\n",
    "            val <- readRDS(val_fp)\n",
    "\n",
    "            seuratfile = extractSeurat(train, exp_slot_name = \"counts\")\n",
    "            train_sampTab = seuratfile$sampTab\n",
    "            train_sampTab$cell_ids = rownames(train_sampTab)\n",
    "            train_expDat = seuratfile$expDat\n",
    "\n",
    "            seuratfile = extractSeurat(val, exp_slot_name = \"counts\")\n",
    "            val_sampTab = seuratfile$sampTab\n",
    "            val_expDat = seuratfile$expDat\n",
    "\n",
    "\n",
    "            class_info<-scn_train(\n",
    "                stTrain = train_sampTab, expTrain = train_expDat, nTopGenes = 10, nRand = 70,\n",
    "                nTrees = 1000, nTopGenePairs = 25, dLevel = \"cell_type\", colName_samp = \"cell_ids\")\n",
    "            classRes_val_all = scn_predict(cnProc=class_info[['cnProc']], expDat=val_expDat, nrand = 50)\n",
    "\n",
    "            txt_fp = paste(\"/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/\", dtype, \"_\", disease, \".txt\", sep = \"\")\n",
    "            write.table(classRes_val_all, file = txt_fp, sep=\"\\t\")\n",
    "        }\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af257b6d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e0651d7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8504bb47",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8579815e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58bdd5a0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5da72f4e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79bc9543",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03807c16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5073b859",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "93fe0f38",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "265fd94a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95c4b1b2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2745a464",
   "metadata": {},
   "outputs": [],
   "source": [
    "train = readRDS('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_train.rds')\n",
    "val = readRDS('/data/pollock/benchmarking/pollock_datasets_with_folds/scRNAseq/brca_fold2_val.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f95168e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "seuratfile = extractSeurat(train, exp_slot_name = \"counts\")\n",
    "train_sampTab = seuratfile$sampTab\n",
    "train_sampTab$cell_ids = rownames(train_sampTab)\n",
    "train_expDat = seuratfile$expDat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bd37868c",
   "metadata": {},
   "outputs": [],
   "source": [
    "seuratfile = extractSeurat(val, exp_slot_name = \"counts\")\n",
    "val_sampTab = seuratfile$sampTab\n",
    "val_expDat = seuratfile$expDat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3968d8ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6385 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell_type</th><th scope=col>barcode</th><th scope=col>sample</th><th scope=col>cell_ids</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>18_GGCGTCATCCAACTAG-1</th><td>CD8 T cell</td><td>GGCGTCATCCAACTAG-1</td><td>18</td><td>18_GGCGTCATCCAACTAG-1</td></tr>\n",
       "\t<tr><th scope=row>0_AGTAACCGTGATACTC-1</th><td>CD8 T cell</td><td>AGTAACCGTGATACTC-1</td><td> 0</td><td>0_AGTAACCGTGATACTC-1 </td></tr>\n",
       "\t<tr><th scope=row>15_ATCGGATGTCGCCTAG-1</th><td>CD8 T cell</td><td>ATCGGATGTCGCCTAG-1</td><td>15</td><td>15_ATCGGATGTCGCCTAG-1</td></tr>\n",
       "\t<tr><th scope=row>28_TTGCATTGTACAGTCT-1</th><td>CD8 T cell</td><td>TTGCATTGTACAGTCT-1</td><td>28</td><td>28_TTGCATTGTACAGTCT-1</td></tr>\n",
       "\t<tr><th scope=row>11_TCAAGTGAGGTACCTT-1</th><td>CD8 T cell</td><td>TCAAGTGAGGTACCTT-1</td><td>11</td><td>11_TCAAGTGAGGTACCTT-1</td></tr>\n",
       "\t<tr><th scope=row>28_CTGCCATGTAACTGCT-1</th><td>CD8 T cell</td><td>CTGCCATGTAACTGCT-1</td><td>28</td><td>28_CTGCCATGTAACTGCT-1</td></tr>\n",
       "\t<tr><th scope=row>2_TCCAGAAAGCAGTACG-1</th><td>CD8 T cell</td><td>TCCAGAAAGCAGTACG-1</td><td> 2</td><td>2_TCCAGAAAGCAGTACG-1 </td></tr>\n",
       "\t<tr><th scope=row>29_ACGGAAGCAAGCGATG-1</th><td>CD8 T cell</td><td>ACGGAAGCAAGCGATG-1</td><td>29</td><td>29_ACGGAAGCAAGCGATG-1</td></tr>\n",
       "\t<tr><th scope=row>2_TAGGTTGGTAGTTAGA-1</th><td>CD8 T cell</td><td>TAGGTTGGTAGTTAGA-1</td><td> 2</td><td>2_TAGGTTGGTAGTTAGA-1 </td></tr>\n",
       "\t<tr><th scope=row>29_ATGGATCGTGAGCAGT-1</th><td>CD8 T cell</td><td>ATGGATCGTGAGCAGT-1</td><td>29</td><td>29_ATGGATCGTGAGCAGT-1</td></tr>\n",
       "\t<tr><th scope=row>16_AGTAACCGTGACTCGC-1</th><td>CD8 T cell</td><td>AGTAACCGTGACTCGC-1</td><td>16</td><td>16_AGTAACCGTGACTCGC-1</td></tr>\n",
       "\t<tr><th scope=row>7_TCATGTTCAGCACAGA-1</th><td>CD8 T cell</td><td>TCATGTTCAGCACAGA-1</td><td> 7</td><td>7_TCATGTTCAGCACAGA-1 </td></tr>\n",
       "\t<tr><th scope=row>27_AACAGGGGTACACGCC-1</th><td>CD8 T cell</td><td>AACAGGGGTACACGCC-1</td><td>27</td><td>27_AACAGGGGTACACGCC-1</td></tr>\n",
       "\t<tr><th scope=row>28_GAGTTGTAGATCACTC-1</th><td>CD8 T cell</td><td>GAGTTGTAGATCACTC-1</td><td>28</td><td>28_GAGTTGTAGATCACTC-1</td></tr>\n",
       "\t<tr><th scope=row>28_GTGCACGTCGCCTAGG-1</th><td>CD8 T cell</td><td>GTGCACGTCGCCTAGG-1</td><td>28</td><td>28_GTGCACGTCGCCTAGG-1</td></tr>\n",
       "\t<tr><th scope=row>29_GGGACCTTCGGTGTTA-1</th><td>CD8 T cell</td><td>GGGACCTTCGGTGTTA-1</td><td>29</td><td>29_GGGACCTTCGGTGTTA-1</td></tr>\n",
       "\t<tr><th scope=row>28_GGCTTGGCAAATACAG-1</th><td>CD8 T cell</td><td>GGCTTGGCAAATACAG-1</td><td>28</td><td>28_GGCTTGGCAAATACAG-1</td></tr>\n",
       "\t<tr><th scope=row>27_ATTTCTGGTGATCATC-1</th><td>CD8 T cell</td><td>ATTTCTGGTGATCATC-1</td><td>27</td><td>27_ATTTCTGGTGATCATC-1</td></tr>\n",
       "\t<tr><th scope=row>4_TCGTCCAAGGCATGCA-1</th><td>CD8 T cell</td><td>TCGTCCAAGGCATGCA-1</td><td> 4</td><td>4_TCGTCCAAGGCATGCA-1 </td></tr>\n",
       "\t<tr><th scope=row>28_ATGAGGGGTGCCAAGA-1</th><td>CD8 T cell</td><td>ATGAGGGGTGCCAAGA-1</td><td>28</td><td>28_ATGAGGGGTGCCAAGA-1</td></tr>\n",
       "\t<tr><th scope=row>28_TTTACGTCAAGAATAC-1</th><td>CD8 T cell</td><td>TTTACGTCAAGAATAC-1</td><td>28</td><td>28_TTTACGTCAAGAATAC-1</td></tr>\n",
       "\t<tr><th scope=row>29_GGTGAAGCAGGAGGAG-1</th><td>CD8 T cell</td><td>GGTGAAGCAGGAGGAG-1</td><td>29</td><td>29_GGTGAAGCAGGAGGAG-1</td></tr>\n",
       "\t<tr><th scope=row>5_TCGGGTGTCATGGGAG-1</th><td>CD8 T cell</td><td>TCGGGTGTCATGGGAG-1</td><td> 5</td><td>5_TCGGGTGTCATGGGAG-1 </td></tr>\n",
       "\t<tr><th scope=row>5_AGATGAACATGGAGAC-1</th><td>CD8 T cell</td><td>AGATGAACATGGAGAC-1</td><td> 5</td><td>5_AGATGAACATGGAGAC-1 </td></tr>\n",
       "\t<tr><th scope=row>28_CCATAAGAGACTCCGC-1</th><td>CD8 T cell</td><td>CCATAAGAGACTCCGC-1</td><td>28</td><td>28_CCATAAGAGACTCCGC-1</td></tr>\n",
       "\t<tr><th scope=row>29_CTACCCAGTAGTGTGG-1</th><td>CD8 T cell</td><td>CTACCCAGTAGTGTGG-1</td><td>29</td><td>29_CTACCCAGTAGTGTGG-1</td></tr>\n",
       "\t<tr><th scope=row>2_CTGTGAACACTACCGG-1</th><td>CD8 T cell</td><td>CTGTGAACACTACCGG-1</td><td> 2</td><td>2_CTGTGAACACTACCGG-1 </td></tr>\n",
       "\t<tr><th scope=row>29_GGGTGAATCAACCTCC-1</th><td>CD8 T cell</td><td>GGGTGAATCAACCTCC-1</td><td>29</td><td>29_GGGTGAATCAACCTCC-1</td></tr>\n",
       "\t<tr><th scope=row>7_AGGAGGTAGTGTTCAC-1</th><td>CD8 T cell</td><td>AGGAGGTAGTGTTCAC-1</td><td> 7</td><td>7_AGGAGGTAGTGTTCAC-1 </td></tr>\n",
       "\t<tr><th scope=row>0_TCCACGTGTCATAGTC-1</th><td>CD8 T cell</td><td>TCCACGTGTCATAGTC-1</td><td> 0</td><td>0_TCCACGTGTCATAGTC-1 </td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>24_TATTGGGTCATAGACC-1</th><td>Erythrocyte</td><td>TATTGGGTCATAGACC-1</td><td>24</td><td>24_TATTGGGTCATAGACC-1</td></tr>\n",
       "\t<tr><th scope=row>24_GTTGTGATCAAGGAGC-1</th><td>Erythrocyte</td><td>GTTGTGATCAAGGAGC-1</td><td>24</td><td>24_GTTGTGATCAAGGAGC-1</td></tr>\n",
       "\t<tr><th scope=row>23_TTATTGCCAACCCTAA-1</th><td>Erythrocyte</td><td>TTATTGCCAACCCTAA-1</td><td>23</td><td>23_TTATTGCCAACCCTAA-1</td></tr>\n",
       "\t<tr><th scope=row>22_CTGTACCCAATAACGA-1</th><td>Erythrocyte</td><td>CTGTACCCAATAACGA-1</td><td>22</td><td>22_CTGTACCCAATAACGA-1</td></tr>\n",
       "\t<tr><th scope=row>27_CCTGCATCAGCTTTCC-1</th><td>Erythrocyte</td><td>CCTGCATCAGCTTTCC-1</td><td>27</td><td>27_CCTGCATCAGCTTTCC-1</td></tr>\n",
       "\t<tr><th scope=row>23_GGGACAAGTACAAAGT-1</th><td>Erythrocyte</td><td>GGGACAAGTACAAAGT-1</td><td>23</td><td>23_GGGACAAGTACAAAGT-1</td></tr>\n",
       "\t<tr><th scope=row>27_TAAGCCATCGTCAACA-1</th><td>Erythrocyte</td><td>TAAGCCATCGTCAACA-1</td><td>27</td><td>27_TAAGCCATCGTCAACA-1</td></tr>\n",
       "\t<tr><th scope=row>25_CCAATTTGTGTTTCTT-1</th><td>Erythrocyte</td><td>CCAATTTGTGTTTCTT-1</td><td>25</td><td>25_CCAATTTGTGTTTCTT-1</td></tr>\n",
       "\t<tr><th scope=row>23_TCGGGCATCAAACCCA-1</th><td>Erythrocyte</td><td>TCGGGCATCAAACCCA-1</td><td>23</td><td>23_TCGGGCATCAAACCCA-1</td></tr>\n",
       "\t<tr><th scope=row>23_GATTGGTAGACTTGTC-1</th><td>Erythrocyte</td><td>GATTGGTAGACTTGTC-1</td><td>23</td><td>23_GATTGGTAGACTTGTC-1</td></tr>\n",
       "\t<tr><th scope=row>27_GTATTGGGTGTGTCGC-1</th><td>Erythrocyte</td><td>GTATTGGGTGTGTCGC-1</td><td>27</td><td>27_GTATTGGGTGTGTCGC-1</td></tr>\n",
       "\t<tr><th scope=row>22_TCTCTGGCACTCCTTG-1</th><td>Erythrocyte</td><td>TCTCTGGCACTCCTTG-1</td><td>22</td><td>22_TCTCTGGCACTCCTTG-1</td></tr>\n",
       "\t<tr><th scope=row>25_TCGGGTGAGAGAGCCT-1</th><td>Erythrocyte</td><td>TCGGGTGAGAGAGCCT-1</td><td>25</td><td>25_TCGGGTGAGAGAGCCT-1</td></tr>\n",
       "\t<tr><th scope=row>24_GTGGCGTCAGTCAGTT-1</th><td>Erythrocyte</td><td>GTGGCGTCAGTCAGTT-1</td><td>24</td><td>24_GTGGCGTCAGTCAGTT-1</td></tr>\n",
       "\t<tr><th scope=row>19_TGACGCGTCAGCGGAA-1</th><td>Erythrocyte</td><td>TGACGCGTCAGCGGAA-1</td><td>19</td><td>19_TGACGCGTCAGCGGAA-1</td></tr>\n",
       "\t<tr><th scope=row>19_TGAATCGTCTCCACTG-1</th><td>Erythrocyte</td><td>TGAATCGTCTCCACTG-1</td><td>19</td><td>19_TGAATCGTCTCCACTG-1</td></tr>\n",
       "\t<tr><th scope=row>19_GGGTCTGGTTACCCTC-1</th><td>Erythrocyte</td><td>GGGTCTGGTTACCCTC-1</td><td>19</td><td>19_GGGTCTGGTTACCCTC-1</td></tr>\n",
       "\t<tr><th scope=row>19_ATCACAGCAGTAACCT-1</th><td>Erythrocyte</td><td>ATCACAGCAGTAACCT-1</td><td>19</td><td>19_ATCACAGCAGTAACCT-1</td></tr>\n",
       "\t<tr><th scope=row>24_TTTGATCAGATGCGAC-1</th><td>Erythrocyte</td><td>TTTGATCAGATGCGAC-1</td><td>24</td><td>24_TTTGATCAGATGCGAC-1</td></tr>\n",
       "\t<tr><th scope=row>4_CAGAGCCAGCAGGGAG-1</th><td>Erythrocyte</td><td>CAGAGCCAGCAGGGAG-1</td><td> 4</td><td>4_CAGAGCCAGCAGGGAG-1 </td></tr>\n",
       "\t<tr><th scope=row>22_TCCTGCAAGTGGAAGA-1</th><td>Erythrocyte</td><td>TCCTGCAAGTGGAAGA-1</td><td>22</td><td>22_TCCTGCAAGTGGAAGA-1</td></tr>\n",
       "\t<tr><th scope=row>24_GAGGCCTGTTAATCGC-1</th><td>Erythrocyte</td><td>GAGGCCTGTTAATCGC-1</td><td>24</td><td>24_GAGGCCTGTTAATCGC-1</td></tr>\n",
       "\t<tr><th scope=row>24_ACATCCCAGTAGGGTC-1</th><td>Erythrocyte</td><td>ACATCCCAGTAGGGTC-1</td><td>24</td><td>24_ACATCCCAGTAGGGTC-1</td></tr>\n",
       "\t<tr><th scope=row>4_GGGTATTTCATGAGGG-1</th><td>Erythrocyte</td><td>GGGTATTTCATGAGGG-1</td><td> 4</td><td>4_GGGTATTTCATGAGGG-1 </td></tr>\n",
       "\t<tr><th scope=row>27_CACAGATGTTGGGCCT-1</th><td>Erythrocyte</td><td>CACAGATGTTGGGCCT-1</td><td>27</td><td>27_CACAGATGTTGGGCCT-1</td></tr>\n",
       "\t<tr><th scope=row>24_TTCTCTCCAGTCAGAG-1</th><td>Erythrocyte</td><td>TTCTCTCCAGTCAGAG-1</td><td>24</td><td>24_TTCTCTCCAGTCAGAG-1</td></tr>\n",
       "\t<tr><th scope=row>24_GGGACTCAGGGACCAT-1</th><td>Erythrocyte</td><td>GGGACTCAGGGACCAT-1</td><td>24</td><td>24_GGGACTCAGGGACCAT-1</td></tr>\n",
       "\t<tr><th scope=row>23_GTGGTTAAGCAATAAC-1</th><td>Erythrocyte</td><td>GTGGTTAAGCAATAAC-1</td><td>23</td><td>23_GTGGTTAAGCAATAAC-1</td></tr>\n",
       "\t<tr><th scope=row>27_GACCTTCAGATAGTGT-1</th><td>Erythrocyte</td><td>GACCTTCAGATAGTGT-1</td><td>27</td><td>27_GACCTTCAGATAGTGT-1</td></tr>\n",
       "\t<tr><th scope=row>19_TGCGGGTTCCAGCAAT-1</th><td>Erythrocyte</td><td>TGCGGGTTCCAGCAAT-1</td><td>19</td><td>19_TGCGGGTTCCAGCAAT-1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6385 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & cell\\_type & barcode & sample & cell\\_ids\\\\\n",
       "  & <fct> & <fct> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t18\\_GGCGTCATCCAACTAG-1 & CD8 T cell & GGCGTCATCCAACTAG-1 & 18 & 18\\_GGCGTCATCCAACTAG-1\\\\\n",
       "\t0\\_AGTAACCGTGATACTC-1 & CD8 T cell & AGTAACCGTGATACTC-1 &  0 & 0\\_AGTAACCGTGATACTC-1 \\\\\n",
       "\t15\\_ATCGGATGTCGCCTAG-1 & CD8 T cell & ATCGGATGTCGCCTAG-1 & 15 & 15\\_ATCGGATGTCGCCTAG-1\\\\\n",
       "\t28\\_TTGCATTGTACAGTCT-1 & CD8 T cell & TTGCATTGTACAGTCT-1 & 28 & 28\\_TTGCATTGTACAGTCT-1\\\\\n",
       "\t11\\_TCAAGTGAGGTACCTT-1 & CD8 T cell & TCAAGTGAGGTACCTT-1 & 11 & 11\\_TCAAGTGAGGTACCTT-1\\\\\n",
       "\t28\\_CTGCCATGTAACTGCT-1 & CD8 T cell & CTGCCATGTAACTGCT-1 & 28 & 28\\_CTGCCATGTAACTGCT-1\\\\\n",
       "\t2\\_TCCAGAAAGCAGTACG-1 & CD8 T cell & TCCAGAAAGCAGTACG-1 &  2 & 2\\_TCCAGAAAGCAGTACG-1 \\\\\n",
       "\t29\\_ACGGAAGCAAGCGATG-1 & CD8 T cell & ACGGAAGCAAGCGATG-1 & 29 & 29\\_ACGGAAGCAAGCGATG-1\\\\\n",
       "\t2\\_TAGGTTGGTAGTTAGA-1 & CD8 T cell & TAGGTTGGTAGTTAGA-1 &  2 & 2\\_TAGGTTGGTAGTTAGA-1 \\\\\n",
       "\t29\\_ATGGATCGTGAGCAGT-1 & CD8 T cell & ATGGATCGTGAGCAGT-1 & 29 & 29\\_ATGGATCGTGAGCAGT-1\\\\\n",
       "\t16\\_AGTAACCGTGACTCGC-1 & CD8 T cell & AGTAACCGTGACTCGC-1 & 16 & 16\\_AGTAACCGTGACTCGC-1\\\\\n",
       "\t7\\_TCATGTTCAGCACAGA-1 & CD8 T cell & TCATGTTCAGCACAGA-1 &  7 & 7\\_TCATGTTCAGCACAGA-1 \\\\\n",
       "\t27\\_AACAGGGGTACACGCC-1 & CD8 T cell & AACAGGGGTACACGCC-1 & 27 & 27\\_AACAGGGGTACACGCC-1\\\\\n",
       "\t28\\_GAGTTGTAGATCACTC-1 & CD8 T cell & GAGTTGTAGATCACTC-1 & 28 & 28\\_GAGTTGTAGATCACTC-1\\\\\n",
       "\t28\\_GTGCACGTCGCCTAGG-1 & CD8 T cell & GTGCACGTCGCCTAGG-1 & 28 & 28\\_GTGCACGTCGCCTAGG-1\\\\\n",
       "\t29\\_GGGACCTTCGGTGTTA-1 & CD8 T cell & GGGACCTTCGGTGTTA-1 & 29 & 29\\_GGGACCTTCGGTGTTA-1\\\\\n",
       "\t28\\_GGCTTGGCAAATACAG-1 & CD8 T cell & GGCTTGGCAAATACAG-1 & 28 & 28\\_GGCTTGGCAAATACAG-1\\\\\n",
       "\t27\\_ATTTCTGGTGATCATC-1 & CD8 T cell & ATTTCTGGTGATCATC-1 & 27 & 27\\_ATTTCTGGTGATCATC-1\\\\\n",
       "\t4\\_TCGTCCAAGGCATGCA-1 & CD8 T cell & TCGTCCAAGGCATGCA-1 &  4 & 4\\_TCGTCCAAGGCATGCA-1 \\\\\n",
       "\t28\\_ATGAGGGGTGCCAAGA-1 & CD8 T cell & ATGAGGGGTGCCAAGA-1 & 28 & 28\\_ATGAGGGGTGCCAAGA-1\\\\\n",
       "\t28\\_TTTACGTCAAGAATAC-1 & CD8 T cell & TTTACGTCAAGAATAC-1 & 28 & 28\\_TTTACGTCAAGAATAC-1\\\\\n",
       "\t29\\_GGTGAAGCAGGAGGAG-1 & CD8 T cell & GGTGAAGCAGGAGGAG-1 & 29 & 29\\_GGTGAAGCAGGAGGAG-1\\\\\n",
       "\t5\\_TCGGGTGTCATGGGAG-1 & CD8 T cell & TCGGGTGTCATGGGAG-1 &  5 & 5\\_TCGGGTGTCATGGGAG-1 \\\\\n",
       "\t5\\_AGATGAACATGGAGAC-1 & CD8 T cell & AGATGAACATGGAGAC-1 &  5 & 5\\_AGATGAACATGGAGAC-1 \\\\\n",
       "\t28\\_CCATAAGAGACTCCGC-1 & CD8 T cell & CCATAAGAGACTCCGC-1 & 28 & 28\\_CCATAAGAGACTCCGC-1\\\\\n",
       "\t29\\_CTACCCAGTAGTGTGG-1 & CD8 T cell & CTACCCAGTAGTGTGG-1 & 29 & 29\\_CTACCCAGTAGTGTGG-1\\\\\n",
       "\t2\\_CTGTGAACACTACCGG-1 & CD8 T cell & CTGTGAACACTACCGG-1 &  2 & 2\\_CTGTGAACACTACCGG-1 \\\\\n",
       "\t29\\_GGGTGAATCAACCTCC-1 & CD8 T cell & GGGTGAATCAACCTCC-1 & 29 & 29\\_GGGTGAATCAACCTCC-1\\\\\n",
       "\t7\\_AGGAGGTAGTGTTCAC-1 & CD8 T cell & AGGAGGTAGTGTTCAC-1 &  7 & 7\\_AGGAGGTAGTGTTCAC-1 \\\\\n",
       "\t0\\_TCCACGTGTCATAGTC-1 & CD8 T cell & TCCACGTGTCATAGTC-1 &  0 & 0\\_TCCACGTGTCATAGTC-1 \\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\t24\\_TATTGGGTCATAGACC-1 & Erythrocyte & TATTGGGTCATAGACC-1 & 24 & 24\\_TATTGGGTCATAGACC-1\\\\\n",
       "\t24\\_GTTGTGATCAAGGAGC-1 & Erythrocyte & GTTGTGATCAAGGAGC-1 & 24 & 24\\_GTTGTGATCAAGGAGC-1\\\\\n",
       "\t23\\_TTATTGCCAACCCTAA-1 & Erythrocyte & TTATTGCCAACCCTAA-1 & 23 & 23\\_TTATTGCCAACCCTAA-1\\\\\n",
       "\t22\\_CTGTACCCAATAACGA-1 & Erythrocyte & CTGTACCCAATAACGA-1 & 22 & 22\\_CTGTACCCAATAACGA-1\\\\\n",
       "\t27\\_CCTGCATCAGCTTTCC-1 & Erythrocyte & CCTGCATCAGCTTTCC-1 & 27 & 27\\_CCTGCATCAGCTTTCC-1\\\\\n",
       "\t23\\_GGGACAAGTACAAAGT-1 & Erythrocyte & GGGACAAGTACAAAGT-1 & 23 & 23\\_GGGACAAGTACAAAGT-1\\\\\n",
       "\t27\\_TAAGCCATCGTCAACA-1 & Erythrocyte & TAAGCCATCGTCAACA-1 & 27 & 27\\_TAAGCCATCGTCAACA-1\\\\\n",
       "\t25\\_CCAATTTGTGTTTCTT-1 & Erythrocyte & CCAATTTGTGTTTCTT-1 & 25 & 25\\_CCAATTTGTGTTTCTT-1\\\\\n",
       "\t23\\_TCGGGCATCAAACCCA-1 & Erythrocyte & TCGGGCATCAAACCCA-1 & 23 & 23\\_TCGGGCATCAAACCCA-1\\\\\n",
       "\t23\\_GATTGGTAGACTTGTC-1 & Erythrocyte & GATTGGTAGACTTGTC-1 & 23 & 23\\_GATTGGTAGACTTGTC-1\\\\\n",
       "\t27\\_GTATTGGGTGTGTCGC-1 & Erythrocyte & GTATTGGGTGTGTCGC-1 & 27 & 27\\_GTATTGGGTGTGTCGC-1\\\\\n",
       "\t22\\_TCTCTGGCACTCCTTG-1 & Erythrocyte & TCTCTGGCACTCCTTG-1 & 22 & 22\\_TCTCTGGCACTCCTTG-1\\\\\n",
       "\t25\\_TCGGGTGAGAGAGCCT-1 & Erythrocyte & TCGGGTGAGAGAGCCT-1 & 25 & 25\\_TCGGGTGAGAGAGCCT-1\\\\\n",
       "\t24\\_GTGGCGTCAGTCAGTT-1 & Erythrocyte & GTGGCGTCAGTCAGTT-1 & 24 & 24\\_GTGGCGTCAGTCAGTT-1\\\\\n",
       "\t19\\_TGACGCGTCAGCGGAA-1 & Erythrocyte & TGACGCGTCAGCGGAA-1 & 19 & 19\\_TGACGCGTCAGCGGAA-1\\\\\n",
       "\t19\\_TGAATCGTCTCCACTG-1 & Erythrocyte & TGAATCGTCTCCACTG-1 & 19 & 19\\_TGAATCGTCTCCACTG-1\\\\\n",
       "\t19\\_GGGTCTGGTTACCCTC-1 & Erythrocyte & GGGTCTGGTTACCCTC-1 & 19 & 19\\_GGGTCTGGTTACCCTC-1\\\\\n",
       "\t19\\_ATCACAGCAGTAACCT-1 & Erythrocyte & ATCACAGCAGTAACCT-1 & 19 & 19\\_ATCACAGCAGTAACCT-1\\\\\n",
       "\t24\\_TTTGATCAGATGCGAC-1 & Erythrocyte & TTTGATCAGATGCGAC-1 & 24 & 24\\_TTTGATCAGATGCGAC-1\\\\\n",
       "\t4\\_CAGAGCCAGCAGGGAG-1 & Erythrocyte & CAGAGCCAGCAGGGAG-1 &  4 & 4\\_CAGAGCCAGCAGGGAG-1 \\\\\n",
       "\t22\\_TCCTGCAAGTGGAAGA-1 & Erythrocyte & TCCTGCAAGTGGAAGA-1 & 22 & 22\\_TCCTGCAAGTGGAAGA-1\\\\\n",
       "\t24\\_GAGGCCTGTTAATCGC-1 & Erythrocyte & GAGGCCTGTTAATCGC-1 & 24 & 24\\_GAGGCCTGTTAATCGC-1\\\\\n",
       "\t24\\_ACATCCCAGTAGGGTC-1 & Erythrocyte & ACATCCCAGTAGGGTC-1 & 24 & 24\\_ACATCCCAGTAGGGTC-1\\\\\n",
       "\t4\\_GGGTATTTCATGAGGG-1 & Erythrocyte & GGGTATTTCATGAGGG-1 &  4 & 4\\_GGGTATTTCATGAGGG-1 \\\\\n",
       "\t27\\_CACAGATGTTGGGCCT-1 & Erythrocyte & CACAGATGTTGGGCCT-1 & 27 & 27\\_CACAGATGTTGGGCCT-1\\\\\n",
       "\t24\\_TTCTCTCCAGTCAGAG-1 & Erythrocyte & TTCTCTCCAGTCAGAG-1 & 24 & 24\\_TTCTCTCCAGTCAGAG-1\\\\\n",
       "\t24\\_GGGACTCAGGGACCAT-1 & Erythrocyte & GGGACTCAGGGACCAT-1 & 24 & 24\\_GGGACTCAGGGACCAT-1\\\\\n",
       "\t23\\_GTGGTTAAGCAATAAC-1 & Erythrocyte & GTGGTTAAGCAATAAC-1 & 23 & 23\\_GTGGTTAAGCAATAAC-1\\\\\n",
       "\t27\\_GACCTTCAGATAGTGT-1 & Erythrocyte & GACCTTCAGATAGTGT-1 & 27 & 27\\_GACCTTCAGATAGTGT-1\\\\\n",
       "\t19\\_TGCGGGTTCCAGCAAT-1 & Erythrocyte & TGCGGGTTCCAGCAAT-1 & 19 & 19\\_TGCGGGTTCCAGCAAT-1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6385 × 4\n",
       "\n",
       "| <!--/--> | cell_type &lt;fct&gt; | barcode &lt;fct&gt; | sample &lt;int&gt; | cell_ids &lt;chr&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 18_GGCGTCATCCAACTAG-1 | CD8 T cell | GGCGTCATCCAACTAG-1 | 18 | 18_GGCGTCATCCAACTAG-1 |\n",
       "| 0_AGTAACCGTGATACTC-1 | CD8 T cell | AGTAACCGTGATACTC-1 |  0 | 0_AGTAACCGTGATACTC-1  |\n",
       "| 15_ATCGGATGTCGCCTAG-1 | CD8 T cell | ATCGGATGTCGCCTAG-1 | 15 | 15_ATCGGATGTCGCCTAG-1 |\n",
       "| 28_TTGCATTGTACAGTCT-1 | CD8 T cell | TTGCATTGTACAGTCT-1 | 28 | 28_TTGCATTGTACAGTCT-1 |\n",
       "| 11_TCAAGTGAGGTACCTT-1 | CD8 T cell | TCAAGTGAGGTACCTT-1 | 11 | 11_TCAAGTGAGGTACCTT-1 |\n",
       "| 28_CTGCCATGTAACTGCT-1 | CD8 T cell | CTGCCATGTAACTGCT-1 | 28 | 28_CTGCCATGTAACTGCT-1 |\n",
       "| 2_TCCAGAAAGCAGTACG-1 | CD8 T cell | TCCAGAAAGCAGTACG-1 |  2 | 2_TCCAGAAAGCAGTACG-1  |\n",
       "| 29_ACGGAAGCAAGCGATG-1 | CD8 T cell | ACGGAAGCAAGCGATG-1 | 29 | 29_ACGGAAGCAAGCGATG-1 |\n",
       "| 2_TAGGTTGGTAGTTAGA-1 | CD8 T cell | TAGGTTGGTAGTTAGA-1 |  2 | 2_TAGGTTGGTAGTTAGA-1  |\n",
       "| 29_ATGGATCGTGAGCAGT-1 | CD8 T cell | ATGGATCGTGAGCAGT-1 | 29 | 29_ATGGATCGTGAGCAGT-1 |\n",
       "| 16_AGTAACCGTGACTCGC-1 | CD8 T cell | AGTAACCGTGACTCGC-1 | 16 | 16_AGTAACCGTGACTCGC-1 |\n",
       "| 7_TCATGTTCAGCACAGA-1 | CD8 T cell | TCATGTTCAGCACAGA-1 |  7 | 7_TCATGTTCAGCACAGA-1  |\n",
       "| 27_AACAGGGGTACACGCC-1 | CD8 T cell | AACAGGGGTACACGCC-1 | 27 | 27_AACAGGGGTACACGCC-1 |\n",
       "| 28_GAGTTGTAGATCACTC-1 | CD8 T cell | GAGTTGTAGATCACTC-1 | 28 | 28_GAGTTGTAGATCACTC-1 |\n",
       "| 28_GTGCACGTCGCCTAGG-1 | CD8 T cell | GTGCACGTCGCCTAGG-1 | 28 | 28_GTGCACGTCGCCTAGG-1 |\n",
       "| 29_GGGACCTTCGGTGTTA-1 | CD8 T cell | GGGACCTTCGGTGTTA-1 | 29 | 29_GGGACCTTCGGTGTTA-1 |\n",
       "| 28_GGCTTGGCAAATACAG-1 | CD8 T cell | GGCTTGGCAAATACAG-1 | 28 | 28_GGCTTGGCAAATACAG-1 |\n",
       "| 27_ATTTCTGGTGATCATC-1 | CD8 T cell | ATTTCTGGTGATCATC-1 | 27 | 27_ATTTCTGGTGATCATC-1 |\n",
       "| 4_TCGTCCAAGGCATGCA-1 | CD8 T cell | TCGTCCAAGGCATGCA-1 |  4 | 4_TCGTCCAAGGCATGCA-1  |\n",
       "| 28_ATGAGGGGTGCCAAGA-1 | CD8 T cell | ATGAGGGGTGCCAAGA-1 | 28 | 28_ATGAGGGGTGCCAAGA-1 |\n",
       "| 28_TTTACGTCAAGAATAC-1 | CD8 T cell | TTTACGTCAAGAATAC-1 | 28 | 28_TTTACGTCAAGAATAC-1 |\n",
       "| 29_GGTGAAGCAGGAGGAG-1 | CD8 T cell | GGTGAAGCAGGAGGAG-1 | 29 | 29_GGTGAAGCAGGAGGAG-1 |\n",
       "| 5_TCGGGTGTCATGGGAG-1 | CD8 T cell | TCGGGTGTCATGGGAG-1 |  5 | 5_TCGGGTGTCATGGGAG-1  |\n",
       "| 5_AGATGAACATGGAGAC-1 | CD8 T cell | AGATGAACATGGAGAC-1 |  5 | 5_AGATGAACATGGAGAC-1  |\n",
       "| 28_CCATAAGAGACTCCGC-1 | CD8 T cell | CCATAAGAGACTCCGC-1 | 28 | 28_CCATAAGAGACTCCGC-1 |\n",
       "| 29_CTACCCAGTAGTGTGG-1 | CD8 T cell | CTACCCAGTAGTGTGG-1 | 29 | 29_CTACCCAGTAGTGTGG-1 |\n",
       "| 2_CTGTGAACACTACCGG-1 | CD8 T cell | CTGTGAACACTACCGG-1 |  2 | 2_CTGTGAACACTACCGG-1  |\n",
       "| 29_GGGTGAATCAACCTCC-1 | CD8 T cell | GGGTGAATCAACCTCC-1 | 29 | 29_GGGTGAATCAACCTCC-1 |\n",
       "| 7_AGGAGGTAGTGTTCAC-1 | CD8 T cell | AGGAGGTAGTGTTCAC-1 |  7 | 7_AGGAGGTAGTGTTCAC-1  |\n",
       "| 0_TCCACGTGTCATAGTC-1 | CD8 T cell | TCCACGTGTCATAGTC-1 |  0 | 0_TCCACGTGTCATAGTC-1  |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| 24_TATTGGGTCATAGACC-1 | Erythrocyte | TATTGGGTCATAGACC-1 | 24 | 24_TATTGGGTCATAGACC-1 |\n",
       "| 24_GTTGTGATCAAGGAGC-1 | Erythrocyte | GTTGTGATCAAGGAGC-1 | 24 | 24_GTTGTGATCAAGGAGC-1 |\n",
       "| 23_TTATTGCCAACCCTAA-1 | Erythrocyte | TTATTGCCAACCCTAA-1 | 23 | 23_TTATTGCCAACCCTAA-1 |\n",
       "| 22_CTGTACCCAATAACGA-1 | Erythrocyte | CTGTACCCAATAACGA-1 | 22 | 22_CTGTACCCAATAACGA-1 |\n",
       "| 27_CCTGCATCAGCTTTCC-1 | Erythrocyte | CCTGCATCAGCTTTCC-1 | 27 | 27_CCTGCATCAGCTTTCC-1 |\n",
       "| 23_GGGACAAGTACAAAGT-1 | Erythrocyte | GGGACAAGTACAAAGT-1 | 23 | 23_GGGACAAGTACAAAGT-1 |\n",
       "| 27_TAAGCCATCGTCAACA-1 | Erythrocyte | TAAGCCATCGTCAACA-1 | 27 | 27_TAAGCCATCGTCAACA-1 |\n",
       "| 25_CCAATTTGTGTTTCTT-1 | Erythrocyte | CCAATTTGTGTTTCTT-1 | 25 | 25_CCAATTTGTGTTTCTT-1 |\n",
       "| 23_TCGGGCATCAAACCCA-1 | Erythrocyte | TCGGGCATCAAACCCA-1 | 23 | 23_TCGGGCATCAAACCCA-1 |\n",
       "| 23_GATTGGTAGACTTGTC-1 | Erythrocyte | GATTGGTAGACTTGTC-1 | 23 | 23_GATTGGTAGACTTGTC-1 |\n",
       "| 27_GTATTGGGTGTGTCGC-1 | Erythrocyte | GTATTGGGTGTGTCGC-1 | 27 | 27_GTATTGGGTGTGTCGC-1 |\n",
       "| 22_TCTCTGGCACTCCTTG-1 | Erythrocyte | TCTCTGGCACTCCTTG-1 | 22 | 22_TCTCTGGCACTCCTTG-1 |\n",
       "| 25_TCGGGTGAGAGAGCCT-1 | Erythrocyte | TCGGGTGAGAGAGCCT-1 | 25 | 25_TCGGGTGAGAGAGCCT-1 |\n",
       "| 24_GTGGCGTCAGTCAGTT-1 | Erythrocyte | GTGGCGTCAGTCAGTT-1 | 24 | 24_GTGGCGTCAGTCAGTT-1 |\n",
       "| 19_TGACGCGTCAGCGGAA-1 | Erythrocyte | TGACGCGTCAGCGGAA-1 | 19 | 19_TGACGCGTCAGCGGAA-1 |\n",
       "| 19_TGAATCGTCTCCACTG-1 | Erythrocyte | TGAATCGTCTCCACTG-1 | 19 | 19_TGAATCGTCTCCACTG-1 |\n",
       "| 19_GGGTCTGGTTACCCTC-1 | Erythrocyte | GGGTCTGGTTACCCTC-1 | 19 | 19_GGGTCTGGTTACCCTC-1 |\n",
       "| 19_ATCACAGCAGTAACCT-1 | Erythrocyte | ATCACAGCAGTAACCT-1 | 19 | 19_ATCACAGCAGTAACCT-1 |\n",
       "| 24_TTTGATCAGATGCGAC-1 | Erythrocyte | TTTGATCAGATGCGAC-1 | 24 | 24_TTTGATCAGATGCGAC-1 |\n",
       "| 4_CAGAGCCAGCAGGGAG-1 | Erythrocyte | CAGAGCCAGCAGGGAG-1 |  4 | 4_CAGAGCCAGCAGGGAG-1  |\n",
       "| 22_TCCTGCAAGTGGAAGA-1 | Erythrocyte | TCCTGCAAGTGGAAGA-1 | 22 | 22_TCCTGCAAGTGGAAGA-1 |\n",
       "| 24_GAGGCCTGTTAATCGC-1 | Erythrocyte | GAGGCCTGTTAATCGC-1 | 24 | 24_GAGGCCTGTTAATCGC-1 |\n",
       "| 24_ACATCCCAGTAGGGTC-1 | Erythrocyte | ACATCCCAGTAGGGTC-1 | 24 | 24_ACATCCCAGTAGGGTC-1 |\n",
       "| 4_GGGTATTTCATGAGGG-1 | Erythrocyte | GGGTATTTCATGAGGG-1 |  4 | 4_GGGTATTTCATGAGGG-1  |\n",
       "| 27_CACAGATGTTGGGCCT-1 | Erythrocyte | CACAGATGTTGGGCCT-1 | 27 | 27_CACAGATGTTGGGCCT-1 |\n",
       "| 24_TTCTCTCCAGTCAGAG-1 | Erythrocyte | TTCTCTCCAGTCAGAG-1 | 24 | 24_TTCTCTCCAGTCAGAG-1 |\n",
       "| 24_GGGACTCAGGGACCAT-1 | Erythrocyte | GGGACTCAGGGACCAT-1 | 24 | 24_GGGACTCAGGGACCAT-1 |\n",
       "| 23_GTGGTTAAGCAATAAC-1 | Erythrocyte | GTGGTTAAGCAATAAC-1 | 23 | 23_GTGGTTAAGCAATAAC-1 |\n",
       "| 27_GACCTTCAGATAGTGT-1 | Erythrocyte | GACCTTCAGATAGTGT-1 | 27 | 27_GACCTTCAGATAGTGT-1 |\n",
       "| 19_TGCGGGTTCCAGCAAT-1 | Erythrocyte | TGCGGGTTCCAGCAAT-1 | 19 | 19_TGCGGGTTCCAGCAAT-1 |\n",
       "\n"
      ],
      "text/plain": [
       "                      cell_type   barcode            sample\n",
       "18_GGCGTCATCCAACTAG-1 CD8 T cell  GGCGTCATCCAACTAG-1 18    \n",
       "0_AGTAACCGTGATACTC-1  CD8 T cell  AGTAACCGTGATACTC-1  0    \n",
       "15_ATCGGATGTCGCCTAG-1 CD8 T cell  ATCGGATGTCGCCTAG-1 15    \n",
       "28_TTGCATTGTACAGTCT-1 CD8 T cell  TTGCATTGTACAGTCT-1 28    \n",
       "11_TCAAGTGAGGTACCTT-1 CD8 T cell  TCAAGTGAGGTACCTT-1 11    \n",
       "28_CTGCCATGTAACTGCT-1 CD8 T cell  CTGCCATGTAACTGCT-1 28    \n",
       "2_TCCAGAAAGCAGTACG-1  CD8 T cell  TCCAGAAAGCAGTACG-1  2    \n",
       "29_ACGGAAGCAAGCGATG-1 CD8 T cell  ACGGAAGCAAGCGATG-1 29    \n",
       "2_TAGGTTGGTAGTTAGA-1  CD8 T cell  TAGGTTGGTAGTTAGA-1  2    \n",
       "29_ATGGATCGTGAGCAGT-1 CD8 T cell  ATGGATCGTGAGCAGT-1 29    \n",
       "16_AGTAACCGTGACTCGC-1 CD8 T cell  AGTAACCGTGACTCGC-1 16    \n",
       "7_TCATGTTCAGCACAGA-1  CD8 T cell  TCATGTTCAGCACAGA-1  7    \n",
       "27_AACAGGGGTACACGCC-1 CD8 T cell  AACAGGGGTACACGCC-1 27    \n",
       "28_GAGTTGTAGATCACTC-1 CD8 T cell  GAGTTGTAGATCACTC-1 28    \n",
       "28_GTGCACGTCGCCTAGG-1 CD8 T cell  GTGCACGTCGCCTAGG-1 28    \n",
       "29_GGGACCTTCGGTGTTA-1 CD8 T cell  GGGACCTTCGGTGTTA-1 29    \n",
       "28_GGCTTGGCAAATACAG-1 CD8 T cell  GGCTTGGCAAATACAG-1 28    \n",
       "27_ATTTCTGGTGATCATC-1 CD8 T cell  ATTTCTGGTGATCATC-1 27    \n",
       "4_TCGTCCAAGGCATGCA-1  CD8 T cell  TCGTCCAAGGCATGCA-1  4    \n",
       "28_ATGAGGGGTGCCAAGA-1 CD8 T cell  ATGAGGGGTGCCAAGA-1 28    \n",
       "28_TTTACGTCAAGAATAC-1 CD8 T cell  TTTACGTCAAGAATAC-1 28    \n",
       "29_GGTGAAGCAGGAGGAG-1 CD8 T cell  GGTGAAGCAGGAGGAG-1 29    \n",
       "5_TCGGGTGTCATGGGAG-1  CD8 T cell  TCGGGTGTCATGGGAG-1  5    \n",
       "5_AGATGAACATGGAGAC-1  CD8 T cell  AGATGAACATGGAGAC-1  5    \n",
       "28_CCATAAGAGACTCCGC-1 CD8 T cell  CCATAAGAGACTCCGC-1 28    \n",
       "29_CTACCCAGTAGTGTGG-1 CD8 T cell  CTACCCAGTAGTGTGG-1 29    \n",
       "2_CTGTGAACACTACCGG-1  CD8 T cell  CTGTGAACACTACCGG-1  2    \n",
       "29_GGGTGAATCAACCTCC-1 CD8 T cell  GGGTGAATCAACCTCC-1 29    \n",
       "7_AGGAGGTAGTGTTCAC-1  CD8 T cell  AGGAGGTAGTGTTCAC-1  7    \n",
       "0_TCCACGTGTCATAGTC-1  CD8 T cell  TCCACGTGTCATAGTC-1  0    \n",
       "⋮                     ⋮           ⋮                  ⋮     \n",
       "24_TATTGGGTCATAGACC-1 Erythrocyte TATTGGGTCATAGACC-1 24    \n",
       "24_GTTGTGATCAAGGAGC-1 Erythrocyte GTTGTGATCAAGGAGC-1 24    \n",
       "23_TTATTGCCAACCCTAA-1 Erythrocyte TTATTGCCAACCCTAA-1 23    \n",
       "22_CTGTACCCAATAACGA-1 Erythrocyte CTGTACCCAATAACGA-1 22    \n",
       "27_CCTGCATCAGCTTTCC-1 Erythrocyte CCTGCATCAGCTTTCC-1 27    \n",
       "23_GGGACAAGTACAAAGT-1 Erythrocyte GGGACAAGTACAAAGT-1 23    \n",
       "27_TAAGCCATCGTCAACA-1 Erythrocyte TAAGCCATCGTCAACA-1 27    \n",
       "25_CCAATTTGTGTTTCTT-1 Erythrocyte CCAATTTGTGTTTCTT-1 25    \n",
       "23_TCGGGCATCAAACCCA-1 Erythrocyte TCGGGCATCAAACCCA-1 23    \n",
       "23_GATTGGTAGACTTGTC-1 Erythrocyte GATTGGTAGACTTGTC-1 23    \n",
       "27_GTATTGGGTGTGTCGC-1 Erythrocyte GTATTGGGTGTGTCGC-1 27    \n",
       "22_TCTCTGGCACTCCTTG-1 Erythrocyte TCTCTGGCACTCCTTG-1 22    \n",
       "25_TCGGGTGAGAGAGCCT-1 Erythrocyte TCGGGTGAGAGAGCCT-1 25    \n",
       "24_GTGGCGTCAGTCAGTT-1 Erythrocyte GTGGCGTCAGTCAGTT-1 24    \n",
       "19_TGACGCGTCAGCGGAA-1 Erythrocyte TGACGCGTCAGCGGAA-1 19    \n",
       "19_TGAATCGTCTCCACTG-1 Erythrocyte TGAATCGTCTCCACTG-1 19    \n",
       "19_GGGTCTGGTTACCCTC-1 Erythrocyte GGGTCTGGTTACCCTC-1 19    \n",
       "19_ATCACAGCAGTAACCT-1 Erythrocyte ATCACAGCAGTAACCT-1 19    \n",
       "24_TTTGATCAGATGCGAC-1 Erythrocyte TTTGATCAGATGCGAC-1 24    \n",
       "4_CAGAGCCAGCAGGGAG-1  Erythrocyte CAGAGCCAGCAGGGAG-1  4    \n",
       "22_TCCTGCAAGTGGAAGA-1 Erythrocyte TCCTGCAAGTGGAAGA-1 22    \n",
       "24_GAGGCCTGTTAATCGC-1 Erythrocyte GAGGCCTGTTAATCGC-1 24    \n",
       "24_ACATCCCAGTAGGGTC-1 Erythrocyte ACATCCCAGTAGGGTC-1 24    \n",
       "4_GGGTATTTCATGAGGG-1  Erythrocyte GGGTATTTCATGAGGG-1  4    \n",
       "27_CACAGATGTTGGGCCT-1 Erythrocyte CACAGATGTTGGGCCT-1 27    \n",
       "24_TTCTCTCCAGTCAGAG-1 Erythrocyte TTCTCTCCAGTCAGAG-1 24    \n",
       "24_GGGACTCAGGGACCAT-1 Erythrocyte GGGACTCAGGGACCAT-1 24    \n",
       "23_GTGGTTAAGCAATAAC-1 Erythrocyte GTGGTTAAGCAATAAC-1 23    \n",
       "27_GACCTTCAGATAGTGT-1 Erythrocyte GACCTTCAGATAGTGT-1 27    \n",
       "19_TGCGGGTTCCAGCAAT-1 Erythrocyte TGCGGGTTCCAGCAAT-1 19    \n",
       "                      cell_ids             \n",
       "18_GGCGTCATCCAACTAG-1 18_GGCGTCATCCAACTAG-1\n",
       "0_AGTAACCGTGATACTC-1  0_AGTAACCGTGATACTC-1 \n",
       "15_ATCGGATGTCGCCTAG-1 15_ATCGGATGTCGCCTAG-1\n",
       "28_TTGCATTGTACAGTCT-1 28_TTGCATTGTACAGTCT-1\n",
       "11_TCAAGTGAGGTACCTT-1 11_TCAAGTGAGGTACCTT-1\n",
       "28_CTGCCATGTAACTGCT-1 28_CTGCCATGTAACTGCT-1\n",
       "2_TCCAGAAAGCAGTACG-1  2_TCCAGAAAGCAGTACG-1 \n",
       "29_ACGGAAGCAAGCGATG-1 29_ACGGAAGCAAGCGATG-1\n",
       "2_TAGGTTGGTAGTTAGA-1  2_TAGGTTGGTAGTTAGA-1 \n",
       "29_ATGGATCGTGAGCAGT-1 29_ATGGATCGTGAGCAGT-1\n",
       "16_AGTAACCGTGACTCGC-1 16_AGTAACCGTGACTCGC-1\n",
       "7_TCATGTTCAGCACAGA-1  7_TCATGTTCAGCACAGA-1 \n",
       "27_AACAGGGGTACACGCC-1 27_AACAGGGGTACACGCC-1\n",
       "28_GAGTTGTAGATCACTC-1 28_GAGTTGTAGATCACTC-1\n",
       "28_GTGCACGTCGCCTAGG-1 28_GTGCACGTCGCCTAGG-1\n",
       "29_GGGACCTTCGGTGTTA-1 29_GGGACCTTCGGTGTTA-1\n",
       "28_GGCTTGGCAAATACAG-1 28_GGCTTGGCAAATACAG-1\n",
       "27_ATTTCTGGTGATCATC-1 27_ATTTCTGGTGATCATC-1\n",
       "4_TCGTCCAAGGCATGCA-1  4_TCGTCCAAGGCATGCA-1 \n",
       "28_ATGAGGGGTGCCAAGA-1 28_ATGAGGGGTGCCAAGA-1\n",
       "28_TTTACGTCAAGAATAC-1 28_TTTACGTCAAGAATAC-1\n",
       "29_GGTGAAGCAGGAGGAG-1 29_GGTGAAGCAGGAGGAG-1\n",
       "5_TCGGGTGTCATGGGAG-1  5_TCGGGTGTCATGGGAG-1 \n",
       "5_AGATGAACATGGAGAC-1  5_AGATGAACATGGAGAC-1 \n",
       "28_CCATAAGAGACTCCGC-1 28_CCATAAGAGACTCCGC-1\n",
       "29_CTACCCAGTAGTGTGG-1 29_CTACCCAGTAGTGTGG-1\n",
       "2_CTGTGAACACTACCGG-1  2_CTGTGAACACTACCGG-1 \n",
       "29_GGGTGAATCAACCTCC-1 29_GGGTGAATCAACCTCC-1\n",
       "7_AGGAGGTAGTGTTCAC-1  7_AGGAGGTAGTGTTCAC-1 \n",
       "0_TCCACGTGTCATAGTC-1  0_TCCACGTGTCATAGTC-1 \n",
       "⋮                     ⋮                    \n",
       "24_TATTGGGTCATAGACC-1 24_TATTGGGTCATAGACC-1\n",
       "24_GTTGTGATCAAGGAGC-1 24_GTTGTGATCAAGGAGC-1\n",
       "23_TTATTGCCAACCCTAA-1 23_TTATTGCCAACCCTAA-1\n",
       "22_CTGTACCCAATAACGA-1 22_CTGTACCCAATAACGA-1\n",
       "27_CCTGCATCAGCTTTCC-1 27_CCTGCATCAGCTTTCC-1\n",
       "23_GGGACAAGTACAAAGT-1 23_GGGACAAGTACAAAGT-1\n",
       "27_TAAGCCATCGTCAACA-1 27_TAAGCCATCGTCAACA-1\n",
       "25_CCAATTTGTGTTTCTT-1 25_CCAATTTGTGTTTCTT-1\n",
       "23_TCGGGCATCAAACCCA-1 23_TCGGGCATCAAACCCA-1\n",
       "23_GATTGGTAGACTTGTC-1 23_GATTGGTAGACTTGTC-1\n",
       "27_GTATTGGGTGTGTCGC-1 27_GTATTGGGTGTGTCGC-1\n",
       "22_TCTCTGGCACTCCTTG-1 22_TCTCTGGCACTCCTTG-1\n",
       "25_TCGGGTGAGAGAGCCT-1 25_TCGGGTGAGAGAGCCT-1\n",
       "24_GTGGCGTCAGTCAGTT-1 24_GTGGCGTCAGTCAGTT-1\n",
       "19_TGACGCGTCAGCGGAA-1 19_TGACGCGTCAGCGGAA-1\n",
       "19_TGAATCGTCTCCACTG-1 19_TGAATCGTCTCCACTG-1\n",
       "19_GGGTCTGGTTACCCTC-1 19_GGGTCTGGTTACCCTC-1\n",
       "19_ATCACAGCAGTAACCT-1 19_ATCACAGCAGTAACCT-1\n",
       "24_TTTGATCAGATGCGAC-1 24_TTTGATCAGATGCGAC-1\n",
       "4_CAGAGCCAGCAGGGAG-1  4_CAGAGCCAGCAGGGAG-1 \n",
       "22_TCCTGCAAGTGGAAGA-1 22_TCCTGCAAGTGGAAGA-1\n",
       "24_GAGGCCTGTTAATCGC-1 24_GAGGCCTGTTAATCGC-1\n",
       "24_ACATCCCAGTAGGGTC-1 24_ACATCCCAGTAGGGTC-1\n",
       "4_GGGTATTTCATGAGGG-1  4_GGGTATTTCATGAGGG-1 \n",
       "27_CACAGATGTTGGGCCT-1 27_CACAGATGTTGGGCCT-1\n",
       "24_TTCTCTCCAGTCAGAG-1 24_TTCTCTCCAGTCAGAG-1\n",
       "24_GGGACTCAGGGACCAT-1 24_GGGACTCAGGGACCAT-1\n",
       "23_GTGGTTAAGCAATAAC-1 23_GTGGTTAAGCAATAAC-1\n",
       "27_GACCTTCAGATAGTGT-1 27_GACCTTCAGATAGTGT-1\n",
       "19_TGCGGGTTCCAGCAAT-1 19_TGCGGGTTCCAGCAAT-1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "train_sampTab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "46db50f3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 5468 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>29_TTCTTCCGTACGATGG-1</th><th scope=col>1_GTGTAACCACCTGCAG-1</th><th scope=col>17_TCGTAGACAGCTGAAG-1</th><th scope=col>7_AGAAATGCACCCATAA-1</th><th scope=col>15_TTTACTGGTTTGAACC-1</th><th scope=col>26_TTGTTCACACGAGGTA-1</th><th scope=col>28_ATATCCTCAGGGCTTC-1</th><th scope=col>0_CAGGTATCAGAGTTCT-1</th><th scope=col>27_CTTCCTTAGCTTGTTG-1</th><th scope=col>28_TCAAGTGCACACTGGC-1</th><th scope=col>⋯</th><th scope=col>24_TTCGATTCAGTTCCAA-1</th><th scope=col>4_AAGGTAATCCGCACGA-1</th><th scope=col>25_TTGACCCAGCGTGTCC-1</th><th scope=col>4_CTATCCGCAACCCGCA-1</th><th scope=col>4_ATCGTGACAGGATTCT-1</th><th scope=col>19_AGCCACGAGTGGTCAG-1</th><th scope=col>19_CACAACATCAGTGGGA-1</th><th scope=col>27_TTTGTTGAGAGTAACT-1</th><th scope=col>22_GCAGCCAAGTCAAGCG-1</th><th scope=col>23_TGTGCGGGTAGACACG-1</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>AL627309.1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AL627309.3</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AL627309.4</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AL732372.1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AL669831.2</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><th scope=row>AL669831.5</th><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>⋯</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 5468 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & 29\\_TTCTTCCGTACGATGG-1 & 1\\_GTGTAACCACCTGCAG-1 & 17\\_TCGTAGACAGCTGAAG-1 & 7\\_AGAAATGCACCCATAA-1 & 15\\_TTTACTGGTTTGAACC-1 & 26\\_TTGTTCACACGAGGTA-1 & 28\\_ATATCCTCAGGGCTTC-1 & 0\\_CAGGTATCAGAGTTCT-1 & 27\\_CTTCCTTAGCTTGTTG-1 & 28\\_TCAAGTGCACACTGGC-1 & ⋯ & 24\\_TTCGATTCAGTTCCAA-1 & 4\\_AAGGTAATCCGCACGA-1 & 25\\_TTGACCCAGCGTGTCC-1 & 4\\_CTATCCGCAACCCGCA-1 & 4\\_ATCGTGACAGGATTCT-1 & 19\\_AGCCACGAGTGGTCAG-1 & 19\\_CACAACATCAGTGGGA-1 & 27\\_TTTGTTGAGAGTAACT-1 & 22\\_GCAGCCAAGTCAAGCG-1 & 23\\_TGTGCGGGTAGACACG-1\\\\\n",
       "\\hline\n",
       "\tAL627309.1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tAL627309.3 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tAL627309.4 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tAL732372.1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tAL669831.2 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\tAL669831.5 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & ⋯ & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 5468 of type dbl\n",
       "\n",
       "| <!--/--> | 29_TTCTTCCGTACGATGG-1 | 1_GTGTAACCACCTGCAG-1 | 17_TCGTAGACAGCTGAAG-1 | 7_AGAAATGCACCCATAA-1 | 15_TTTACTGGTTTGAACC-1 | 26_TTGTTCACACGAGGTA-1 | 28_ATATCCTCAGGGCTTC-1 | 0_CAGGTATCAGAGTTCT-1 | 27_CTTCCTTAGCTTGTTG-1 | 28_TCAAGTGCACACTGGC-1 | ⋯ | 24_TTCGATTCAGTTCCAA-1 | 4_AAGGTAATCCGCACGA-1 | 25_TTGACCCAGCGTGTCC-1 | 4_CTATCCGCAACCCGCA-1 | 4_ATCGTGACAGGATTCT-1 | 19_AGCCACGAGTGGTCAG-1 | 19_CACAACATCAGTGGGA-1 | 27_TTTGTTGAGAGTAACT-1 | 22_GCAGCCAAGTCAAGCG-1 | 23_TGTGCGGGTAGACACG-1 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| AL627309.1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| AL627309.3 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| AL627309.4 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| AL732372.1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| AL669831.2 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "| AL669831.5 | 0 | 0 | 1 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | ⋯ | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "           29_TTCTTCCGTACGATGG-1 1_GTGTAACCACCTGCAG-1 17_TCGTAGACAGCTGAAG-1\n",
       "AL627309.1 0                     0                    0                    \n",
       "AL627309.3 0                     0                    0                    \n",
       "AL627309.4 0                     0                    0                    \n",
       "AL732372.1 0                     0                    0                    \n",
       "AL669831.2 0                     0                    0                    \n",
       "AL669831.5 0                     0                    1                    \n",
       "           7_AGAAATGCACCCATAA-1 15_TTTACTGGTTTGAACC-1 26_TTGTTCACACGAGGTA-1\n",
       "AL627309.1 0                    0                     0                    \n",
       "AL627309.3 0                    0                     0                    \n",
       "AL627309.4 0                    0                     0                    \n",
       "AL732372.1 0                    0                     0                    \n",
       "AL669831.2 0                    0                     0                    \n",
       "AL669831.5 0                    0                     0                    \n",
       "           28_ATATCCTCAGGGCTTC-1 0_CAGGTATCAGAGTTCT-1 27_CTTCCTTAGCTTGTTG-1\n",
       "AL627309.1 0                     0                    0                    \n",
       "AL627309.3 0                     0                    0                    \n",
       "AL627309.4 0                     0                    0                    \n",
       "AL732372.1 0                     0                    0                    \n",
       "AL669831.2 0                     0                    0                    \n",
       "AL669831.5 0                     0                    0                    \n",
       "           28_TCAAGTGCACACTGGC-1 ⋯ 24_TTCGATTCAGTTCCAA-1 4_AAGGTAATCCGCACGA-1\n",
       "AL627309.1 0                     ⋯ 0                     0                   \n",
       "AL627309.3 0                     ⋯ 0                     0                   \n",
       "AL627309.4 0                     ⋯ 0                     0                   \n",
       "AL732372.1 0                     ⋯ 0                     0                   \n",
       "AL669831.2 0                     ⋯ 0                     0                   \n",
       "AL669831.5 0                     ⋯ 0                     0                   \n",
       "           25_TTGACCCAGCGTGTCC-1 4_CTATCCGCAACCCGCA-1 4_ATCGTGACAGGATTCT-1\n",
       "AL627309.1 0                     0                    0                   \n",
       "AL627309.3 0                     0                    0                   \n",
       "AL627309.4 0                     0                    0                   \n",
       "AL732372.1 0                     0                    0                   \n",
       "AL669831.2 0                     0                    0                   \n",
       "AL669831.5 0                     0                    0                   \n",
       "           19_AGCCACGAGTGGTCAG-1 19_CACAACATCAGTGGGA-1 27_TTTGTTGAGAGTAACT-1\n",
       "AL627309.1 0                     0                     0                    \n",
       "AL627309.3 0                     0                     0                    \n",
       "AL627309.4 0                     0                     0                    \n",
       "AL732372.1 0                     0                     0                    \n",
       "AL669831.2 0                     0                     0                    \n",
       "AL669831.5 0                     0                     0                    \n",
       "           22_GCAGCCAAGTCAAGCG-1 23_TGTGCGGGTAGACACG-1\n",
       "AL627309.1 0                     0                    \n",
       "AL627309.3 0                     0                    \n",
       "AL627309.4 0                     0                    \n",
       "AL732372.1 0                     0                    \n",
       "AL669831.2 0                     0                    \n",
       "AL669831.5 0                     0                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(val_expDat)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "75f24c70",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 3</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>cell_type</th><th scope=col>barcode</th><th scope=col>sample</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>29_TTCTTCCGTACGATGG-1</th><td>CD8 T cell</td><td>TTCTTCCGTACGATGG-1</td><td>29</td></tr>\n",
       "\t<tr><th scope=row>1_GTGTAACCACCTGCAG-1</th><td>CD8 T cell</td><td>GTGTAACCACCTGCAG-1</td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>17_TCGTAGACAGCTGAAG-1</th><td>CD8 T cell</td><td>TCGTAGACAGCTGAAG-1</td><td>17</td></tr>\n",
       "\t<tr><th scope=row>7_AGAAATGCACCCATAA-1</th><td>CD8 T cell</td><td>AGAAATGCACCCATAA-1</td><td> 7</td></tr>\n",
       "\t<tr><th scope=row>15_TTTACTGGTTTGAACC-1</th><td>CD8 T cell</td><td>TTTACTGGTTTGAACC-1</td><td>15</td></tr>\n",
       "\t<tr><th scope=row>26_TTGTTCACACGAGGTA-1</th><td>CD8 T cell</td><td>TTGTTCACACGAGGTA-1</td><td>26</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 3\n",
       "\\begin{tabular}{r|lll}\n",
       "  & cell\\_type & barcode & sample\\\\\n",
       "  & <fct> & <fct> & <int>\\\\\n",
       "\\hline\n",
       "\t29\\_TTCTTCCGTACGATGG-1 & CD8 T cell & TTCTTCCGTACGATGG-1 & 29\\\\\n",
       "\t1\\_GTGTAACCACCTGCAG-1 & CD8 T cell & GTGTAACCACCTGCAG-1 &  1\\\\\n",
       "\t17\\_TCGTAGACAGCTGAAG-1 & CD8 T cell & TCGTAGACAGCTGAAG-1 & 17\\\\\n",
       "\t7\\_AGAAATGCACCCATAA-1 & CD8 T cell & AGAAATGCACCCATAA-1 &  7\\\\\n",
       "\t15\\_TTTACTGGTTTGAACC-1 & CD8 T cell & TTTACTGGTTTGAACC-1 & 15\\\\\n",
       "\t26\\_TTGTTCACACGAGGTA-1 & CD8 T cell & TTGTTCACACGAGGTA-1 & 26\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 3\n",
       "\n",
       "| <!--/--> | cell_type &lt;fct&gt; | barcode &lt;fct&gt; | sample &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| 29_TTCTTCCGTACGATGG-1 | CD8 T cell | TTCTTCCGTACGATGG-1 | 29 |\n",
       "| 1_GTGTAACCACCTGCAG-1 | CD8 T cell | GTGTAACCACCTGCAG-1 |  1 |\n",
       "| 17_TCGTAGACAGCTGAAG-1 | CD8 T cell | TCGTAGACAGCTGAAG-1 | 17 |\n",
       "| 7_AGAAATGCACCCATAA-1 | CD8 T cell | AGAAATGCACCCATAA-1 |  7 |\n",
       "| 15_TTTACTGGTTTGAACC-1 | CD8 T cell | TTTACTGGTTTGAACC-1 | 15 |\n",
       "| 26_TTGTTCACACGAGGTA-1 | CD8 T cell | TTGTTCACACGAGGTA-1 | 26 |\n",
       "\n"
      ],
      "text/plain": [
       "                      cell_type  barcode            sample\n",
       "29_TTCTTCCGTACGATGG-1 CD8 T cell TTCTTCCGTACGATGG-1 29    \n",
       "1_GTGTAACCACCTGCAG-1  CD8 T cell GTGTAACCACCTGCAG-1  1    \n",
       "17_TCGTAGACAGCTGAAG-1 CD8 T cell TCGTAGACAGCTGAAG-1 17    \n",
       "7_AGAAATGCACCCATAA-1  CD8 T cell AGAAATGCACCCATAA-1  7    \n",
       "15_TTTACTGGTTTGAACC-1 CD8 T cell TTTACTGGTTTGAACC-1 15    \n",
       "26_TTGTTCACACGAGGTA-1 CD8 T cell TTGTTCACACGAGGTA-1 26    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(val_sampTab)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "205d22b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample table has been prepared\n",
      "Expression data has been normalized\n",
      "Finding classification genes\n",
      "Done testing\n",
      "There are  210  classification genes\n",
      "Finding top pairs\n",
      "nPairs =  190  for  CD8 T cell \n",
      "nPairs =  190  for  Endothelial \n",
      "nPairs =  190  for  Fibroblast \n",
      "nPairs =  190  for  Malignant \n",
      "nPairs =  190  for  NK \n",
      "nPairs =  190  for  Monocyte \n",
      "nPairs =  190  for  Treg \n",
      "nPairs =  190  for  CD4 T cell \n",
      "nPairs =  190  for  B cell \n",
      "nPairs =  190  for  Plasma \n",
      "nPairs =  190  for  Mast \n",
      "nPairs =  190  for  Dendritic \n",
      "nPairs =  190  for  Erythrocyte \n",
      "There are 324 top gene pairs\n",
      "Finished pair transforming the data\n",
      "Number of missing genes  0 \n",
      "All Done\n"
     ]
    }
   ],
   "source": [
    "# class_info<-scn_train(\n",
    "#     stTrain = stTrain, expTrain = expTrain, nTopGenes = 10, nRand = 70,\n",
    "#     nTrees = 1000, nTopGenePairs = 25, dLevel = \"newAnn\", colName_samp = \"cell\")\n",
    "class_info<-scn_train(\n",
    "    stTrain = train_sampTab, expTrain = train_expDat, nTopGenes = 10, nRand = 70,\n",
    "    nTrees = 1000, nTopGenePairs = 25, dLevel = \"cell_type\", colName_samp = \"cell_ids\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "1dbe0c66",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loaded in the cnProc\n",
      "All Done\n"
     ]
    }
   ],
   "source": [
    "#validate data\n",
    "# stTestList = splitCommon(sampTab=stList[[2]], ncells=100, dLevel=\"newAnn\")\n",
    "# stTest = stTestList[[1]]\n",
    "# expTest = expTMraw[commonGenes,rownames(stTest)]\n",
    "\n",
    "#predict\n",
    "classRes_val_all = scn_predict(cnProc=class_info[['cnProc']], expDat=val_expDat, nrand = 50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fc6d04bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 14 × 5518 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>29_TTCTTCCGTACGATGG-1</th><th scope=col>1_GTGTAACCACCTGCAG-1</th><th scope=col>17_TCGTAGACAGCTGAAG-1</th><th scope=col>7_AGAAATGCACCCATAA-1</th><th scope=col>15_TTTACTGGTTTGAACC-1</th><th scope=col>26_TTGTTCACACGAGGTA-1</th><th scope=col>28_ATATCCTCAGGGCTTC-1</th><th scope=col>0_CAGGTATCAGAGTTCT-1</th><th scope=col>27_CTTCCTTAGCTTGTTG-1</th><th scope=col>28_TCAAGTGCACACTGGC-1</th><th scope=col>⋯</th><th scope=col>rand_41</th><th scope=col>rand_42</th><th scope=col>rand_43</th><th scope=col>rand_44</th><th scope=col>rand_45</th><th scope=col>rand_46</th><th scope=col>rand_47</th><th scope=col>rand_48</th><th scope=col>rand_49</th><th scope=col>rand_50</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B cell</th><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.000</td><td>0.000</td><td>0.000</td><td>⋯</td><td>0.067</td><td>0.038</td><td>0.002</td><td>0.012</td><td>0.089</td><td>0.081</td><td>0.053</td><td>0.035</td><td>0.113</td><td>0.124</td></tr>\n",
       "\t<tr><th scope=row>CD4 T cell</th><td>0.090</td><td>0.142</td><td>0.038</td><td>0.040</td><td>0.203</td><td>0.484</td><td>0.042</td><td>0.043</td><td>0.128</td><td>0.076</td><td>⋯</td><td>0.023</td><td>0.011</td><td>0.053</td><td>0.017</td><td>0.034</td><td>0.055</td><td>0.080</td><td>0.021</td><td>0.014</td><td>0.059</td></tr>\n",
       "\t<tr><th scope=row>CD8 T cell</th><td>0.701</td><td>0.776</td><td>0.720</td><td>0.900</td><td>0.515</td><td>0.292</td><td>0.053</td><td>0.678</td><td>0.364</td><td>0.840</td><td>⋯</td><td>0.025</td><td>0.008</td><td>0.017</td><td>0.008</td><td>0.017</td><td>0.014</td><td>0.018</td><td>0.019</td><td>0.005</td><td>0.037</td></tr>\n",
       "\t<tr><th scope=row>Dendritic</th><td>0.002</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.001</td><td>0.002</td><td>0.002</td><td>0.000</td><td>0.009</td><td>0.000</td><td>⋯</td><td>0.097</td><td>0.176</td><td>0.025</td><td>0.042</td><td>0.078</td><td>0.073</td><td>0.074</td><td>0.076</td><td>0.098</td><td>0.066</td></tr>\n",
       "\t<tr><th scope=row>Endothelial</th><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.002</td><td>0.001</td><td>0.000</td><td>0.000</td><td>0.003</td><td>0.000</td><td>⋯</td><td>0.087</td><td>0.092</td><td>0.035</td><td>0.081</td><td>0.051</td><td>0.054</td><td>0.073</td><td>0.088</td><td>0.029</td><td>0.027</td></tr>\n",
       "\t<tr><th scope=row>Erythrocyte</th><td>0.045</td><td>0.004</td><td>0.009</td><td>0.010</td><td>0.001</td><td>0.002</td><td>0.004</td><td>0.000</td><td>0.003</td><td>0.076</td><td>⋯</td><td>0.172</td><td>0.142</td><td>0.137</td><td>0.192</td><td>0.135</td><td>0.111</td><td>0.092</td><td>0.121</td><td>0.106</td><td>0.054</td></tr>\n",
       "\t<tr><th scope=row>Fibroblast</th><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.000</td><td>0.001</td><td>0.000</td><td>0.006</td><td>0.000</td><td>⋯</td><td>0.040</td><td>0.078</td><td>0.064</td><td>0.136</td><td>0.049</td><td>0.047</td><td>0.085</td><td>0.120</td><td>0.011</td><td>0.022</td></tr>\n",
       "\t<tr><th scope=row>Malignant</th><td>0.005</td><td>0.001</td><td>0.000</td><td>0.000</td><td>0.010</td><td>0.001</td><td>0.005</td><td>0.000</td><td>0.007</td><td>0.000</td><td>⋯</td><td>0.130</td><td>0.093</td><td>0.226</td><td>0.119</td><td>0.119</td><td>0.096</td><td>0.122</td><td>0.095</td><td>0.082</td><td>0.069</td></tr>\n",
       "\t<tr><th scope=row>Mast</th><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.000</td><td>⋯</td><td>0.037</td><td>0.032</td><td>0.006</td><td>0.044</td><td>0.077</td><td>0.006</td><td>0.061</td><td>0.058</td><td>0.041</td><td>0.008</td></tr>\n",
       "\t<tr><th scope=row>Monocyte</th><td>0.001</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.002</td><td>0.002</td><td>0.000</td><td>0.000</td><td>0.010</td><td>0.000</td><td>⋯</td><td>0.121</td><td>0.163</td><td>0.109</td><td>0.143</td><td>0.072</td><td>0.145</td><td>0.040</td><td>0.125</td><td>0.031</td><td>0.108</td></tr>\n",
       "\t<tr><th scope=row>NK</th><td>0.134</td><td>0.069</td><td>0.203</td><td>0.006</td><td>0.215</td><td>0.094</td><td>0.003</td><td>0.268</td><td>0.430</td><td>0.002</td><td>⋯</td><td>0.058</td><td>0.022</td><td>0.038</td><td>0.015</td><td>0.058</td><td>0.059</td><td>0.073</td><td>0.027</td><td>0.025</td><td>0.044</td></tr>\n",
       "\t<tr><th scope=row>Plasma</th><td>0.004</td><td>0.001</td><td>0.001</td><td>0.000</td><td>0.002</td><td>0.007</td><td>0.002</td><td>0.000</td><td>0.021</td><td>0.000</td><td>⋯</td><td>0.025</td><td>0.013</td><td>0.150</td><td>0.088</td><td>0.098</td><td>0.115</td><td>0.059</td><td>0.069</td><td>0.266</td><td>0.286</td></tr>\n",
       "\t<tr><th scope=row>rand</th><td>0.002</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.000</td><td>0.000</td><td>0.000</td><td>0.001</td><td>0.000</td><td>⋯</td><td>0.101</td><td>0.118</td><td>0.113</td><td>0.101</td><td>0.116</td><td>0.116</td><td>0.145</td><td>0.133</td><td>0.165</td><td>0.074</td></tr>\n",
       "\t<tr><th scope=row>Treg</th><td>0.016</td><td>0.007</td><td>0.029</td><td>0.043</td><td>0.047</td><td>0.115</td><td>0.887</td><td>0.011</td><td>0.017</td><td>0.006</td><td>⋯</td><td>0.017</td><td>0.014</td><td>0.025</td><td>0.002</td><td>0.007</td><td>0.028</td><td>0.025</td><td>0.013</td><td>0.014</td><td>0.022</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 14 × 5518 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & 29\\_TTCTTCCGTACGATGG-1 & 1\\_GTGTAACCACCTGCAG-1 & 17\\_TCGTAGACAGCTGAAG-1 & 7\\_AGAAATGCACCCATAA-1 & 15\\_TTTACTGGTTTGAACC-1 & 26\\_TTGTTCACACGAGGTA-1 & 28\\_ATATCCTCAGGGCTTC-1 & 0\\_CAGGTATCAGAGTTCT-1 & 27\\_CTTCCTTAGCTTGTTG-1 & 28\\_TCAAGTGCACACTGGC-1 & ⋯ & rand\\_41 & rand\\_42 & rand\\_43 & rand\\_44 & rand\\_45 & rand\\_46 & rand\\_47 & rand\\_48 & rand\\_49 & rand\\_50\\\\\n",
       "\\hline\n",
       "\tB cell & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.001 & 0.000 & 0.000 & 0.000 & ⋯ & 0.067 & 0.038 & 0.002 & 0.012 & 0.089 & 0.081 & 0.053 & 0.035 & 0.113 & 0.124\\\\\n",
       "\tCD4 T cell & 0.090 & 0.142 & 0.038 & 0.040 & 0.203 & 0.484 & 0.042 & 0.043 & 0.128 & 0.076 & ⋯ & 0.023 & 0.011 & 0.053 & 0.017 & 0.034 & 0.055 & 0.080 & 0.021 & 0.014 & 0.059\\\\\n",
       "\tCD8 T cell & 0.701 & 0.776 & 0.720 & 0.900 & 0.515 & 0.292 & 0.053 & 0.678 & 0.364 & 0.840 & ⋯ & 0.025 & 0.008 & 0.017 & 0.008 & 0.017 & 0.014 & 0.018 & 0.019 & 0.005 & 0.037\\\\\n",
       "\tDendritic & 0.002 & 0.000 & 0.000 & 0.001 & 0.001 & 0.002 & 0.002 & 0.000 & 0.009 & 0.000 & ⋯ & 0.097 & 0.176 & 0.025 & 0.042 & 0.078 & 0.073 & 0.074 & 0.076 & 0.098 & 0.066\\\\\n",
       "\tEndothelial & 0.000 & 0.000 & 0.000 & 0.000 & 0.002 & 0.001 & 0.000 & 0.000 & 0.003 & 0.000 & ⋯ & 0.087 & 0.092 & 0.035 & 0.081 & 0.051 & 0.054 & 0.073 & 0.088 & 0.029 & 0.027\\\\\n",
       "\tErythrocyte & 0.045 & 0.004 & 0.009 & 0.010 & 0.001 & 0.002 & 0.004 & 0.000 & 0.003 & 0.076 & ⋯ & 0.172 & 0.142 & 0.137 & 0.192 & 0.135 & 0.111 & 0.092 & 0.121 & 0.106 & 0.054\\\\\n",
       "\tFibroblast & 0.000 & 0.000 & 0.000 & 0.000 & 0.001 & 0.000 & 0.001 & 0.000 & 0.006 & 0.000 & ⋯ & 0.040 & 0.078 & 0.064 & 0.136 & 0.049 & 0.047 & 0.085 & 0.120 & 0.011 & 0.022\\\\\n",
       "\tMalignant & 0.005 & 0.001 & 0.000 & 0.000 & 0.010 & 0.001 & 0.005 & 0.000 & 0.007 & 0.000 & ⋯ & 0.130 & 0.093 & 0.226 & 0.119 & 0.119 & 0.096 & 0.122 & 0.095 & 0.082 & 0.069\\\\\n",
       "\tMast & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.000 & 0.001 & 0.000 & ⋯ & 0.037 & 0.032 & 0.006 & 0.044 & 0.077 & 0.006 & 0.061 & 0.058 & 0.041 & 0.008\\\\\n",
       "\tMonocyte & 0.001 & 0.000 & 0.000 & 0.000 & 0.002 & 0.002 & 0.000 & 0.000 & 0.010 & 0.000 & ⋯ & 0.121 & 0.163 & 0.109 & 0.143 & 0.072 & 0.145 & 0.040 & 0.125 & 0.031 & 0.108\\\\\n",
       "\tNK & 0.134 & 0.069 & 0.203 & 0.006 & 0.215 & 0.094 & 0.003 & 0.268 & 0.430 & 0.002 & ⋯ & 0.058 & 0.022 & 0.038 & 0.015 & 0.058 & 0.059 & 0.073 & 0.027 & 0.025 & 0.044\\\\\n",
       "\tPlasma & 0.004 & 0.001 & 0.001 & 0.000 & 0.002 & 0.007 & 0.002 & 0.000 & 0.021 & 0.000 & ⋯ & 0.025 & 0.013 & 0.150 & 0.088 & 0.098 & 0.115 & 0.059 & 0.069 & 0.266 & 0.286\\\\\n",
       "\trand & 0.002 & 0.000 & 0.000 & 0.000 & 0.001 & 0.000 & 0.000 & 0.000 & 0.001 & 0.000 & ⋯ & 0.101 & 0.118 & 0.113 & 0.101 & 0.116 & 0.116 & 0.145 & 0.133 & 0.165 & 0.074\\\\\n",
       "\tTreg & 0.016 & 0.007 & 0.029 & 0.043 & 0.047 & 0.115 & 0.887 & 0.011 & 0.017 & 0.006 & ⋯ & 0.017 & 0.014 & 0.025 & 0.002 & 0.007 & 0.028 & 0.025 & 0.013 & 0.014 & 0.022\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 14 × 5518 of type dbl\n",
       "\n",
       "| <!--/--> | 29_TTCTTCCGTACGATGG-1 | 1_GTGTAACCACCTGCAG-1 | 17_TCGTAGACAGCTGAAG-1 | 7_AGAAATGCACCCATAA-1 | 15_TTTACTGGTTTGAACC-1 | 26_TTGTTCACACGAGGTA-1 | 28_ATATCCTCAGGGCTTC-1 | 0_CAGGTATCAGAGTTCT-1 | 27_CTTCCTTAGCTTGTTG-1 | 28_TCAAGTGCACACTGGC-1 | ⋯ | rand_41 | rand_42 | rand_43 | rand_44 | rand_45 | rand_46 | rand_47 | rand_48 | rand_49 | rand_50 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| B cell | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.001 | 0.000 | 0.000 | 0.000 | ⋯ | 0.067 | 0.038 | 0.002 | 0.012 | 0.089 | 0.081 | 0.053 | 0.035 | 0.113 | 0.124 |\n",
       "| CD4 T cell | 0.090 | 0.142 | 0.038 | 0.040 | 0.203 | 0.484 | 0.042 | 0.043 | 0.128 | 0.076 | ⋯ | 0.023 | 0.011 | 0.053 | 0.017 | 0.034 | 0.055 | 0.080 | 0.021 | 0.014 | 0.059 |\n",
       "| CD8 T cell | 0.701 | 0.776 | 0.720 | 0.900 | 0.515 | 0.292 | 0.053 | 0.678 | 0.364 | 0.840 | ⋯ | 0.025 | 0.008 | 0.017 | 0.008 | 0.017 | 0.014 | 0.018 | 0.019 | 0.005 | 0.037 |\n",
       "| Dendritic | 0.002 | 0.000 | 0.000 | 0.001 | 0.001 | 0.002 | 0.002 | 0.000 | 0.009 | 0.000 | ⋯ | 0.097 | 0.176 | 0.025 | 0.042 | 0.078 | 0.073 | 0.074 | 0.076 | 0.098 | 0.066 |\n",
       "| Endothelial | 0.000 | 0.000 | 0.000 | 0.000 | 0.002 | 0.001 | 0.000 | 0.000 | 0.003 | 0.000 | ⋯ | 0.087 | 0.092 | 0.035 | 0.081 | 0.051 | 0.054 | 0.073 | 0.088 | 0.029 | 0.027 |\n",
       "| Erythrocyte | 0.045 | 0.004 | 0.009 | 0.010 | 0.001 | 0.002 | 0.004 | 0.000 | 0.003 | 0.076 | ⋯ | 0.172 | 0.142 | 0.137 | 0.192 | 0.135 | 0.111 | 0.092 | 0.121 | 0.106 | 0.054 |\n",
       "| Fibroblast | 0.000 | 0.000 | 0.000 | 0.000 | 0.001 | 0.000 | 0.001 | 0.000 | 0.006 | 0.000 | ⋯ | 0.040 | 0.078 | 0.064 | 0.136 | 0.049 | 0.047 | 0.085 | 0.120 | 0.011 | 0.022 |\n",
       "| Malignant | 0.005 | 0.001 | 0.000 | 0.000 | 0.010 | 0.001 | 0.005 | 0.000 | 0.007 | 0.000 | ⋯ | 0.130 | 0.093 | 0.226 | 0.119 | 0.119 | 0.096 | 0.122 | 0.095 | 0.082 | 0.069 |\n",
       "| Mast | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.000 | 0.001 | 0.000 | ⋯ | 0.037 | 0.032 | 0.006 | 0.044 | 0.077 | 0.006 | 0.061 | 0.058 | 0.041 | 0.008 |\n",
       "| Monocyte | 0.001 | 0.000 | 0.000 | 0.000 | 0.002 | 0.002 | 0.000 | 0.000 | 0.010 | 0.000 | ⋯ | 0.121 | 0.163 | 0.109 | 0.143 | 0.072 | 0.145 | 0.040 | 0.125 | 0.031 | 0.108 |\n",
       "| NK | 0.134 | 0.069 | 0.203 | 0.006 | 0.215 | 0.094 | 0.003 | 0.268 | 0.430 | 0.002 | ⋯ | 0.058 | 0.022 | 0.038 | 0.015 | 0.058 | 0.059 | 0.073 | 0.027 | 0.025 | 0.044 |\n",
       "| Plasma | 0.004 | 0.001 | 0.001 | 0.000 | 0.002 | 0.007 | 0.002 | 0.000 | 0.021 | 0.000 | ⋯ | 0.025 | 0.013 | 0.150 | 0.088 | 0.098 | 0.115 | 0.059 | 0.069 | 0.266 | 0.286 |\n",
       "| rand | 0.002 | 0.000 | 0.000 | 0.000 | 0.001 | 0.000 | 0.000 | 0.000 | 0.001 | 0.000 | ⋯ | 0.101 | 0.118 | 0.113 | 0.101 | 0.116 | 0.116 | 0.145 | 0.133 | 0.165 | 0.074 |\n",
       "| Treg | 0.016 | 0.007 | 0.029 | 0.043 | 0.047 | 0.115 | 0.887 | 0.011 | 0.017 | 0.006 | ⋯ | 0.017 | 0.014 | 0.025 | 0.002 | 0.007 | 0.028 | 0.025 | 0.013 | 0.014 | 0.022 |\n",
       "\n"
      ],
      "text/plain": [
       "            29_TTCTTCCGTACGATGG-1 1_GTGTAACCACCTGCAG-1 17_TCGTAGACAGCTGAAG-1\n",
       "B cell      0.000                 0.000                0.000                \n",
       "CD4 T cell  0.090                 0.142                0.038                \n",
       "CD8 T cell  0.701                 0.776                0.720                \n",
       "Dendritic   0.002                 0.000                0.000                \n",
       "Endothelial 0.000                 0.000                0.000                \n",
       "Erythrocyte 0.045                 0.004                0.009                \n",
       "Fibroblast  0.000                 0.000                0.000                \n",
       "Malignant   0.005                 0.001                0.000                \n",
       "Mast        0.000                 0.000                0.000                \n",
       "Monocyte    0.001                 0.000                0.000                \n",
       "NK          0.134                 0.069                0.203                \n",
       "Plasma      0.004                 0.001                0.001                \n",
       "rand        0.002                 0.000                0.000                \n",
       "Treg        0.016                 0.007                0.029                \n",
       "            7_AGAAATGCACCCATAA-1 15_TTTACTGGTTTGAACC-1 26_TTGTTCACACGAGGTA-1\n",
       "B cell      0.000                0.000                 0.000                \n",
       "CD4 T cell  0.040                0.203                 0.484                \n",
       "CD8 T cell  0.900                0.515                 0.292                \n",
       "Dendritic   0.001                0.001                 0.002                \n",
       "Endothelial 0.000                0.002                 0.001                \n",
       "Erythrocyte 0.010                0.001                 0.002                \n",
       "Fibroblast  0.000                0.001                 0.000                \n",
       "Malignant   0.000                0.010                 0.001                \n",
       "Mast        0.000                0.000                 0.000                \n",
       "Monocyte    0.000                0.002                 0.002                \n",
       "NK          0.006                0.215                 0.094                \n",
       "Plasma      0.000                0.002                 0.007                \n",
       "rand        0.000                0.001                 0.000                \n",
       "Treg        0.043                0.047                 0.115                \n",
       "            28_ATATCCTCAGGGCTTC-1 0_CAGGTATCAGAGTTCT-1 27_CTTCCTTAGCTTGTTG-1\n",
       "B cell      0.001                 0.000                0.000                \n",
       "CD4 T cell  0.042                 0.043                0.128                \n",
       "CD8 T cell  0.053                 0.678                0.364                \n",
       "Dendritic   0.002                 0.000                0.009                \n",
       "Endothelial 0.000                 0.000                0.003                \n",
       "Erythrocyte 0.004                 0.000                0.003                \n",
       "Fibroblast  0.001                 0.000                0.006                \n",
       "Malignant   0.005                 0.000                0.007                \n",
       "Mast        0.000                 0.000                0.001                \n",
       "Monocyte    0.000                 0.000                0.010                \n",
       "NK          0.003                 0.268                0.430                \n",
       "Plasma      0.002                 0.000                0.021                \n",
       "rand        0.000                 0.000                0.001                \n",
       "Treg        0.887                 0.011                0.017                \n",
       "            28_TCAAGTGCACACTGGC-1 ⋯ rand_41 rand_42 rand_43 rand_44 rand_45\n",
       "B cell      0.000                 ⋯ 0.067   0.038   0.002   0.012   0.089  \n",
       "CD4 T cell  0.076                 ⋯ 0.023   0.011   0.053   0.017   0.034  \n",
       "CD8 T cell  0.840                 ⋯ 0.025   0.008   0.017   0.008   0.017  \n",
       "Dendritic   0.000                 ⋯ 0.097   0.176   0.025   0.042   0.078  \n",
       "Endothelial 0.000                 ⋯ 0.087   0.092   0.035   0.081   0.051  \n",
       "Erythrocyte 0.076                 ⋯ 0.172   0.142   0.137   0.192   0.135  \n",
       "Fibroblast  0.000                 ⋯ 0.040   0.078   0.064   0.136   0.049  \n",
       "Malignant   0.000                 ⋯ 0.130   0.093   0.226   0.119   0.119  \n",
       "Mast        0.000                 ⋯ 0.037   0.032   0.006   0.044   0.077  \n",
       "Monocyte    0.000                 ⋯ 0.121   0.163   0.109   0.143   0.072  \n",
       "NK          0.002                 ⋯ 0.058   0.022   0.038   0.015   0.058  \n",
       "Plasma      0.000                 ⋯ 0.025   0.013   0.150   0.088   0.098  \n",
       "rand        0.000                 ⋯ 0.101   0.118   0.113   0.101   0.116  \n",
       "Treg        0.006                 ⋯ 0.017   0.014   0.025   0.002   0.007  \n",
       "            rand_46 rand_47 rand_48 rand_49 rand_50\n",
       "B cell      0.081   0.053   0.035   0.113   0.124  \n",
       "CD4 T cell  0.055   0.080   0.021   0.014   0.059  \n",
       "CD8 T cell  0.014   0.018   0.019   0.005   0.037  \n",
       "Dendritic   0.073   0.074   0.076   0.098   0.066  \n",
       "Endothelial 0.054   0.073   0.088   0.029   0.027  \n",
       "Erythrocyte 0.111   0.092   0.121   0.106   0.054  \n",
       "Fibroblast  0.047   0.085   0.120   0.011   0.022  \n",
       "Malignant   0.096   0.122   0.095   0.082   0.069  \n",
       "Mast        0.006   0.061   0.058   0.041   0.008  \n",
       "Monocyte    0.145   0.040   0.125   0.031   0.108  \n",
       "NK          0.059   0.073   0.027   0.025   0.044  \n",
       "Plasma      0.115   0.059   0.069   0.266   0.286  \n",
       "rand        0.116   0.145   0.133   0.165   0.074  \n",
       "Treg        0.028   0.025   0.013   0.014   0.022  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "classRes_val_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "49dea472",
   "metadata": {},
   "outputs": [],
   "source": [
    "dtype = 'scRNAseq'\n",
    "disease = 'brca_fold2'\n",
    "txt_fp = paste(\"/data/pollock/benchmarking/results/singlecellnet/interdataset_with_folds/\", dtype, \"_\", disease, \".txt\", sep = \"\")\n",
    "write.table(classRes_val_all, file = txt_fp, sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fac655c6",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
